diff --git a/XMonad/Util/DebugWindow.hs b/XMonad/Util/DebugWindow.hs
index d4f5138..866c5b4 100644
--- a/XMonad/Util/DebugWindow.hs
+++ b/XMonad/Util/DebugWindow.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 -----------------------------------------------------------------------------
@@ -43,3 +44,7 @@ debugWindow w =  do
       return $ "(deleted window " ++ wx ++ ")"
-    Just (WindowAttributes x y wid ht bw m o) -> do
+    Just (WindowAttributes x y wid ht bw
+#if MIN_VERSION_X11(1,7,0)
+                                      _cm _mi
+#endif
+                                      m o) -> do
       c' <- withDisplay $ \d ->
