diff -rN -u old-xmonad-contrib-0.8/XMonad/Hooks/UrgencyHook.hs new-xmonad-contrib-0.8/XMonad/Hooks/UrgencyHook.hs
--- old-xmonad-contrib-0.8/XMonad/Hooks/UrgencyHook.hs	2008-12-29 18:20:15.000000000 +0100
+++ new-xmonad-contrib-0.8/XMonad/Hooks/UrgencyHook.hs	2008-12-29 18:20:15.000000000 +0100
@@ -356,7 +356,7 @@
 -- Defaults to a duration of five seconds, and no extra args to dzen.
 -- See 'DzenUrgencyHook'.
 dzenUrgencyHook :: DzenUrgencyHook
-dzenUrgencyHook = DzenUrgencyHook { duration = (5 `seconds`), args = [] }
+dzenUrgencyHook = DzenUrgencyHook { duration = (seconds 5), args = [] }
 
 -- | Spawn a commandline thing, appending the window id to the prefix string
 -- you provide. (Make sure to add a space if you need it.) Do your crazy

