--- bindings-bfd-1.2.0.0-orig/src/Bindings/Bfd.hsc	2012-07-17 01:32:14.000000000 +1000
+++ bindings-bfd-1.2.0.0/src/Bindings/Bfd.hsc	2012-10-12 12:15:21.811374158 +1100
@@ -73,7 +73,7 @@
 import Data.Bits  ( (.&.), bit )
 import Data.Word  ( Word )
 
-import Foreign.C        ( CInt, CLong, CString, CUInt, newCString, peekCString
+import Foreign.C        ( CInt(..), CLong(..), CString, CUInt(..), newCString, peekCString
                         , withCString )
 import Foreign.Marshal  ( free, mallocArray, peekArray, toBool )
 import Foreign.Ptr      ( FunPtr, Ptr, nullPtr, wordPtrToPtr )
--- bindings-bfd-1.2.0.0-orig/src/Bindings/Bfd/Target.hsc	2012-07-17 01:32:14.000000000 +1000
+++ bindings-bfd-1.2.0.0/src/Bindings/Bfd/Target.hsc	2012-10-12 12:16:01.722440973 +1100
@@ -57,7 +57,7 @@
 import Data.Bits ( (.&.), bit )
 import Data.Word ( Word )
 
-import Foreign.C        ( CInt, CLong, CString, CUInt, newCString, peekCString
+import Foreign.C        ( CInt(..), CLong, CString, CUInt, newCString, peekCString
                         , withCString )
 import Foreign.Marshal  ( free, peekArray0, toBool )
 import Foreign.Ptr      ( FunPtr, Ptr, castPtrToFunPtr, nullPtr, wordPtrToPtr )
