--- hamlet-1.0.1.2-orig/hamlet.cabal	2012-04-29 16:39:47.000000000 +1000
+++ hamlet-1.0.1.2/hamlet.cabal	2012-05-05 14:59:42.638605783 +1000
@@ -82,7 +82,7 @@
                  , containers       >= 0.2     && < 0.5
                  , text             >= 0.7     && < 1
                  , HUnit
-                 , hspec            >= 0.8     && < 0.10
+                 , hspec            >= 0.8     && < 1.1
                  , blaze-html       >= 0.5     && < 0.6
                  , blaze-markup     >= 0.5.1   && < 0.6
 
