diff --git a/hamlet.cabal b/hamlet.cabal
index 1ca6a43..e30fc77 100644
--- a/hamlet.cabal
+++ b/hamlet.cabal
@@ -60,24 +60,6 @@ library
                      Text.Css
     ghc-options:     -Wall
 
-test-suite runtests
-    hs-source-dirs: tests
-    main-is: runtests.hs
-    type: exitcode-stdio-1.0
-
-    ghc-options:   -Wall
-    build-depends: hamlet >= 0.7.2,
-                   base             >= 4       && < 5,
-                   parsec           >= 2       && < 4,
-                   containers       >= 0.2     && < 0.5,
-                   json-types       >= 0.1     && < 0.2,
-                   text             >= 0.7     && < 0.12,
-                   HUnit,
-                   QuickCheck >= 2 && < 3,
-                   test-framework-hunit,
-                   test-framework
-
-
 source-repository head
   type:     git
   location: git://github.com/snoyberg/hamlet.git
