From a0ebe34207569b5f8abfafa6d42f75bafc495007 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@protonmail.com>
Date: Sat, 14 Oct 2023 17:07:30 -0600
Subject: [PATCH 1/1] Disable timeouts

Signed-off-by: hololeap <hololeap@protonmail.com>
---
 src/Test/Hls.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Test/Hls.hs b/src/Test/Hls.hs
index a52e46f..24a84cd 100644
--- a/src/Test/Hls.hs
+++ b/src/Test/Hls.hs
@@ -152,7 +152,7 @@ instance Pretty LogTestHarness where
 
 -- | Run 'defaultMainWithRerun', limiting each single test case running at most 10 minutes
 defaultTestRunner :: TestTree -> IO ()
-defaultTestRunner = defaultMainWithRerun . adjustOption (const $ NumThreads 1) . adjustOption (const $ mkTimeout 600000000)
+defaultTestRunner = defaultMainWithRerun . adjustOption (const $ NumThreads 1) . adjustOption (const NoTimeout)
 
 gitDiff :: FilePath -> FilePath -> [String]
 gitDiff fRef fNew = ["git", "-c", "core.fileMode=false", "diff", "--no-index", "--text", "--exit-code", fRef, fNew]
-- 
2.41.0

