diff --git a/data-memocombinators.cabal b/data-memocombinators.cabal
index 901965a..cb683d6 100644
--- a/data-memocombinators.cabal
+++ b/data-memocombinators.cabal
@@ -10,6 +10,6 @@ Author: Luke Palmer
 Homepage: http://github.com/luqui/data-memocombinators
 Maintainer: lrpalmer@gmail.com
 Build-Type: Simple
-Build-Depends: base >= 3 && < 4.4, array, data-inttrie
+Build-Depends: base >= 3 && < 4.5, array, data-inttrie
 Exposed-Modules: Data.MemoCombinators
 Extensions: Rank2Types, ScopedTypeVariables
