--- highlighting-kate-0.3-orig/highlighting-kate.cabal	2011-11-27 13:08:38.000000000 +1100
+++ highlighting-kate-0.3/highlighting-kate.cabal	2011-11-27 14:56:52.672932074 +1100
@@ -41,6 +41,7 @@
                      xml/boo.xml
                      xml/c.xml
                      xml/changelog.xml
+                     xml/clojure.xml
                      xml/cmake.xml
                      xml/coffeescript.xml
                      xml/coldfusion.xml
@@ -166,6 +167,7 @@
                      Text.Highlighting.Kate.Syntax.Boo
                      Text.Highlighting.Kate.Syntax.C
                      Text.Highlighting.Kate.Syntax.Changelog
+                     Text.Highlighting.Kate.Syntax.Clojure
                      Text.Highlighting.Kate.Syntax.Cmake
                      Text.Highlighting.Kate.Syntax.Coffeescript
                      Text.Highlighting.Kate.Syntax.Coldfusion
