diff --git a/src/Data/RDF/PatriciaTreeGraph.hs b/src/Data/RDF/PatriciaTreeGraph.hs
index c127d4e..843d0f7 100644
--- a/src/Data/RDF/PatriciaTreeGraph.hs
+++ b/src/Data/RDF/PatriciaTreeGraph.hs
@@ -8,3 +8,3 @@ import Data.RDF.Types
 
-import Control.DeepSeq (NFData)
+import Control.DeepSeq (NFData(rnf))
 import qualified Data.Graph.Inductive.Graph as G
@@ -20,2 +20,8 @@ newtype PatriciaTreeGraph = PatriciaTreeGraph (PT.Gr Node Node,IntMap.IntMap Nod
 
+-- workaround until fgl releases
+--    7616e7135c9401e98f4c7350c9f60108ea73e456
+-- with proper instance
+instance NFData (PT.Gr Node Node)
+  where rnf x = seq x ()
+
 instance RDF PatriciaTreeGraph where
