diff --git a/Network/SimpleIRC/Core.hs b/Network/SimpleIRC/Core.hs
index 9242a23..13d2d2e 100644
--- a/Network/SimpleIRC/Core.hs
+++ b/Network/SimpleIRC/Core.hs
@@ -56,3 +56,3 @@ import System.Timeout
 import Data.Time
-import System.Locale
+import Data.Time.Locale.Compat
 import qualified Data.ByteString.Char8 as B
diff --git a/simpleirc.cabal b/simpleirc.cabal
index bbb0151..05744be 100644
--- a/simpleirc.cabal
+++ b/simpleirc.cabal
@@ -32,3 +32,4 @@ Library
     time >= 1.1.4,
-    old-locale >= 1.0.0.2
+    old-locale >= 1.0.0.2,
+    time-locale-compat
   Exposed-modules:
