--- a/src/Data/Iteratee/Base.hs
+++ b/src/Data/Iteratee/Base.hs
@@ -59,2 +59,3 @@ import qualified Control.Exception as E
 import Data.Data
+import qualified Data.Semigroup as S
 
@@ -81,2 +82,7 @@ instance (Eq c) => Eq (Stream c) where
 
+instance Monoid c => S.Semigroup (Stream c) where
+  (<>) (EOF mErr) _ = EOF mErr
+  (<>) _ (EOF mErr) = EOF mErr
+  (<>) (Chunk s1) (Chunk s2) = Chunk (s1 `mappend` s2)
+
 instance Monoid c => Monoid (Stream c) where
