--- HaXml-1.22.2-orig/src/Text/XML/HaXml/Schema/Schema.hs	2011-07-25 19:17:45.000000000 +1000
+++ HaXml-1.22.2/src/Text/XML/HaXml/Schema/Schema.hs	2011-07-27 20:35:42.846696468 +1000
@@ -47,7 +47,7 @@
 
 -- | A type t can extend another type s by the addition of extra elements
 --   and/or attributes.  s is therefore the supertype of t.
-class Extension t s {- | t -> s -} where  -- fundep ill-advised.
+class Extension t s where  -- fundep ill-advised.
     supertype :: t -> s
 
 -- | A type t can restrict another type s, that is, t admits fewer values
