--- HaXml-1.22.4-orig/src/Text/XML/HaXml/Schema/Schema.hs	2011-08-12 00:59:31.000000000 +1000
+++ HaXml-1.22.4/src/Text/XML/HaXml/Schema/Schema.hs	2011-08-12 12:25:39.605043566 +1000
@@ -47,7 +47,7 @@
 
 -- | A type t can extend another type s by the addition of extra elements
 --   and/or attributes.  s is therefore the supertype of t.
-class Extension t s {- | t -> s -} where  -- fundep ill-advised.
+class Extension t s {- - | t -> s -} where  -- fundep ill-advised.
     supertype :: t -> s
 
 -- | A type t can restrict another type s, that is, t admits fewer values
