--- HaXml-1.22.5-orig/HaXml.cabal	2011-08-13 22:46:39.000000000 +1000
+++ HaXml-1.22.5/HaXml.cabal	2011-08-14 10:05:17.796998165 +1000
@@ -78,49 +78,49 @@
   cpp-options: -DMAJOR=1 -DMINOR=22
   nhc98-options: -K10M
 
-Executable Canonicalise
+Executable Canonicalise-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: Canonicalise.hs
 
-Executable CanonicaliseLazy
+Executable CanonicaliseLazy-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: CanonicaliseLazy.hs
 
-Executable Xtract
+Executable Xtract-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: Xtract.hs
 
-Executable Validate
+Executable Validate-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: Validate.hs
 
-Executable MkOneOf
+Executable MkOneOf-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: MkOneOf.hs
 
-Executable DtdToHaskell
+Executable DtdToHaskell-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
   cpp-options: -DMAJOR=1 -DMINOR=22
   Main-Is: DtdToHaskell.hs
 
-Executable XsdToHaskell
+Executable XsdToHaskell-1.22
   GHC-Options: -Wall
   Extensions:  CPP
   Hs-Source-Dirs: src/tools, src
