diff --git a/src-25/Data/Lenses/Template.hs b/src-25/Data/Lenses/Template.hs
index 3ee2873..88ec56d 100644
--- a/src-25/Data/Lenses/Template.hs
+++ b/src-25/Data/Lenses/Template.hs
@@ -100,4 +100,9 @@ nameDeriveLenses t namer = do
     (params, cons) <- case reified of
+#if MIN_VERSION_template_haskell(2,11,0)
+                 DataD _ _ params _ cons' _ -> return (params, cons')
+                 NewtypeD _ _ params _ con' _ -> return (params, [con'])
+#else
                  DataD _ _ params cons' _ -> return (params, cons')
                  NewtypeD _ _ params con' _ -> return (params, [con'])
+#endif
                  _ -> fail errmsg
