--- hledger-web-0.18-orig/hledger-web.cabal	2012-05-30 04:35:45.000000000 +1000
+++ hledger-web-0.18/hledger-web.cabal	2012-05-30 22:38:28.820093468 +1000
@@ -141,7 +141,8 @@
                    , yesod-core
                    , yesod-default
                    , yesod-static
-                   , blaze-html                    < 0.5
+                   , blaze-html                    < 0.6
+                   , blaze-markup
                    , clientsession
                    , hamlet
                    , network-conduit
--- hledger-web-0.18-orig/Hledger/Web/Handlers.hs	2012-05-30 04:35:45.000000000 +1000
+++ hledger-web-0.18/Hledger/Web/Handlers.hs	2012-05-30 22:48:02.285123265 +1000
@@ -52,7 +52,8 @@
 import System.FilePath (takeFileName)
 import System.IO.Storage (putValue, getValue)
 import System.Locale (defaultTimeLocale)
-import Text.Blaze (preEscapedString, toHtml)
+import Text.Blaze.Html (toHtml)
+import Text.Blaze.Internal (preEscapedString)
 import Text.Hamlet hiding (hamlet)
 import Text.Printf
 import Yesod.Core
