--- a/src/Chart/Hud.hs
+++ b/src/Chart/Hud.hs
@@ -57,7 +57,7 @@ import Diagrams.Prelude
 import qualified Diagrams.TwoD.Size as D
 import Formatting
 import NumHask.Pair
-import NumHask.Prelude hiding (max)
+import NumHask.Prelude hiding (max, lines)
 import NumHask.Range
 import NumHask.Rect
 import NumHask.Space
--- a/src/Chart/Line.hs
+++ b/src/Chart/Line.hs
@@ -23,7 +23,7 @@ import Data.Default (Default(..))
 import Diagrams.Prelude hiding ((<>))
 import GHC.Generics
 import NumHask.Pair
-import NumHask.Prelude
+import NumHask.Prelude hiding (lines)
 import NumHask.Rect
 
 -- | The main features of a line (that distinguish it from a glyph say) is that:
