diff --git a/src/Main.hs b/src/Main.hs
index b9ba1a8..b972807 100644
--- a/src/Main.hs
+++ b/src/Main.hs
@@ -5,2 +5,3 @@ module Main where
 
+import Lambdabot.Config (ID(..))
 import Lambdabot.Main
@@ -19,3 +20,3 @@ import System.IO
 
-flags :: [OptDescr (IO (DSum Config))]
+flags :: [OptDescr (IO (DSum Config ID))]
 flags = 
@@ -29,4 +30,4 @@ flags =
     ] where 
-        arg :: String -> Config t -> (String -> IO t) -> ArgDescr (IO (DSum Config))
-        arg descr key fn = ReqArg (fmap (key :=>) . fn) descr
+        arg :: String -> Config t -> (String -> IO t) -> ArgDescr (IO (DSum Config ID))
+        arg descr key fn = ReqArg (fmap ((key :=>) . ID) . fn) descr
         
@@ -41,3 +42,3 @@ flags =
         
-        noinsult = return (enableInsults :=> False)
+        noinsult = return (enableInsults :=> ID False)
 
@@ -74,3 +75,3 @@ main = do
     exitWith <=< lambdabotMain modulesInfo $
-        [dataDir :=> dir, lbVersion :=> P.version] ++ config'
+        [dataDir :=> ID dir, lbVersion :=> ID P.version] ++ config'
 
@@ -81,2 +82,2 @@ online strs = do
     void $ lambdabotMain modulesInfo
-        [dataDir :=> dir, lbVersion :=> P.version, onStartupCmds :=> strs]
+        [dataDir :=> ID dir, lbVersion :=> ID P.version, onStartupCmds :=> ID strs]
