diff --git a/Plugin/Activity.hs b/Plugin/Activity.hs
index 5f6f6e6..e6bd4dc 100644
--- a/Plugin/Activity.hs
+++ b/Plugin/Activity.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Logging an IRC channel..
 module Plugin.Activity (theModule) where
 
diff --git a/Plugin/Base.hs b/Plugin/Base.hs
index 8bd8f86..71a8942 100644
--- a/Plugin/Base.hs
+++ b/Plugin/Base.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, PatternGuards, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Lambdabot base module. Controls message send and receive
 module Plugin.Base (theModule) where
 
diff --git a/Plugin/Djinn.hs b/Plugin/Djinn.hs
index 5f10ea1..67de282 100644
--- a/Plugin/Djinn.hs
+++ b/Plugin/Djinn.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, PatternGuards, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- Copyright (c) 2005 Donald Bruce Stewart - http://www.cse.unsw.edu.au/~dons
 -- GPL version 2 or later (see http://www.gnu.org/copyleft/gpl.html)
 
diff --git a/Plugin/Fact.hs b/Plugin/Fact.hs
index 1ff711c..dd62492 100644
--- a/Plugin/Fact.hs
+++ b/Plugin/Fact.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- Module    : Fact
 -- Copyright : 2003 Shae Erisson
 -- Copyright : 2005-06 Don Stewart
diff --git a/Plugin/Haddock.hs b/Plugin/Haddock.hs
index 2e1ddde..eb5689e 100644
--- a/Plugin/Haddock.hs
+++ b/Plugin/Haddock.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Hackish Haddock module.
 module Plugin.Haddock (theModule) where
 
diff --git a/Plugin/Hoogle.hs b/Plugin/Hoogle.hs
index 84bc261..1e89e87 100644
--- a/Plugin/Hoogle.hs
+++ b/Plugin/Hoogle.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 -- Copyright (c) 2004-5 Don Stewart - http://www.cse.unsw.edu.au/~dons
 -- GPL version 2 or later (see http://www.gnu.org/copyleft/gpl.html)
diff --git a/Plugin/Karma.hs b/Plugin/Karma.hs
index a2c841d..37a852c 100644
--- a/Plugin/Karma.hs
+++ b/Plugin/Karma.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Karma
 module Plugin.Karma (theModule) where
 
diff --git a/Plugin/Localtime.hs b/Plugin/Localtime.hs
index 872f160..f822142 100644
--- a/Plugin/Localtime.hs
+++ b/Plugin/Localtime.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- Copyright (c) 2005 Don Stewart - http://www.cse.unsw.edu.au/~dons
 -- GPL version 2 or later (see http://www.gnu.org/copyleft/gpl.html)
 
diff --git a/Plugin/More.hs b/Plugin/More.hs
index ced0e63..858270f 100644
--- a/Plugin/More.hs
+++ b/Plugin/More.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Support for more(1) buffering
 module Plugin.More (theModule) where
 
diff --git a/Plugin/Pl.hs b/Plugin/Pl.hs
index 6d1f6fa..d3b5e9f 100644
--- a/Plugin/Pl.hs
+++ b/Plugin/Pl.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 -- | Pointfree programming fun
 --
diff --git a/Plugin/Poll.hs b/Plugin/Poll.hs
index c67bd87..484bf9b 100644
--- a/Plugin/Poll.hs
+++ b/Plugin/Poll.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Module: Vote
 -- | Support for voting
 -- |
diff --git a/Plugin/Quote.hs b/Plugin/Quote.hs
index 12345f4..c1a76f4 100644
--- a/Plugin/Quote.hs
+++ b/Plugin/Quote.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, CPP, MultiParamTypeClasses, PatternGuards, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Support for quotes
 module Plugin.Quote (theModule) where
 
diff --git a/Plugin/Source.hs b/Plugin/Source.hs
index 88d7db4..94bb0e8 100644
--- a/Plugin/Source.hs
+++ b/Plugin/Source.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- Plugin.Source
 -- Display source for specified identifiers
 module Plugin.Source (theModule) where
diff --git a/Plugin/State.hs b/Plugin/State.hs
index fded7b8..fbe7b06 100644
--- a/Plugin/State.hs
+++ b/Plugin/State.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | Persistent state
 -- A demo plugin
 module Plugin.State (theModule) where
diff --git a/Plugin/Tell.hs b/Plugin/Tell.hs
index a60f9ed..19fc4a2 100644
--- a/Plugin/Tell.hs
+++ b/Plugin/Tell.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 {- Leave a message with lambdabot, the faithful secretary
 
 > 17:11 < davidhouse> @tell dmhouse foo
diff --git a/Plugin/Todo.hs b/Plugin/Todo.hs
index 86ec3c5..d30f473 100644
--- a/Plugin/Todo.hs
+++ b/Plugin/Todo.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, PatternGuards, ScopedTypeVariables, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -- | A todo list
 --
 -- (c) 2005 Samuel Bronson
diff --git a/Plugin/Where.hs b/Plugin/Where.hs
index 157bc61..c06416c 100644
--- a/Plugin/Where.hs
+++ b/Plugin/Where.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TemplateHaskell, MultiParamTypeClasses, TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 -- |
 -- Module    : Where
