--- ghc-mod-1.0.4-orig/Info.hs	2011-12-02 15:10:05.000000000 +1100
+++ ghc-mod-1.0.4/Info.hs	2011-12-21 16:08:49.211987913 +1100
@@ -108,7 +108,12 @@
 setContextFromTarget = do
     ms <- depanal [] False
     top <- map ms_mod <$> filterM isTop ms
+#if __GLASGOW_HASKELL__ >= 704
+    let itop = map IIModule top
+    setContext itop
+#else
     setContext top []
+#endif
     return (top /= [])
   where
     isTop ms = lookupMod `gcatch` returnFalse
