diff --git a/aclocal.m4 b/aclocal.m4
index 4678774..7f947f0 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -212,7 +212,7 @@ fi
 changequote([, ])dnl
 ])
 FP_COMPARE_VERSIONS([$fptools_cv_greencard_version],[-lt],[$1],
-  [AC_MSG_ERROR([greencard version $1 or later is required (found '$fptools_cv_greencard_version')])])[]dnl
+  [AC_MSG_ERROR([greencard version $1 or later is required (found '$fptools_cv_greencard_version')])])[]
 GreenCardVersion=$fptools_cv_greencard_version
 AC_SUBST(GreenCardVersion)
 ])
@@ -247,7 +247,7 @@ changequote([, ])dnl
 if test ! -f compiler/parser/Parser.hs || test ! -f compiler/main/ParsePkgConf.hs || test ! -f compiler/cmm/CmmParse.hs || test ! -f compiler/parser/ParserCore.hs
 then
     FP_COMPARE_VERSIONS([$fptools_cv_happy_version],[-lt],[1.15],
-      [AC_MSG_ERROR([Happy version 1.15 or later is required to compile GHC.])])[]dnl
+      [AC_MSG_ERROR([Happy version 1.15 or later is required to compile GHC.])])[]
 fi
 HappyVersion=$fptools_cv_happy_version;
 AC_SUBST(HappyVersion)
@@ -282,7 +282,7 @@ changequote([, ])dnl
 if test ! -f compiler/cmm/CmmLex.hs || test ! -f compiler/parser/Lexer.hs
 then
     FP_COMPARE_VERSIONS([$fptools_cv_alex_version],[-lt],[2.1.0],
-      [AC_MSG_ERROR([Alex version 2.1.0 or later is required to compile GHC.])])[]dnl
+      [AC_MSG_ERROR([Alex version 2.1.0 or later is required to compile GHC.])])[]
 fi
 AlexVersion=$fptools_cv_alex_version;
 AC_SUBST(AlexVersion)
diff --git a/configure.ac b/configure.ac
index 4d39e74..e3573c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -795,7 +795,7 @@ if test "$BootingFromHc" = "NO" -a -d "$srcdir/compiler"; then
      AC_MSG_ERROR([GHC is required unless bootstrapping from .hc files.])
   fi
   FP_COMPARE_VERSIONS([$GhcVersion],[-lt],[6.4],
-    [AC_MSG_ERROR([GHC version 6.4 or later is required to compile GHC.])])dnl
+    [AC_MSG_ERROR([GHC version 6.4 or later is required to compile GHC.])])
 fi;
 
 # This uses GHC, so put it after the "GHC is required" check above:
