--- ghc-7.7.20121101-orig/rules/build-package.mk	2012-11-02 05:00:11.000000000 +1100
+++ ghc-7.7.20121101/rules/build-package.mk	2012-11-03 00:40:20.787547845 +1100
@@ -104,6 +104,16 @@
 endif
 endif
 
+ifneq "$3" "0"
+ifeq "$$(BuildSharedLibs)" "YES"
+ifeq "$$(TargetOS_CPP)" "linux"
+$1_$2_GHC_LD_OPTS = \
+    -fno-use-rpaths \
+    $$(foreach d,$$($1_$2_TRANSITIVE_DEPS),-optl-Wl$$(comma)-rpath -optl-Wl$$(comma)'$$$$ORIGIN/../$$d')
+endif
+endif
+endif
+
 $(call hs-sources,$1,$2)
 $(call c-sources,$1,$2)
 $(call includes-sources,$1,$2)
--- ghc-7.7.20121101-orig/rules/build-package-way.mk	2012-11-02 05:00:11.000000000 +1100
+++ ghc-7.7.20121101/rules/build-package-way.mk	2012-11-03 00:34:29.800589305 +1100
@@ -70,7 +70,7 @@
 ifeq "$$(HostOS_CPP)" "mingw32"
 $$($1_$2_$3_LIB) : $$($1_$2_$3_ALL_OBJS) $$(ALL_RTS_LIBS) $$($1_$2_$3_DEPS_LIBS)
 	$$(call cmd,$1_$2_HC) $$($1_$2_$3_ALL_HC_OPTS) $$($1_$2_$3_ALL_OBJS) \
-         -shared -dynamic -dynload deploy \
+         -shared -dynamic -dynload deploy $$($1_$2_GHC_LD_OPTS) \
 	 $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES)) \
          -no-auto-link-packages \
          -o $$@
@@ -80,7 +80,7 @@
 else
 $$($1_$2_$3_LIB) : $$($1_$2_$3_ALL_OBJS) $$(ALL_RTS_LIBS) $$($1_$2_$3_DEPS_LIBS)
 	$$(call cmd,$1_$2_HC) $$($1_$2_$3_ALL_HC_OPTS) $$($1_$2_$3_ALL_OBJS) \
-         -shared -dynamic -dynload deploy \
+         -shared -dynamic -dynload deploy $$($1_$2_GHC_LD_OPTS) \
 	 $$(addprefix -l,$$($1_$2_EXTRA_LIBRARIES)) $$(addprefix -L,$$($1_$2_EXTRA_LIBDIRS)) \
          -no-auto-link-packages \
          -o $$@
