diff --git a/libraries/base/System/Environment/ExecutablePath.hsc b/libraries/base/System/Environment/ExecutablePath.hsc
index 448cade49a..db4a1fb8ed 100644
--- a/libraries/base/System/Environment/ExecutablePath.hsc
+++ b/libraries/base/System/Environment/ExecutablePath.hsc
@@ -169,7 +169,7 @@ getExecutablePath = go 2048  -- plenty, PATH_MAX is 512 under Win32
 getFinalPath :: FilePath -> IO FilePath
 getFinalPath path = withCWString path $ \s ->
   bracket (createFile s) c_closeHandle $ \h -> do
-    let invalid = h == wordPtrToPtr (#const (intptr_t)INVALID_HANDLE_VALUE)
+    let invalid = h == wordPtrToPtr (#const (intptr_t)(-1))
     if invalid then pure path else go h bufSize
 
   where go h sz = allocaArray (fromIntegral sz) $ \outPath -> do
