gentoo's eclasses don't contain reference to 'origin' remote.
diff --git a/boot b/boot
index b6ecc09f28..9cb22ea5cb 100755
--- a/boot
+++ b/boot
@@ -24,6 +24,7 @@ def die(mesg):
     sys.exit(1)
 
 def check_for_url_rewrites():
+  try:
     if os.path.isdir('.git') and \
        subprocess.check_output('git config remote.origin.url'.split()).find(b'github.com') != -1 and \
        subprocess.call(['git', 'config', '--get-regexp', '^url.*github.com/.*/packages-.insteadOf']) != 0:
@@ -56,6 +57,8 @@ def check_for_url_rewrites():
               * https://ghc.haskell.org/trac/ghc/wiki/Newcomers or
               * https://ghc.haskell.org/trac/ghc/wiki/Building/GettingTheSources#CloningfromGitHub
         """)
+  except Exception as e:
+    print_err('check_for_url_rewrites() failed as: %s. Ignoring.' % e)
 
 def check_boot_packages():
     # Check that we have all boot packages.
