--- a/rts/RtsUtils.c
+++ b/rts/RtsUtils.c
@@ -152,6 +152,7 @@ reportHeapOverflow(void)
                             (W_)RtsFlags.GcFlags.maxHeapSize * BLOCK_SIZE);
 }
 
+#ifndef _WIN32
 /* -----------------------------------------------------------------------------
    Sleep for the given period of time.
    -------------------------------------------------------------------------- */
@@ -169,6 +170,7 @@ int rtsSleep(Time t)
     } while (ret == -1 && errno == EINTR);
     return ret;
 }
+#endif
 
 /* -----------------------------------------------------------------------------
    Get the current time as a string.  Used in profiling reports.
