commit 243c304273a1254418aeacf4d81cadb2d59945ad
Author: Erik de Castro Lopo <erikd@mega-nerd.com>
Date:   Mon Apr 4 22:40:15 2011 +1000

    Fix compiling on linux-powerpc.

diff --git a/rts/Linker.c b/rts/Linker.c
index 5285ec6..c840857 100644
--- a/rts/Linker.c
+++ b/rts/Linker.c
@@ -2572,7 +2572,11 @@ static void
 ocFlushInstructionCache( ObjectCode *oc )
 {
     /* The main object code */
-    ocFlushInstructionCacheFrom(oc->image + oc->misalignment, oc->fileSize);
+    ocFlushInstructionCacheFrom(oc->image
+#ifdef darwin_HOST_OS
+            + oc->misalignment
+#endif
+            , oc->fileSize);
 
     /* Jump Islands */
     ocFlushInstructionCacheFrom(oc->symbol_extras, sizeof(SymbolExtra) * oc->n_symbol_extras);
