--- ghc-6.4.2/ghc/rts/Linker.c	2006-01-04 15:24:10.000000000 +0000
+++ ghc-6.4.2.new/ghc/rts/Linker.c	2006-05-15 21:06:45.000000000 +0100
@@ -2695,6 +2695,9 @@
    IF_DEBUG(linker,debugBelch( "Architecture is " ));
    switch (ehdr->e_machine) {
       case EM_386:   IF_DEBUG(linker,debugBelch( "x86" )); break;
+#ifdef EM_SPARC32PLUS
+      case EM_SPARC32PLUS:
+#endif
       case EM_SPARC: IF_DEBUG(linker,debugBelch( "sparc" )); break;
 #ifdef EM_IA_64
       case EM_IA_64: IF_DEBUG(linker,debugBelch( "ia64" )); break;
