--- ghc-7.10.0.20141222-orig/libraries/gen_contents_index	2014-12-23 02:55:30.000000000 +1100
+++ ghc-7.10.0.20141222/libraries/gen_contents_index	2015-01-25 13:46:59.726679969 +1100
@@ -34,7 +34,7 @@
 
     HADDOCK=../../inplace/bin/haddock
     HADDOCK_ARGS="-p ../prologue.txt"
-    for REPO in `grep '^libraries/[^ ]*  *- ' ../../packages | sed -e 's#libraries/##' -e 's/ .*//'`
+    for REPO in `grep '^libraries/[^ ]*  *- ' ../../packages | sed -e 's#libraries/##' -e 's/ .*//'` base integer-gmp integer-gmp2 integer-simple template-haskell
     do
         if [ -f "../$REPO/ghc-packages" ]
         then
@@ -54,7 +54,7 @@
                 # It's easier to portably remove tabs with tr than to try to get
                 # sed to do what we want
                 VERSION=`grep -i '^version:' $LIBPATH/$NAME.cabal | sed 's/.*://' | tr -d ' \t'`
-                HADDOCK_ARG="--read-interface=${NAME}-${VERSION},$HADDOCK_FILE"
+                HADDOCK_ARG="--package-name=${NAME} --package-version=${VERSION} --read-interface=${NAME}-${VERSION},$HADDOCK_FILE"
                 HADDOCK_ARGS="$HADDOCK_ARGS $HADDOCK_ARG"
             fi
         done
