diff --git a/aclocal.m4 b/aclocal.m4
index 697cba5690..7ad9c36453 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -479,8 +479,18 @@ AC_DEFUN([FP_SETTINGS],
         SettingsLdCommand="$LdCmd"
         SettingsArCommand="$ArCmd"
         SettingsPerlCommand="$PerlCmd"
-        SettingsDllWrapCommand="/bin/false"
-        SettingsWindresCommand="/bin/false"
+        if test -z "$DllWrap"
+        then
+            SettingsDllWrapCommand="/bin/false"
+        else
+            SettingsDllWrapCommand="$DllWrap"
+        fi
+        if test -z "$Windres"
+        then
+            SettingsWindresCommand="/bin/false"
+        else
+            SettingsWindresCommand="$Windres"
+        fi
         SettingsLibtoolCommand="libtool"
         SettingsTouchCommand='touch'
     fi
