diff --git a/libraries/ghc-prim/cbits/atomic.c b/libraries/ghc-prim/cbits/atomic.c
index b238041feb..a8f9b164ba 100644
--- a/libraries/ghc-prim/cbits/atomic.c
+++ b/libraries/ghc-prim/cbits/atomic.c
@@ -317,2 +317,3 @@ hs_atomicread32(StgWord x)
 
+#if WORD_SIZE_IN_BITS == 64
 extern StgWord64 hs_atomicread64(StgWord x);
@@ -327,2 +328,3 @@ hs_atomicread64(StgWord x)
 }
+#endif
 
@@ -365,2 +367,3 @@ hs_atomicwrite32(StgWord x, StgWord val)
 
+#if WORD_SIZE_IN_BITS == 64
 extern void hs_atomicwrite64(StgWord x, StgWord64 val);
@@ -375 +378,2 @@ hs_atomicwrite64(StgWord x, StgWord64 val)
 }
+#endif
