diff --git a/libraries/network/Network/Socket.hsc b/libraries/network/Network/Socket.hsc
index ae35942..6c6d06a 100644
--- a/libraries/network/Network/Socket.hsc
+++ b/libraries/network/Network/Socket.hsc
@@ -69,10 +69,12 @@ module Network.Socket (
     getPeerName,	-- :: Socket -> IO SockAddr
     getSocketName,	-- :: Socket -> IO SockAddr
 
+#if 0
 #ifdef SO_PEERCRED
 	-- get the credentials of our domain socket peer.
     getPeerCred,         -- :: Socket -> IO (CUInt{-pid-}, CUInt{-uid-}, CUInt{-gid-})
 #endif
+#endif /* 0 */
 
     socketPort,		-- :: Socket -> IO PortNumber
 
@@ -985,7 +987,7 @@ getSocketOption (MkSocket s _ _ _ _) so = do
 	 c_getsockopt s (socketOptLevel so) (packSocketOption so) ptr_v ptr_sz
        fromIntegral `liftM` peek ptr_v
 
-
+#if 0
 #ifdef SO_PEERCRED
 -- | Returns the processID, userID and groupID of the socket's peer.
 --
@@ -1004,6 +1006,7 @@ getPeerCred sock = do
      gid <- (#peek struct ucred, gid) ptr_cr
      return (pid, uid, gid)
 #endif
+#endif /* 0 */
 
 #if defined(DOMAIN_SOCKET_SUPPORT)
 -- sending/receiving ancillary socket data; low-level mechanism
