https://gcc.gnu.org/PR81426
--- a/mk/config.mk.in
+++ b/mk/config.mk.in
@@ -928,3 +928,11 @@ CONF_CC_OPTS_STAGE2 += -mno-gpopt
 CONF_HC_OPTS_STAGE1 += -optc-mno-gpopt
 CONF_HC_OPTS_STAGE2 += -optc-mno-gpopt
 endif
+
+# workaround gcc bug: https://gcc.gnu.org/PR81426
+ifeq "$(TargetArch_CPP)" "sh4"
+CONF_CC_OPTS_STAGE1 += -mlra
+CONF_CC_OPTS_STAGE2 += -mlra
+CONF_HC_OPTS_STAGE1 += -optc-mlra
+CONF_HC_OPTS_STAGE2 += -optc-mlra
+endif
