diff -rN -u old-ghc-1/compiler/main/DriverPipeline.hs new-ghc/compiler/main/DriverPipeline.hs
--- old-ghc-1/compiler/main/DriverPipeline.hs	2006-09-13 10:33:23.000000000 +0200
+++ new-ghc/compiler/main/DriverPipeline.hs	2006-09-13 10:33:23.000000000 +0200
@@ -1014,6 +1014,10 @@
 	let ld_r args = SysTools.runLink dflags ([ 
 				SysTools.Option "-nostdlib",
 				SysTools.Option "-nodefaultlibs",
+#ifdef sparc_TARGET_ARCH
+				-- options '--relax' and '-r' are incompatible
+				SysTools.Option "-mno-relax",
+#endif
 				SysTools.Option "-Wl,-r", 
 				SysTools.Option ld_x_flag, 
 				SysTools.Option "-o", 

