diff --git a/rts/linker/PEi386.c b/rts/linker/PEi386.c
index 3dcf8c4281..364f7780b2 100644
--- a/rts/linker/PEi386.c
+++ b/rts/linker/PEi386.c
@@ -159 +159 @@ static uint8_t* cstring_from_COFF_symbol_name(
-#include <Psapi.h>
+#include <psapi.h>
diff --git a/rts/package.conf.in b/rts/package.conf.in
index 52d7ef8be2..1746af59e2 100644
--- a/rts/package.conf.in
+++ b/rts/package.conf.in
@@ -48,2 +48,2 @@ extra-libraries:
-                              ,"Dbghelp"    /* for crash dump */
-                              ,"Psapi"      /* for process information.  */
+                              ,"dbghelp"    /* for crash dump */
+                              ,"psapi"      /* for process information.  */
diff --git a/rts/rts.cabal.in b/rts/rts.cabal.in
index 71aef3df0e..53b6271b10 100644
--- a/rts/rts.cabal.in
+++ b/rts/rts.cabal.in
@@ -61 +61 @@ library
-          Dbghelp
+          dbghelp
@@ -63 +63 @@ library
-          Psapi
+          psapi
diff --git a/rts/win32/veh_excn.c b/rts/win32/veh_excn.c
index fd50562448..4b7d29a1a1 100644
--- a/rts/win32/veh_excn.c
+++ b/rts/win32/veh_excn.c
@@ -23 +23 @@
-#include <Dbghelp.h>
+#include <dbghelp.h>
