--- ghc-6.6/compiler/Makefile	2006-10-10 14:03:51.000000000 -0500
+++ ghc-6.6.patched/compiler/Makefile	2007-04-04 11:57:06.000000000 -0500
@@ -415,7 +415,9 @@
 PKG_DEPENDS += template-haskell
 
 # Use threaded RTS with GHCi, so threads don't get blocked at the prompt.
+ifneq "$(GhcNotThreaded)" "YES"
 SRC_HC_OPTS += -threaded
+endif
 
 ALL_DIRS += ghci
 
--- ghc-6.6/compiler/Makefile.ghcbin	2006-10-10 14:03:52.000000000 -0500
+++ ghc-6.6.patched/compiler/Makefile.ghcbin	2007-04-04 11:57:30.000000000 -0500
@@ -26,7 +26,9 @@
 SRC_HC_OPTS += -DGHCI -DBREAKPOINT
 endif
 
+ifneq "$(GhcNotThreaded)" "YES"
 SRC_HC_OPTS += -threaded
+endif
 
 odir=stage$(stage)
 
