--- a/llvm-targets
+++ b/llvm-targets
@@ -3,1 +3,2 @@
 ,("armv6-unknown-linux-gnueabihf", ("e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64", "arm1136jf-s", "+strict-align"))
+,("armv6j-unknown-linux-gnueabihf", ("e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64", "arm1176jzf-s", "+strict-align"))
--- a/utils/llvm-targets/gen-data-layout.sh
+++ b/utils/llvm-targets/gen-data-layout.sh
@@ -34,1 +34,2 @@ TARGETS=(
     "armv6-unknown-linux-gnueabihf"
+    "armv6j-unknown-linux-gnueabihf"
