--- old-ghc-6.12.0.20091010/configure.ac	2009-11-07 14:11:31.747192438 +0100
+++ new-ghc-6.12.0.20091010/configure.ac	2009-11-07 14:11:31.752194402 +0100
@@ -219,6 +219,8 @@
         exit 1
     fi
 fi
+build=`echo "$build" | sed -e 's/linux-gnu/linux/' -e 's/-pc-/-unknown-/'`
+[build=`echo "$build" | sed -e 's/i[4-6]86/i386/'`] # escaping square bracket
 
 if test "$host" = ""
 then
@@ -231,6 +233,8 @@
         exit 1
     fi
 fi
+host=`echo "$host" | sed -e 's/linux-gnu/linux/' -e 's/-pc-/-unknown-/'`
+[host=`echo "$host" | sed -e 's/i[4-6]86/i386/'`] # escaping square bracket
 
 if test "$target" = ""
 then
@@ -243,6 +247,8 @@
         exit 1
     fi
 fi
+target=`echo "$target" | sed -e 's/linux-gnu/linux/' -e 's/-pc-/-unknown-/'`
+[target=`echo "$target" | sed -e 's/i[4-6]86/i386/'`] # escaping square bracket
 
 exeext=''
 soext='.so'
