diff --git a/libraries/base/GHC/IOBase.lhs b/libraries/base/GHC/IOBase.lhs
index 112268e..deee8f5 100644
--- a/libraries/base/GHC/IOBase.lhs
+++ b/libraries/base/GHC/IOBase.lhs
@@ -53,6 +53,7 @@ import Data.Maybe  ( Maybe(..) )
 import GHC.Show
 import GHC.List
 import GHC.Read
+import GHC.Int (Int32) -- for FD (ideally we need CInt)
 
 #ifndef __HADDOCK__
 import {-# SOURCE #-} Data.Dynamic
@@ -323,7 +324,7 @@ instance Eq Handle where
  (DuplexHandle _ h1 _) == (DuplexHandle _ h2 _) = h1 == h2
  _ == _ = False 
 
-type FD = Int -- XXX ToDo: should be CInt
+type FD = Int32 -- XXX ToDo: should be CInt (causes circular deps)
 
 data Handle__
   = Handle__ {
