--- a/compiler/ghc.mk
+++ b/compiler/ghc.mk
@@ -382,6 +382,13 @@ compiler/stage1/build/Parser_HC_OPTS += -O0 -fno-ignore-interface-pragmas -fcmm-
 compiler/stage2/build/Parser_HC_OPTS += -O0 -fno-ignore-interface-pragmas -fcmm-sink
 compiler/stage3/build/Parser_HC_OPTS += -O0 -fno-ignore-interface-pragmas -fcmm-sink
 
+# HsInstances overflows GOT at least on alpha
+ifeq "$(GhcUnregisterised)" "NO"
+else
+compiler/hsSyn/HsInstances_HC_OPTS   += -O0
+compiler/GHC/Hs/Instances_HC_OPTS    += -O0
+endif
+
 ifeq "$(GhcProfiled)" "YES"
 # If we're profiling GHC then we want SCCs.  However, adding -auto-all
 # everywhere tends to give a hard-to-read profile, and adds lots of
