--- a/libraries/integer-gmp/gmp/ghc.mk
+++ b/libraries/integer-gmp/gmp/ghc.mk
@@ -126,6 +126,12 @@ libraries/integer-gmp/gmp/libgmp.a libraries/integer-gmp/gmp/gmp.h:
 	cd libraries/integer-gmp/gmp && $(PATCH_CMD) -p0 < gmpsrc.patch
 	chmod +x libraries/integer-gmp/gmp/ln
 
+	# Gentoo hack: change ABI to GMPABI variable to avoid conflict with gentoo's variable
+	$(MV) libraries/integer-gmp/gmp/gmpbuild/configure libraries/integer-gmp/gmp/gmpbuild/configure.wrapped
+	echo '#!/usr/bin/env sh' > libraries/integer-gmp/gmp/gmpbuild/configure
+	echo 'exec env ABI="${GMPABI}" "$$0.wrapped" "$$@"' >> libraries/integer-gmp/gmp/gmpbuild/configure
+	chmod a+rx libraries/integer-gmp/gmp/gmpbuild/configure libraries/integer-gmp/gmp/gmpbuild/configure.wrapped
+
 	# Note: We must pass `TARGETPLATFORM` to the `--host` argument of GMP's
 	#       `./configure`, not `HOSTPLATFORM`: the 'host' on which GMP will
 	#       run is the 'target' platform of the compiler we're building.
