--- uu-interleaved-0.1.0.0-orig/src/Control/Applicative/Interleaved.hs	2013-05-14 23:43:46.000000000 +1000
+++ uu-interleaved-0.1.0.0/src/Control/Applicative/Interleaved.hs	2015-01-30 19:36:47.654872721 +1100
@@ -1,6 +1,7 @@
 {-# LANGUAGE ExistentialQuantification,
              ScopedTypeVariables,
-             FlexibleInstances #-}
+             FlexibleInstances,
+             CPP #-}
 
 -- | This module contains the additional data types, instance definitions and functions to run parsers in an interleaved way.
 --   If all the interleaved parsers recognise a single connected piece of the input text this incorporates the permutation parsers.
@@ -26,8 +27,11 @@
 
 -- import Text.ParserCombinators.UU.Core
 import Control.Applicative
+#if MIN_VERSION_base(4,8,0)
+import Data.Monoid hiding (Alt)
+#else
 import Data.Monoid
-
+#endif
 
 infixl 4  <||>
 infixl 4  <<||> 
