--- a/src/Lambdabot/Command.hs
+++ b/src/Lambdabot/Command.hs
@@ -33,6 +33,7 @@ import Control.Monad.Identity
 import Control.Monad.Reader
 import Control.Monad.Trans.Control
 import Control.Monad.Writer
+import qualified Control.Monad.Fail as CMF
 
 data CmdArgs = forall a. Msg.Message a => CmdArgs
     { _message  :: a
@@ -50,6 +51,9 @@ instance Monad m => Monad (Cmd m) where
     return = Cmd . return
     Cmd x >>= f = Cmd (x >>= (unCmd . f))
     fail = lift . fail
+instance Monad m => CMF.MonadFail (Cmd m) where
+    fail = lift . fail
+
 instance MonadIO m => MonadIO (Cmd m) where
     liftIO = lift . liftIO
 instance MonadBase b m => MonadBase b (Cmd m) where
