commit 0f3f8d2190909ebcae1cc20551370fbbf2a922de
Author: Ian Lynagh <igloo@earth.li>
Date:   Sun Jun 24 14:28:08 2012 +0100

    Fix build with GHC HEAD

diff --git a/Control/Sequential/STM.hs b/Control/Sequential/STM.hs
index 328132c..468e7bd 100644
--- a/Control/Sequential/STM.hs
+++ b/Control/Sequential/STM.hs
@@ -14,7 +14,9 @@ module Control.Sequential.STM (
 	TVar, newTVar, newTVarIO, readTVar, readTVarIO, writeTVar
     ) where
 
+#if __GLASGOW_HASKELL__ < 705
 import Prelude hiding (catch)
+#endif
 import Control.Applicative (Applicative(pure, (<*>)))
 import Control.Exception
 import Data.IORef

