diff -rN -u old-darcs-2.5.2/darcs.cabal new-darcs-2.5.2/darcs.cabal
--- old-darcs-2.5.2/darcs.cabal	2011-06-03 20:24:32.079482374 +0200
+++ new-darcs-2.5.2/darcs.cabal	2011-06-03 20:24:32.080482374 +0200
@@ -166,11 +166,11 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && < 0.96
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && < 0.96,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -368,14 +368,14 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && < 0.96
 
     if os(solaris)
       cc-options:     -DHAVE_SIGINFO_H
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && < 0.96,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -516,14 +516,14 @@
     cpp-options:    -DWIN32
     c-sources:      src/win32/send_email.c
     build-depends:  unix-compat >= 0.1.2,
-                    regex-posix >= 0.94.4 && < 0.95
+                    regex-posix >= 0.94.4 && < 0.96
 
   if os(solaris)
     cc-options:     -DHAVE_SIGINFO_H
 
   build-depends:   base          < 5,
                    extensible-exceptions >= 0.1 && < 0.2,
-                   regex-compat >= 0.71 && < 0.94,
+                   regex-compat >= 0.71 && < 0.96,
                    mtl          >= 1.0 && < 2.1,
                    parsec       >= 2.0 && < 3.2,
                    html         == 1.0.*,
@@ -619,7 +619,7 @@
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
-                     regex-compat >= 0.71 && < 0.94,
+                     regex-compat >= 0.71 && < 0.96,
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
@@ -672,7 +672,7 @@
       cpp-options:    -DWIN32
       c-sources:      src/win32/send_email.c
       build-depends:  unix-compat >= 0.1.2,
-                      regex-posix >= 0.94.4 && < 0.95
+                      regex-posix >= 0.94.4 && < 0.96
 
     if os(solaris)
       cc-options:     -DHAVE_SIGINFO_H
