diff --git a/Distribution/ShellHarness.hs b/Distribution/ShellHarness.hs
index 78a8a9b..949b752 100644
--- a/Distribution/ShellHarness.hs
+++ b/Distribution/ShellHarness.hs
@@ -12,10 +12,9 @@ import System.Environment ( getEnv, getEnvironment )
 import System.Exit ( ExitCode (..) )
 import System.FilePath
 import System.IO
-import System( system )
 import System.Process ( ProcessHandle,
                         runInteractiveProcess, waitForProcess,
-                        getProcessExitCode )
+                        getProcessExitCode, system )
 import Data.Maybe
 import Data.List ( isInfixOf, isPrefixOf, (\\), nubBy, isSuffixOf )
 import Control.Concurrent
diff --git a/darcs.cabal b/darcs.cabal
index 887a27b..56e36fa 100644
--- a/darcs.cabal
+++ b/darcs.cabal
@@ -124,7 +124,7 @@ Executable          witnesses
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.2
+    build-tools: ghc >= 6.10 && < 7.4
 
     hs-source-dirs:   src
     include-dirs:     src
@@ -182,7 +182,7 @@ Executable          witnesses
                      text == 0.11.*,
                      old-time   == 1.0.*,
                      directory  >= 1.0.0.0 && < 1.2.0.0,
-                     process    == 1.0.*,
+                     process    >= 1.0 && < 1.2,
                      containers >= 0.1 && < 0.5,
                      array      >= 0.1 && < 0.4,
                      random     == 1.0.*,
@@ -192,7 +192,7 @@ Executable          witnesses
                      test-framework             >= 0.2.2,
                      test-framework-quickcheck2 >= 0.2.2
     if !os(windows)
-      build-depends: unix >= 1.0 && < 2.5
+      build-depends: unix >= 1.0 && < 2.6
     if flag(http)
         build-depends:    network >= 2.2 && < 2.4,
                           HTTP    >= 3000.0 && < 4000.2
@@ -208,7 +208,7 @@ Library
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.2
+    build-tools: ghc >= 6.10 && < 7.4
 
     hs-source-dirs:   src
     include-dirs:     src
@@ -385,14 +385,14 @@ Library
                      tar          == 0.3.*
 
     if !os(windows)
-      build-depends: unix >= 1.0 && < 2.5
+      build-depends: unix >= 1.0 && < 2.6
 
     build-depends: base >= 3,
                    bytestring >= 0.9.0 && < 0.10,
                    text == 0.11.*,
                    old-time   == 1.0.*,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
-                   process    == 1.0.*,
+                   process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
                    array      >= 0.1 && < 0.4,
                    random     == 1.0.*
@@ -483,7 +483,7 @@ Executable          darcs
                     src/umask.c
                     src/Crypt/sha2.c
 
-  build-tools: ghc >= 6.10 && < 7.2
+  build-tools: ghc >= 6.10 && < 7.4
 
   -- We need optimizations, regardless of what Hackage says
   -- Note: "if true" works around a cabal bug with order of flag composition
@@ -533,14 +533,14 @@ Executable          darcs
                    tar          == 0.3.*
 
   if !os(windows)
-    build-depends: unix >= 1.0 && < 2.5
+    build-depends: unix >= 1.0 && < 2.6
 
   build-depends: base >= 3,
                  bytestring >= 0.9.0 && < 0.10,
                  text == 0.11.*,
                    old-time   == 1.0.*,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
-                   process    == 1.0.*,
+                   process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
                    array      >= 0.1 && < 0.4,
                    random     == 1.0.*
@@ -615,7 +615,7 @@ Executable          unit
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.2
+    build-tools: ghc >= 6.10 && < 7.4
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
@@ -678,7 +678,7 @@ Executable          unit
       cc-options:     -DHAVE_SIGINFO_H
 
     if !os(windows)
-      build-depends: unix >= 1.0 && < 2.5
+      build-depends: unix >= 1.0 && < 2.6
 
     build-depends: base >= 3,
                    bytestring >= 0.9.0 && < 0.10,
@@ -686,7 +686,7 @@ Executable          unit
                    text == 0.11.*,
                    old-time   == 1.0.*,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
-                   process    == 1.0.*,
+                   process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
                    array      >= 0.1 && < 0.4,
                    hashed-storage >= 0.5.2 && < 0.6,
diff --git a/src/Darcs/RepoPath.hs b/src/Darcs/RepoPath.hs
index aa00156..8c5432b 100644
--- a/src/Darcs/RepoPath.hs
+++ b/src/Darcs/RepoPath.hs
@@ -60,11 +60,9 @@ import qualified Darcs.Patch.FileName as PatchFileName ( FileName, fp2fn, fn2fp
 #include "impossible.h"
 
 class FilePathOrURL a where
- {-# INLINE toPath #-}
  toPath :: a -> String
 
 class FilePathOrURL a => FilePathLike a where
- {-# INLINE toFilePath #-}
  toFilePath :: a -> FilePath
 
 -- | Paths which are relative to the local darcs repository and normalized.
