diff --git a/darcs.cabal b/darcs.cabal
index 060a0bb..186b17d 100644
--- a/darcs.cabal
+++ b/darcs.cabal
@@ -424,3 +424,3 @@ Library
                      async        >= 2.0.1.4 && < 2.2,
-                     sandi        >= 0.2 && < 0.4,
+                     sandi        >= 0.2,
                      unix-compat  >= 0.1.2 && < 0.5,
diff --git a/src/Darcs/Util/Hash.hs b/src/Darcs/Util/Hash.hs
index 6c8bb9d..e6ccdf4 100644
--- a/src/Darcs/Util/Hash.hs
+++ b/src/Darcs/Util/Hash.hs
@@ -3,3 +3,3 @@
 --  BSD3
-{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveDataTypeable, CPP #-}
 module Darcs.Util.Hash( Hash(..)
@@ -15,2 +15,7 @@ import qualified Codec.Binary.Base16 as B16
 
+#if MIN_VERSION_sandi(0,4,0)
+#  define b16_dec b16Dec
+#  define b16_enc b16Enc
+#endif
+
 import Data.Maybe( isJust, fromJust )
