diff --git a/tests/hspwd.hs b/tests/hspwd.hs
index b93a7d5..f6fe2dd 100644
--- a/tests/hspwd.hs
+++ b/tests/hspwd.hs
@@ -1,5 +1,5 @@
 module Main where
 
-import Directory ( getCurrentDirectory )
+import System.Directory ( getCurrentDirectory )
 
 main = getCurrentDirectory >>= putStr
diff --git a/tests/trackdown-bisect-helper.hs b/tests/trackdown-bisect-helper.hs
index 5c325c4..7fcb4a4 100644
--- a/tests/trackdown-bisect-helper.hs
+++ b/tests/trackdown-bisect-helper.hs
@@ -15,8 +15,9 @@ patches with the linear implementation.
 
 
 import Control.Monad
+import System.Environment
 import System.IO
-import System
+import System.Process
 import System.Random
 import Data.List
 import Control.Exception
