diff -rN -u old-darcs-screened/src/Darcs/UI/SelectChanges.hs new-darcs-screened/src/Darcs/UI/SelectChanges.hs
--- old-darcs-screened/src/Darcs/UI/SelectChanges.hs	2022-01-21 14:50:08.028020677 +1100
+++ new-darcs-screened/src/Darcs/UI/SelectChanges.hs	2022-01-21 14:50:08.029020691 +1100
@@ -346,7 +346,11 @@
   -- after selecting with a splitter, the results may not be canonical
   canonizeAfterSplitter :: (FL p :> FL p) wA wB -> (FL p :> FL p) wA wB
   canonizeAfterSplitter (x :> y) =
-    let canonizeIfNeeded = maybe id canonizeSplit mspl
+    let
+      canonizeIfNeeded =
+        case mspl of
+          Just s -> canonizeSplit s
+          Nothing -> id
     in canonizeIfNeeded x :> canonizeIfNeeded y
 
   -- retrieve the results of patch selection
patch 821fbe1391f263d5becc87e1dea077a46f1eb2ba
Author: Ganesh Sittampalam <ganesh@earth.li>
Date:   Sun May 16 00:55:25 AEST 2021
  * fix GHC 9.0 compile error
  
  I suspect that it was caused by the changes to
  impredicativity in GHC 9.