--- darcs-2.5.2-orig/darcs.cabal	2011-12-19 20:44:08.644140399 +1100
+++ darcs-2.5.2/darcs.cabal	2011-12-24 10:03:14.325396692 +1100
@@ -124,7 +124,7 @@
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.4
+    build-tools: ghc >= 6.10 && < 7.6
 
     hs-source-dirs:   src
     include-dirs:     src
@@ -174,17 +174,17 @@
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
-                     filepath     >= 1.1.0.0 && < 1.3.0.0,
+                     filepath     >= 1.1.0.0 && < 1.5.0.0,
                      haskeline    >= 0.6.2.2 && < 0.7,
                      hashed-storage >= 0.5.2 && < 0.6,
                      base >= 3,
                      bytestring >= 0.9.0 && < 0.10,
                      text == 0.11.*,
-                     old-time   == 1.0.*,
+                     old-time   >= 1.0 && < 1.2,
                      directory  >= 1.0.0.0 && < 1.2.0.0,
                      process    >= 1.0 && < 1.2,
                      containers >= 0.1 && < 0.5,
-                     array      >= 0.1 && < 0.4,
+                     array      >= 0.1 && < 0.5,
                      random     == 1.0.*,
                      tar          == 0.3.*,
                      zlib >= 0.5.1.0 && < 0.6.0.0,
@@ -208,7 +208,7 @@
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.4
+    build-tools: ghc >= 6.10 && < 7.6
 
     hs-source-dirs:   src
     include-dirs:     src
@@ -379,7 +379,7 @@
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
-                     filepath     >= 1.1.0.0 && < 1.3.0.0,
+                     filepath     >= 1.1.0.0 && < 1.5.0.0,
                      haskeline    >= 0.6.2.2 && < 0.7,
                      hashed-storage >= 0.5.2 && < 0.6,
                      tar          == 0.3.*
@@ -390,11 +390,11 @@
     build-depends: base >= 3,
                    bytestring >= 0.9.0 && < 0.10,
                    text == 0.11.*,
-                   old-time   == 1.0.*,
+                   old-time   >= 1.0 && < 1.2,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
                    process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
-                   array      >= 0.1 && < 0.4,
+                   array      >= 0.1 && < 0.5,
                    random     == 1.0.*
 
 
@@ -483,7 +483,7 @@
                     src/umask.c
                     src/Crypt/sha2.c
 
-  build-tools: ghc >= 6.10 && < 7.4
+  build-tools: ghc >= 6.10 && < 7.6
 
   -- We need optimizations, regardless of what Hackage says
   -- Note: "if true" works around a cabal bug with order of flag composition
@@ -527,7 +527,7 @@
                    mtl          >= 1.0 && < 2.1,
                    parsec       >= 2.0 && < 3.2,
                    html         == 1.0.*,
-                   filepath     >= 1.1.0.0 && < 1.3.0.0,
+                   filepath     >= 1.1.0.0 && < 1.5.0.0,
                    haskeline    >= 0.6.2.2 && < 0.7,
                    hashed-storage >= 0.5.2 && < 0.6,
                    tar          == 0.3.*
@@ -538,11 +538,11 @@
   build-depends: base >= 3,
                  bytestring >= 0.9.0 && < 0.10,
                  text == 0.11.*,
-                   old-time   == 1.0.*,
+                   old-time   >= 1.0 && < 1.2,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
                    process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
-                   array      >= 0.1 && < 0.4,
+                   array      >= 0.1 && < 0.5,
                    random     == 1.0.*
 
   if flag(curl)
@@ -615,7 +615,7 @@
   else
     buildable: True
 
-    build-tools: ghc >= 6.10 && < 7.4
+    build-tools: ghc >= 6.10 && < 7.6
 
     build-depends:   base          < 5,
                      extensible-exceptions >= 0.1 && < 0.2,
@@ -623,7 +623,7 @@
                      mtl          >= 1.0 && < 2.1,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
-                     filepath     >= 1.1.0.0 && < 1.3.0.0,
+                     filepath     >= 1.1.0.0 && < 1.5.0.0,
                      QuickCheck   >= 2.1.0.0,
                      HUnit        >= 1.0,
                      test-framework             >= 0.2.2,
@@ -684,11 +684,11 @@
                    bytestring >= 0.9.0 && < 0.10,
                    haskeline    >= 0.6.2.2 && < 0.7,
                    text == 0.11.*,
-                   old-time   == 1.0.*,
+                   old-time   >= 1.0 && < 1.2,
                    directory  >= 1.0.0.0 && < 1.2.0.0,
                    process    >= 1.0 && < 1.2,
                    containers >= 0.1 && < 0.5,
-                   array      >= 0.1 && < 0.4,
+                   array      >= 0.1 && < 0.5,
                    hashed-storage >= 0.5.2 && < 0.6,
                    random     == 1.0.*
 
--- darcs-2.5.2-orig/src/Darcs/Utils.hs	2011-03-14 09:40:46.000000000 +1100
+++ darcs-2.5.2/src/Darcs/Utils.hs	2011-12-24 10:07:16.125342597 +1100
@@ -62,7 +62,7 @@
                            , TreeMonad, tree )
 import Storage.Hashed.Tree( Tree, listImmediate, findTree )
 
-showHexLen :: (Integral a) => Int -> a -> String
+showHexLen :: (Integral a, Show a) => Int -> a -> String
 showHexLen n x = let s = showHex x ""
                  in replicate (n - length s) ' ' ++ s
 
