--- darcs-2.8.0-orig/darcs.cabal	2012-04-23 01:14:14.000000000 +1000
+++ darcs-2.8.0/darcs.cabal	2012-05-05 15:27:31.999138292 +1000
@@ -364,7 +364,7 @@
 
     build-depends:   extensible-exceptions >= 0.1 && < 0.2,
                      regex-compat >= 0.95.1,
-                     mtl          >= 1.0 && < 2.1,
+                     mtl          >= 1.0 && < 2.2,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
                      filepath     >= 1.1.0.0 && < 1.5.0.0,
@@ -525,7 +525,7 @@
 
   build-depends:   extensible-exceptions >= 0.1 && < 0.2,
                    regex-compat >= 0.95.1,
-                   mtl          >= 1.0 && < 2.1,
+                   mtl          >= 1.0 && < 2.2,
                    parsec       >= 2.0 && < 3.2,
                    html         == 1.0.*,
                    filepath     >= 1.1.0.0 && < 1.5.0.0,
@@ -612,7 +612,7 @@
 
     build-depends:   extensible-exceptions >= 0.1 && < 0.2,
                      regex-compat >= 0.95.1,
-                     mtl          >= 1.0 && < 2.1,
+                     mtl          >= 1.0 && < 2.2,
                      parsec       >= 2.0 && < 3.2,
                      html         == 1.0.*,
                      filepath     >= 1.1.0.0 && < 1.5.0.0,
