--- a/src/Diagrams/Backend/PGF/Render.hs
+++ b/src/Diagrams/Backend/PGF/Render.hs
@@ -55,2 +55,4 @@ import Prelude
 
+import qualified Data.Semigroup as S
+
 -- | This data declaration is simply used as a token to distinguish
@@ -99,2 +101,5 @@ instance Fractional n => Default (Options PGF V2 n) where
 
+instance S.Semigroup (Render PGF V2 n) where
+  R ra <> R rb = R $ ra >> rb
+
 instance Monoid (Render PGF V2 n) where
