--- th-lift-0.5.3-orig/th-lift.cabal	2010-12-17 23:15:57.000000000 +1100
+++ th-lift-0.5.3/th-lift.cabal	2011-08-01 18:26:59.517014049 +1000
@@ -16,7 +16,7 @@
 
 Library
   Exposed-modules: Language.Haskell.TH.Lift
-  Extensions:      CPP, TemplateHaskell, MagicHash, TypeSynonymInstances
+  Extensions:      CPP, TemplateHaskell, MagicHash, TypeSynonymInstances, FlexibleInstances
   Hs-Source-Dirs:  src
   Build-Depends:   base >= 3 && < 5
 
@@ -24,7 +24,7 @@
     Build-Depends: packedstring == 0.1.*,
                    template-haskell >= 2.2 && < 2.4
   else
-    Build-Depends: template-haskell >= 2.4 && < 2.6
+    Build-Depends: template-haskell >= 2.4 && < 2.7
 
 source-repository head
   type:     git
--- th-lift-0.5.3-orig/src/Language/Haskell/TH/Lift.hs	2010-12-17 23:15:57.000000000 +1100
+++ th-lift-0.5.3/src/Language/Haskell/TH/Lift.hs	2011-08-01 18:23:20.913228648 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE CPP, TemplateHaskell, MagicHash, TypeSynonymInstances #-}
+{-# LANGUAGE CPP, TemplateHaskell, MagicHash, TypeSynonymInstances, FlexibleInstances #-}
 module Language.Haskell.TH.Lift (deriveLift, deriveLiftMany, deriveLift', deriveLiftMany', Lift(..)) where
 
 #if !(MIN_VERSION_template_haskell(2,4,0))
