diff --git a/hs-src/Language/Egison/Types.hs b/hs-src/Language/Egison/Types.hs
index 38d17e5..23052d4 100644
--- a/hs-src/Language/Egison/Types.hs
+++ b/hs-src/Language/Egison/Types.hs
@@ -1728,9 +1728,10 @@ modifyCounter m = do
   return result
 
 data RuntimeState = RuntimeState
-    -- | index counter for generating fresh variable
-      { indexCounter :: Int
-    -- | names of called functions for improved error message
+      {
+        -- | index counter for generating fresh variable
+        indexCounter :: Int
+        -- | names of called functions for improved error message
       , funcNameStack :: [String]
       }
 
