--- a/test/Common/Test.hs
+++ b/test/Common/Test.hs
@@ -20,6 +20,8 @@
            , ScopedTypeVariables
            , TypeSynonymInstances
  #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 
 module Common.Test
     ( tests
--- a/test/PostgreSQL/MigrateJSON.hs
+++ b/test/PostgreSQL/MigrateJSON.hs
@@ -10,6 +10,9 @@
            , TypeFamilies
            , UndecidableInstances
  #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
 module PostgreSQL.MigrateJSON where
 
 import Control.Monad.Trans.Resource (ResourceT)
