--- cautious-file-1.0.1-orig/cautious-file.cabal	2012-03-22 07:05:54.000000000 +1100
+++ cautious-file-1.0.1/cautious-file.cabal	2012-04-01 00:31:23.897007905 +1100
@@ -25,7 +25,7 @@
 
 Library
   hs-source-dirs:       src
-  build-Depends:	base >= 4, base < 5, directory == 1.1.*, filepath >= 1.2, filepath < 1.4, bytestring == 0.9.*
+  build-Depends:	base >= 4, base < 5, directory >= 1.0, directory < 1.2, filepath >= 1.1, filepath < 1.4, bytestring == 0.9.*
   if flag(posix) && !os(Windows)
     cpp-options: -D_POSIX
     build-Depends: unix
