diff --git a/src/Language/Javascript/JSaddle/WebSockets.hs b/src/Language/Javascript/JSaddle/WebSockets.hs
index 94718c0..e60ee89 100644
--- a/src/Language/Javascript/JSaddle/WebSockets.hs
+++ b/src/Language/Javascript/JSaddle/WebSockets.hs
@@ -2,2 +2,3 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP #-}
 -----------------------------------------------------------------------------
@@ -51,3 +52,7 @@ jsaddleOr opts entryPoint = websocketsOr opts wsApp
             receiveDataMessage conn >>= \case
+#if MIN_VERSION_websockets(0,12,0)
+                (WS.Text t _cache) ->
+#else
                 (WS.Text t) ->
+#endif
                     case decode t of
