diff --git a/src/Control/Lens/Action/Reified.hs b/src/Control/Lens/Action/Reified.hs
index 1837963..0116c91 100644
--- a/src/Control/Lens/Action/Reified.hs
+++ b/src/Control/Lens/Action/Reified.hs
@@ -26,2 +26,3 @@ import Control.Monad.Reader.Class
 import Data.Functor.Bind
+import Data.Functor.Contravariant (phantom)
 import Data.Functor.Plus
@@ -50,6 +51,6 @@ instance Strong (ReifiedMonadicFold m) where
   first' l = MonadicFold $ \f (s,c) ->
-    coerce $ runMonadicFold l (dimap (flip (,) c) coerce f) s
+    phantom $ runMonadicFold l (dimap (flip (,) c) phantom f) s
   {-# INLINE first' #-}
   second' l = MonadicFold $ \f (c,s) ->
-    coerce $ runMonadicFold l (dimap ((,) c) coerce f) s
+    phantom $ runMonadicFold l (dimap ((,) c) phantom f) s
   {-# INLINE second' #-}
