diff --git a/UI/HSCurses/Curses.hsc b/UI/HSCurses/Curses.hsc
index eee44f4..c721c5d 100644
--- a/UI/HSCurses/Curses.hsc
+++ b/UI/HSCurses/Curses.hsc
@@ -241,7 +241,7 @@ resetParams = do
 fi :: (Integral a, Num b) => a -> b
 fi = fromIntegral
 
-throwIfErr :: Num a => String -> IO a -> IO a
+throwIfErr :: (Num a, Show a, Eq a) => String -> IO a -> IO a
 --throwIfErr name act = do
 --    res <- act
 --    if res == (cERR)
@@ -249,7 +249,7 @@ throwIfErr :: Num a => String -> IO a -> IO a
 --        else return res
 throwIfErr s = throwIf (== (#const ERR)) (\a -> "Curses[" ++ show a ++ "]:"  ++ s)
 
-throwIfErr_ :: Num a => String -> IO a -> IO ()
+throwIfErr_ :: (Num a, Show a, Eq a) => String -> IO a -> IO ()
 throwIfErr_ name act = void $ throwIfErr name act
 
 errI :: IO CInt -> IO ()
diff --git a/hscurses.cabal b/hscurses.cabal
index d19e5b0..b9e4549 100644
--- a/hscurses.cabal
+++ b/hscurses.cabal
@@ -36,7 +36,7 @@ Source-Repository head
 
 Library
   Build-depends:  base == 4.*, mtl,
-                  old-time == 1.0.*, old-locale == 1.0.*
+                  old-time >= 1.0 && < 1.2, old-locale == 1.0.*
   if !os(windows)
     Build-depends: unix >= 2.4 && < 2.6
   Exposed-modules:
