diff --git a/UI/HSCurses/Curses.hsc b/UI/HSCurses/Curses.hsc
index 8e45b03..8724801 100644
--- a/UI/HSCurses/Curses.hsc
+++ b/UI/HSCurses/Curses.hsc
@@ -745,13 +745,13 @@ attrBold = (#const A_BOLD)
 
 ------------------------------------------------------------------------
 
-foreign import ccall threadsafe
+foreign import ccall safe 
     waddch :: Window -> ChType -> IO CInt
 
-foreign import ccall threadsafe
+foreign import ccall safe
     waddchnstr :: Window -> CString -> CInt -> IO CInt
 
-foreign import ccall threadsafe "static curses.h mvaddch" mvaddch_c :: CInt -> CInt -> ChType -> IO ()
+foreign import ccall safe "static curses.h mvaddch" mvaddch_c :: CInt -> CInt -> ChType -> IO ()
 
 mvWAddStr :: Window -> Int -> Int -> String -> IO ()
 mvWAddStr w y x str = wMove w y x >> wAddStr w str
@@ -828,12 +828,12 @@ wAddStr _   [] = return ()
 wAddStr win s  = throwIfErr_ ("waddnstr: <" ++ s ++ ">") $
     withLCStringLen (s) (\(ws,len) -> waddnstr win ws (fi len))
 
-foreign import ccall threadsafe
+foreign import ccall safe
     waddnstr :: Window -> CString -> CInt -> IO CInt
 
 #endif
 
-foreign import ccall threadsafe
+foreign import ccall safe
     vline  :: Char -> Int -> IO ()
 
 ------------------------------------------------------------------------
diff --git a/hscurses.cabal b/hscurses.cabal
index ce30519..b20d2bb 100644
--- a/hscurses.cabal
+++ b/hscurses.cabal
@@ -31,7 +31,7 @@ Data-files:
     cbits/HSCurses.h, cbits/HSCursesUtils.h
 
 Library
-  Build-depends:  base == 4.*, unix == 2.4.*, mtl,
+  Build-depends:  base == 4.*, unix >= 2.4 && < 2.6, mtl,
                   old-time == 1.0.*, old-locale == 1.0.*
   Exposed-modules:
       UI.HSCurses.Curses, UI.HSCurses.CursesHelper, UI.HSCurses.Widgets,
