diff --git a/tests/Instances.hs b/tests/Instances.hs
index 7e34ae3..ecfeebf 100644
--- a/tests/Instances.hs
+++ b/tests/Instances.hs
@@ -209,2 +209,3 @@ instance Arbitrary a => Arbitrary (DList.DList a) where
 
+#if !MIN_VERSION_QuickCheck(2,9,1)
 -- would like to have -- https://github.com/nick8325/quickcheck/issues/73
@@ -212,2 +213,3 @@ instance Arbitrary a => Arbitrary (Identity a) where
     arbitrary = Identity <$> arbitrary
+#endif
 
