diff --git a/tests/Instances.hs b/tests/Instances.hs
index 23a0e79..87fcf1a 100644
--- a/tests/Instances.hs
+++ b/tests/Instances.hs
@@ -144,2 +144,3 @@ instance (ApproxEq a) => ApproxEq [a] where
 
+#if ! MIN_VERSION_QuickCheck(2,8,3)
 instance Arbitrary Version where
@@ -150,2 +151,6 @@ instance Arbitrary a => Arbitrary (NonEmpty a) where
 
+instance Arbitrary a => Arbitrary (Const a b) where
+    arbitrary = Const <$> arbitrary
+#endif
+
 -- Version tags are deprecated, so we avoid using them in the Arbitrary
@@ -170,4 +175 @@ instance Arbitrary b => Arbitrary (Tagged a b) where
     arbitrary = Tagged <$> arbitrary
-
-instance Arbitrary a => Arbitrary (Const a b) where
-    arbitrary = Const <$> arbitrary
