diff --git a/Data/Aeson/Encoding/Builder.hs b/Data/Aeson/Encoding/Builder.hs
index 832b30a..a5c3863 100644
--- a/Data/Aeson/Encoding/Builder.hs
+++ b/Data/Aeson/Encoding/Builder.hs
@@ -40,3 +40,3 @@ module Data.Aeson.Encoding.Builder
 import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
 
diff --git a/Data/Aeson/Text.hs b/Data/Aeson/Text.hs
index 2786d7e..a179f53 100644
--- a/Data/Aeson/Text.hs
+++ b/Data/Aeson/Text.hs
@@ -23,3 +23,3 @@ module Data.Aeson.Text
 import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
 
diff --git a/Data/Aeson/Types/FromJSON.hs b/Data/Aeson/Types/FromJSON.hs
index a02dbcf..e0445ee 100644
--- a/Data/Aeson/Types/FromJSON.hs
+++ b/Data/Aeson/Types/FromJSON.hs
@@ -72,3 +72,3 @@ module Data.Aeson.Types.FromJSON
 import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
 
diff --git a/Data/Aeson/Types/ToJSON.hs b/Data/Aeson/Types/ToJSON.hs
index caa78d8..4d763c4 100644
--- a/Data/Aeson/Types/ToJSON.hs
+++ b/Data/Aeson/Types/ToJSON.hs
@@ -59,3 +59,3 @@ module Data.Aeson.Types.ToJSON
 import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
 
