diff --git a/tests/Instances.hs b/tests/Instances.hs
index e47ddb4..bf84f22 100644
--- a/tests/Instances.hs
+++ b/tests/Instances.hs
@@ -188,17 +188,21 @@ instance Arbitrary Natural where
   arbitrary = fromInteger . abs <$> arbitrary
 #endif
 
+#if !MIN_VERSION_QuickCheck(2,10,0)
 instance Arbitrary (Proxy a) where
     arbitrary = pure Proxy
 
 instance Arbitrary b => Arbitrary (Tagged a b) where
     arbitrary = Tagged <$> arbitrary
+#endif
 
 instance Arbitrary a => Arbitrary (DList.DList a) where
     arbitrary = DList.fromList <$> arbitrary
 
+#if !MIN_VERSION_QuickCheck(2,10,0)
 instance Arbitrary (f (g a)) => Arbitrary (Compose f g a) where
     arbitrary = Compose <$> arbitrary
+#endif
 
 #if !MIN_VERSION_QuickCheck(2,9,0)
 instance Arbitrary a => Arbitrary (Const a b) where
