--- cereal-0.3.5.1-orig/src/Data/Serialize.hs	2012-01-14 11:46:16.000000000 +1100
+++ cereal-0.3.5.1/src/Data/Serialize.hs	2012-05-13 16:13:26.703441393 +1000
@@ -514,7 +514,7 @@
 
 ------------------------------------------------------------------------
 
-checkGetSum :: (Ord word, Bits word, GetSum f) => word -> word -> Get (f a)
+checkGetSum :: (Ord word, Bits word, Num word, GetSum f) => word -> word -> Get (f a)
 checkGetSum size code | code < size = getSum code size
                       | otherwise   = fail "Unknown encoding for constructor"
 {-# INLINE checkGetSum #-}
