From https://github.com/winterland1989/word24/pull/7
From: Xavier Dectot <xavier@wheredoibegin.fr>
Date: Mon, 10 Mar 2025 18:37:19 +0100
Subject: [PATCH 3/3] Fixing a small ommission in test-suite

--- a/word24.cabal
+++ b/word24.cabal
@@ -44,7 +44,8 @@ Test-Suite test
     word24,
     QuickCheck                 >= 2 && < 3,
     test-framework             >= 0.2 && < 0.9,
-    test-framework-quickcheck2 >= 0.2 && < 0.7
+    test-framework-quickcheck2 >= 0.2 && < 0.7,
+    word-compat                >= 0.0 && <0.1
 
 benchmark bench24
   default-language: Haskell2010
-- 
2.48.1

