--- abstract-deque-0.1.5-orig/abstract-deque.cabal	2012-01-07 21:40:46.000000000 +1100
+++ abstract-deque-0.1.5/abstract-deque.cabal	2012-05-13 22:52:47.399627314 +1000
@@ -47,20 +47,21 @@
   build-depends:     base >= 4 && < 5, 
                      containers, HUnit
 
-  if impl( ghc >= 7.2) {
-    build-depends:   IORefCAS >= 0.2 
-    cpp-options:  -DUSE_CAS -DDEFAULT_SIGNATURES
---    extensions: DefaultSignatures
---    GHC-Options: -XDefaultSignatures
-  }
+-- IORefCAS does not exist on hackage, fix compilation failure with
+-- ghc 7.4.1 and 7.5.20120511
+--   if impl( ghc >= 7.2) {
+--     build-depends:   IORefCAS >= 0.2 
+--     cpp-options:  -DUSE_CAS -DDEFAULT_SIGNATURES
+-- --    extensions: DefaultSignatures
+-- --    GHC-Options: -XDefaultSignatures
+--   }
 
   extensions: CPP
   ghc-options: -O2
 
 
-
-Test-Suite test-abstract-deque
-    type:       exitcode-stdio-1.0
-    main-is:    Test.hs
-    build-depends:     base >= 4 && < 5, IORefCAS >= 0.2, containers, HUnit
-    ghc-options: -O2 -threaded -rtsopts 
+-- Test-Suite test-abstract-deque
+--     type:       exitcode-stdio-1.0
+--     main-is:    Test.hs
+--     build-depends:     base >= 4 && < 5, IORefCAS >= 0.2, containers, HUnit
+--     ghc-options: -O2 -threaded -rtsopts 
