From de29b5aed3957881c15dbede74f3c7d45e88588b Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@users.noreply.github.com>
Date: Tue, 13 Dec 2022 21:12:25 -0700
Subject: [PATCH] Disable doctests

Signed-off-by: hololeap <hololeap@users.noreply.github.com>
---
 hledger-lib.cabal | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hledger-lib.cabal b/hledger-lib.cabal
index fd38a3e..895fb10 100644
--- a/hledger-lib.cabal
+++ b/hledger-lib.cabal
@@ -192,8 +192,7 @@ test-suite doctest
     , unordered-containers >=0.2
     , utf8-string >=0.3.5
   default-language: Haskell2010
-  if impl(ghc >= 9.0) && impl(ghc < 9.2)
-    buildable: False
+  buildable: False
 
 test-suite unittest
   type: exitcode-stdio-1.0
-- 
2.37.4

