diff --git a/test/main.hs b/test/main.hs
index 72f4886..6f6e3f4 100644
--- a/test/main.hs
+++ b/test/main.hs
@@ -8,2 +8,3 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE CPP #-}
 import Test.Hspec
@@ -416,4 +417,2 @@ instance Exception DummyException
 
-instance Arbitrary (Map Int Char) where
-    arbitrary = mapFromList <$> arbitrary
 instance Arbitrary (HashMap Int Char) where
@@ -424,4 +423,10 @@ instance Arbitrary (UVector Int) where
     arbitrary = fromList <$> arbitrary
+#if ! MIN_VERSION_QuickCheck(2,8,2)
+instance Arbitrary (Map Int Char) where
+    arbitrary = mapFromList <$> arbitrary
 instance Arbitrary (Set Int) where
     arbitrary = setFromList <$> arbitrary
+instance Arbitrary (Seq Int) where
+    arbitrary = fromList <$> arbitrary
+#endif
 instance Arbitrary (HashSet Int) where
@@ -436,3 +441 @@ instance Arbitrary LText where
     arbitrary = fromList <$> arbitrary
-instance Arbitrary (Seq Int) where
-    arbitrary = fromList <$> arbitrary
