diff -Naur c2hs-0.14.1.orig/c2hs.cabal c2hs-0.14.1/c2hs.cabal
--- c2hs-0.14.1.orig/c2hs.cabal	2005-07-15 14:10:47.000000000 +0200
+++ c2hs-0.14.1/c2hs.cabal	2005-08-08 19:15:23.000000000 +0200
@@ -15,6 +15,9 @@
 	libraries.  It extracts C interface information from vanilla header 
 	files and generates marshaling and signature code in Haskell.
 Category:       Development tool
+Build-Depends:		base, haskell98
+Hs-Source-Dirs:	c2hs/lib
+Exposed-Modules: C2HS
 
 Executable:     c2hs
 Hs-Source-Dirs:	base/admin 
