diff -Naur c2hs-0.14.1.orig/c2hs.cabal c2hs-0.14.1/c2hs.cabal
--- c2hs-0.14.1.orig/c2hs.cabal 2005-07-15 14:10:47.000000000 +0000
+++ c2hs-0.14.1/c2hs.cabal      2005-12-04 21:30:55.000000000 +0000
@@ -15,6 +15,7 @@
 	libraries.  It extracts C interface information from vanilla header 
 	files and generates marshaling and signature code in Haskell.
 Category:       Development tool
+Build-Depends:  base, haskell98
 
 Executable:     c2hs
 Hs-Source-Dirs:	base/admin 
