--- vcswrapper-0.0.1-orig/src/VCSWrapper/Common/TemporaryFiles.hs	2012-02-06 08:42:52.000000000 +1100
+++ vcswrapper-0.0.1/src/VCSWrapper/Common/TemporaryFiles.hs	2013-04-01 23:27:13.664585469 +1100
@@ -1,3 +1,4 @@
+{-# LANGUAGE ScopedTypeVariables #-}
 -----------------------------------------------------------------------------
 --
 -- Module      :  VCSWrapper.Common.TemporaryFiles
@@ -18,8 +19,7 @@
 
 import System.IO
 import System.Directory(getTemporaryDirectory, removeFile)
-import System.IO.Error(catch)
-import Control.Exception(finally)
+import Control.Exception(catch, finally, IOException)
 
 {- |
     Executes given function using a tempory file.
@@ -31,7 +31,7 @@
              -> IO a
 withTempFile pattern func =
     do
-       tempdir <- catch (getTemporaryDirectory) (\_ -> return ".")
+       tempdir <- catch (getTemporaryDirectory) (\(_::IOException) -> return ".")
        putStrLn $ "Obtained temporary directory: "++tempdir
        (file, handle) <- openTempFile tempdir pattern
        putStrLn $ "Opened file: "++file++", handle: "++show handle
--- vcswrapper-0.0.1-orig/src/VCSWrapper/Svn.hs	2012-02-06 08:42:52.000000000 +1100
+++ vcswrapper-0.0.1/src/VCSWrapper/Svn.hs	2013-04-01 23:23:38.072579455 +1100
@@ -42,11 +42,11 @@
 
 import VCSWrapper.Common.TemporaryFiles
 import Control.Monad.Reader
-import Maybe
+import Data.Maybe
 import System.IO
 
 import Data.List.Utils(startswith)
-import Monad (filterM)
+import Control.Monad (filterM)
 import System.Directory(doesFileExist, getDirectoryContents)
 import System.FilePath(combine, splitFileName)
 
--- vcswrapper-0.0.1-orig/src/VCSWrapper/Mercurial.hs	2012-02-06 08:42:52.000000000 +1100
+++ vcswrapper-0.0.1/src/VCSWrapper/Mercurial.hs	2013-04-01 23:22:37.308577761 +1100
@@ -34,7 +34,7 @@
 
 import System.IO
 import Control.Monad.Reader
-import Maybe
+import Data.Maybe
 
 {- |
     Add all new files, delete all missing files. Executes @hg addremove@.
--- vcswrapper-0.0.1-orig/vcswrapper.cabal	2012-02-06 08:42:52.000000000 +1100
+++ vcswrapper-0.0.1/vcswrapper.cabal	2013-04-02 00:02:49.129645033 +1100
@@ -1,5 +1,5 @@
 name: vcswrapper
-version: 0.0.1
+version: 0.0.2
 cabal-version: >= 1.8
 build-type: Simple
 license: GPL
@@ -19,16 +19,14 @@
 data-dir: ""
 
 library
-    build-depends: MissingH >=1.1.0.3 && <1.2,
-               base >=4.0.0.0 && <4.5,
-               directory >=1.1.0.0 && <1.2,
-               haddock >=2.9.0 && <2.10,
-               haskell98 >=1.1.0.1 && <2.1,
-               hxt >=9.1.2 && <9.2,
-               mtl >=2.0.1.0 && <2.1,
+    build-depends: MissingH >=1.1.0.3 && <1.3,
+               base >=4.0.0.0 && <5,
+               directory >=1.1.0.0 && <1.3,
+               hxt >=9.1.2 && <9.4,
+               mtl >=2.0.1.0 && <2.2,
                parsec >=3.1.1 && <3.2,
                process >=1.0.1.5 && <1.2,
-               filepath >=1.2.0.0 && < 1.3
+               filepath >=1.2.0.0 && < 1.4
     exposed-modules: VCSWrapper.Common VCSWrapper.Git VCSWrapper.Svn VCSWrapper.Mercurial
     other-modules: 
                VCSWrapper.Svn.Types VCSWrapper.Svn.Process VCSWrapper.Svn.Parsers 
@@ -41,16 +39,14 @@
 
 executable vcswrapper
     main-is: Main.hs
-    build-depends: MissingH >=1.1.0.3 && <1.2,
-               base >=4.0.0.0 && <4.5,
-               directory >=1.1.0.0 && <1.2,
-               haddock >=2.9.0 && <2.10,
-               haskell98 >=1.1.0.1 && <2.1,
-               hxt >=9.1.2 && <9.2,
-               mtl >=2.0.1.0 && <2.1,
+    build-depends: MissingH >=1.1.0.3 && <1.3,
+               base >=4.0.0.0 && <5,
+               directory >=1.1.0.0 && <1.3,
+               hxt >=9.1.2 && <9.4,
+               mtl >=2.0.1.0 && <2.2,
                parsec >=3.1.1 && <3.2,
                process >=1.0.1.5 && <1.2,
-               filepath >=1.2.0.0 && < 1.3
+               filepath >=1.2.0.0 && < 1.4
     buildable: True
     hs-source-dirs: src
     other-modules:
