From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>
Bug: https://github.com/chrisdone/lucid/issues/107

Bug has been fixed upstream so hashable is no longer needed as a
dependency.

--- old/dhall-docs.cabal
+++ new/dhall-docs.cabal
@@ -137,12 +137,6 @@
         dhall                             ,
         dhall-docs                        ,
         foldl        < 1.5                ,
-
-        -- hashable isn't used directly, but lucid's rendering output depends
-        -- on the Hashable Text instance defined in hashable.
-        -- See https://github.com/chrisdone/lucid/issues/107.
-        hashable     < 1.3.1              ,
-
         path                              ,
         path-io                           ,
         pretty       >= 1.1.1.1           ,
