diff --git a/cabal-dev.cabal b/cabal-dev.cabal
index f1b7f90..2454875 100644
--- a/cabal-dev.cabal
+++ b/cabal-dev.cabal
@@ -70,7 +70,7 @@ Executable ghc-pkg-6_8-compat
 Executable cabal-dev-test
   Main-is: RunTests.hs
   GHC-Options: -Wall
-  HS-Source-Dirs: src, test
+  HS-Source-Dirs: src, test, cabal-install-0.8.2
   if flag(no-cabal-dev) || !flag(build-tests)
     Buildable: False
   else
@@ -129,9 +129,6 @@ Executable cabal-dev
   if os(windows)
      build-depends: Win32 >= 2.1  && < 2.3
 
-  Build-tools:
-    cabal >= 0.8.2
-
   Other-modules:
     Distribution.Dev.AddSource,
     Distribution.Dev.Sandbox,
