--- Makefile.in	2010-11-13 17:26:28.514100423 +0100
+++ Makefile.in.new	2010-11-13 17:32:27.770849138 +0100
@@ -1,7 +1,8 @@
 CC = @CC@
 SED = @SED@
 CFLAGS = @CFLAGS@ @GLIB_CFLAGS@ @DBUS_GLIB_CFLAGS@ @LIBPANEL_CFLAGS@
-LDFLAGS = @LDFLAGS@ @GLIB_LIBS@ @DBUS_GLIB_LIBS@ @LIBPANEL_LIBS@
+LDFLAGS = @LDFLAGS@
+LIBS = @GLIB_LIBS@ @DBUS_GLIB_LIBS@ @LIBPANEL_LIBS@
 INSTALL_SH = @INSTALL@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -22,7 +23,7 @@
 	$(INSTALL_SH) -m 0644 xmonad48.png $(DESTDIR)$(datarootdir)/icons/hicolor/48x48/apps
 
 xmonad-log-applet: main.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ main.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ main.c $(LIBS)
 
 xmonad-log-applet.server: xmonad-log-applet.server.in
 	$(SED) -e 's!@LIBEXECDIR@!${libdir}/gnome-applets!g' \
