--- a/src/Database/HaskellDB/DBSpec/DBInfo.hs
+++ b/src/Database/HaskellDB/DBSpec/DBInfo.hs
@@ -26,2 +26,3 @@ import Data.Char (toLower, isAlpha)
 import Text.PrettyPrint.HughesPJ
+import Prelude hiding ((<>))
 
--- a/src/Database/HaskellDB/DBSpec/DBSpecToDBDirect.hs
+++ b/src/Database/HaskellDB/DBSpec/DBSpecToDBDirect.hs
@@ -36,2 +36,3 @@ import System.Directory (createDirectory, doesDirectoryExist)
 import Text.PrettyPrint.HughesPJ
+import Prelude hiding ((<>))
 
--- a/src/Database/HaskellDB/PrintQuery.hs
+++ b/src/Database/HaskellDB/PrintQuery.hs
@@ -29,2 +29,3 @@ import Database.HaskellDB.Sql.Print as Sql (ppSql)
 import Text.PrettyPrint.HughesPJ
+import Prelude hiding((<>))
 
--- a/src/Database/HaskellDB/Sql/Print.hs
+++ b/src/Database/HaskellDB/Sql/Print.hs
@@ -28,2 +28,3 @@ import Data.List (intersperse)
 import Text.PrettyPrint.HughesPJ
+import Prelude hiding((<>))
 
