diff --git a/src/Database/HaskellDB/BoundedList.hs b/src/Database/HaskellDB/BoundedList.hs
index 63d05ee..cf83bc2 100644
--- a/src/Database/HaskellDB/BoundedList.hs
+++ b/src/Database/HaskellDB/BoundedList.hs
@@ -103,3 +103,3 @@ instance Size N2 where size _ = 2
 instance Less N1 N2
-instance Less a N1 => Less a N2
+instance (Size a, Less a N1) => Less a N2
 
@@ -108,3 +108,3 @@ instance Size N3 where size _ = 3
 instance Less N2 N3
-instance Less a N2 => Less a N3
+instance (Size a, Less a N2) => Less a N3
 
@@ -113,3 +113,3 @@ instance Size N4 where size _ = 4
 instance Less N3 N4
-instance Less a N3 => Less a N4
+instance (Size a, Less a N3) => Less a N4
 
@@ -118,3 +118,3 @@ instance Size N5 where size _ = 5
 instance Less N4 N5
-instance Less a N4 => Less a N5
+instance (Size a, Less a N4) => Less a N5
 
@@ -123,3 +123,3 @@ instance Size N6 where size _ = 6
 instance Less N5 N6
-instance Less a N5 => Less a N6
+instance (Size a, Less a N5) => Less a N6
 
@@ -128,3 +128,3 @@ instance Size N7 where size _ = 7
 instance Less N6 N7
-instance Less a N6 => Less a N7
+instance (Size a, Less a N6) => Less a N7
 
@@ -133,3 +133,3 @@ instance Size N8 where size _ = 8
 instance Less N7 N8
-instance Less a N7 => Less a N8
+instance (Size a, Less a N7) => Less a N8
 
@@ -138,3 +138,3 @@ instance Size N9 where size _ = 9
 instance Less N8 N9
-instance Less a N8 => Less a N9
+instance (Size a, Less a N8) => Less a N9
 
@@ -143,3 +143,3 @@ instance Size N10 where size _ = 10
 instance Less N9 N10
-instance Less a N9 => Less a N10
+instance (Size a, Less a N9) => Less a N10
 
@@ -148,3 +148,3 @@ instance Size N11 where size _ = 11
 instance Less N10 N11
-instance Less a N10 => Less a N11
+instance (Size a, Less a N10) => Less a N11
 
@@ -153,3 +153,3 @@ instance Size N12 where size _ = 12
 instance Less N11 N12
-instance Less a N11 => Less a N12
+instance (Size a, Less a N11) => Less a N12
 
@@ -158,3 +158,3 @@ instance Size N13 where size _ = 13
 instance Less N12 N13
-instance Less a N12 => Less a N13
+instance (Size a, Less a N12) => Less a N13
 
@@ -163,3 +163,3 @@ instance Size N14 where size _ = 14
 instance Less N13 N14
-instance Less a N13 => Less a N14
+instance (Size a, Less a N13) => Less a N14
 
@@ -168,3 +168,3 @@ instance Size N15 where size _ = 15
 instance Less N14 N15
-instance Less a N14 => Less a N15
+instance (Size a, Less a N14) => Less a N15
 
@@ -173,3 +173,3 @@ instance Size N16 where size _ = 16
 instance Less N15 N16
-instance Less a N15 => Less a N16
+instance (Size a, Less a N15) => Less a N16
 
@@ -178,3 +178,3 @@ instance Size N17 where size _ = 17
 instance Less N16 N17
-instance Less a N16 => Less a N17
+instance (Size a, Less a N16) => Less a N17
 
@@ -183,3 +183,3 @@ instance Size N18 where size _ = 18
 instance Less N17 N18
-instance Less a N17 => Less a N18
+instance (Size a, Less a N17) => Less a N18
 
@@ -188,3 +188,3 @@ instance Size N19 where size _ = 19
 instance Less N18 N19
-instance Less a N18 => Less a N19
+instance (Size a, Less a N18) => Less a N19
 
@@ -193,3 +193,3 @@ instance Size N20 where size _ = 20
 instance Less N19 N20
-instance Less a N19 => Less a N20
+instance (Size a, Less a N19) => Less a N20
 
@@ -198,3 +198,3 @@ instance Size N21 where size _ = 21
 instance Less N20 N21
-instance Less a N20 => Less a N21
+instance (Size a, Less a N20) => Less a N21
 
@@ -203,3 +203,3 @@ instance Size N22 where size _ = 22
 instance Less N21 N22
-instance Less a N21 => Less a N22
+instance (Size a, Less a N21) => Less a N22
 
@@ -208,3 +208,3 @@ instance Size N23 where size _ = 23
 instance Less N22 N23
-instance Less a N22 => Less a N23
+instance (Size a, Less a N22) => Less a N23
 
@@ -213,3 +213,3 @@ instance Size N24 where size _ = 24
 instance Less N23 N24
-instance Less a N23 => Less a N24
+instance (Size a, Less a N23) => Less a N24
 
@@ -218,3 +218,3 @@ instance Size N25 where size _ = 25
 instance Less N24 N25
-instance Less a N24 => Less a N25
+instance (Size a, Less a N24) => Less a N25
 
@@ -223,3 +223,3 @@ instance Size N26 where size _ = 26
 instance Less N25 N26
-instance Less a N25 => Less a N26
+instance (Size a, Less a N25) => Less a N26
 
@@ -228,3 +228,3 @@ instance Size N27 where size _ = 27
 instance Less N26 N27
-instance Less a N26 => Less a N27
+instance (Size a, Less a N26) => Less a N27
 
@@ -233,3 +233,3 @@ instance Size N28 where size _ = 28
 instance Less N27 N28
-instance Less a N27 => Less a N28
+instance (Size a, Less a N27) => Less a N28
 
@@ -238,3 +238,3 @@ instance Size N29 where size _ = 29
 instance Less N28 N29
-instance Less a N28 => Less a N29
+instance (Size a, Less a N28) => Less a N29
 
@@ -243,3 +243,3 @@ instance Size N30 where size _ = 30
 instance Less N29 N30
-instance Less a N29 => Less a N30
+instance (Size a, Less a N29) => Less a N30
 
@@ -248,3 +248,3 @@ instance Size N31 where size _ = 31
 instance Less N30 N31
-instance Less a N30 => Less a N31
+instance (Size a, Less a N30) => Less a N31
 
@@ -253,3 +253,3 @@ instance Size N32 where size _ = 32
 instance Less N31 N32
-instance Less a N31 => Less a N32
+instance (Size a, Less a N31) => Less a N32
 
@@ -258,3 +258,3 @@ instance Size N33 where size _ = 33
 instance Less N32 N33
-instance Less a N32 => Less a N33
+instance (Size a, Less a N32) => Less a N33
 
@@ -263,3 +263,3 @@ instance Size N34 where size _ = 34
 instance Less N33 N34
-instance Less a N33 => Less a N34
+instance (Size a, Less a N33) => Less a N34
 
@@ -268,3 +268,3 @@ instance Size N35 where size _ = 35
 instance Less N34 N35
-instance Less a N34 => Less a N35
+instance (Size a, Less a N34) => Less a N35
 
@@ -273,3 +273,3 @@ instance Size N36 where size _ = 36
 instance Less N35 N36
-instance Less a N35 => Less a N36
+instance (Size a, Less a N35) => Less a N36
 
@@ -278,3 +278,3 @@ instance Size N37 where size _ = 37
 instance Less N36 N37
-instance Less a N36 => Less a N37
+instance (Size a, Less a N36) => Less a N37
 
@@ -283,3 +283,3 @@ instance Size N38 where size _ = 38
 instance Less N37 N38
-instance Less a N37 => Less a N38
+instance (Size a, Less a N37) => Less a N38
 
@@ -288,3 +288,3 @@ instance Size N39 where size _ = 39
 instance Less N38 N39
-instance Less a N38 => Less a N39
+instance (Size a, Less a N38) => Less a N39
 
@@ -293,3 +293,3 @@ instance Size N40 where size _ = 40
 instance Less N39 N40
-instance Less a N39 => Less a N40
+instance (Size a, Less a N39) => Less a N40
 
@@ -298,3 +298,3 @@ instance Size N41 where size _ = 41
 instance Less N40 N41
-instance Less a N40 => Less a N41
+instance (Size a, Less a N40) => Less a N41
 
@@ -303,3 +303,3 @@ instance Size N42 where size _ = 42
 instance Less N41 N42
-instance Less a N41 => Less a N42
+instance (Size a, Less a N41) => Less a N42
 
@@ -308,3 +308,3 @@ instance Size N43 where size _ = 43
 instance Less N42 N43
-instance Less a N42 => Less a N43
+instance (Size a, Less a N42) => Less a N43
 
@@ -313,3 +313,3 @@ instance Size N44 where size _ = 44
 instance Less N43 N44
-instance Less a N43 => Less a N44
+instance (Size a, Less a N43) => Less a N44
 
@@ -318,3 +318,3 @@ instance Size N45 where size _ = 45
 instance Less N44 N45
-instance Less a N44 => Less a N45
+instance (Size a, Less a N44) => Less a N45
 
@@ -323,3 +323,3 @@ instance Size N46 where size _ = 46
 instance Less N45 N46
-instance Less a N45 => Less a N46
+instance (Size a, Less a N45) => Less a N46
 
@@ -328,3 +328,3 @@ instance Size N47 where size _ = 47
 instance Less N46 N47
-instance Less a N46 => Less a N47
+instance (Size a, Less a N46) => Less a N47
 
@@ -333,3 +333,3 @@ instance Size N48 where size _ = 48
 instance Less N47 N48
-instance Less a N47 => Less a N48
+instance (Size a, Less a N47) => Less a N48
 
@@ -338,3 +338,3 @@ instance Size N49 where size _ = 49
 instance Less N48 N49
-instance Less a N48 => Less a N49
+instance (Size a, Less a N48) => Less a N49
 
@@ -343,3 +343,3 @@ instance Size N50 where size _ = 50
 instance Less N49 N50
-instance Less a N49 => Less a N50
+instance (Size a, Less a N49) => Less a N50
 
@@ -348,3 +348,3 @@ instance Size N51 where size _ = 51
 instance Less N50 N51
-instance Less a N50 => Less a N51
+instance (Size a, Less a N50) => Less a N51
 
@@ -353,3 +353,3 @@ instance Size N52 where size _ = 52
 instance Less N51 N52
-instance Less a N51 => Less a N52
+instance (Size a, Less a N51) => Less a N52
 
@@ -358,3 +358,3 @@ instance Size N53 where size _ = 53
 instance Less N52 N53
-instance Less a N52 => Less a N53
+instance (Size a, Less a N52) => Less a N53
 
@@ -363,3 +363,3 @@ instance Size N54 where size _ = 54
 instance Less N53 N54
-instance Less a N53 => Less a N54
+instance (Size a, Less a N53) => Less a N54
 
@@ -368,3 +368,3 @@ instance Size N55 where size _ = 55
 instance Less N54 N55
-instance Less a N54 => Less a N55
+instance (Size a, Less a N54) => Less a N55
 
@@ -373,3 +373,3 @@ instance Size N56 where size _ = 56
 instance Less N55 N56
-instance Less a N55 => Less a N56
+instance (Size a, Less a N55) => Less a N56
 
@@ -378,3 +378,3 @@ instance Size N57 where size _ = 57
 instance Less N56 N57
-instance Less a N56 => Less a N57
+instance (Size a, Less a N56) => Less a N57
 
@@ -383,3 +383,3 @@ instance Size N58 where size _ = 58
 instance Less N57 N58
-instance Less a N57 => Less a N58
+instance (Size a, Less a N57) => Less a N58
 
@@ -388,3 +388,3 @@ instance Size N59 where size _ = 59
 instance Less N58 N59
-instance Less a N58 => Less a N59
+instance (Size a, Less a N58) => Less a N59
 
@@ -393,3 +393,3 @@ instance Size N60 where size _ = 60
 instance Less N59 N60
-instance Less a N59 => Less a N60
+instance (Size a, Less a N59) => Less a N60
 
@@ -398,3 +398,3 @@ instance Size N61 where size _ = 61
 instance Less N60 N61
-instance Less a N60 => Less a N61
+instance (Size a, Less a N60) => Less a N61
 
@@ -403,3 +403,3 @@ instance Size N62 where size _ = 62
 instance Less N61 N62
-instance Less a N61 => Less a N62
+instance (Size a, Less a N61) => Less a N62
 
@@ -408,3 +408,3 @@ instance Size N63 where size _ = 63
 instance Less N62 N63
-instance Less a N62 => Less a N63
+instance (Size a, Less a N62) => Less a N63
 
@@ -413,3 +413,3 @@ instance Size N64 where size _ = 64
 instance Less N63 N64
-instance Less a N63 => Less a N64
+instance (Size a, Less a N63) => Less a N64
 
@@ -418,3 +418,3 @@ instance Size N65 where size _ = 65
 instance Less N64 N65
-instance Less a N64 => Less a N65
+instance (Size a, Less a N64) => Less a N65
 
@@ -423,3 +423,3 @@ instance Size N66 where size _ = 66
 instance Less N65 N66
-instance Less a N65 => Less a N66
+instance (Size a, Less a N65) => Less a N66
 
@@ -428,3 +428,3 @@ instance Size N67 where size _ = 67
 instance Less N66 N67
-instance Less a N66 => Less a N67
+instance (Size a, Less a N66) => Less a N67
 
@@ -433,3 +433,3 @@ instance Size N68 where size _ = 68
 instance Less N67 N68
-instance Less a N67 => Less a N68
+instance (Size a, Less a N67) => Less a N68
 
@@ -438,3 +438,3 @@ instance Size N69 where size _ = 69
 instance Less N68 N69
-instance Less a N68 => Less a N69
+instance (Size a, Less a N68) => Less a N69
 
@@ -443,3 +443,3 @@ instance Size N70 where size _ = 70
 instance Less N69 N70
-instance Less a N69 => Less a N70
+instance (Size a, Less a N69) => Less a N70
 
@@ -448,3 +448,3 @@ instance Size N71 where size _ = 71
 instance Less N70 N71
-instance Less a N70 => Less a N71
+instance (Size a, Less a N70) => Less a N71
 
@@ -453,3 +453,3 @@ instance Size N72 where size _ = 72
 instance Less N71 N72
-instance Less a N71 => Less a N72
+instance (Size a, Less a N71) => Less a N72
 
@@ -458,3 +458,3 @@ instance Size N73 where size _ = 73
 instance Less N72 N73
-instance Less a N72 => Less a N73
+instance (Size a, Less a N72) => Less a N73
 
@@ -463,3 +463,3 @@ instance Size N74 where size _ = 74
 instance Less N73 N74
-instance Less a N73 => Less a N74
+instance (Size a, Less a N73) => Less a N74
 
@@ -468,3 +468,3 @@ instance Size N75 where size _ = 75
 instance Less N74 N75
-instance Less a N74 => Less a N75
+instance (Size a, Less a N74) => Less a N75
 
@@ -473,3 +473,3 @@ instance Size N76 where size _ = 76
 instance Less N75 N76
-instance Less a N75 => Less a N76
+instance (Size a, Less a N75) => Less a N76
 
@@ -478,3 +478,3 @@ instance Size N77 where size _ = 77
 instance Less N76 N77
-instance Less a N76 => Less a N77
+instance (Size a, Less a N76) => Less a N77
 
@@ -483,3 +483,3 @@ instance Size N78 where size _ = 78
 instance Less N77 N78
-instance Less a N77 => Less a N78
+instance (Size a, Less a N77) => Less a N78
 
@@ -488,3 +488,3 @@ instance Size N79 where size _ = 79
 instance Less N78 N79
-instance Less a N78 => Less a N79
+instance (Size a, Less a N78) => Less a N79
 
@@ -493,3 +493,3 @@ instance Size N80 where size _ = 80
 instance Less N79 N80
-instance Less a N79 => Less a N80
+instance (Size a, Less a N79) => Less a N80
 
@@ -498,3 +498,3 @@ instance Size N81 where size _ = 81
 instance Less N80 N81
-instance Less a N80 => Less a N81
+instance (Size a, Less a N80) => Less a N81
 
@@ -503,3 +503,3 @@ instance Size N82 where size _ = 82
 instance Less N81 N82
-instance Less a N81 => Less a N82
+instance (Size a, Less a N81) => Less a N82
 
@@ -508,3 +508,3 @@ instance Size N83 where size _ = 83
 instance Less N82 N83
-instance Less a N82 => Less a N83
+instance (Size a, Less a N82) => Less a N83
 
@@ -513,3 +513,3 @@ instance Size N84 where size _ = 84
 instance Less N83 N84
-instance Less a N83 => Less a N84
+instance (Size a, Less a N83) => Less a N84
 
@@ -518,3 +518,3 @@ instance Size N85 where size _ = 85
 instance Less N84 N85
-instance Less a N84 => Less a N85
+instance (Size a, Less a N84) => Less a N85
 
@@ -523,3 +523,3 @@ instance Size N86 where size _ = 86
 instance Less N85 N86
-instance Less a N85 => Less a N86
+instance (Size a, Less a N85) => Less a N86
 
@@ -528,3 +528,3 @@ instance Size N87 where size _ = 87
 instance Less N86 N87
-instance Less a N86 => Less a N87
+instance (Size a, Less a N86) => Less a N87
 
@@ -533,3 +533,3 @@ instance Size N88 where size _ = 88
 instance Less N87 N88
-instance Less a N87 => Less a N88
+instance (Size a, Less a N87) => Less a N88
 
@@ -538,3 +538,3 @@ instance Size N89 where size _ = 89
 instance Less N88 N89
-instance Less a N88 => Less a N89
+instance (Size a, Less a N88) => Less a N89
 
@@ -543,3 +543,3 @@ instance Size N90 where size _ = 90
 instance Less N89 N90
-instance Less a N89 => Less a N90
+instance (Size a, Less a N89) => Less a N90
 
@@ -548,3 +548,3 @@ instance Size N91 where size _ = 91
 instance Less N90 N91
-instance Less a N90 => Less a N91
+instance (Size a, Less a N90) => Less a N91
 
@@ -553,3 +553,3 @@ instance Size N92 where size _ = 92
 instance Less N91 N92
-instance Less a N91 => Less a N92
+instance (Size a, Less a N91) => Less a N92
 
@@ -558,3 +558,3 @@ instance Size N93 where size _ = 93
 instance Less N92 N93
-instance Less a N92 => Less a N93
+instance (Size a, Less a N92) => Less a N93
 
@@ -563,3 +563,3 @@ instance Size N94 where size _ = 94
 instance Less N93 N94
-instance Less a N93 => Less a N94
+instance (Size a, Less a N93) => Less a N94
 
@@ -568,3 +568,3 @@ instance Size N95 where size _ = 95
 instance Less N94 N95
-instance Less a N94 => Less a N95
+instance (Size a, Less a N94) => Less a N95
 
@@ -573,3 +573,3 @@ instance Size N96 where size _ = 96
 instance Less N95 N96
-instance Less a N95 => Less a N96
+instance (Size a, Less a N95) => Less a N96
 
@@ -578,3 +578,3 @@ instance Size N97 where size _ = 97
 instance Less N96 N97
-instance Less a N96 => Less a N97
+instance (Size a, Less a N96) => Less a N97
 
@@ -583,3 +583,3 @@ instance Size N98 where size _ = 98
 instance Less N97 N98
-instance Less a N97 => Less a N98
+instance (Size a, Less a N97) => Less a N98
 
@@ -588,3 +588,3 @@ instance Size N99 where size _ = 99
 instance Less N98 N99
-instance Less a N98 => Less a N99
+instance (Size a, Less a N98) => Less a N99
 
@@ -593,3 +593,3 @@ instance Size N100 where size _ = 100
 instance Less N99 N100
-instance Less a N99 => Less a N100
+instance (Size a, Less a N99) => Less a N100
 
@@ -598,3 +598,3 @@ instance Size N101 where size _ = 101
 instance Less N100 N101
-instance Less a N100 => Less a N101
+instance (Size a, Less a N100) => Less a N101
 
@@ -603,3 +603,3 @@ instance Size N102 where size _ = 102
 instance Less N101 N102
-instance Less a N101 => Less a N102
+instance (Size a, Less a N101) => Less a N102
 
@@ -608,3 +608,3 @@ instance Size N103 where size _ = 103
 instance Less N102 N103
-instance Less a N102 => Less a N103
+instance (Size a, Less a N102) => Less a N103
 
@@ -613,3 +613,3 @@ instance Size N104 where size _ = 104
 instance Less N103 N104
-instance Less a N103 => Less a N104
+instance (Size a, Less a N103) => Less a N104
 
@@ -618,3 +618,3 @@ instance Size N105 where size _ = 105
 instance Less N104 N105
-instance Less a N104 => Less a N105
+instance (Size a, Less a N104) => Less a N105
 
@@ -623,3 +623,3 @@ instance Size N106 where size _ = 106
 instance Less N105 N106
-instance Less a N105 => Less a N106
+instance (Size a, Less a N105) => Less a N106
 
@@ -628,3 +628,3 @@ instance Size N107 where size _ = 107
 instance Less N106 N107
-instance Less a N106 => Less a N107
+instance (Size a, Less a N106) => Less a N107
 
@@ -633,3 +633,3 @@ instance Size N108 where size _ = 108
 instance Less N107 N108
-instance Less a N107 => Less a N108
+instance (Size a, Less a N107) => Less a N108
 
@@ -638,3 +638,3 @@ instance Size N109 where size _ = 109
 instance Less N108 N109
-instance Less a N108 => Less a N109
+instance (Size a, Less a N108) => Less a N109
 
@@ -643,3 +643,3 @@ instance Size N110 where size _ = 110
 instance Less N109 N110
-instance Less a N109 => Less a N110
+instance (Size a, Less a N109) => Less a N110
 
@@ -648,3 +648,3 @@ instance Size N111 where size _ = 111
 instance Less N110 N111
-instance Less a N110 => Less a N111
+instance (Size a, Less a N110) => Less a N111
 
@@ -653,3 +653,3 @@ instance Size N112 where size _ = 112
 instance Less N111 N112
-instance Less a N111 => Less a N112
+instance (Size a, Less a N111) => Less a N112
 
@@ -658,3 +658,3 @@ instance Size N113 where size _ = 113
 instance Less N112 N113
-instance Less a N112 => Less a N113
+instance (Size a, Less a N112) => Less a N113
 
@@ -663,3 +663,3 @@ instance Size N114 where size _ = 114
 instance Less N113 N114
-instance Less a N113 => Less a N114
+instance (Size a, Less a N113) => Less a N114
 
@@ -668,3 +668,3 @@ instance Size N115 where size _ = 115
 instance Less N114 N115
-instance Less a N114 => Less a N115
+instance (Size a, Less a N114) => Less a N115
 
@@ -673,3 +673,3 @@ instance Size N116 where size _ = 116
 instance Less N115 N116
-instance Less a N115 => Less a N116
+instance (Size a, Less a N115) => Less a N116
 
@@ -678,3 +678,3 @@ instance Size N117 where size _ = 117
 instance Less N116 N117
-instance Less a N116 => Less a N117
+instance (Size a, Less a N116) => Less a N117
 
@@ -683,3 +683,3 @@ instance Size N118 where size _ = 118
 instance Less N117 N118
-instance Less a N117 => Less a N118
+instance (Size a, Less a N117) => Less a N118
 
@@ -688,3 +688,3 @@ instance Size N119 where size _ = 119
 instance Less N118 N119
-instance Less a N118 => Less a N119
+instance (Size a, Less a N118) => Less a N119
 
@@ -693,3 +693,3 @@ instance Size N120 where size _ = 120
 instance Less N119 N120
-instance Less a N119 => Less a N120
+instance (Size a, Less a N119) => Less a N120
 
@@ -698,3 +698,3 @@ instance Size N121 where size _ = 121
 instance Less N120 N121
-instance Less a N120 => Less a N121
+instance (Size a, Less a N120) => Less a N121
 
@@ -703,3 +703,3 @@ instance Size N122 where size _ = 122
 instance Less N121 N122
-instance Less a N121 => Less a N122
+instance (Size a, Less a N121) => Less a N122
 
@@ -708,3 +708,3 @@ instance Size N123 where size _ = 123
 instance Less N122 N123
-instance Less a N122 => Less a N123
+instance (Size a, Less a N122) => Less a N123
 
@@ -713,3 +713,3 @@ instance Size N124 where size _ = 124
 instance Less N123 N124
-instance Less a N123 => Less a N124
+instance (Size a, Less a N123) => Less a N124
 
@@ -718,3 +718,3 @@ instance Size N125 where size _ = 125
 instance Less N124 N125
-instance Less a N124 => Less a N125
+instance (Size a, Less a N124) => Less a N125
 
@@ -723,3 +723,3 @@ instance Size N126 where size _ = 126
 instance Less N125 N126
-instance Less a N125 => Less a N126
+instance (Size a, Less a N125) => Less a N126
 
@@ -728,3 +728,3 @@ instance Size N127 where size _ = 127
 instance Less N126 N127
-instance Less a N126 => Less a N127
+instance (Size a, Less a N126) => Less a N127
 
@@ -733,3 +733,3 @@ instance Size N128 where size _ = 128
 instance Less N127 N128
-instance Less a N127 => Less a N128
+instance (Size a, Less a N127) => Less a N128
 
@@ -738,3 +738,3 @@ instance Size N129 where size _ = 129
 instance Less N128 N129
-instance Less a N128 => Less a N129
+instance (Size a, Less a N128) => Less a N129
 
@@ -743,3 +743,3 @@ instance Size N130 where size _ = 130
 instance Less N129 N130
-instance Less a N129 => Less a N130
+instance (Size a, Less a N129) => Less a N130
 
@@ -748,3 +748,3 @@ instance Size N131 where size _ = 131
 instance Less N130 N131
-instance Less a N130 => Less a N131
+instance (Size a, Less a N130) => Less a N131
 
@@ -753,3 +753,3 @@ instance Size N132 where size _ = 132
 instance Less N131 N132
-instance Less a N131 => Less a N132
+instance (Size a, Less a N131) => Less a N132
 
@@ -758,3 +758,3 @@ instance Size N133 where size _ = 133
 instance Less N132 N133
-instance Less a N132 => Less a N133
+instance (Size a, Less a N132) => Less a N133
 
@@ -763,3 +763,3 @@ instance Size N134 where size _ = 134
 instance Less N133 N134
-instance Less a N133 => Less a N134
+instance (Size a, Less a N133) => Less a N134
 
@@ -768,3 +768,3 @@ instance Size N135 where size _ = 135
 instance Less N134 N135
-instance Less a N134 => Less a N135
+instance (Size a, Less a N134) => Less a N135
 
@@ -773,3 +773,3 @@ instance Size N136 where size _ = 136
 instance Less N135 N136
-instance Less a N135 => Less a N136
+instance (Size a, Less a N135) => Less a N136
 
@@ -778,3 +778,3 @@ instance Size N137 where size _ = 137
 instance Less N136 N137
-instance Less a N136 => Less a N137
+instance (Size a, Less a N136) => Less a N137
 
@@ -783,3 +783,3 @@ instance Size N138 where size _ = 138
 instance Less N137 N138
-instance Less a N137 => Less a N138
+instance (Size a, Less a N137) => Less a N138
 
@@ -788,3 +788,3 @@ instance Size N139 where size _ = 139
 instance Less N138 N139
-instance Less a N138 => Less a N139
+instance (Size a, Less a N138) => Less a N139
 
@@ -793,3 +793,3 @@ instance Size N140 where size _ = 140
 instance Less N139 N140
-instance Less a N139 => Less a N140
+instance (Size a, Less a N139) => Less a N140
 
@@ -798,3 +798,3 @@ instance Size N141 where size _ = 141
 instance Less N140 N141
-instance Less a N140 => Less a N141
+instance (Size a, Less a N140) => Less a N141
 
@@ -803,3 +803,3 @@ instance Size N142 where size _ = 142
 instance Less N141 N142
-instance Less a N141 => Less a N142
+instance (Size a, Less a N141) => Less a N142
 
@@ -808,3 +808,3 @@ instance Size N143 where size _ = 143
 instance Less N142 N143
-instance Less a N142 => Less a N143
+instance (Size a, Less a N142) => Less a N143
 
@@ -813,3 +813,3 @@ instance Size N144 where size _ = 144
 instance Less N143 N144
-instance Less a N143 => Less a N144
+instance (Size a, Less a N143) => Less a N144
 
@@ -818,3 +818,3 @@ instance Size N145 where size _ = 145
 instance Less N144 N145
-instance Less a N144 => Less a N145
+instance (Size a, Less a N144) => Less a N145
 
@@ -823,3 +823,3 @@ instance Size N146 where size _ = 146
 instance Less N145 N146
-instance Less a N145 => Less a N146
+instance (Size a, Less a N145) => Less a N146
 
@@ -828,3 +828,3 @@ instance Size N147 where size _ = 147
 instance Less N146 N147
-instance Less a N146 => Less a N147
+instance (Size a, Less a N146) => Less a N147
 
@@ -833,3 +833,3 @@ instance Size N148 where size _ = 148
 instance Less N147 N148
-instance Less a N147 => Less a N148
+instance (Size a, Less a N147) => Less a N148
 
@@ -838,3 +838,3 @@ instance Size N149 where size _ = 149
 instance Less N148 N149
-instance Less a N148 => Less a N149
+instance (Size a, Less a N148) => Less a N149
 
@@ -843,3 +843,3 @@ instance Size N150 where size _ = 150
 instance Less N149 N150
-instance Less a N149 => Less a N150
+instance (Size a, Less a N149) => Less a N150
 
@@ -848,3 +848,3 @@ instance Size N151 where size _ = 151
 instance Less N150 N151
-instance Less a N150 => Less a N151
+instance (Size a, Less a N150) => Less a N151
 
@@ -853,3 +853,3 @@ instance Size N152 where size _ = 152
 instance Less N151 N152
-instance Less a N151 => Less a N152
+instance (Size a, Less a N151) => Less a N152
 
@@ -858,3 +858,3 @@ instance Size N153 where size _ = 153
 instance Less N152 N153
-instance Less a N152 => Less a N153
+instance (Size a, Less a N152) => Less a N153
 
@@ -863,3 +863,3 @@ instance Size N154 where size _ = 154
 instance Less N153 N154
-instance Less a N153 => Less a N154
+instance (Size a, Less a N153) => Less a N154
 
@@ -868,3 +868,3 @@ instance Size N155 where size _ = 155
 instance Less N154 N155
-instance Less a N154 => Less a N155
+instance (Size a, Less a N154) => Less a N155
 
@@ -873,3 +873,3 @@ instance Size N156 where size _ = 156
 instance Less N155 N156
-instance Less a N155 => Less a N156
+instance (Size a, Less a N155) => Less a N156
 
@@ -878,3 +878,3 @@ instance Size N157 where size _ = 157
 instance Less N156 N157
-instance Less a N156 => Less a N157
+instance (Size a, Less a N156) => Less a N157
 
@@ -883,3 +883,3 @@ instance Size N158 where size _ = 158
 instance Less N157 N158
-instance Less a N157 => Less a N158
+instance (Size a, Less a N157) => Less a N158
 
@@ -888,3 +888,3 @@ instance Size N159 where size _ = 159
 instance Less N158 N159
-instance Less a N158 => Less a N159
+instance (Size a, Less a N158) => Less a N159
 
@@ -893,3 +893,3 @@ instance Size N160 where size _ = 160
 instance Less N159 N160
-instance Less a N159 => Less a N160
+instance (Size a, Less a N159) => Less a N160
 
@@ -898,3 +898,3 @@ instance Size N161 where size _ = 161
 instance Less N160 N161
-instance Less a N160 => Less a N161
+instance (Size a, Less a N160) => Less a N161
 
@@ -903,3 +903,3 @@ instance Size N162 where size _ = 162
 instance Less N161 N162
-instance Less a N161 => Less a N162
+instance (Size a, Less a N161) => Less a N162
 
@@ -908,3 +908,3 @@ instance Size N163 where size _ = 163
 instance Less N162 N163
-instance Less a N162 => Less a N163
+instance (Size a, Less a N162) => Less a N163
 
@@ -913,3 +913,3 @@ instance Size N164 where size _ = 164
 instance Less N163 N164
-instance Less a N163 => Less a N164
+instance (Size a, Less a N163) => Less a N164
 
@@ -918,3 +918,3 @@ instance Size N165 where size _ = 165
 instance Less N164 N165
-instance Less a N164 => Less a N165
+instance (Size a, Less a N164) => Less a N165
 
@@ -923,3 +923,3 @@ instance Size N166 where size _ = 166
 instance Less N165 N166
-instance Less a N165 => Less a N166
+instance (Size a, Less a N165) => Less a N166
 
@@ -928,3 +928,3 @@ instance Size N167 where size _ = 167
 instance Less N166 N167
-instance Less a N166 => Less a N167
+instance (Size a, Less a N166) => Less a N167
 
@@ -933,3 +933,3 @@ instance Size N168 where size _ = 168
 instance Less N167 N168
-instance Less a N167 => Less a N168
+instance (Size a, Less a N167) => Less a N168
 
@@ -938,3 +938,3 @@ instance Size N169 where size _ = 169
 instance Less N168 N169
-instance Less a N168 => Less a N169
+instance (Size a, Less a N168) => Less a N169
 
@@ -943,3 +943,3 @@ instance Size N170 where size _ = 170
 instance Less N169 N170
-instance Less a N169 => Less a N170
+instance (Size a, Less a N169) => Less a N170
 
@@ -948,3 +948,3 @@ instance Size N171 where size _ = 171
 instance Less N170 N171
-instance Less a N170 => Less a N171
+instance (Size a, Less a N170) => Less a N171
 
@@ -953,3 +953,3 @@ instance Size N172 where size _ = 172
 instance Less N171 N172
-instance Less a N171 => Less a N172
+instance (Size a, Less a N171) => Less a N172
 
@@ -958,3 +958,3 @@ instance Size N173 where size _ = 173
 instance Less N172 N173
-instance Less a N172 => Less a N173
+instance (Size a, Less a N172) => Less a N173
 
@@ -963,3 +963,3 @@ instance Size N174 where size _ = 174
 instance Less N173 N174
-instance Less a N173 => Less a N174
+instance (Size a, Less a N173) => Less a N174
 
@@ -968,3 +968,3 @@ instance Size N175 where size _ = 175
 instance Less N174 N175
-instance Less a N174 => Less a N175
+instance (Size a, Less a N174) => Less a N175
 
@@ -973,3 +973,3 @@ instance Size N176 where size _ = 176
 instance Less N175 N176
-instance Less a N175 => Less a N176
+instance (Size a, Less a N175) => Less a N176
 
@@ -978,3 +978,3 @@ instance Size N177 where size _ = 177
 instance Less N176 N177
-instance Less a N176 => Less a N177
+instance (Size a, Less a N176) => Less a N177
 
@@ -983,3 +983,3 @@ instance Size N178 where size _ = 178
 instance Less N177 N178
-instance Less a N177 => Less a N178
+instance (Size a, Less a N177) => Less a N178
 
@@ -988,3 +988,3 @@ instance Size N179 where size _ = 179
 instance Less N178 N179
-instance Less a N178 => Less a N179
+instance (Size a, Less a N178) => Less a N179
 
@@ -993,3 +993,3 @@ instance Size N180 where size _ = 180
 instance Less N179 N180
-instance Less a N179 => Less a N180
+instance (Size a, Less a N179) => Less a N180
 
@@ -998,3 +998,3 @@ instance Size N181 where size _ = 181
 instance Less N180 N181
-instance Less a N180 => Less a N181
+instance (Size a, Less a N180) => Less a N181
 
@@ -1003,3 +1003,3 @@ instance Size N182 where size _ = 182
 instance Less N181 N182
-instance Less a N181 => Less a N182
+instance (Size a, Less a N181) => Less a N182
 
@@ -1008,3 +1008,3 @@ instance Size N183 where size _ = 183
 instance Less N182 N183
-instance Less a N182 => Less a N183
+instance (Size a, Less a N182) => Less a N183
 
@@ -1013,3 +1013,3 @@ instance Size N184 where size _ = 184
 instance Less N183 N184
-instance Less a N183 => Less a N184
+instance (Size a, Less a N183) => Less a N184
 
@@ -1018,3 +1018,3 @@ instance Size N185 where size _ = 185
 instance Less N184 N185
-instance Less a N184 => Less a N185
+instance (Size a, Less a N184) => Less a N185
 
@@ -1023,3 +1023,3 @@ instance Size N186 where size _ = 186
 instance Less N185 N186
-instance Less a N185 => Less a N186
+instance (Size a, Less a N185) => Less a N186
 
@@ -1028,3 +1028,3 @@ instance Size N187 where size _ = 187
 instance Less N186 N187
-instance Less a N186 => Less a N187
+instance (Size a, Less a N186) => Less a N187
 
@@ -1033,3 +1033,3 @@ instance Size N188 where size _ = 188
 instance Less N187 N188
-instance Less a N187 => Less a N188
+instance (Size a, Less a N187) => Less a N188
 
@@ -1038,3 +1038,3 @@ instance Size N189 where size _ = 189
 instance Less N188 N189
-instance Less a N188 => Less a N189
+instance (Size a, Less a N188) => Less a N189
 
@@ -1043,3 +1043,3 @@ instance Size N190 where size _ = 190
 instance Less N189 N190
-instance Less a N189 => Less a N190
+instance (Size a, Less a N189) => Less a N190
 
@@ -1048,3 +1048,3 @@ instance Size N191 where size _ = 191
 instance Less N190 N191
-instance Less a N190 => Less a N191
+instance (Size a, Less a N190) => Less a N191
 
@@ -1053,3 +1053,3 @@ instance Size N192 where size _ = 192
 instance Less N191 N192
-instance Less a N191 => Less a N192
+instance (Size a, Less a N191) => Less a N192
 
@@ -1058,3 +1058,3 @@ instance Size N193 where size _ = 193
 instance Less N192 N193
-instance Less a N192 => Less a N193
+instance (Size a, Less a N192) => Less a N193
 
@@ -1063,3 +1063,3 @@ instance Size N194 where size _ = 194
 instance Less N193 N194
-instance Less a N193 => Less a N194
+instance (Size a, Less a N193) => Less a N194
 
@@ -1068,3 +1068,3 @@ instance Size N195 where size _ = 195
 instance Less N194 N195
-instance Less a N194 => Less a N195
+instance (Size a, Less a N194) => Less a N195
 
@@ -1073,3 +1073,3 @@ instance Size N196 where size _ = 196
 instance Less N195 N196
-instance Less a N195 => Less a N196
+instance (Size a, Less a N195) => Less a N196
 
@@ -1078,3 +1078,3 @@ instance Size N197 where size _ = 197
 instance Less N196 N197
-instance Less a N196 => Less a N197
+instance (Size a, Less a N196) => Less a N197
 
@@ -1083,3 +1083,3 @@ instance Size N198 where size _ = 198
 instance Less N197 N198
-instance Less a N197 => Less a N198
+instance (Size a, Less a N197) => Less a N198
 
@@ -1088,3 +1088,3 @@ instance Size N199 where size _ = 199
 instance Less N198 N199
-instance Less a N198 => Less a N199
+instance (Size a, Less a N198) => Less a N199
 
@@ -1093,3 +1093,3 @@ instance Size N200 where size _ = 200
 instance Less N199 N200
-instance Less a N199 => Less a N200
+instance (Size a, Less a N199) => Less a N200
 
@@ -1098,3 +1098,3 @@ instance Size N201 where size _ = 201
 instance Less N200 N201
-instance Less a N200 => Less a N201
+instance (Size a, Less a N200) => Less a N201
 
@@ -1103,3 +1103,3 @@ instance Size N202 where size _ = 202
 instance Less N201 N202
-instance Less a N201 => Less a N202
+instance (Size a, Less a N201) => Less a N202
 
@@ -1108,3 +1108,3 @@ instance Size N203 where size _ = 203
 instance Less N202 N203
-instance Less a N202 => Less a N203
+instance (Size a, Less a N202) => Less a N203
 
@@ -1113,3 +1113,3 @@ instance Size N204 where size _ = 204
 instance Less N203 N204
-instance Less a N203 => Less a N204
+instance (Size a, Less a N203) => Less a N204
 
@@ -1118,3 +1118,3 @@ instance Size N205 where size _ = 205
 instance Less N204 N205
-instance Less a N204 => Less a N205
+instance (Size a, Less a N204) => Less a N205
 
@@ -1123,3 +1123,3 @@ instance Size N206 where size _ = 206
 instance Less N205 N206
-instance Less a N205 => Less a N206
+instance (Size a, Less a N205) => Less a N206
 
@@ -1128,3 +1128,3 @@ instance Size N207 where size _ = 207
 instance Less N206 N207
-instance Less a N206 => Less a N207
+instance (Size a, Less a N206) => Less a N207
 
@@ -1133,3 +1133,3 @@ instance Size N208 where size _ = 208
 instance Less N207 N208
-instance Less a N207 => Less a N208
+instance (Size a, Less a N207) => Less a N208
 
@@ -1138,3 +1138,3 @@ instance Size N209 where size _ = 209
 instance Less N208 N209
-instance Less a N208 => Less a N209
+instance (Size a, Less a N208) => Less a N209
 
@@ -1143,3 +1143,3 @@ instance Size N210 where size _ = 210
 instance Less N209 N210
-instance Less a N209 => Less a N210
+instance (Size a, Less a N209) => Less a N210
 
@@ -1148,3 +1148,3 @@ instance Size N211 where size _ = 211
 instance Less N210 N211
-instance Less a N210 => Less a N211
+instance (Size a, Less a N210) => Less a N211
 
@@ -1153,3 +1153,3 @@ instance Size N212 where size _ = 212
 instance Less N211 N212
-instance Less a N211 => Less a N212
+instance (Size a, Less a N211) => Less a N212
 
@@ -1158,3 +1158,3 @@ instance Size N213 where size _ = 213
 instance Less N212 N213
-instance Less a N212 => Less a N213
+instance (Size a, Less a N212) => Less a N213
 
@@ -1163,3 +1163,3 @@ instance Size N214 where size _ = 214
 instance Less N213 N214
-instance Less a N213 => Less a N214
+instance (Size a, Less a N213) => Less a N214
 
@@ -1168,3 +1168,3 @@ instance Size N215 where size _ = 215
 instance Less N214 N215
-instance Less a N214 => Less a N215
+instance (Size a, Less a N214) => Less a N215
 
@@ -1173,3 +1173,3 @@ instance Size N216 where size _ = 216
 instance Less N215 N216
-instance Less a N215 => Less a N216
+instance (Size a, Less a N215) => Less a N216
 
@@ -1178,3 +1178,3 @@ instance Size N217 where size _ = 217
 instance Less N216 N217
-instance Less a N216 => Less a N217
+instance (Size a, Less a N216) => Less a N217
 
@@ -1183,3 +1183,3 @@ instance Size N218 where size _ = 218
 instance Less N217 N218
-instance Less a N217 => Less a N218
+instance (Size a, Less a N217) => Less a N218
 
@@ -1188,3 +1188,3 @@ instance Size N219 where size _ = 219
 instance Less N218 N219
-instance Less a N218 => Less a N219
+instance (Size a, Less a N218) => Less a N219
 
@@ -1193,3 +1193,3 @@ instance Size N220 where size _ = 220
 instance Less N219 N220
-instance Less a N219 => Less a N220
+instance (Size a, Less a N219) => Less a N220
 
@@ -1198,3 +1198,3 @@ instance Size N221 where size _ = 221
 instance Less N220 N221
-instance Less a N220 => Less a N221
+instance (Size a, Less a N220) => Less a N221
 
@@ -1203,3 +1203,3 @@ instance Size N222 where size _ = 222
 instance Less N221 N222
-instance Less a N221 => Less a N222
+instance (Size a, Less a N221) => Less a N222
 
@@ -1208,3 +1208,3 @@ instance Size N223 where size _ = 223
 instance Less N222 N223
-instance Less a N222 => Less a N223
+instance (Size a, Less a N222) => Less a N223
 
@@ -1213,3 +1213,3 @@ instance Size N224 where size _ = 224
 instance Less N223 N224
-instance Less a N223 => Less a N224
+instance (Size a, Less a N223) => Less a N224
 
@@ -1218,3 +1218,3 @@ instance Size N225 where size _ = 225
 instance Less N224 N225
-instance Less a N224 => Less a N225
+instance (Size a, Less a N224) => Less a N225
 
@@ -1223,3 +1223,3 @@ instance Size N226 where size _ = 226
 instance Less N225 N226
-instance Less a N225 => Less a N226
+instance (Size a, Less a N225) => Less a N226
 
@@ -1228,3 +1228,3 @@ instance Size N227 where size _ = 227
 instance Less N226 N227
-instance Less a N226 => Less a N227
+instance (Size a, Less a N226) => Less a N227
 
@@ -1233,3 +1233,3 @@ instance Size N228 where size _ = 228
 instance Less N227 N228
-instance Less a N227 => Less a N228
+instance (Size a, Less a N227) => Less a N228
 
@@ -1238,3 +1238,3 @@ instance Size N229 where size _ = 229
 instance Less N228 N229
-instance Less a N228 => Less a N229
+instance (Size a, Less a N228) => Less a N229
 
@@ -1243,3 +1243,3 @@ instance Size N230 where size _ = 230
 instance Less N229 N230
-instance Less a N229 => Less a N230
+instance (Size a, Less a N229) => Less a N230
 
@@ -1248,3 +1248,3 @@ instance Size N231 where size _ = 231
 instance Less N230 N231
-instance Less a N230 => Less a N231
+instance (Size a, Less a N230) => Less a N231
 
@@ -1253,3 +1253,3 @@ instance Size N232 where size _ = 232
 instance Less N231 N232
-instance Less a N231 => Less a N232
+instance (Size a, Less a N231) => Less a N232
 
@@ -1258,3 +1258,3 @@ instance Size N233 where size _ = 233
 instance Less N232 N233
-instance Less a N232 => Less a N233
+instance (Size a, Less a N232) => Less a N233
 
@@ -1263,3 +1263,3 @@ instance Size N234 where size _ = 234
 instance Less N233 N234
-instance Less a N233 => Less a N234
+instance (Size a, Less a N233) => Less a N234
 
@@ -1268,3 +1268,3 @@ instance Size N235 where size _ = 235
 instance Less N234 N235
-instance Less a N234 => Less a N235
+instance (Size a, Less a N234) => Less a N235
 
@@ -1273,3 +1273,3 @@ instance Size N236 where size _ = 236
 instance Less N235 N236
-instance Less a N235 => Less a N236
+instance (Size a, Less a N235) => Less a N236
 
@@ -1278,3 +1278,3 @@ instance Size N237 where size _ = 237
 instance Less N236 N237
-instance Less a N236 => Less a N237
+instance (Size a, Less a N236) => Less a N237
 
@@ -1283,3 +1283,3 @@ instance Size N238 where size _ = 238
 instance Less N237 N238
-instance Less a N237 => Less a N238
+instance (Size a, Less a N237) => Less a N238
 
@@ -1288,3 +1288,3 @@ instance Size N239 where size _ = 239
 instance Less N238 N239
-instance Less a N238 => Less a N239
+instance (Size a, Less a N238) => Less a N239
 
@@ -1293,3 +1293,3 @@ instance Size N240 where size _ = 240
 instance Less N239 N240
-instance Less a N239 => Less a N240
+instance (Size a, Less a N239) => Less a N240
 
@@ -1298,3 +1298,3 @@ instance Size N241 where size _ = 241
 instance Less N240 N241
-instance Less a N240 => Less a N241
+instance (Size a, Less a N240) => Less a N241
 
@@ -1303,3 +1303,3 @@ instance Size N242 where size _ = 242
 instance Less N241 N242
-instance Less a N241 => Less a N242
+instance (Size a, Less a N241) => Less a N242
 
@@ -1308,3 +1308,3 @@ instance Size N243 where size _ = 243
 instance Less N242 N243
-instance Less a N242 => Less a N243
+instance (Size a, Less a N242) => Less a N243
 
@@ -1313,3 +1313,3 @@ instance Size N244 where size _ = 244
 instance Less N243 N244
-instance Less a N243 => Less a N244
+instance (Size a, Less a N243) => Less a N244
 
@@ -1318,3 +1318,3 @@ instance Size N245 where size _ = 245
 instance Less N244 N245
-instance Less a N244 => Less a N245
+instance (Size a, Less a N244) => Less a N245
 
@@ -1323,3 +1323,3 @@ instance Size N246 where size _ = 246
 instance Less N245 N246
-instance Less a N245 => Less a N246
+instance (Size a, Less a N245) => Less a N246
 
@@ -1328,3 +1328,3 @@ instance Size N247 where size _ = 247
 instance Less N246 N247
-instance Less a N246 => Less a N247
+instance (Size a, Less a N246) => Less a N247
 
@@ -1333,3 +1333,3 @@ instance Size N248 where size _ = 248
 instance Less N247 N248
-instance Less a N247 => Less a N248
+instance (Size a, Less a N247) => Less a N248
 
@@ -1338,3 +1338,3 @@ instance Size N249 where size _ = 249
 instance Less N248 N249
-instance Less a N248 => Less a N249
+instance (Size a, Less a N248) => Less a N249
 
@@ -1343,3 +1343,3 @@ instance Size N250 where size _ = 250
 instance Less N249 N250
-instance Less a N249 => Less a N250
+instance (Size a, Less a N249) => Less a N250
 
@@ -1348,3 +1348,3 @@ instance Size N251 where size _ = 251
 instance Less N250 N251
-instance Less a N250 => Less a N251
+instance (Size a, Less a N250) => Less a N251
 
@@ -1353,3 +1353,3 @@ instance Size N252 where size _ = 252
 instance Less N251 N252
-instance Less a N251 => Less a N252
+instance (Size a, Less a N251) => Less a N252
 
@@ -1358,3 +1358,3 @@ instance Size N253 where size _ = 253
 instance Less N252 N253
-instance Less a N252 => Less a N253
+instance (Size a, Less a N252) => Less a N253
 
@@ -1363,3 +1363,3 @@ instance Size N254 where size _ = 254
 instance Less N253 N254
-instance Less a N253 => Less a N254
+instance (Size a, Less a N253) => Less a N254
 
@@ -1368,3 +1368,3 @@ instance Size N255 where size _ = 255
 instance Less N254 N255
-instance Less a N254 => Less a N255
+instance (Size a, Less a N254) => Less a N255
 
@@ -1373,3 +1373,3 @@ instance Size N65535 where size _ = 65535
 instance Less N255 N65535
-instance Less a N255 => Less a N65535
+instance (Size a, Less a N255) => Less a N65535
 
