From e0465512092b3ea3ba1f84fef199731ae7c3d6c2 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sun, 29 May 2011 12:19:38 +0300
Subject: [PATCH] updated up to latest haskell-src-exts-1.11.*

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 Main.hs           |    2 +-
 hothasktags.cabal |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Main.hs b/Main.hs
index fc3c5d6..0d54e2e 100644
--- a/Main.hs
+++ b/Main.hs
@@ -185,7 +185,7 @@ makeDatabase files = do
         L.extensions = [L.MultiParamTypeClasses, L.ExistentialQuantification, L.FlexibleContexts],
         L.ignoreLanguagePragmas = False,
         L.ignoreLinePragmas = False,
-        L.fixities = []
+        L.fixities = Just []
       }
 
 moduleFile :: L.Module L.SrcSpanInfo -> FilePath
diff --git a/hothasktags.cabal b/hothasktags.cabal
index e3bc69e..98ab31c 100644
--- a/hothasktags.cabal
+++ b/hothasktags.cabal
@@ -30,7 +30,7 @@ executable hothasktags
     build-depends: 
         base == 4.*,
         containers,
-        haskell-src-exts >= 1.8 && < 1.11,
+        haskell-src-exts == 1.11.*,
         cpphs == 1.11.*
     main-is: Main.hs
     ghc-options: -W
-- 
1.7.3.4

