--- logging-effect-1.2.5-orig/src/Control/Monad/Log.hs	2018-03-22 03:25:28.000000000 +1100
+++ logging-effect-1.2.5/src/Control/Monad/Log.hs	2018-04-12 20:36:43.428987584 +1000
@@ -92,6 +92,7 @@
 import Control.Monad.Trans.State.Strict (StateT(..))
 import Control.Monad.Writer.Class (MonadWriter(..))
 import Data.Monoid
+import Data.Semigroup ()
 import Data.Time (UTCTime, getCurrentTime)
 #if !MIN_VERSION_base(4, 9, 0)
 import GHC.SrcLoc (SrcLoc, showSrcLoc)
@@ -399,11 +400,13 @@
 
 newtype Ap m = Ap { runAp :: m () }
 
+instance Applicative m => Semigroup (Ap m) where
+  Ap l <> Ap r = Ap (l *> r)
+  {-# INLINEABLE (<>) #-}
+
 instance Applicative m => Monoid (Ap m) where
   mempty = Ap (pure ())
   {-# INLINEABLE mempty #-}
-  Ap l `mappend` Ap r = Ap (l *> r)
-  {-# INLINEABLE mappend #-}
 
 -- | The main instance of 'MonadLog', which replaces calls to 'logMessage' with calls to a 'Handler'.
 instance Monad m => MonadLog message (LoggingT message m) where
