--- logging-effect-1.2.5-orig/src/Control/Monad/Log.hs	2018-03-22 03:25:28.000000000 +1100
+++ logging-effect-1.2.5/src/Control/Monad/Log.hs	2018-04-15 22:20:19.635495657 +1000
@@ -92,6 +92,7 @@
 import Control.Monad.Trans.State.Strict (StateT(..))
 import Control.Monad.Writer.Class (MonadWriter(..))
 import Data.Monoid
+import qualified Data.Semigroup as S (Semigroup(..), (<>))
 import Data.Time (UTCTime, getCurrentTime)
 #if !MIN_VERSION_base(4, 9, 0)
 import GHC.SrcLoc (SrcLoc, showSrcLoc)
@@ -399,10 +400,14 @@
 
 newtype Ap m = Ap { runAp :: m () }
 
+instance Applicative m => S.Semigroup (Ap m) where
+  Ap l <> Ap r = Ap (l *> r)
+  {-# INLINEABLE (<>) #-}
+
 instance Applicative m => Monoid (Ap m) where
   mempty = Ap (pure ())
   {-# INLINEABLE mempty #-}
-  Ap l `mappend` Ap r = Ap (l *> r)
+  mappend = (S.<>)
   {-# INLINEABLE mappend #-}
 
 -- | The main instance of 'MonadLog', which replaces calls to 'logMessage' with calls to a 'Handler'.
