--- mvc-1.1.6-orig/mvc.cabal	2017-12-01 15:03:06.000000000 +1100
+++ mvc-1.1.6/mvc.cabal	2018-04-18 21:31:00.647834175 +1000
@@ -28,11 +28,11 @@
     Hs-Source-Dirs: src
     Build-Depends:
         base              >= 4       && < 5  ,
-        async             >= 2.0.0   && < 2.2,
+        async             >= 2.0.0   && < 2.3,
         contravariant                   < 1.5,
         foldl             >= 1.1     && < 1.4,
         managed                         < 1.1,
-        mmorph            >= 1.0.2   && < 1.1,
+        mmorph            >= 1.0.2   && < 1.2,
         pipes             >= 4.1.7   && < 4.4,
         pipes-concurrency >= 2.0.3   && < 2.1,
         transformers      >= 0.2.0.0 && < 0.6
