--- mvc-1.1.6-orig/mvc.cabal	2017-12-01 15:03:06.000000000 +1100
+++ mvc-1.1.6/mvc.cabal	2018-04-18 21:38:51.898635640 +1000
@@ -36,6 +36,8 @@
         pipes             >= 4.1.7   && < 4.4,
         pipes-concurrency >= 2.0.3   && < 2.1,
         transformers      >= 0.2.0.0 && < 0.6
+    if !impl(ghc >= 8.0)
+        Build-depends: semigroups == 0.18.*
     Exposed-Modules:
         MVC,
         MVC.Prelude
--- mvc-1.1.6-orig/src/MVC.hs	2017-12-01 15:03:06.000000000 +1100
+++ mvc-1.1.6/src/MVC.hs	2018-04-18 20:56:31.229769412 +1000
@@ -115,6 +115,7 @@
 import Data.Functor.Contravariant (Contravariant(contramap))
 import Data.Monoid (Monoid(mempty, mappend, mconcat), (<>), First)
 import qualified Data.Monoid as M
+import qualified Data.Semigroup as S
 import Pipes
 import Pipes.Concurrent
 import Pipes.Prelude (foldM, loop)
@@ -161,9 +162,13 @@
 instance Functor Controller where
     fmap f (AsInput i) = AsInput (fmap f i)
 
+-- Deriving `Semigroup`
+instance S.Semigroup (Controller a) where
+    (AsInput i1) <> (AsInput i2) = AsInput (i1 S.<> i2)
+
 -- Deriving `Monoid`
 instance Monoid (Controller a) where
-    mappend (AsInput i1) (AsInput i2) = AsInput (mappend i1 i2)
+    mappend = (<>)
 
     mempty = AsInput mempty
 
@@ -254,9 +259,12 @@
 -}
 newtype View a = AsFold (FoldM IO a ())
 
+instance S.Semigroup (View a) where
+    (AsFold fold1) <> (AsFold fold2) = AsFold (fold1 S.<> fold2)
+
 instance Monoid (View a) where
     mempty = AsFold mempty
-    mappend (AsFold fold1) (AsFold fold2) = AsFold (mappend fold1 fold2)
+    mappend = (<>)
 
 instance Contravariant View where
     contramap f (AsFold fold) = AsFold (premapM f fold)
