diff --git a/Database/HSparql/QueryGenerator.hs b/Database/HSparql/QueryGenerator.hs
index 1bc022c..129e79e 100644
--- a/Database/HSparql/QueryGenerator.hs
+++ b/Database/HSparql/QueryGenerator.hs
@@ -351,3 +351,3 @@ notExpr = NegatedExpr . expr
 -- Builtin Functions
-type BuiltinFunc1 = (TermLike a) => a -> Expr
+type BuiltinFunc1 = forall a . (TermLike a) => a -> Expr
 builtinFunc1 :: Function -> BuiltinFunc1
@@ -355,3 +355,3 @@ builtinFunc1 f x = BuiltinCall f [expr x]
 
-type BuiltinFunc2 = (TermLike a, TermLike b) => a -> b -> Expr
+type BuiltinFunc2 = forall a b . (TermLike a, TermLike b) => a -> b -> Expr
 builtinFunc2 :: Function -> BuiltinFunc2
