diff --git a/ShowQ.hs b/ShowQ.hs
index a7663d9..fdb75d2 100644
--- a/ShowQ.hs
+++ b/ShowQ.hs
@@ -24,9 +24,9 @@ tests prop ntest stamps =
   do result <- quickCheckWithResult stdArgs prop
      case result of
        NoExpectedFailure _ _ _ -> done "Arguments exhausted after" (numTests result) stamps
-       GaveUp _ _ _ -> done "Arguments exhausted after" (numTests result) stamps
-       Success _ _ _  -> done "OK, passed" (numTests result) stamps
-       Failure _ _ _ _ _ _ _ -> return $ "Falsifiable, after "
+       GaveUp{} -> done "Arguments exhausted after" (numTests result) stamps
+       Success{}  -> done "OK, passed" (numTests result) stamps
+       Failure{} -> return $ "Falsifiable, after "
                                 ++ show ntest
                                 ++ " tests:\n"
                                 ++ reason result
