--- haskell-updater-1.2.0.5-orig/Distribution/Gentoo/PkgManager.hs	2012-02-12 07:08:49.000000000 +1100
+++ haskell-updater-1.2.0.5/Distribution/Gentoo/PkgManager.hs	2012-06-29 22:06:38.858341358 +1000
@@ -26,7 +26,7 @@
 import Data.Maybe(mapMaybe, fromMaybe)
 import qualified Data.Map as M
 import Data.Map(Map)
-import System.IO.Error(try)
+import Control.Exception(try, SomeException)
 import System.Environment(getEnv)
 
 -- -----------------------------------------------------------------------------
@@ -48,7 +48,7 @@
 --   "portage".  Note that even if that environment variable is
 --   defined, if it is unknown then it won't be used.
 defaultPM :: IO PkgManager
-defaultPM = do eDPM <- try $ getEnv "PACKAGE_MANAGER"
+defaultPM = do eDPM <- (try $ getEnv "PACKAGE_MANAGER") :: IO (Either SomeException String)
                let dPM = either (const defaultPMName) id eDPM
                    mPM = dPM `M.lookup` pmNameMap
                return $ fromMaybe knownDef mPM
