diff -rN -u old-GF/src/compiler/GF/System/Console.hs new-GF/src/compiler/GF/System/Console.hs
--- old-GF/src/compiler/GF/System/Console.hs	2013-03-10 00:50:45.261666860 +0300
+++ new-GF/src/compiler/GF/System/Console.hs	2013-03-10 00:50:45.712666883 +0300
@@ -11,6 +11,9 @@
     do codepage <- getACP
        setCP codepage
        setEncoding ("CP"++show codepage)
+#else
+    do -- warnings actually contain utf-8 symbols, so we have no choice
+       hSetEncoding stderr utf8
 #endif
        return () :: IO ()
 
