diff --git a/src/compiler/GF.hs b/src/compiler/GF.hs
index 052e02c..37fb86f 100644
--- a/src/compiler/GF.hs
+++ b/src/compiler/GF.hs
@@ -28,6 +28,9 @@ main = do
   hSetEncoding stdin  enc
   hSetEncoding stdout enc
   hSetEncoding stderr enc
+#else
+  -- warnings actually contain utf-8 symbols, so we have no choice
+  hSetEncoding stderr utf8
 #endif
   args <- getArgs
   case parseOptions args of
