--- bytestring-nums-0.3.2-orig/Data/ByteString/Nums/Careless/Hex.hs	2010-06-01 10:13:08.000000000 +1000
+++ bytestring-nums-0.3.2/Data/ByteString/Nums/Careless/Hex.hs	2011-07-27 19:11:37.315526433 +1000
@@ -2,6 +2,7 @@
 
 {-# LANGUAGE MultiParamTypeClasses
            , TypeSynonymInstances
+           , FlexibleInstances
   #-}
 
 
--- bytestring-nums-0.3.2-orig/Data/ByteString/Nums/Careless/Int.hs	2010-06-01 10:13:08.000000000 +1000
+++ bytestring-nums-0.3.2/Data/ByteString/Nums/Careless/Int.hs	2011-07-27 19:12:02.681308150 +1000
@@ -2,6 +2,7 @@
 
 {-# LANGUAGE MultiParamTypeClasses
            , TypeSynonymInstances
+           , FlexibleInstances
   #-}
 
 
--- bytestring-nums-0.3.2-orig/Data/ByteString/Nums/Careless/Float.hs	2010-06-01 10:13:08.000000000 +1000
+++ bytestring-nums-0.3.2/Data/ByteString/Nums/Careless/Float.hs	2011-07-27 19:13:19.459674616 +1000
@@ -2,6 +2,7 @@
 
 {-# LANGUAGE MultiParamTypeClasses
            , TypeSynonymInstances
+           , FlexibleInstances
   #-}
 
 
