--- a/Distribution/Superdoc/Hooks.hs
+++ b/Distribution/Superdoc/Hooks.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 -- | This module defines Superdoc's parameters, as well as the Cabal
@@ -154,3 +155,7 @@ posthaddock sdargs args flags desc info = do
   -- it fails to run the postHscolour hook. So we run it explicitly.  
+#if MIN_VERSION_Cabal(2,2,0)
+  when (haddockLinkedSource flags == Flag True) $ do
+#else
   when (haddockHscolour flags == Flag True) $ do
+#endif
     do_posthscolour docdir
