From 10cfecc9ab7292f65692d5c6ad7900dfe29f1568 Mon Sep 17 00:00:00 2001
From: Jack Todaro <jackmtodaro@gmail.com>
Date: Sun, 5 Aug 2018 23:03:40 +1000
Subject: [PATCH] GHC 8.4 compatibility

---
 src/UU/PPrint.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UU/PPrint.hs b/src/UU/PPrint.hs
index f7d02ab..0d7d92e 100644
--- a/src/UU/PPrint.hs
+++ b/src/UU/PPrint.hs
@@ -60,7 +60,7 @@ module UU.PPrint
 import System.IO      (Handle,hPutStr,hPutChar,stdout)
 
 #if __GLASGOW_HASKELL__ >= 710
-import Prelude hiding ((<$>))
+import Prelude hiding ((<$>),(<>))
 #endif
 
 infixr 5 </>,<//>,<$>,<$$>
-- 
2.18.0

