--- miniplex-0.3.4-orig/miniplex.cabal	2009-03-07 04:13:09.000000000 +1100
+++ miniplex-0.3.4/miniplex.cabal	2011-08-12 21:53:37.954058753 +1000
@@ -23,7 +23,7 @@
   hs-source-dirs: lib
   if flag(small_base)
     build-depends: directory
-  build-depends: base <4, network, stm, unix, mtl, colock >=0.2.2
+  build-depends: base >=4 && <5, network, stm, unix, mtl, colock >=0.2.2
   ghc-options: -O2 -Wall
 
 executable plox-read
--- miniplex-0.3.4-orig/lib/System/Miniplex/Sekrit.hs	2009-03-07 04:13:09.000000000 +1100
+++ miniplex-0.3.4/lib/System/Miniplex/Sekrit.hs	2011-08-12 21:54:17.414060393 +1000
@@ -10,7 +10,7 @@
     , mode644
 ) where
 
-import Control.Exception
+import Control.OldException
 import Control.Monad
 
 import Foreign.C.Error
--- miniplex-0.3.4-orig/lib/System/Miniplex/Sink.hs	2009-03-07 04:13:09.000000000 +1100
+++ miniplex-0.3.4/lib/System/Miniplex/Sink.hs	2011-08-12 21:57:19.447679152 +1000
@@ -1,4 +1,4 @@
-{-# OPTIONS_GHC -fglasgow-exts #-}
+{-# LANGUAGE DeriveDataTypeable #-}
 
 -- | Asynchronous message broadcasting.
 module System.Miniplex.Sink (
@@ -22,7 +22,7 @@
 import Data.List
 import Control.Concurrent
 import Control.Concurrent.STM
-import Control.Exception
+import Control.OldException
 import Control.Monad
 import Control.Monad.Fix
 import System.Directory
--- miniplex-0.3.4-orig/lib/System/Miniplex/Source.hs	2009-03-07 04:13:09.000000000 +1100
+++ miniplex-0.3.4/lib/System/Miniplex/Source.hs	2011-08-12 21:58:57.108155854 +1000
@@ -1,4 +1,4 @@
-{-# OPTIONS_GHC -fglasgow-exts #-}
+{-# LANGUAGE DeriveDataTypeable #-}
 
 -- | Synchronous message receiving.
 module System.Miniplex.Source (
@@ -17,7 +17,7 @@
 import Prelude hiding (read, catch)
 
 import Control.Concurrent
-import Control.Exception
+import Control.OldException
 import Control.Monad
 import Control.Monad.Fix
 import Control.Monad.Reader ()
