diff --git a/Distribution/Client/GZipUtils.hs b/Distribution/Client/GZipUtils.hs
index ce1b7e4..56313ae 100644
--- a/Distribution/Client/GZipUtils.hs
+++ b/Distribution/Client/GZipUtils.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 -----------------------------------------------------------------------------
@@ -32,3 +33,7 @@ import Codec.Compression.Zlib.Internal
 maybeDecompress :: ByteString -> ByteString
-maybeDecompress bytes = foldStream $ decompressWithErrors gzipOrZlibFormat defaultDecompressParams bytes
+maybeDecompress bytes =
+#if MIN_VERSION_zlib(0,6,0)
+    Codec.Compression.GZip.decompress bytes
+#else
+    foldStream $ decompressWithErrors gzipOrZlibFormat defaultDecompressParams bytes
   where
@@ -44 +49,2 @@ maybeDecompress bytes = foldStream $ decompressWithErrors gzipOrZlibFormat defau
     doFold (StreamError _ msg)  = error $ "Codec.Compression.Zlib: " ++ msg
+#endif
diff --git a/cabal-install.cabal b/cabal-install.cabal
index 5c121f8..b34e50d 100644
--- a/cabal-install.cabal
+++ b/cabal-install.cabal
@@ -140,3 +140,3 @@ executable cabal
         time       >= 1.1      && < 1.6,
-        zlib       >= 0.5.3    && < 0.6
+        zlib       >= 0.5.3    && < 0.7
 
