--- unordered-containers-0.1.4.0-orig/unordered-containers.cabal	2011-06-24 07:42:14.000000000 +1000
+++ unordered-containers-0.1.4.0/unordered-containers.cabal	2011-07-07 19:50:45.974541707 +1000
@@ -62,8 +62,8 @@
     base,
     hashable >= 1.0.1.1 && < 1.2,
     QuickCheck >= 2.4.0.1,
-    test-framework >= 0.3.3 && < 0.4,
-    test-framework-quickcheck2 >= 0.2.9 && < 0.3,
+    test-framework >= 0.3.3,
+    test-framework-quickcheck2 >= 0.2.9,
     unordered-containers
 
   ghc-options: -Wall
@@ -79,8 +79,8 @@
     containers,
     hashable >= 1.0.1.1 && < 1.2,
     QuickCheck >= 2.4.0.1,
-    test-framework >= 0.3.3 && < 0.4,
-    test-framework-quickcheck2 >= 0.2.9 && < 0.3,
+    test-framework >= 0.3.3,
+    test-framework-quickcheck2 >= 0.2.9,
     unordered-containers
 
   ghc-options: -Wall
