diff --git a/SourceGraph.cabal b/SourceGraph.cabal
index 781d625..c95f7d0 100644
--- a/SourceGraph.cabal
+++ b/SourceGraph.cabal
@@ -77,6 +77,6 @@ Executable SourceGraph {
                         fgl >= 5.4.2.3 && < 5.5,
                         Graphalyze == 0.13.*,
                         graphviz >= 2999.14.1.0 && < 2999.15,
-                        Cabal >= 1.8 && < 1.15,
+                        Cabal >= 1.8 && < 1.17,
                         haskell-src-exts >= 1.5.0 && < 1.14.0
 }
