diff --git a/hint.cabal b/hint.cabal
index d6d2d86..26fadee 100644
--- a/hint.cabal
+++ b/hint.cabal
@@ -47,3 +47,3 @@ Library
   build-depends: base == 4.*,
-                 ghc >= 7.6 && < 8,
+                 ghc >= 7.6 && < 8.2,
                  ghc-paths,
diff --git a/src/Hint/GHC.hs b/src/Hint/GHC.hs
index 562dc19..ca16957 100644
--- a/src/Hint/GHC.hs
+++ b/src/Hint/GHC.hs
@@ -1,3 +1,7 @@
 module Hint.GHC (
-    Message, module X
+      Message
+    , module X
+#if __GLASGOW_HASKELL__ >= 800
+    , WarnReason(..)
+#endif
 ) where
@@ -21,2 +25,5 @@ import FastString as X (fsLit)
 
+#if __GLASGOW_HASKELL__ >= 800
+import DynFlags as X(WarnReason(..))
+#endif
 #if __GLASGOW_HASKELL__ >= 710
diff --git a/src/Hint/InterpreterT.hs b/src/Hint/InterpreterT.hs
index b555173..ebb0f8f 100644
--- a/src/Hint/InterpreterT.hs
+++ b/src/Hint/InterpreterT.hs
@@ -184,3 +184,7 @@ newSessionData a =
 mkLogHandler :: IORef [GhcError] -> GhcErrLogger
+#if __GLASGOW_HASKELL__ >= 800
+mkLogHandler r df _wr _ src style msg =
+#else
 mkLogHandler r df _ src style msg =
+#endif
     let renderErrMsg = GHC.showSDoc df
diff --git a/src/Hint/Parsers.hs b/src/Hint/Parsers.hs
index 52eff39..8ccffaa 100644
--- a/src/Hint/Parsers.hs
+++ b/src/Hint/Parsers.hs
@@ -51,2 +51,5 @@ failOnParseError parser expr = mayFail go
                              let logger'  = logger dflags
+#if __GLASGOW_HASKELL__ >= 800
+                                                   GHC.NoReason
+#endif
                                  errStyle = GHC.defaultErrStyle dflags
