diff --git a/src/Data/VectorSpace/WrappedNum.hs b/src/Data/VectorSpace/WrappedNum.hs
index d7df2aa..fffa766 100644
--- a/src/Data/VectorSpace/WrappedNum.hs
+++ b/src/Data/VectorSpace/WrappedNum.hs
@@ -2,2 +2,3 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE RankNTypes #-}
 module Data.VectorSpace.WrappedNum
@@ -21,3 +22,3 @@ newtype WrappedNum a = WrapNum { unwrapNum :: a }
 derivingUnbox "Wrapped"
-    [t| (U.Unbox a) => WrappedNum a -> a |] [| unwrapNum |] [| \ a -> WrapNum a |]
+    [t| forall a . (U.Unbox a) => WrappedNum a -> a |] [| unwrapNum |] [| \ a -> WrapNum a |]
 
