--- silently-1.2.3-orig/silently.cabal	2012-10-19 13:22:53.000000000 +1100
+++ silently-1.2.3/silently.cabal	2012-10-20 14:27:35.576833434 +1100
@@ -62,6 +62,7 @@
     , HUnit
     , directory
     , transformers
+    , nanospec
 
 -- This tests the generic implementation, that should work on all platforms.
 test-suite spec-generic
@@ -79,3 +80,4 @@
       base
     , directory
     , deepseq
+    , nanospec
--- silently-1.2.3-orig/test/Spec.hs	2012-10-19 13:22:53.000000000 +1100
+++ silently-1.2.3/test/Spec.hs	2012-10-20 14:27:46.366047496 +1100
@@ -1,6 +1,6 @@
 module Main (main) where
 
-import           NanoSpec
+import           Test.Hspec
 import           System.IO
 import           System.IO.Silently
 import           System.Directory
