--- hexpat-0.20.2-orig/hexpat.cabal	2012-07-24 13:23:08.000000000 +1000
+++ hexpat-0.20.2/hexpat.cabal	2012-10-20 16:53:30.938318772 +1100
@@ -123,7 +123,7 @@
     deepseq >= 1.1.0.0 && < 1.4.0.0,
     containers,
     extensible-exceptions == 0.1.*,
-    List >= 0.4.2 && < 0.5
+    List >= 0.4.2 && < 0.6
   Exposed-Modules:
     Text.XML.Expat.Annotated,
     Text.XML.Expat.Cursor,
--- hexpat-0.20.2-orig/Text/XML/Expat/Extended.hs	2012-07-24 13:23:08.000000000 +1000
+++ hexpat-0.20.2/Text/XML/Expat/Extended.hs	2012-10-20 16:51:42.816881157 +1100
@@ -80,7 +80,7 @@
 import Control.DeepSeq
 import qualified Data.ByteString as B
 import qualified Data.ByteString.Lazy as L
-import Data.List.Class
+import Data.List.Class (List(..), foldlL, joinM)
 import Data.Maybe
 import Data.Monoid
 
