commit c005f5322ec5324e53b4fe9558e7670ed9e792bc
Author: RyanGlScott <ryan.gl.scott@gmail.com>
Date:   Mon Feb 8 09:28:09 2016 -0500

    Fix conduit and resourcet on GHC 8.0

diff --git a/resourcet/Control/Monad/Trans/Resource.hs b/resourcet/Control/Monad/Trans/Resource.hs
index d168ec4..90d7e76 100644
--- a/resourcet/Control/Monad/Trans/Resource.hs
+++ b/resourcet/Control/Monad/Trans/Resource.hs
@@ -6,7 +6,6 @@
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE DeriveDataTypeable #-}
-{-# LANGUAGE ImpredicativeTypes #-}
 #if __GLASGOW_HASKELL__ >= 704
 {-# LANGUAGE ConstraintKinds #-}
 #endif
@@ -97,12 +96,12 @@ register = liftResourceT . registerRIO
 --
 -- Since 0.3.0
 release :: MonadIO m => ReleaseKey -> m ()
-release (ReleaseKey istate rk) = liftIO $ release' istate rk  
+release (ReleaseKey istate rk) = liftIO $ release' istate rk
     (maybe (return ()) id)
 
 -- | Unprotect resource from cleanup actions, this allowes you to send
 -- resource into another resourcet process and reregister it there.
--- It returns an release action that should be run in order to clean 
+-- It returns an release action that should be run in order to clean
 -- resource or Nothing in case if resource is already freed.
 --
 -- Since 0.4.5
@@ -129,7 +128,7 @@ allocate a = liftResourceT . allocateRIO a
 --
 -- Since 0.3.0
 resourceMask :: MonadResource m => ((forall a. ResourceT IO a -> ResourceT IO a) -> ResourceT IO b) -> m b
-resourceMask = liftResourceT . resourceMaskRIO
+resourceMask r = liftResourceT (resourceMaskRIO r)
 
 allocateRIO :: IO a -> (a -> IO ()) -> ResourceT IO (ReleaseKey, a)
 allocateRIO acquire rel = ResourceT $ \istate -> liftIO $ E.mask $ \restore -> do
diff --git a/resourcet/resourcet.cabal b/resourcet/resourcet.cabal
index b1785ca..98ffada 100644
--- a/resourcet/resourcet.cabal
+++ b/resourcet/resourcet.cabal
@@ -22,7 +22,7 @@ Library
                      , transformers-base        >= 0.4.4        && < 0.5
                      , monad-control            >= 0.3.1        && < 1.1
                      , containers
-                     , transformers             >= 0.2.2        && < 0.5
+                     , transformers             >= 0.2.2        && < 0.6
                      , transformers-compat      >= 0.3          && < 0.6
                      , mtl                      >= 2.0          && < 2.3
                      , mmorph
