--- strict-concurrency-0.2.2/strict-concurrency.cabal	2010-09-08 12:04:49.000000000 +0200
+++ strict-concurrency.cabal	2010-09-08 11:25:49.000000000 +0200
@@ -15,7 +15,7 @@
 Maintainer:     Don Stewart <dons@galois.com>
 Copyright:      (c) 2007-8 Don Stewart
 Homepage:       http://code.haskell.org/~dons/code/strict-concurrency
-build-depends:  base >= 3 && < 4, parallel
+build-depends:  base >= 3 && < 4, parallel < 3
 ghc-options:    -Wall -fglasgow-exts
 extensions:     CPP, BangPatterns
 build-type:     Simple
