diff --git a/Setup.lhs b/Setup.lhs
index e5e7786..7aaf0a0 100644
--- a/Setup.lhs
+++ b/Setup.lhs
@@ -29,20 +29,6 @@ main = defaultMainWithHooks simpleUserHooks {
 
--- 'ConstOrId' is a @Cabal-1.16@ vs @Cabal-1.18@ compatibility hack,
--- 'programFindLocation' has a new (unused in this case)
--- parameter. 'ConstOrId' adds this parameter when types say it is
--- mandatory.
-class ConstOrId a b where
-    constOrId :: a -> b
-
-instance ConstOrId a a where
-    constOrId = id
-
-instance ConstOrId a (b -> a) where
-    constOrId = const
-
 mysqlConfigProgram = (simpleProgram "mysql_config") {
-    programFindLocation = \verbosity -> constOrId $ do
-      mysql_config  <- findProgramOnPath "mysql_config"  verbosity
-      mysql_config5 <- findProgramOnPath "mysql_config5" verbosity
-      return (mysql_config `mplus` mysql_config5)
+    programFindLocation = \verbosity path -> liftM2 mplus
+      (findProgramOnSearchPath verbosity path "mysql_config")
+      (findProgramOnSearchPath verbosity path "mysql_config5")
   }
