diff --git a/src/Server.hs b/src/Server.hs
index 66cd3a6..2ea85d2 100644
--- a/src/Server.hs
+++ b/src/Server.hs
@@ -238,2 +238,6 @@ getManager opt = H.newManager H.defaultManagerSettings {
       | opt_proxy_timeout opt == 0 = H.managerResponseTimeout H.defaultManagerSettings
+#if MIN_VERSION_http_client(0,5,0)
+      | otherwise                  = H.responseTimeoutMicro (opt_proxy_timeout opt * 1000000)
+#else
       | otherwise                  = Just (opt_proxy_timeout opt * 1000000) -- micro seconds
+#endif
