diff --git a/reflex-vty.cabal b/reflex-vty.cabal
index b614c1e..482445e 100644
--- a/reflex-vty.cabal
+++ b/reflex-vty.cabal
@@ -20,6 +20,10 @@ extra-source-files:
 extra-doc-files: doc/tasks.png
 tested-with: GHC ==8.8.4 || ==8.6.5 || ==8.4.4
 
+flag examples
+    description:    Build demonstration program
+    default:        False
+
 library
   exposed-modules: Reflex.Vty
                  , Reflex.Vty.Host
@@ -53,7 +57,7 @@ library
   default-language: Haskell2010
   ghc-options: -Wall
 
-executable example
+executable reflex-vty-example
   hs-source-dirs: src-bin
   main-is: example.hs
   ghc-options: -threaded
@@ -67,6 +71,8 @@ executable example
     transformers,
     vty
   default-language: Haskell2010
+  if !flag(examples)
+      buildable: False
 
 source-repository head
   type: git
