diff --git a/Text/Pandoc/Builder.hs b/Text/Pandoc/Builder.hs
index 8d84051..7bbd76b 100644
--- a/Text/Pandoc/Builder.hs
+++ b/Text/Pandoc/Builder.hs
@@ -1,6 +1,7 @@
 {-# LANGUAGE TypeSynonymInstances, FlexibleInstances, MultiParamTypeClasses,
     DeriveDataTypeable, GeneralizedNewtypeDeriving, CPP, StandaloneDeriving,
     DeriveGeneric #-}
+{-# LANGUAGE DeriveTraversable #-}
 {-
 Copyright (C) 2010-2012 John MacFarlane <jgm@berkeley.edu>
 
