--- happstack-server-7.0.1-orig/happstack-server.cabal	2012-04-21 09:23:18.000000000 +1000
+++ happstack-server-7.0.1/happstack-server.cabal	2012-04-23 12:36:24.139008744 +1000
@@ -23,6 +23,10 @@
     Description: Choose newer network library with merged in network-bytestring.
     Default: True
 
+Flag blaze_html_0_5
+    Description: Use blaze-html 0.5 and blaze-markup 0.5
+    Default: False
+
 Flag tests
     Description: Build the testsuite, and include the tests in the library
     Default: False
@@ -82,7 +86,6 @@
 
   Build-Depends:       base,
                        base64-bytestring == 0.1.*,
-                       blaze-html >= 0.3 && < 0.5,
                        bytestring,
                        containers,
                        directory,
@@ -111,6 +114,14 @@
   else
     Build-Depends:     network < 2.2.3, network-bytestring
 
+  if flag(blaze_html_0_5)
+    build-depends:
+                       blaze-html >= 0.5 && < 0.6,
+                       blaze-markup >= 0.5.1 && < 0.6
+  else
+    build-depends:
+                       blaze-html >= 0.4 && < 0.5
+
   if (flag(template_haskell) && !(arch(arm)))
     Build-Depends:     template-haskell
     cpp-options:       -DTEMPLATE_HASKELL
--- happstack-server-7.0.1-orig/src/Happstack/Server/Response.hs	2012-04-21 09:23:18.000000000 +1000
+++ happstack-server-7.0.1/src/Happstack/Server/Response.hs	2012-04-23 18:32:09.722378967 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE FlexibleContexts, FlexibleInstances, TypeSynonymInstances, ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts, FlexibleInstances, TypeSynonymInstances, ScopedTypeVariables, CPP #-}
 -- | Functions and classes related to generating a 'Response' and setting the response code. For detailed instruction see the Happstack Crash Course: <http://happstack.com/docs/crashcourse/HelloWorld.html#response_code>
 module Happstack.Server.Response 
     ( -- * Converting values to a 'Response'
@@ -39,8 +39,13 @@
 import           Happstack.Server.SURI           (ToSURI)
 import           System.Locale                   (defaultTimeLocale)
 import           System.Time                     (CalendarTime, formatCalendarTime)
+#if MIN_VERSION_blaze_html(0,5,0)
+import qualified Text.Blaze.Html                 as Blaze
+import qualified Text.Blaze.Html.Renderer.Utf8   as Blaze
+#else
 import qualified Text.Blaze                      as Blaze
 import qualified Text.Blaze.Renderer.Utf8        as Blaze
+#endif
 import           Text.Html                       (Html, renderHtml)
 import qualified Text.XHtml                      as XHtml (Html, renderHtml)
 
