diff --git a/src/Happstack/Server/Internal/SocketTH.hs b/src/Happstack/Server/Internal/SocketTH.hs
index 19ff276..51ed184 100644
--- a/src/Happstack/Server/Internal/SocketTH.hs
+++ b/src/Happstack/Server/Internal/SocketTH.hs
@@ -22,3 +22,7 @@ supportsIPv6 = $(let c = ["Network.Socket.SockAddrInet6", "Network.Socket.Intern
                     case info of
+#if MIN_VERSION_template_haskell(2,11,0)
+                      TyConI (DataD _ _ _ _ cs _) ->
+#else
                       TyConI (DataD _ _ _ cs _) ->
+#endif
                         if any isInet6 cs
