--- configurator-0.2.0.0-orig/Data/Configurator.hs	2011-11-22 09:17:06.000000000 +1100
+++ configurator-0.2.0.0/Data/Configurator.hs	2012-07-07 21:33:52.704727186 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE BangPatterns, OverloadedStrings, RecordWildCards,
+{-# LANGUAGE BangPatterns, CPP, OverloadedStrings, RecordWildCards,
     ScopedTypeVariables, TupleSections #-}
 
 -- |
@@ -74,7 +74,11 @@
 import Data.Text.Lazy.Builder.Int (decimal)
 import Data.Text.Lazy.Builder.RealFloat (realFloat)
 import Data.Ratio (denominator, numerator)
-import Prelude hiding (catch, lookup)
+import Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+  catch,
+#endif
+  lookup)
 import System.Environment (getEnv)
 import System.IO (hPutStrLn, stderr)
 import System.IO.Unsafe (unsafePerformIO)
