diff --git a/src/Data/Lens/Light/Template.hs b/src/Data/Lens/Light/Template.hs
index 34ef897..16c5ee3 100644
--- a/src/Data/Lens/Light/Template.hs
+++ b/src/Data/Lens/Light/Template.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE CPP #-}
 
@@ -86,4 +87,9 @@ decMakeLens t dec namer = do
     (params, cons) <- case dec of
+#if MIN_VERSION_template_haskell(2,11,0)
+                 DataD _ _ params _ cons' _ -> return (params, cons')
+                 NewtypeD _ _ params _ con' _ -> return (params, [con'])
+#else
                  DataD _ _ params cons' _ -> return (params, cons')
                  NewtypeD _ _ params con' _ -> return (params, [con'])
+#endif
                  _ -> fail $ errmsg t
