--- HTF-0.8.2.1-orig/HTF.cabal	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/HTF.cabal	2012-08-18 19:56:20.735022562 +1000
@@ -39,7 +39,7 @@
 Executable htfpp
   Main-Is:          HTFPP.hs
   Build-Depends:    cpphs >= 1.12,
-                    haskell-src-exts == 1.11.*,
+                    haskell-src-exts >= 1.11 && < 1.14,
                     base == 4.*
   Other-Modules:
     Test.Framework.Preprocessor
@@ -53,13 +53,13 @@
                     containers >= 0.3 && < 0.6,
                     process >= 1.0 && < 1.2,
                     directory >= 1.0 && < 1.2,
-                    mtl >= 1.1 && < 2.1,
+                    mtl >= 1.1 && < 2.2,
                     pretty >= 1.0 && < 1.2,
                     Diff >= 0.1.3 && < 0.2,
-                    unix >= 2.4 && < 2.6,
-                    bytestring == 0.9.*,
+                    unix >= 2.4 && < 2.7,
+                    bytestring >= 0.9 && < 0.11,
                     regex-compat >= 0.92 && < 0.96,
-                    haskell-src-exts == 1.11.*
+                    haskell-src-exts >= 1.11 && < 1.14
   Exposed-Modules:
     Test.Framework
     Test.Framework.HUnitWrapper
--- HTF-0.8.2.1-orig/Test/Framework/Diff.hs	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/Test/Framework/Diff.hs	2012-08-18 20:02:37.322123708 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 --
 -- Copyright (c) 2011   Stefan Wehr - http://www.stefanwehr.de
 --
@@ -24,7 +25,9 @@
 
 ) where
 
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding (catch)
+#endif
 
 import Control.Exception (catch, finally, IOException)
 import qualified Data.List as List
--- HTF-0.8.2.1-orig/Test/Framework/BlackBoxTest.hs	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/Test/Framework/BlackBoxTest.hs	2012-08-18 20:03:37.441054933 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 --
 -- Copyright (c) 2005,2009   Stefan Wehr - http://www.stefanwehr.de
 --
@@ -34,7 +35,9 @@
 
 ) where
 
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding ( catch )
+#endif
 
 import System.IO
 import System.Exit
--- HTF-0.8.2.1-orig/Test/Framework/QuickCheckWrapper.hs	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/Test/Framework/QuickCheckWrapper.hs	2012-08-18 20:05:40.845019160 +1000
@@ -1,5 +1,5 @@
 {-# LANGUAGE FlexibleInstances,OverlappingInstances,ExistentialQuantification,
-             DeriveDataTypeable,ScopedTypeVariables #-}
+             DeriveDataTypeable,ScopedTypeVariables,CPP #-}
 
 --
 -- Copyright (c) 2005,2009-2011 Stefan Wehr - http://www.stefanwehr.de
@@ -32,7 +32,9 @@
 
 import qualified Data.Map as Map
 import Control.Concurrent.MVar
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding ( catch )
+#endif
 import Control.Exception ( SomeException, Exception, Handler(..),
                            throw, catch, catches, evaluate )
 import System.IO
--- HTF-0.8.2.1-orig/Test/Framework/HaskellParser.hs	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/Test/Framework/HaskellParser.hs	2012-08-18 20:07:28.637481940 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ScopedTypeVariables,CPP #-}
 
 -- This library is free software; you can redistribute it and/or
 -- modify it under the terms of the GNU Lesser General Public
@@ -20,7 +20,9 @@
 import Data.Maybe
 import Data.Char ( isSpace )
 import Control.Exception ( evaluate, catch, SomeException )
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding ( catch )
+#endif
 
 import qualified Language.Haskell.Exts as Exts
 import qualified Language.Haskell.Exts.Parser as Parser
--- HTF-0.8.2.1-orig/HTFPP.hs	2012-08-18 18:40:23.000000000 +1000
+++ HTF-0.8.2.1/HTFPP.hs	2012-08-18 20:09:38.795663219 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ScopedTypeVariables,CPP #-}
 
 --
 -- Copyright (c) 2009   Stefan Wehr - http://www.stefanwehr.de
@@ -18,7 +18,9 @@
 -- Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA
 --
 
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding ( catch )
+#endif
 import System.IO
 import System.Environment
 import System.Exit
