diff --git a/HTF.cabal b/HTF.cabal
index ffa381e..643ebca 100644
--- a/HTF.cabal
+++ b/HTF.cabal
@@ -39,7 +39,7 @@ Source-Repository head
 Executable htfpp
   Main-Is:          HTFPP.hs
   Build-Depends:    cpphs >= 1.12,
-                    haskell-src-exts == 1.11.*,
+                    haskell-src-exts >= 1.11 && < 1.14,
                     base == 4.*
   Other-Modules:
     Test.Framework.Preprocessor
diff --git a/HTF.cabal b/HTF.cabal
index 643ebca..77a2985 100644
--- a/HTF.cabal
+++ b/HTF.cabal
@@ -59,7 +59,7 @@ Library
                     unix >= 2.4 && < 2.6,
                     bytestring == 0.9.*,
                     regex-compat >= 0.92 && < 0.96,
-                    haskell-src-exts == 1.11.*
+                    haskell-src-exts >= 1.11 && < 1.14
   Exposed-Modules:
     Test.Framework
     Test.Framework.HUnitWrapper
