diff --git a/src/Graphics/Vty/Widgets/Borders.hs b/src/Graphics/Vty/Widgets/Borders.hs
index 4cb738a..100bdca 100644
--- a/src/Graphics/Vty/Widgets/Borders.hs
+++ b/src/Graphics/Vty/Widgets/Borders.hs
@@ -24,3 +24,3 @@ import qualified Data.Text as T
 
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/Centering.hs b/src/Graphics/Vty/Widgets/Centering.hs
index fe14e2b..f792e57 100644
--- a/src/Graphics/Vty/Widgets/Centering.hs
+++ b/src/Graphics/Vty/Widgets/Centering.hs
@@ -14,3 +14,3 @@ where
 import Graphics.Vty.Widgets.Core
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Util
diff --git a/src/Graphics/Vty/Widgets/Core.hs b/src/Graphics/Vty/Widgets/Core.hs
index 89c6bbe..e5a2a5a 100644
--- a/src/Graphics/Vty/Widgets/Core.hs
+++ b/src/Graphics/Vty/Widgets/Core.hs
@@ -81,3 +81,3 @@ import Control.Monad
 import Control.Exception
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth, regionHeight)
 import Graphics.Vty.Widgets.Util
diff --git a/src/Graphics/Vty/Widgets/Edit.hs b/src/Graphics/Vty/Widgets/Edit.hs
index 97f9917..13e87d5 100644
--- a/src/Graphics/Vty/Widgets/Edit.hs
+++ b/src/Graphics/Vty/Widgets/Edit.hs
@@ -66,3 +66,3 @@ import Data.Maybe (isJust)
 import qualified Data.Text as T
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth, regionHeight)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/EventLoop.hs b/src/Graphics/Vty/Widgets/EventLoop.hs
index dffddf7..8a496ec 100644
--- a/src/Graphics/Vty/Widgets/EventLoop.hs
+++ b/src/Graphics/Vty/Widgets/EventLoop.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE DeriveDataTypeable, ExistentialQuantification #-}
+{-# LANGUAGE CPP #-}
 -- |This module provides the main event loop functionality for this
@@ -48,3 +49,7 @@ runUi :: Collection -> RenderContext -> IO ()
 runUi collection ctx = do
+#if MIN_VERSION_vty(5,15,0)
+  vty <- mkVty mempty
+#else
   vty <- mkVty def
+#endif
 
diff --git a/src/Graphics/Vty/Widgets/Fills.hs b/src/Graphics/Vty/Widgets/Fills.hs
index ac629e0..9e45f6b 100644
--- a/src/Graphics/Vty/Widgets/Fills.hs
+++ b/src/Graphics/Vty/Widgets/Fills.hs
@@ -11,3 +11,3 @@ where
 
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth, regionHeight)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/Fixed.hs b/src/Graphics/Vty/Widgets/Fixed.hs
index aa55924..26d42db 100644
--- a/src/Graphics/Vty/Widgets/Fixed.hs
+++ b/src/Graphics/Vty/Widgets/Fixed.hs
@@ -21,3 +21,3 @@ where
 import Control.Monad
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/List.hs b/src/Graphics/Vty/Widgets/List.hs
index 385fcb2..2cccca9 100644
--- a/src/Graphics/Vty/Widgets/List.hs
+++ b/src/Graphics/Vty/Widgets/List.hs
@@ -54,3 +54,3 @@ import qualified Data.Text as T
 import qualified Data.Vector as V
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/Padding.hs b/src/Graphics/Vty/Widgets/Padding.hs
index 8950917..876fcf0 100644
--- a/src/Graphics/Vty/Widgets/Padding.hs
+++ b/src/Graphics/Vty/Widgets/Padding.hs
@@ -22,3 +22,3 @@ where
 import Data.Monoid
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/ProgressBar.hs b/src/Graphics/Vty/Widgets/ProgressBar.hs
index 0d8c075..95295c0 100644
--- a/src/Graphics/Vty/Widgets/ProgressBar.hs
+++ b/src/Graphics/Vty/Widgets/ProgressBar.hs
@@ -18,3 +18,3 @@ import Control.Monad
 import qualified Data.Text as T
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/Table.hs b/src/Graphics/Vty/Widgets/Table.hs
index 2960ac7..6c07770 100644
--- a/src/Graphics/Vty/Widgets/Table.hs
+++ b/src/Graphics/Vty/Widgets/Table.hs
@@ -34,3 +34,3 @@ import Control.Exception
 import Control.Monad
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth, regionHeight)
 import Graphics.Vty.Widgets.Core
diff --git a/src/Graphics/Vty/Widgets/Util.hs b/src/Graphics/Vty/Widgets/Util.hs
index b9aea1c..46a632a 100644
--- a/src/Graphics/Vty/Widgets/Util.hs
+++ b/src/Graphics/Vty/Widgets/Util.hs
@@ -28,3 +28,3 @@ import Control.Applicative
 import qualified Data.Text as T
-import Graphics.Vty
+import Graphics.Vty hiding (regionWidth, regionHeight)
 
diff --git a/test/src/Tests/FormattedText.hs b/test/src/Tests/FormattedText.hs
index 575d509..4809d46 100644
--- a/test/src/Tests/FormattedText.hs
+++ b/test/src/Tests/FormattedText.hs
@@ -8,3 +8,3 @@ import qualified Data.Text as T
 
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight)
 import Graphics.Vty.Widgets.Text
diff --git a/test/src/Tests/Util.hs b/test/src/Tests/Util.hs
index d6b03de..23e0b1c 100644
--- a/test/src/Tests/Util.hs
+++ b/test/src/Tests/Util.hs
@@ -4,3 +4,3 @@ import Control.Applicative
 import qualified Data.Text as T
-import Graphics.Vty
+import Graphics.Vty hiding (regionHeight, regionWidth)
 import Graphics.Vty.Widgets.Core
