diff --git a/src/Graphics/Vty/Widgets/DirBrowser.hs b/src/Graphics/Vty/Widgets/DirBrowser.hs
index a05e2e0..9d05a83 100644
--- a/src/Graphics/Vty/Widgets/DirBrowser.hs
+++ b/src/Graphics/Vty/Widgets/DirBrowser.hs
@@ -236,5 +236,5 @@ builtInAnnotations cur sk =
       , defAttr)
-    , (\_ s -> isSymbolicLink s,
+    , (\_ s -> System.Posix.Files.isSymbolicLink s,
        (\p stat -> do
-          linkDest <- if not $ isSymbolicLink stat
+          linkDest <- if not $ System.Posix.Files.isSymbolicLink stat
                       then return ""
