diff --git a/test/src/Tests/Instances.hs b/test/src/Tests/Instances.hs
index 2a1cca0..4fc710c 100644
--- a/test/src/Tests/Instances.hs
+++ b/test/src/Tests/Instances.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
@@ -22,3 +23,7 @@ instance Arbitrary Color where
 instance Arbitrary Attr where
+#if MIN_VERSION_vty(5,18,1)
+    arbitrary = Attr <$> arbitrary <*> arbitrary <*> arbitrary <*> pure Default
+#else
     arbitrary = Attr <$> arbitrary <*> arbitrary <*> arbitrary
+#endif
 
