--- a/src/Graphics/Vty/Widgets/Padding.hs
+++ b/src/Graphics/Vty/Widgets/Padding.hs
@@ -25,2 +25,3 @@ import Graphics.Vty.Widgets.Core
 import Graphics.Vty.Widgets.Util
+import qualified Data.Semigroup as S
 
@@ -39,2 +40,6 @@ instance Show Padded where
 
+instance S.Semigroup Padding where
+    (<>) (Padding a1 a2 a3 a4) (Padding b1 b2 b3 b4) =
+        Padding (a1 + b1) (a2 + b2) (a3 + b3) (a4 + b4)
+
 instance Monoid Padding where
--- a/src/Graphics/Vty/Widgets/Table.hs
+++ b/src/Graphics/Vty/Widgets/Table.hs
@@ -45,2 +45,3 @@ import Graphics.Vty.Widgets.Box
 import Graphics.Vty.Widgets.Alignment
+import qualified Data.Semigroup as S
 
@@ -155,2 +156,5 @@ instance (RowLike a) => RowLike [a] where
 
+instance S.Semigroup TableRow where
+    (TableRow as) <> (TableRow bs) = TableRow $ as ++ bs
+
 instance Monoid TableRow where
--- a/src/Graphics/Vty/Widgets/Text.hs
+++ b/src/Graphics/Vty/Widgets/Text.hs
@@ -36,2 +36,3 @@ import Text.Trans.Tokenize
 import Graphics.Vty.Widgets.Util
+import qualified Data.Semigroup as S
 
@@ -44,2 +45,6 @@ newtype Formatter = Formatter (DisplayRegion -> TextStream Attr -> IO (TextStrea
 
+instance S.Semigroup Formatter where
+    (<>) (Formatter f1) (Formatter f2) =
+        Formatter (\sz t -> f1 sz t >>= f2 sz)
+
 instance Monoid Formatter where
