--- LambdaShell-0.9.1-orig/LambdaShell.cabal	2007-11-30 04:06:25.000000000 +1100
+++ LambdaShell-0.9.1/LambdaShell.cabal	2012-08-05 12:46:39.048335543 +1000
@@ -33,7 +33,7 @@
      readline
   GHC-Options: -O
   Build-Depends:
-     base, haskell98, parsec,
+     base, parsec,
      mtl, readline,
      Shellac >= 0.9,
      Shellac-readline >= 0.9
--- LambdaShell-0.9.1-orig/src/LambdaCmdLine.hs	2007-11-30 04:06:25.000000000 +1100
+++ LambdaShell-0.9.1/src/LambdaCmdLine.hs	2012-08-05 12:46:59.378809686 +1000
@@ -24,7 +24,7 @@
 
 import Data.IORef
 import Numeric
-import Maybe
+import Data.Maybe
 import Data.List
 import Data.Char
 import qualified Data.Map as Map
