--- PostgreSQL/PostgreSQL.cabal.old	2006-01-05 19:32:52.000000000 +0100
+++ PostgreSQL/PostgreSQL.cabal	2006-01-05 19:36:19.000000000 +0100
@@ -7,4 +7,5 @@
 exposed-modules: Database.HSQL.PostgreSQL
 build-depends: 	 base, hsql
 extensions:      ForeignFunctionInterface, CPP
-extra-libraries: pq
\ Kein Zeilenumbruch am Dateiende.
+extra-libraries: pq
+include-dirs:	 /usr/include/postgresql/server
