diff -Naur probability-0.2.2.1/src/Numeric/Probability/Distribution.hs probability-0.2.2.1-r1/src/Numeric/Probability/Distribution.hs
--- probability-0.2.2.1/src/Numeric/Probability/Distribution.hs	2010-06-03 15:11:08.000000000 +0300
+++ probability-0.2.2.1-r1/src/Numeric/Probability/Distribution.hs	2011-11-17 04:17:17.454512383 +0200
@@ -12,6 +12,7 @@
 
 import Prelude hiding (map, filter)
 
+import Data.Ord(comparing)
 
 -- * Events
 type Event a = a -> Bool
@@ -132,7 +133,7 @@
 sortP = List.sortBy (\x y -> compare (snd y) (snd x))
 
 sortElem :: Ord a => [(a,prob)] -> [(a,prob)]
-sortElem = List.sortBy (\x y -> compare (fst y) (fst x))
+sortElem = List.sortBy (comparing fst)
 
 
 -- ** Normalization = grouping
