--- a/ShellCheck/AnalyzerLib.hs
+++ b/ShellCheck/AnalyzerLib.hs
@@ -41,2 +41,3 @@ import Test.QuickCheck.All (forAllProperties)
 import Test.QuickCheck.Test (quickCheckWithResult, stdArgs, maxSuccess)
+import qualified Data.Semigroup as S
 
@@ -59,2 +60,8 @@ runChecker params checker = notes
 
+instance S.Semigroup Checker where
+    (<>) x y = Checker {
+        perScript = perScript x `composeAnalyzers` perScript y,
+        perToken = perToken x `composeAnalyzers` perToken y
+        }
+
 instance Monoid Checker where
--- a/shellcheck.hs
+++ b/shellcheck.hs
@@ -48,2 +48,3 @@ import System.Exit
 import System.IO
+import qualified Data.Semigroup as S
 
@@ -58,2 +59,5 @@ data Status =
 
+instance S.Semigroup Status where
+    (<>) = max
+
 instance Monoid Status where
