From 4b9842c3e3d653be69af19778970337775e2404d Mon Sep 17 00:00:00 2001
From: Alex Washburn <github@recursion.ninja>
Date: Mon, 16 Nov 2020 21:02:16 -0500
Subject: [PATCH] Adding compatability with version 0.16.* of
 'optparse-applicative'

diff --git a/src/Diagrams/Backend/CmdLine.hs b/src/Diagrams/Backend/CmdLine.hs
index f82a5dc..1c912c1 100644
--- a/src/Diagrams/Backend/CmdLine.hs
+++ b/src/Diagrams/Backend/CmdLine.hs
@@ -220,11 +220,17 @@ diagramLoopOpts = DiagramLoopOpts
 --   Taken from Options.Applicative.Extra but without the
 --   short option 'h'.  We want the 'h' for Height.
 helper' :: Parser (a -> a)
-helper' = abortOption ShowHelpText $ mconcat
+helper' = abortOption param $ mconcat
   [ long "help"
   , short '?'
   , help "Show this help text"
   ]
+  where
+#if MIN_VERSION_optparse_applicative(0,16,0)
+    param = ShowHelpText Nothing
+#else
+    param = ShowHelpText 
+#endif
 
 -- | Apply a parser to the command line that includes the standard
 --   program description and help behavior.  Results in parsed commands
-- 
2.26.2

