diff --git a/Text/PrettyPrint/Leijen.hs b/Text/PrettyPrint/Leijen.hs
index b4a8423..3e411d3 100644
--- a/Text/PrettyPrint/Leijen.hs
+++ b/Text/PrettyPrint/Leijen.hs
@@ -119,8 +119,10 @@ import Prelude hiding ((<$>))
 #endif
 
 infixr 5 </>,<//>,<$>,<$$>
-infixr 6 <>,<+>
-
+infixr 6 <+>
+#if !MIN_VERSION_base(4,11,0)
+infixr 6 <>
+#endif
 
 -----------------------------------------------------------
 -- list, tupled and semiBraces pretty print a list of
@@ -309,8 +311,13 @@ fold f ds       = foldr1 f ds
 -- | The document @(x \<\> y)@ concatenates document @x@ and document
 -- @y@. It is an associative operation having 'empty' as a left and
 -- right unit.  (infixr 6)
+#if MIN_VERSION_base(4,11,0)
+instance Semigroup Doc where
+  x <> y          = x `beside` y
+#else
 (<>) :: Doc -> Doc -> Doc
 x <> y          = x `beside` y
+#endif
 
 -- | The document @(x \<+\> y)@ concatenates document @x@ and @y@ with a
 -- @space@ in between.  (infixr 6)

commit d63b507f5d675d03f2314170fcda30b882c318d4
Author: Víctor López Juan <victor@lopezjuan.com>
Date:   Sat Mar 31 12:00:48 2018 +0200

    Add testing for multiple GHC versions

diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..e41145c
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,24 @@
+.PHONY: build install
+
+build:
+       stack build
+
+install:
+       stack install
+
+# Testing
+.PHONY: test test-all test-default test-%
+
+test-all: test-default test-7.10.3 test-7.8.4 test-8.0.2 test-8.2.2
+
+test: test-default
+
+test-default:
+       stack clean
+       stack build
+       stack haddock --no-haddock-deps
+
+test-%:
+       stack --stack-yaml stack-$*.yaml clean
+       stack --stack-yaml stack-$*.yaml build
+       stack --stack-yaml stack-$*.yaml haddock --no-haddock-deps
diff --git a/stack-7.10.3.yaml b/stack-7.10.3.yaml
new file mode 100644
index 0000000..33e45d4
--- /dev/null
+++ b/stack-7.10.3.yaml
@@ -0,0 +1,6 @@
+resolver: ghc-7.10.3
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
diff --git a/stack-7.8.4.yaml b/stack-7.8.4.yaml
new file mode 100644
index 0000000..032e189
--- /dev/null
+++ b/stack-7.8.4.yaml
@@ -0,0 +1,6 @@
+resolver: ghc-7.8.4
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
diff --git a/stack-8.0.2.yaml b/stack-8.0.2.yaml
new file mode 100644
index 0000000..e373f2e
--- /dev/null
+++ b/stack-8.0.2.yaml
@@ -0,0 +1,6 @@
+resolver: ghc-8.0.2
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
diff --git a/stack-8.2.2.yaml b/stack-8.2.2.yaml
new file mode 100644
index 0000000..825faa5
--- /dev/null
+++ b/stack-8.2.2.yaml
@@ -0,0 +1,6 @@
+resolver: ghc-8.2.2
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []
diff --git a/stack.yaml b/stack.yaml
new file mode 100644
index 0000000..cdf5b21
--- /dev/null
+++ b/stack.yaml
@@ -0,0 +1,6 @@
+resolver: ghc-8.4.1
+packages:
+- '.'
+extra-deps: []
+flags: {}
+extra-package-dbs: []

