commit ae800077d4a9316e73a98f819db4bb038c477e14
Author: Víctor López Juan <victor@lopezjuan.com>
Date:   Sat Mar 31 12:42:01 2018 +0200

    `Monoid` instance for `Doc`

diff --git a/Text/PrettyPrint/Leijen.hs b/Text/PrettyPrint/Leijen.hs
index e17ce4d..ec054b9 100644
--- a/Text/PrettyPrint/Leijen.hs
+++ b/Text/PrettyPrint/Leijen.hs
@@ -119,6 +119,10 @@ import System.IO (Handle,hPutStr,hPutChar,stdout)
 import Prelude hiding ((<$>))
 #endif
 
+#if !MIN_VERSION_base(4,11,0)
+import Data.Monoid (Monoid(..))
+#endif
+
 infixr 5 </>,<//>,<$>,<$$>
 infixr 6 <+>
 #if !MIN_VERSION_base(4,11,0)
@@ -320,6 +324,10 @@ instance Semigroup Doc where
 x <> y          = x `beside` y
 #endif
 
+instance Monoid Doc where
+  mempty = empty
+  mappend = (<>)
+
 -- | The document @(x \<+\> y)@ concatenates document @x@ and @y@ with a
 -- @space@ in between.  (infixr 6)
 (<+>) :: Doc -> Doc -> Doc

commit 17b9ea525ca21caf04b33f071431721aaba16e77
Author: Víctor López Juan <victor@lopezjuan.com>
Date:   Sat Mar 31 12:21:11 2018 +0200

    Move `OverlappingInstances` from `.cabal` file to Haskell source file
    
    This language extension is deprecated, we should eventually do without
    it.

diff --git a/Text/PrettyPrint/Leijen.hs b/Text/PrettyPrint/Leijen.hs
index 3e411d3..e17ce4d 100644
--- a/Text/PrettyPrint/Leijen.hs
+++ b/Text/PrettyPrint/Leijen.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE OverlappingInstances #-}
 -----------------------------------------------------------------------------
 -- |
 -- Module      :  Text.PrettyPrint.Leijen
diff --git a/wl-pprint.cabal b/wl-pprint.cabal
index 0284d1b..5f0e3f5 100644
--- a/wl-pprint.cabal
+++ b/wl-pprint.cabal
@@ -16,7 +16,6 @@ Build-Type:          Simple
 Library
   Build-Depends:       base < 5
   Exposed-Modules:     Text.PrettyPrint.Leijen
-  Extensions:          OverlappingInstances
 
 source-repository head
   type: git

commit 69646bc717172ae6d9ff85f111439a95d00e4b8c
Author: Víctor López Juan <victor@lopezjuan.com>
Date:   Sat Mar 31 12:20:01 2018 +0200

    Add `Semigroup` instance for Doc if base >= 4.11
    
    This also avoids the conflict with Prelude.(<>).

