--- th-reify-many-0.1.4-orig/src/Language/Haskell/TH/ReifyMany/Internal.hs	2016-01-19 10:36:08.000000000 +1100
+++ th-reify-many-0.1.4/src/Language/Haskell/TH/ReifyMany/Internal.hs	2016-04-24 20:47:51.239949539 +1000
@@ -76,7 +76,11 @@
         _ -> fail $ "Error in getInstances: " ++ show clz ++ " isn't a class"
   where
 #if MIN_VERSION_template_haskell(2,7,0)
-    convertDec (InstanceD ctxt typ decs) = do
+    convertDec (InstanceD
+#if MIN_VERSION_template_haskell(2,11,0)
+                Nothing
+#endif
+                ctxt typ decs) = do
         typ' <- expandSyns typ
         return $ Just (TypeclassInstance ctxt typ' decs)
     convertDec _ = return Nothing
