--- a/lib/Pdf/Toolbox/Content/FontInfo.hs
+++ b/lib/Pdf/Toolbox/Content/FontInfo.hs
@@ -38,2 +38,3 @@ import qualified Pdf.Toolbox.Content.Encoding.WinAnsi as WinAnsi
 import qualified Pdf.Toolbox.Content.Encoding.MacRoman as MacRoman
+import qualified Data.Semigroup as S
 
@@ -84,2 +85,8 @@ data CIDFontWidths = CIDFontWidths {
 
+instance S.Semigroup CIDFontWidths where
+  w1 <> w2 = CIDFontWidths {
+    cidFontWidthsChars = cidFontWidthsChars w1 `mappend` cidFontWidthsChars w2,
+    cidFontWidthsRanges = cidFontWidthsRanges w1 `mappend` cidFontWidthsRanges w2
+    }
+
 instance Monoid CIDFontWidths where
