--- PSQueue-1.1-orig/Data/PSQueue.hs	2008-10-15 02:47:27.000000000 +1100
+++ PSQueue-1.1/Data/PSQueue.hs	2018-04-10 10:18:58.354211094 +1000
@@ -14,7 +14,8 @@
 <http://citeseer.ist.psu.edu/hinze01simple.html>
 
 -}
-
+{-# LANGUAGE CPP                 #-}
+  
 -- Some modifications by Scott Dillard
 
 
@@ -62,7 +63,11 @@
     , foldl
 ) where
 
-import Prelude hiding (lookup,null,foldl,foldr)
+import Prelude hiding (lookup,null,foldl,foldr
+#if MIN_VERSION_base(4,11,0)
+                      ,(<>)
+#endif
+                      )
 import qualified Prelude as P
 
 {-
