commit e0686e310c948fbcbde9e64421563c91a42dc0b3
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Apr 19 09:25:00 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/Cheapskate/ParserCombinators.hs b/Cheapskate/ParserCombinators.hs
index d60699d..77efb09 100644
--- a/Cheapskate/ParserCombinators.hs
+++ b/Cheapskate/ParserCombinators.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 module Cheapskate.ParserCombinators (
     Position(..)
   , Parser
@@ -35,6 +36,7 @@ import Prelude hiding (takeWhile)
 import Data.Text (Text)
 import qualified Data.Text as T
 import Control.Monad
+import qualified Control.Monad.Fail as Fail
 import Control.Applicative
 import qualified Data.Set as Set
 
@@ -109,7 +111,9 @@ instance Alternative Parser where
 
 instance Monad Parser where
   return x = Parser $ \st -> Right (st, x)
-  fail e = Parser $ \st -> Left $ ParseError (position st) e
+#if !(MIN_VERSION_base(4,13,0))
+  fail = Fail.fail
+#endif
   p >>= g = Parser $ \st ->
     case evalParser p st of
          Left e        -> Left e
@@ -117,6 +121,9 @@ instance Monad Parser where
   {-# INLINE return #-}
   {-# INLINE (>>=) #-}
 
+instance Fail.MonadFail Parser where
+  fail e = Parser $ \st -> Left $ ParseError (position st) e
+
 instance MonadPlus Parser where
   mzero = Parser $ \st -> Left $ ParseError (position st) "(mzero)"
   mplus p1 p2 = Parser $ \st ->
