diff --git a/Agda.cabal b/Agda.cabal
index 9d476f5..715090f 100644
--- a/Agda.cabal
+++ b/Agda.cabal
@@ -40,7 +40,6 @@ extra-source-files: src/full/Agda/undefined.h
                     doc/release-notes/*.txt
 data-dir:           src/data
 data-files:         Agda.css
-                    emacs-mode/*.el
                     EpicInclude/AgdaPrelude.e
                     EpicInclude/stdagda.c
                     EpicInclude/stdagda.h
@@ -332,16 +331,3 @@ library
 
   if flag(debug)
     ghc-options:    -ddump-simpl-stats
-
-executable agda-mode
-  hs-source-dirs:   src/agda-mode
-  main-is:          Main.hs
-  other-modules:    Paths_Agda
-  if flag(use-locale)
-    build-depends:  base >= 4.2 && < 4.5
-  else
-    build-depends:  base == 4.1.*,
-                    utf8-string == 0.3.*
-  build-depends:    filepath >= 1.1 && < 1.3,
-                    process >= 1.0.1.0 && < 1.2,
-                    directory >= 1.0 && < 1.2
