--- agda-9999-orig/Agda.cabal	2011-09-18 18:59:32.000000000 +1000
+++ agda-9999/Agda.cabal	2011-10-01 22:33:04.381943543 +1000
@@ -40,7 +40,6 @@
                     doc/release-notes/*.txt
 data-dir:           src/data
 data-files:         Agda.css
-                    emacs-mode/*.el
                     EpicInclude/AgdaPrelude.e
                     EpicInclude/stdagda.c
                     EpicInclude/stdagda.h
@@ -335,15 +334,3 @@
 
   if flag(debug)
     ghc-options:    -ddump-simpl-stats
-executable agda-mode
-  hs-source-dirs:   src/agda-mode
-  main-is:          Main.hs
-  other-modules:    Paths_Agda
-  if flag(use-locale)
-    build-depends:  base >= 4.2 && < 4.5
-  else
-    build-depends:  base == 4.1.*,
-                    utf8-string == 0.3.*
-  build-depends:    filepath >= 1.1 && < 1.3,
-                    process >= 1.0.1.0 && < 1.2,
-                    directory >= 1.0 && < 1.2
