--- Agda-2.3.0.1-orig/Agda.cabal	2012-04-01 11:54:47.565833931 +1000
+++ Agda-2.3.0.1/Agda.cabal	2012-04-12 01:24:54.004394444 +1000
@@ -75,7 +75,7 @@
                     utf8-string == 0.3.*
   if flag(epic)
     build-depends:  epic >= 0.1.13 && < 0.10
-  build-depends:    mtl == 2.0.*,
+  build-depends:    mtl >= 2.0 && < 2.2,
                     QuickCheck >= 2.3 && < 2.5,
                     haskell-src-exts >= 1.9.6 && < 1.14,
                     containers >= 0.1 && < 0.5,
