diff -urN Agda-2.6.2.1/Agda.cabal Agda-2.6.2.1-r1/Agda.cabal
--- Agda-2.6.2.1/Agda.cabal	2022-03-17 23:42:57.826935880 -0600
+++ Agda-2.6.2.1-r1/Agda.cabal	2022-03-17 23:43:51.266935906 -0600
@@ -92,8 +92,7 @@
                     stack-9.2.1.yaml
 
 data-dir:           src/data
-data-files:         emacs-mode/*.el
-                    html/Agda.css
+data-files:         html/Agda.css
                     html/highlight-hover.js
                     JS/agda-rts.js
                     latex/agda.sty
