From a5a706f6d7f6943a02837a4c26a2dc3d52f8ee94 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Mon, 21 Feb 2011 22:13:43 +0200
Subject: [PATCH] Agda-2.2.6: loosen dep on QuickCheck

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 Agda.cabal |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Agda.cabal b/Agda.cabal
index c63c576..2c2269e 100644
--- a/Agda.cabal
+++ b/Agda.cabal
@@ -64,7 +64,7 @@ library
     build-depends: base == 4.1.*,
                    utf8-string == 0.3.*
   build-depends:   mtl >= 1.1 && < 2,
-                   QuickCheck >= 2.1.0.2 && < 2.2,
+                   QuickCheck >= 2.1.0.2 && < 3,
                    haskell-src >= 1.0.1.1 && < 2,
                    containers >= 0.1.0 && < 1,
                    pretty >= 1 && < 2,
-- 
1.7.3.4

