--- agda-9999-orig/Agda.cabal	2011-11-26 00:25:45.382153242 +1100
+++ agda-9999/Agda.cabal	2011-11-26 17:43:50.076056097 +1100
@@ -40,7 +40,6 @@
                     doc/release-notes/*.txt
 data-dir:           src/data
 data-files:         Agda.css
-                    emacs-mode/*.el
                     EpicInclude/AgdaPrelude.e
                     EpicInclude/stdagda.c
                     EpicInclude/stdagda.h
@@ -323,16 +322,3 @@
   if impl(ghc >= 7.2)
     ghc-options:    -fwarn-identities
   ghc-prof-options: -auto-all
-
-executable agda-mode
-  hs-source-dirs:   src/agda-mode
-  main-is:          Main.hs
-  other-modules:    Paths_Agda
-  if flag(use-locale)
-    build-depends:  base >= 4.2 && < 4.5
-  else
-    build-depends:  base == 4.1.*,
-                    utf8-string == 0.3.*
-  build-depends:    filepath >= 1.1 && < 1.3,
-                    process >= 1.0.1.0 && < 1.2,
-                    directory >= 1.0 && < 1.2
