commit fd2740c92d33cd9017df38ab1360f21c372ff0f8
Author: Simon Hengel <sol@typeful.net>
Date:   Mon Oct 2 10:38:02 2017 +0800

    Glob 0.9.0 compat

diff --git a/hpack.cabal b/hpack.cabal
diff --git a/src/Hpack/Util.hs b/src/Hpack/Util.hs
index a9e00b5..9a00261 100644
--- a/src/Hpack/Util.hs
+++ b/src/Hpack/Util.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 module Hpack.Util (
   List(..)
@@ -116,14 +117,29 @@ toPosixFilePath = Posix.joinPath . splitDirectories
 
 expandGlobs :: String -> FilePath -> [String] -> IO ([String], [FilePath])
 expandGlobs name dir patterns = do
-  files <- (fst <$> globDir compiledPatterns dir) >>= mapM removeDirectories
+  files <- globDir_ compiledPatterns dir >>= mapM removeDirectories
   let warnings = [warn pattern | ([], pattern) <- zip files patterns]
   return (warnings, combineResults files)
   where
+    globDir_ :: [Pattern] -> FilePath -> IO [[FilePath]]
+#if MIN_VERSION_Glob(0,9,0)
+    globDir_ = globDir
+#else
+    globDir_ xs = fmap fst . globDir xs
+#endif
+    combineResults :: [[FilePath]] -> [FilePath]
     combineResults = nub . sort . map (toPosixFilePath . makeRelative dir) . concat
+
+    warn :: String -> String
     warn pattern = "Specified pattern " ++ show pattern ++ " for " ++ name ++ " does not match any files"
+
+    compiledPatterns :: [Pattern]
     compiledPatterns = map (compileWith options) patterns
+
+    removeDirectories :: [FilePath] -> IO [FilePath]
     removeDirectories = filterM doesFileExist
+
+    options :: CompOptions
     options = CompOptions {
         characterClasses = False
       , characterRanges = False

