--- LDAP-0.6.8-orig/LDAP/Modify.hsc	2012-08-27 14:19:53.000000000 +1000
+++ LDAP-0.6.8/LDAP/Modify.hsc	2012-09-15 12:04:19.265703368 +1000
@@ -34,6 +34,9 @@
 import LDAP.Data
 import Foreign
 import Foreign.C.String
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
 import LDAP.Result
 import Control.Exception(finally)
 import Data.Bits
--- LDAP-0.6.8-orig/LDAP/Search.hsc	2012-08-27 14:19:53.000000000 +1000
+++ LDAP-0.6.8/LDAP/Search.hsc	2012-09-15 12:04:06.433361217 +1000
@@ -32,6 +32,9 @@
 import LDAP.Data
 import Foreign
 import Foreign.C.String
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
 import LDAP.Result
 import Control.Exception(finally)
 
--- LDAP-0.6.8-orig/LDAP/Result.hsc	2012-08-27 14:19:53.000000000 +1000
+++ LDAP-0.6.8/LDAP/Result.hsc	2012-09-15 12:03:30.552408835 +1000
@@ -27,6 +27,9 @@
 import LDAP.Utils
 import LDAP.Types
 import Foreign
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
 
 #include <ldap.h>
 
