diff --git a/src/StatusNotifier/Tray.hs b/src/StatusNotifier/Tray.hs
index 5d4170c..2aa04b5 100644
--- a/src/StatusNotifier/Tray.hs
+++ b/src/StatusNotifier/Tray.hs
@@ -123,7 +123,7 @@ getIconPixbufByName size name themePath = do
     maybeFile <- if fileExists
     then return $ Just nameString
     else fmap join $ sequenceA $ getIconPathFromThemePath nameString <$> themePath
-    sequenceA $ pixbufNewFromFile <$> maybeFile
+    fromJust <$> (sequenceA $ pixbufNewFromFile <$> maybeFile)
 
 getIconPathFromThemePath :: String -> String -> IO (Maybe String)
 getIconPathFromThemePath name themePath = if name == "" then return Nothing else do
