--- haskeline-0.6.4.5-orig/haskeline.cabal	2011-10-20 04:41:58.000000000 +1100
+++ haskeline-0.6.4.5/haskeline.cabal	2011-12-24 09:43:20.449830215 +1100
@@ -50,7 +50,7 @@
     }
     else {
         if impl(ghc>=6.11) {
-            Build-depends: base >=4.1 && < 4.5, containers>=0.1 && < 0.5, directory>=1.0 && < 1.2,
+            Build-depends: base >=4.1 && < 4.6, containers>=0.1 && < 0.5, directory>=1.0 && < 1.2,
                            bytestring==0.9.*
         }
         else {
@@ -58,7 +58,7 @@
                            bytestring==0.9.*
         }
     }
-    Build-depends:  filepath >= 1.1 && < 1.3, mtl >= 1.1 && < 2.1,
+    Build-depends:  filepath >= 1.1 && < 1.4, mtl >= 1.1 && < 2.1,
                     utf8-string==0.3.* && >=0.3.6,
                     extensible-exceptions==0.1.* && >=0.1.1.0
     Extensions:     ForeignFunctionInterface, Rank2Types, FlexibleInstances,
