diff --git a/haskeline.cabal b/haskeline.cabal
index a3a200f..119bb3b 100644
--- a/haskeline.cabal
+++ b/haskeline.cabal
@@ -50,7 +50,7 @@ Library
     }
     else {
         if impl(ghc>=6.11) {
-            Build-depends: base >=4.1 && < 4.4, containers>=0.1 && < 0.5, directory>=1.0 && < 1.2,
+            Build-depends: base >=4.1 && < 4.5, containers>=0.1 && < 0.5, directory>=1.0 && < 1.2,
                            bytestring==0.9.*
         }
         else {
@@ -102,7 +102,7 @@ Library
         install-includes: win_console.h
         cpp-options: -DMINGW
     } else {
-        Build-depends: unix>=2.0 && < 2.5
+        Build-depends: unix>=2.0 && < 2.6
                         -- unix-2.3 doesn't build on ghc-6.8.1 or earlier
         c-sources: cbits/h_iconv.c
                    cbits/h_wcwidth.c
diff --git a/setup b/setup
index 4cd648f..b2cdb6e 100755
Binary files a/setup and b/setup differ
