--- graphviz-2999.14.1.0-orig/graphviz.cabal	2012-08-29 21:05:46.000000000 +1000
+++ graphviz-2999.14.1.0/graphviz.cabal	2012-08-29 22:56:54.876467899 +1000
@@ -60,7 +60,7 @@
                            fgl == 5.4.*,
                            filepath,
                            polyparse >= 1.7 && < 1.9,
-                           bytestring == 0.9.*,
+                           bytestring >= 0.9 && < 0.11,
                            colour == 2.3.*,
                            transformers >= 0.2 && < 0.4,
                            text,
@@ -115,6 +115,7 @@
         Build-Depends:     base,
                            containers,
                            process,
+                           directory,
                            fgl,
                            filepath,
                            polyparse,
@@ -124,7 +125,7 @@
                            text,
                            wl-pprint-text,
                            dlist,
-                           QuickCheck >= 2.3 && < 2.5
+                           QuickCheck >= 2.3 && < 2.6
 
         hs-Source-Dirs:    . tests
 
--- graphviz-2999.14.1.0-orig/Data/GraphViz/Commands.hs	2012-08-29 21:05:46.000000000 +1000
+++ graphviz-2999.14.1.0/Data/GraphViz/Commands.hs	2012-08-29 22:58:55.180436409 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP, OverloadedStrings #-}
 
 {- |
    Module      : Data.GraphViz.Commands
@@ -46,7 +46,9 @@
     ) where
 
 -- Want to use the extensible-exception version
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding (catch)
+#endif
 
 import Data.GraphViz.Types
 -- This is here just for Haddock linking purposes.
