--- snap-core-0.9.0-orig/src/Snap/Util/GZip.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Util/GZip.hs	2012-07-07 23:48:25.907949060 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP                       #-}
 {-# LANGUAGE DeriveDataTypeable        #-}
 {-# LANGUAGE ExistentialQuantification #-}
 {-# LANGUAGE OverloadedStrings         #-}
@@ -24,7 +25,11 @@
 import qualified Data.Set as Set
 import           Data.Set (Set)
 import           Data.Typeable
-import           Prelude hiding (catch, takeWhile)
+import           Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+                                 catch,
+#endif
+                                 takeWhile)
 
 
 ----------------------------------------------------------------------------
--- snap-core-0.9.0-orig/src/Snap/Util/FileUploads.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Util/FileUploads.hs	2012-07-07 23:47:36.180929134 +1000
@@ -91,7 +91,11 @@
 import           Data.Text (Text)
 import qualified Data.Text.Encoding as TE
 import           Data.Typeable
-import           Prelude hiding (catch, getLine, takeWhile)
+import           Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+                                 catch,
+#endif
+                                 getLine, takeWhile)
 import           System.Directory
 import           System.IO hiding (isEOF)
 ------------------------------------------------------------------------------
--- snap-core-0.9.0-orig/src/Snap/Util/FileServe.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Util/FileServe.hs	2012-07-07 23:47:09.193375491 +1000
@@ -44,7 +44,11 @@
 import           Data.Monoid
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
-import           Prelude hiding (catch, show, Show)
+import           Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+                                 catch,
+#endif
+                                 show, Show)
 import qualified Prelude
 import           System.Directory
 import           System.FilePath
--- snap-core-0.9.0-orig/src/Snap/Internal/Instances.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Internal/Instances.hs	2012-07-07 23:46:38.973755450 +1000
@@ -1,4 +1,5 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE PackageImports #-}
 
 module Snap.Internal.Instances where
@@ -15,7 +16,9 @@
 import qualified Control.Monad.State.Lazy    as LState
 import           Control.Monad.Writer.Strict hiding (pass)
 import qualified Control.Monad.Writer.Lazy   as LWriter
+#if !MIN_VERSION_base(4,6,0)
 import           Prelude                     hiding (catch)
+#endif
 
 ------------------------------------------------------------------------------
 import           Snap.Internal.Types
--- snap-core-0.9.0-orig/src/Snap/Internal/Types.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Internal/Types.hs	2012-07-07 23:46:04.051038754 +1000
@@ -1,4 +1,5 @@
 {-# LANGUAGE BangPatterns              #-}
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DeriveDataTypeable        #-}
 {-# LANGUAGE ExistentialQuantification #-}
 {-# LANGUAGE FlexibleInstances         #-}
@@ -31,7 +32,11 @@
 import qualified Data.Text as T
 import qualified Data.Text.Lazy as LT
 import           Data.Typeable
-import           Prelude hiding (catch, take)
+import           Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+                                 catch,
+#endif
+                                 take)
 ------------------------------------------------------------------------------
 import           Snap.Internal.Http.Types
 import           Snap.Internal.Exceptions
--- snap-core-0.9.0-orig/src/Snap/Iteratee.hs	2012-06-18 08:09:07.000000000 +1000
+++ snap-core-0.9.0/src/Snap/Iteratee.hs	2012-07-07 23:45:01.266749937 +1000
@@ -127,7 +127,11 @@
 import           Foreign hiding (peek)
 import           Foreign.C.Types
 import           GHC.ForeignPtr
-import           Prelude hiding (catch, drop, head, take)
+import           Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+                                 catch,
+#endif
+                                 drop, head, take)
 import           System.IO
 
 #ifndef PORTABLE
