commit e4c888ec62da07419179affd5dda755c7d20bc0e
Author: Ryan Scott <ryan.gl.scott@gmail.com>
Date:   Fri Mar 15 20:33:25 2019 -0400

    Adapt to base-4.13.0.0

diff --git a/src/Snap/Internal/Core.hs b/src/Snap/Internal/Core.hs
index ceb81dd..738d762 100644
--- a/src/Snap/Internal/Core.hs
+++ b/src/Snap/Internal/Core.hs
@@ -311,7 +311,9 @@ instance Monad Snap where
     return = pure
     {-# INLINE return #-}
 #endif
+#if !(MIN_VERSION_base(4,13,0))
     fail   = Fail.fail
+#endif
 
 instance Fail.MonadFail Snap where
     fail   = snapFail
