diff --git a/Graphics/Rendering/Chart/Event.hs b/Graphics/Rendering/Chart/Event.hs
index bd2b79a..bf36321 100644
--- a/Graphics/Rendering/Chart/Event.hs
+++ b/Graphics/Rendering/Chart/Event.hs
@@ -74,3 +74,3 @@ instance ToPlot PlotEvent where
 
-renderPlotLegendEvent :: PlotEvent t e -> Rect -> ChartBackend ()
+renderPlotLegendEvent :: PlotEvent t e -> Rect -> BackendProgram ()
 renderPlotLegendEvent p r = return ()
@@ -78,6 +78,6 @@ renderPlotLegendEvent p r = return ()
 
-filledRect :: FillStyle -> Rect -> ChartBackend ()
+filledRect :: FillStyle -> Rect -> BackendProgram ()
 filledRect fs r = withFillStyle fs $ fillPath (rectPath r)
 
-framedRect :: LineStyle -> Rect -> ChartBackend ()
+framedRect :: LineStyle -> Rect -> BackendProgram ()
 framedRect ls r = withLineStyle ls $ strokePath (rectPath r)
@@ -87,3 +87,3 @@ pulseHeight = 15
 
-renderPlotEvent :: PlotEvent t e -> PointMapFn t e  -> ChartBackend ()
+renderPlotEvent :: PlotEvent t e -> PointMapFn t e  -> BackendProgram ()
 renderPlotEvent p pmap = do
