diff --git a/src/Protolude/Either.hs b/src/Protolude/Either.hs
index 28eab61..ba3d81d 100644
--- a/src/Protolude/Either.hs
+++ b/src/Protolude/Either.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE Safe #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 
@@ -8,12 +9,27 @@ module Protolude.Either (
 , rightToMaybe
 , maybeEmpty
 , maybeToEither
+, fromLeft
+, fromRight
 ) where
 
 import Data.Function (const)
 import Data.Monoid (Monoid, mempty)
 import Data.Maybe (Maybe(..), maybe)
 import Data.Either (Either(..), either)
+#if MIN_VERSION_base(4,10,0)
+import Data.Either (fromLeft, fromRight)
+#else
+-- | Return the contents of a 'Right'-value or a default value otherwise.
+fromLeft :: a -> Either a b -> a
+fromLeft _ (Left a) = a
+fromLeft a _        = a
+
+-- | Return the contents of a 'Right'-value or a default value otherwise.
+fromRight :: b -> Either a b -> b
+fromRight _ (Right b) = b
+fromRight b _         = b
+#endif
 
 leftToMaybe :: Either l r -> Maybe l
 leftToMaybe = either Just (const Nothing)

commit 8a8eb8c7282adf0fbb360806e32ad7f664018fef
Author: Michael Snoyman <michael@snoyman.com>
Date:   Mon May 28 13:21:12 2018 +0300

    Add extra-source-files (#87)

diff --git a/protolude.cabal b/protolude.cabal
index e87335c..16b5463 100644
--- a/protolude.cabal
+++ b/protolude.cabal
@@ -26,6 +26,7 @@ tested-with:
   GHC == 8.0.1,
   GHC == 8.2.1,
   GHC == 8.4.1
+extra-source-files: README.md CHANGES.md
 
 Source-Repository head
     type: git
