diff --git a/hoodle-core.cabal b/hoodle-core.cabal
index a2706b0..50b6809 100644
--- a/hoodle-core.cabal
+++ b/hoodle-core.cabal
@@ -85,2 +85,3 @@ Library
                    time >= 1.2, 
+                   time-locale-compat,
                    transformers >= 0.3,
diff --git a/src/Hoodle/Coroutine/ContextMenu.hs b/src/Hoodle/Coroutine/ContextMenu.hs
index 82a165c..e669263 100644
--- a/src/Hoodle/Coroutine/ContextMenu.hs
+++ b/src/Hoodle/Coroutine/ContextMenu.hs
@@ -4,2 +4,3 @@
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
 
diff --git a/src/Hoodle/Util.hs b/src/Hoodle/Util.hs
index 31b833f..7c0f8a9 100644
--- a/src/Hoodle/Util.hs
+++ b/src/Hoodle/Util.hs
@@ -24,2 +24,3 @@ import Data.Time.Clock
 import Data.Time.Format
+import qualified Data.Time.Locale.Compat as DTLC
 import Data.UUID.V4 (nextRandom)
@@ -89,3 +90,3 @@ errorlog str = do
   utctime <- getCurrentTime 
-  let timestr = formatTime defaultTimeLocale "%F %H:%M:%S %Z" utctime
+  let timestr = formatTime DTLC.defaultTimeLocale "%F %H:%M:%S %Z" utctime
   hPutStr outh (timestr ++ " : " )  
