diff --git a/src/Hoodle/View/Draw.hs b/src/Hoodle/View/Draw.hs
index ad34a49..fb99e36 100644
--- a/src/Hoodle/View/Draw.hs
+++ b/src/Hoodle/View/Draw.hs
@@ -67,3 +67,3 @@ newtype SinglePageDraw a =
                      -> CanvasId
-                     -> Bool                               -- ^ isCurrentCanvas
+                     -> Bool                               -- isCurrentCanvas
                      -> (Gtk.DrawingArea, Maybe Cairo.Surface) 
@@ -80,3 +80,3 @@ newtype ContPageDraw a =
   { unContPageDraw :: RenderCache
-                   -> Bool                                 -- ^ isCurrentCanvas 
+                   -> Bool                                 -- isCurrentCanvas
                    -> CanvasInfo ContinuousPage 
