diff --git a/src/Hoodle/GUI.hs b/src/Hoodle/GUI.hs
index 3e1ad41..489e767 100644
--- a/src/Hoodle/GUI.hs
+++ b/src/Hoodle/GUI.hs
@@ -60,10 +60,9 @@ startGUI mfname mhook = do
     setTitleFromFileName st0
-    -- need for refactoring
-    mapM_ (\(x,y :: Simple Lens Settings Bool) -> lensSetToggleUIForFlag x (settings.y) st0 )
-      [ ("UXINPUTA", doesUseXInput) 
-      , ("HANDA"   , doesUseTouch)
-      , ("POPMENUA", doesUsePopUpMenu)
-      , ("EBDIMGA" , doesEmbedImage)
-      , ("EBDPDFA" , doesEmbedPDF)
-      ] 
+
+    lensSetToggleUIForFlag "UXINPUTA" (settings.doesUseXInput) st0
+    lensSetToggleUIForFlag "HANDA"    (settings.doesUseTouch) st0
+    lensSetToggleUIForFlag "POPMENUA" (settings.doesUsePopUpMenu) st0
+    lensSetToggleUIForFlag "EBDIMGA"  (settings.doesEmbedImage) st0
+    lensSetToggleUIForFlag "EBDPDFA"  (settings.doesEmbedPDF) st0
+
     setToggleUIForFlag "TOGGLENETSRCA" False st0
