--- a/src/Control/Monad/Trans/Either.hs
+++ b/src/Control/Monad/Trans/Either.hs
@@ -14,2 +14,3 @@ import Control.Monad.Base
 import Control.Monad.Trans.Control
+import qualified Data.Semigroup as S
 
@@ -52,2 +53,5 @@ instance (MonadIO m) => MonadIO (EitherT l m) where
 
+instance (Applicative m, Monoid a) => S.Semigroup (EitherT l m a) where
+  (<>) = liftA2 mappend
+
 instance (Applicative m, Monoid a) => Monoid (EitherT l m a) where
