diff --git a/Check.hs b/Check.hs
index c5a74cd..c9e77ab 100644
--- a/Check.hs
+++ b/Check.hs
@@ -22,7 +22,7 @@ import qualified Data.Map as M
 import Control.Monad.State
 
 import Control.Concurrent
-import Control.Exception  (handle, finally)
+import qualified Control.Exception as E
 
 import System.Console.GetOpt
 import System.Environment
@@ -67,7 +67,7 @@ main = time $ do
 fork k cs f = flip mapM_ [1..k] $ \n -> do
     mv <- newEmptyMVar
     modifyMVar_ cs (return . (mv :))
-    forkIO (f n `finally` putMVar mv ())
+    forkIO (f n `E.finally` putMVar mv ())
 
 --
 -- just print out failed urls as they arrive
@@ -112,7 +112,7 @@ thread errs queue bad proxy n = loop
 
     run url = case parseURI url of
         Just uri -> do
-            mn <- handle (return . Left . show) (getStatus uri proxy)
+            mn <- (E.handle :: (E.IOException -> IO a) -> IO a -> IO a) (return . Left . show) (getStatus uri proxy)
             case mn of
                 Right 200 -> return ()
                 Right n   -> next (show n)
@@ -191,7 +191,7 @@ parseArgs = do
     die errs   = dump (concat errs ++ info) >> exitWith (ExitFailure 1)
     help       = dump info                  >> exitWith ExitSuccess
 
-getProxy = handle (\_ -> return Nothing) $ do
+getProxy = (E.handle :: (E.IOException -> IO a) -> IO a -> IO a) (\_ -> return Nothing) $ do
     env <- M.fromList `fmap` getEnvironment
     return $! do
         s <- M.lookup "http_proxy" env
