--- Setup.hs	2005-07-14 09:58:25.000000000 +0200
+++ Setup.hs	2005-07-28 19:06:11.000000000 +0200
@@ -8,15 +8,5 @@
 import Distribution.Setup
 import Distribution.Simple.LocalBuildInfo
 
-main = defaultMainWithHooks defaultUserHooks {postInst = addWrapperAndLib}
+main = defaultMainWithHooks defaultUserHooks
 
--- Install the c2hs shell script wrapper that passes the --data options as
--- well as the `C2HS' library module.
---
--- * We need to do this via a shell script that has been munged by ./configure,
---   as Cabal doesn't give us enough info (eg, no package name and version) to
---   do it all in Haskell.
---
-addWrapperAndLib :: Args -> InstallFlags -> LocalBuildInfo -> IO ExitCode
-addWrapperAndLib _ _ _ =
-  system "./postInst.sh"
