diff --git a/src/Repeats.lhs b/src/Repeats.lhs
index a22ee17..03662c7 100644
--- a/src/Repeats.lhs
+++ b/src/Repeats.lhs
@@ -77,3 +77,3 @@ clusters labels seqs = map (map mylookup) labels
 -- extract words from clusters
-debruijn :: Integral w => Int -> [Sequence a] -> S.Set w
+debruijn :: (Show w, Integral w) => Int -> [Sequence a] -> S.Set w
 debruijn k = foldl1' S.union . map (S.fromList . keys k . seqdata)
@@ -81,2 +81,3 @@ debruijn k = foldl1' S.union . map (S.fromList . keys k . seqdata)
 
+keys :: (Show w, Integral w) => Int -> SeqData -> [w]
 keys k = map fst . hashes (rcontig k)
@@ -97,3 +98,3 @@ toMap = M.fromList . map (\x -> (x,1)) . S.toList
 -- given word length k, calculate repeats from clusters
-repeats :: Integral w => Int -> [[Sequence a]] -> [(w,Int)]
+repeats :: (Show w, Integral w) => Int -> [[Sequence a]] -> [(w,Int)]
 repeats k = filter ((>1).snd) . M.toList . freqs . map (debruijn k)
