--- yesod-bin-1.2.11-orig/Devel.hs	2014-06-18 15:02:50.000000000 +1000
+++ yesod-bin-1.2.11/Devel.hs	2014-07-02 11:38:24.274462800 +1000
@@ -232,7 +232,7 @@
     mainOuterLoop iappPort filesModified = do
       ghcVer <- liftIO ghcVersion
       cabal  <- liftIO $ D.findPackageDesc "."
-      gpd    <- liftIO $ D.readPackageDescription D.normal cabal
+      gpd    <- liftIO $ D.readPackageDescription D.normal (either id id cabal)
       ldar   <- liftIO lookupLdAr
       (hsSourceDirs, _) <- liftIO $ checkCabalFile gpd
       liftIO $ removeFileIfExists (bd </> "setup-config")
@@ -244,8 +244,8 @@
                                                , "yesod-devel/arargs.txt"
                                                , "yesod-devel/ldargs.txt"
                                                ]
-             rebuild <- liftIO $ mkRebuild ghcVer cabal opts ldar
-             mainInnerLoop iappPort hsSourceDirs filesModified cabal rebuild
+             rebuild <- liftIO $ mkRebuild ghcVer (either id id cabal) opts ldar
+             mainInnerLoop iappPort hsSourceDirs filesModified (either id id cabal) rebuild
            else do
              liftIO (threadDelay 5000000)
              mainOuterLoop iappPort filesModified
