--- tagged-0.7-orig/src/Data/Tagged.hs	2013-11-24 12:30:34.678648455 +1100
+++ tagged-0.7/src/Data/Tagged.hs	2013-11-24 13:26:46.574616298 +1100
@@ -2,7 +2,7 @@
 #ifdef LANGUAGE_DeriveDataTypeable
 {-# LANGUAGE DeriveDataTypeable #-}
 #endif
-#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 706
+#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 706 && __GLASGOW_HASKELL__ < 707
 {-# LANGUAGE PolyKinds #-}
 #endif
 #if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ >= 702
