diff --git a/CONTRIBUTORS b/CONTRIBUTORS
index 080d3d0..c5dbd80 100644
--- a/CONTRIBUTORS
+++ b/CONTRIBUTORS
@@ -5,3 +5,4 @@ Edward Kmett
 Will Langstroth <will@langstroth.com>
 Shane O'Brien <shane@duairc.com>
 James Sanders <jimmyjazz14@gmail.com>
+Mark Wright <gienah@gentoo.org>
diff --git a/heist.cabal b/heist.cabal
index 931616d..b5b73a5 100644
--- a/heist.cabal
+++ b/heist.cabal
@@ -88,7 +88,7 @@ Library
     attoparsec                >= 0.10  && < 0.11,
     base                      >= 4     && < 5,
     blaze-builder             >= 0.2   && < 0.4,
-    blaze-html                >= 0.4   && < 0.5,
+    blaze-html                >= 0.4   && < 0.6,
     bytestring,
     containers                >= 0.2   && < 0.5,
     directory,
