diff --git a/flower.cabal b/flower.cabal
index 6f9a450..8e832e9 100644
--- a/flower.cabal
+++ b/flower.cabal
@@ -34,7 +34,7 @@ Executable     flower
     Other-Modules:  Print, Metrics, Options, Fork
     Hs-Source-Dirs: src
     Ghc-Options:    -Wall -threaded
-    Build-Depends:  bio >= 0.4.9, base >=3 && <5, array >= 0.1, bytestring >= 0.9.1, binary == 0.4.*, random, cmdargs >= 0.5, containers, mtl
+    Build-Depends:  bio >= 0.4.9, base >=3 && <5, array >= 0.1, bytestring >= 0.9.1, binary >= 0.4 && <0.8, random, cmdargs >= 0.5, containers, mtl
 
 Executable     flowselect
     Main-Is:        FlowSelect.hs
