diff --git a/OpenCLWrappers.cabal b/OpenCLWrappers.cabal
index 5ec09ab..1be4ef0 100644
--- a/OpenCLWrappers.cabal
+++ b/OpenCLWrappers.cabal
@@ -41,7 +41,7 @@ Library
                    System.OpenCL.Wrappers
   build-depends: base < 4.6,
                  bytestring -any,
-                 mtl >= 2 && < 2.1
+                 mtl >= 2 && < 2.2
   exposed: True
   buildable: True
   hs-source-dirs: .
