--- cpphs.cabal	2006-05-08 16:04:18.000000000 +0100
+++ cpphs.cabal.new	2006-05-08 19:51:04.000000000 +0100
@@ -23,9 +23,5 @@
 	Language.Preprocessor.Cpphs
 	Language.Preprocessor.Unlit
-
-Executable: cpphs
-Main-Is: cpphs.hs
 Other-Modules:
-	Language.Preprocessor.Cpphs,
 	Language.Preprocessor.Cpphs.CppIfdef,
 	Language.Preprocessor.Cpphs.HashDefine,
@@ -37,5 +33,6 @@
 	Language.Preprocessor.Cpphs.SymTab,
 	Language.Preprocessor.Cpphs.Tokenise,
-	Language.Preprocessor.Unlit
 	Text.ParserCombinators.HuttonMeijer
 
+Executable: cpphs
+Main-Is: cpphs.hs
