diff --git a/representable-functors.cabal b/representable-functors.cabal
index b8ffb76..5e2c440 100644
--- a/representable-functors.cabal
+++ b/representable-functors.cabal
@@ -42,17 +42,17 @@ library
   build-depends:
     array                >= 0.3.0.2 && < 0.6,
     base                 >= 4       && < 5,
-    comonad              >= 3       && < 4,
-    comonad-transformers >= 3       && < 4,
-    comonads-fd          >= 3       && < 4,
+    comonad              >= 4       && < 5,
+    comonad-transformers >= 3       && < 5,
+    comonads-fd          >= 3       && < 5,
     containers           >= 0.3     && < 0.6,
     contravariant        >= 0.4.1   && < 1,
     distributive         >= 0.2.2   && < 1,
-    free                 >= 3       && < 4,
+    free                 >= 3       && < 5,
     keys                 >= 3       && < 4,
     mtl                  >= 2.0.1.0 && < 2.2,
     semigroups           >= 0.8.3.1 && < 1,
-    semigroupoids        >= 3       && < 4,
+    semigroupoids        >= 3       && < 5,
     transformers         >= 0.2     && < 0.4
 
   exposed-modules:
diff --git a/src/Control/Comonad/Representable/Store.hs b/src/Control/Comonad/Representable/Store.hs
index 168efbc..79589ba 100644
--- a/src/Control/Comonad/Representable/Store.hs
+++ b/src/Control/Comonad/Representable/Store.hs
@@ -109,7 +109,7 @@ instance Indexable g => ComonadTrans (StoreT g) where
   lower (StoreT w s) = fmap (`index` s) w
 
 instance ComonadHoist (StoreT g) where
-  cohoist (StoreT w s) = StoreT (Identity (extract w)) s
+  cohoist l (StoreT w s) = StoreT (l w) s
 
 instance (ComonadTraced m w, Representable g) => ComonadTraced m (StoreT g w) where
   trace m = trace m . lower
