diff --git a/templates/home.hamlet b/templates/home.hamlet
--- /dev/null
+++ b/templates/home.hamlet
@@ -0,0 +1,41 @@
+<h1>Workers
+
+<table.table.table-striped.table-bordered.table-hover>
+  <thead>
+    <tr>
+  <tbody>
+    <tr>
+      <th>Processes
+      <td>
+        <ul>
+          $forall proc <- sProcs
+            <li>#{proc}
+    <tr>
+      <th>Complete
+      <td>#{sDone}
+    <tr>
+      <th>Failed
+      <td>#{sFailed}
+    <tr>
+      <th>Busy
+      <td>#{get "busy"}
+    <tr>
+      <th>Last Checkin
+      <td>#{get "beat"}
+    <tr>
+      <th>Info
+      <td>
+        <pre>#{get "info"}
+
+<h2>Queues
+
+<table.table.table-striped.table-bordered.table-hover>
+  <thead>
+    <tr>
+      <th>Name
+      <th>Size
+  <tbody>
+    $forall (q,n) <- sQueues
+      <tr>
+        <td>#{q}
+        <td>#{n}
\ No newline at end of file
