--- a/src/Cabal.hs
+++ b/src/Cabal.hs
@@ -20,3 +20,3 @@ import Data.Monoid
 import Prelude
-
+import qualified Data.Semigroup as S
 
@@ -55,2 +55,5 @@ data Cabal = Cabal
 
+instance S.Semigroup Cabal where
+    (<>) (Cabal x1 x2) (Cabal y1 y2) = Cabal (x1?:y1) (x2++y2)
+
 instance Monoid Cabal where
@@ -91,2 +94,6 @@ data CabalSection = CabalSection
 
+instance S.Semigroup CabalSection where
+    (<>) (CabalSection x1 x2 x3 x4 x5 x6) (CabalSection y1 y2 y3 y4 y5 y6) =
+        CabalSection x1 (x2?:y2) (x3<>y3) (x4<>y4) (x5<>y5) (x6<>y6)
+
 instance Monoid CabalSection where
--- a/src/Hi.hs
+++ b/src/Hi.hs
@@ -27,2 +27,3 @@ import System.IO.Extra
 import Prelude
+import qualified Data.Semigroup as S
 
@@ -57,2 +58,16 @@ instance NFData Hi
 
+instance S.Semigroup Hi where
+    (<>) x y = Hi
+        {hiModuleName = f (?:) hiModuleName
+        ,hiImportPackage = f (<>) hiImportPackage
+        ,hiExportIdent = f (<>) hiExportIdent
+        ,hiImportIdent = f (<>) hiImportIdent
+        ,hiImportModule = f (<>) hiImportModule
+        ,hiImportPackageModule = f (<>) hiImportPackageModule
+        ,hiImportOrphan = f (<>) hiImportOrphan
+        ,hiSignatures = f (Map.unionWith (<>)) hiSignatures
+        ,hiFieldName = f (<>) hiFieldName
+        }
+        where f op sel = sel x `op` sel y
+
 instance Monoid Hi where
