From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Disable tests that depend on connecting to the internet

diff -urN dhall-1.40.1/dhall.cabal dhall-1.40.1-new1/dhall.cabal
--- dhall-1.40.1/dhall.cabal	2021-10-31 01:27:47.044398924 -0600
+++ dhall-1.40.1-new1/dhall.cabal	2021-10-31 01:46:26.061998071 -0600
@@ -696,9 +696,6 @@
         turtle                                   < 1.6 ,
         unordered-containers                           ,
         vector                    >= 0.11.0.0 && < 0.13
-    if flag(with-http)
-      CPP-Options:
-        -DWITH_HTTP
     Default-Language: Haskell2010
 
 Test-Suite doctest

diff -urN dhall-1.40.1-new2/src/Dhall/Tutorial.hs dhall-1.40.1-new3/src/Dhall/Tutorial.hs
--- dhall-1.40.1-new2/src/Dhall/Tutorial.hs	2001-09-08 19:46:40.000000000 -0600
+++ dhall-1.40.1-new3/src/Dhall/Tutorial.hs	2021-10-31 02:33:56.005732603 -0600
@@ -375,13 +375,13 @@
 --
 -- ... and you can reference that expression either directly:
 --
--- >>> input auto "https://raw.githubusercontent.com/dhall-lang/dhall-haskell/18e4e9a18dc53271146df3ccf5b4177c3552236b/examples/True" :: IO Bool
--- True
+-- > input auto "https://raw.githubusercontent.com/dhall-lang/dhall-haskell/18e4e9a18dc53271146df3ccf5b4177c3552236b/examples/True" :: IO Bool
+-- > True
 --
 -- ... or inside of a larger expression:
 --
--- >>> input auto "False == https://raw.githubusercontent.com/dhall-lang/dhall-haskell/18e4e9a18dc53271146df3ccf5b4177c3552236b/examples/True" :: IO Bool
--- False
+-- > input auto "False == https://raw.githubusercontent.com/dhall-lang/dhall-haskell/18e4e9a18dc53271146df3ccf5b4177c3552236b/examples/True" :: IO Bool
+-- > False
 --
 -- You're not limited to hosting Dhall expressions on GitHub.  You can host a
 -- Dhall expression anywhere that you can host UTF8-encoded text on the web,
