From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Disable buggy tests
* success/unit/asLocation/RemoteChainEnvA.dhall

diff -urN dhall-1.40.1-new3/tests/Dhall/Test/Import.hs dhall-1.40.1-new4/tests/Dhall/Test/Import.hs
--- dhall-1.40.1-new3/tests/Dhall/Test/Import.hs	2021-10-31 01:45:59.804635519 -0600
+++ dhall-1.40.1-new4/tests/Dhall/Test/Import.hs	2021-10-31 02:54:31.081674159 -0600
@@ -45,6 +45,7 @@
             , importDirectory </> "success/unit/asLocation/RemoteChain2A.dhall"
             , importDirectory </> "success/unit/asLocation/RemoteChain3A.dhall"
             , importDirectory </> "success/unit/asLocation/RemoteChainMissingA.dhall"
+            , importDirectory </> "success/unit/asLocation/RemoteChainEnvA.dhall"
             ]
 
     successTests <- Test.Util.discover (Turtle.chars <* "A.dhall") successTest (do
