From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

These patch allows the tests to compile.

diff -urN semver-range-0.2.8/tests/Unit.hs semver-range-0.2.8-new/tests/Unit.hs
--- semver-range-0.2.8/tests/Unit.hs	2019-06-24 11:47:23.000000000 -0600
+++ semver-range-0.2.8-new/tests/Unit.hs	2021-05-13 20:37:10.221048799 -0600
@@ -16,6 +16,7 @@
 import Test.Hspec
 import Test.QuickCheck (property, Arbitrary(..), oneof)
 import qualified Data.Text as T
+import Control.Monad.Fail (fail)
 
 import Data.SemVer
 
