--- pool-0.1.2-orig/pool.cabal	2011-12-05 22:31:06.000000000 +1100
+++ pool-0.1.2/pool.cabal	2011-12-12 00:00:53.218829554 +1100
@@ -16,6 +16,7 @@
     build-depends:   base                     >= 4         && < 5
                    , transformers             >= 0.2.1     && < 0.3
                    , monad-control            >= 0.2       && < 0.4
+                   , lifted-base              >= 0.1       && < 0.2
     exposed-modules: Data.Pool
     ghc-options:     -Wall
 
--- pool-0.1.2-orig/Data/Pool.hs	2011-12-05 22:31:06.000000000 +1100
+++ pool-0.1.2/Data/Pool.hs	2011-12-12 00:00:15.760938479 +1100
@@ -23,6 +23,7 @@
 import Data.Typeable
 #if MIN_VERSION_monad_control(0, 3, 0)
 import qualified Control.Monad.Trans.Control as I
+import qualified Control.Exception.Lifted as I
 #else
 import qualified Control.Monad.IO.Control as I
 import qualified Control.Exception.Control as I
