diff --git a/pool.cabal b/pool.cabal
index c006617..17a336b 100644
--- a/pool.cabal
+++ b/pool.cabal
@@ -14,7 +14,7 @@ homepage:        http://www.yesodweb.com/book/persistent
 
 library
     build-depends:   base                     >= 4         && < 5
-                   , transformers             >= 0.2.1     && < 0.3
+                   , transformers             >= 0.2.1     && < 0.4
                    , monad-control            >= 0.2       && < 0.4
                    , lifted-base              >= 0.1       && < 0.2
     exposed-modules: Data.Pool
