diff --git a/blaze-textual.cabal b/blaze-textual.cabal
index c35b506..6d05d80 100644
--- a/blaze-textual.cabal
+++ b/blaze-textual.cabal
@@ -48,7 +48,7 @@ library
 
   if impl(ghc >= 6.11)
     cpp-options: -DINTEGER_GMP
-    build-depends: integer-gmp >= 0.2 && < 0.3
+    build-depends: integer-gmp >= 0.2 && < 0.4
 
   if impl(ghc >= 6.9) && impl(ghc < 6.11)
     cpp-options: -DINTEGER_GMP
diff --git a/setup b/setup
index 5b401dc..7c35894 100755
Binary files a/setup and b/setup differ
