diff --git a/src/Fay/Compiler/Prelude.hs b/src/Fay/Compiler/Prelude.hs
index 45f067c..ac6f149 100644
--- a/src/Fay/Compiler/Prelude.hs
+++ b/src/Fay/Compiler/Prelude.hs
@@ -43,5 +43,5 @@ import           Data.Monoid          (Monoid (..), (<>))
 import           Data.Ord
 import           Data.Traversable
-import           Prelude.Compat       hiding (exp, mod)
+import           Prelude.Compat       hiding (exp, mod, (<>))
 import           Safe
 
diff --git a/src/haskell-names/Language/Haskell/Names/Open/Derived.hs b/src/haskell-names/Language/Haskell/Names/Open/Derived.hs
index c2ecb76..b219a56 100644
--- a/src/haskell-names/Language/Haskell/Names/Open/Derived.hs
+++ b/src/haskell-names/Language/Haskell/Names/Open/Derived.hs
@@ -5,4 +5,5 @@
 {-# LANGUAGE TemplateHaskell       #-}
 {-# LANGUAGE UndecidableInstances  #-}
+{-# LANGUAGE CPP  #-}
 module Language.Haskell.Names.Open.Derived where
 
@@ -86,2 +87,6 @@ deriveGTraversable ''Unpackedness
 deriveGTraversable ''ResultSig
 deriveGTraversable ''InjectivityInfo
+#if MIN_VERSION_haskell_src_exts(1,20,0)
+deriveGTraversable ''DerivStrategy
+deriveGTraversable ''MaybePromotedName
+#endif
