commit c0874849815ada4d1da25fd92247e821ade3367d
Author: Adam Bergmark <adam@bergmark.nl>
Date:   Tue Aug 16 10:40:01 2016 +0200

    Allow optparse-applicative 0.13.*

diff --git a/fay.cabal b/fay.cabal
index 5832e03..2defa86 100644
--- a/fay.cabal
+++ b/fay.cabal
@@ -177,7 +177,7 @@ executable fay
       base
     , fay
     , mtl
-    , optparse-applicative >= 0.11 && < 0.13
+    , optparse-applicative >= 0.11 && < 0.14
     , split
 
 executable fay-tests
diff --git a/src/main/Main.hs b/src/main/Main.hs
index 0109cf8..6e28cda 100644
--- a/src/main/Main.hs
+++ b/src/main/Main.hs
@@ -3,16 +3,20 @@
 module Main where
 
 import           Fay
-import           Paths_fay (version)
+import           Paths_fay                 (version)
 
+import           Control.Applicative       ((<|>))
+import qualified Control.Exception         as E
 import           Control.Monad
 import           Control.Monad.Reader
-import           Data.List.Split (wordsBy)
+import           Data.List.Split           (wordsBy)
 import           Data.Maybe
-import           Data.Version (showVersion)
-import           Options.Applicative
-import           Options.Applicative.Types
-import qualified Control.Exception as E
+import           Data.Monoid               ((<>))
+import           Data.Version              (showVersion)
+import           Options.Applicative       (Mod, OptionFields, Parser, argument, execParser, fullDesc, header, help,
+                                            helper, info, long, many, metavar, option, optional, short, strOption,
+                                            switch, value)
+import           Options.Applicative.Types (ReadM (ReadM))
 
 -- | Options and help.
 data FayCompilerOptions = FayCompilerOptions
