--- keter-1.4.3.2-orig/Keter/Main.hs	2015-12-20 04:17:16.000000000 +1100
+++ keter-1.4.3.2/Keter/Main.hs	2018-04-16 13:08:15.686780658 +1000
@@ -38,8 +38,11 @@
 import qualified Data.Text.Read
 import           Data.Time                 (getCurrentTime)
 import           Data.Yaml.FilePath
-import qualified Network.HTTP.Conduit      as HTTP (conduitManagerSettings,
-                                                    newManager)
+#if MIN_VERSION_http_conduit(2, 3, 0)
+import qualified Network.HTTP.Client       as HTTP (defaultManagerSettings, newManager)
+#else
+import qualified Network.HTTP.Conduit      as HTTP (conduitManagerSettings, newManager)
+#endif
 import           Prelude                   hiding (FilePath, log)
 import           System.Directory          (createDirectoryIfMissing,
                                             createDirectoryIfMissing,
@@ -208,7 +211,11 @@
 
 startListening :: KeterConfig -> HostMan.HostManager -> IO ()
 startListening KeterConfig {..} hostman = do
+#if MIN_VERSION_http_conduit(2, 3, 0)
+    manager <- HTTP.newManager HTTP.defaultManagerSettings
+#else
     manager <- HTTP.newManager HTTP.conduitManagerSettings
+#endif
     runAndBlock kconfigListeners $ Proxy.reverseProxy
         kconfigIpFromHeader
         -- calculate the number of microseconds since the
