--- a/Keter/Main.hs
+++ b/Keter/Main.hs
@@ -155,6 +155,12 @@ getIncoming kc = kconfigDir kc </> "incoming"
 isKeter :: FilePath -> Bool
 isKeter fp = takeExtension fp == ".keter"
 
+#if MIN_VERSION_fsnotify(0,3,0)
+#  define UNUSED_ARG _
+#else
+#  define UNUSED_ARG
+#endif
+
 startWatching :: KeterConfig -> AppMan.AppManager -> (LogMessage -> IO ()) -> IO ()
 startWatching kc@KeterConfig {..} appMan log = do
     -- File system watching
@@ -162,13 +168,13 @@ startWatching kc@KeterConfig {..} appMan log = do
     _ <- FSN.watchTree wm (fromString incoming) (const True) $ \e -> do
         e' <-
             case e of
-                FSN.Removed fp _ -> do
+                FSN.Removed fp _ UNUSED_ARG -> do
                     log $ WatchedFile "removed" (fromFilePath fp)
                     return $ Left $ fromFilePath fp
-                FSN.Added fp _ -> do
+                FSN.Added fp _ UNUSED_ARG -> do
                     log $ WatchedFile "added" (fromFilePath fp)
                     return $ Right $ fromFilePath fp
-                FSN.Modified fp _ -> do
+                FSN.Modified fp _ UNUSED_ARG -> do
                     log $ WatchedFile "modified" (fromFilePath fp)
                     return $ Right $ fromFilePath fp
         case e' of
