--- keter-1.4.3.2-orig/Data/Conduit/Process/Unix.hs	2016-08-17 09:25:08.000000000 +1000
+++ keter-1.4.3.2/Data/Conduit/Process/Unix.hs	2018-04-16 12:51:17.954089572 +1000
@@ -61,7 +61,9 @@
                                                   ProcessHandle__ (..))
 
 processHandleMVar :: ProcessHandle -> MVar ProcessHandle__
-#if MIN_VERSION_process(1, 2, 0)
+#if MIN_VERSION_process(1, 6, 0)
+processHandleMVar (ProcessHandle m _ _) = m
+#elif MIN_VERSION_process(1, 2, 0)
 processHandleMVar (ProcessHandle m _) = m
 #else
 processHandleMVar (ProcessHandle m) = m
