diff --git a/Text/Megaparsec/Error.hs b/Text/Megaparsec/Error.hs
index 911c55e..68cfd5c 100644
--- a/Text/Megaparsec/Error.hs
+++ b/Text/Megaparsec/Error.hs
@@ -44,2 +44,3 @@ import Prelude hiding (concat)
 import Test.QuickCheck hiding (label)
+import Test.QuickCheck.Instances () -- NonEmpty
 import qualified Data.List.NonEmpty as NE
diff --git a/megaparsec.cabal b/megaparsec.cabal
index 1fa4b66..a0271b8 100644
--- a/megaparsec.cabal
+++ b/megaparsec.cabal
@@ -63,2 +63,3 @@ library
   build-depends:      QuickCheck   >= 2.7   && < 3.0
+                    , quickcheck-instances
                     , base         >= 4.6   && < 5.0
