--- tagsoup-0.12.6-orig/tagsoup.cabal	2011-12-12 02:21:42.000000000 +1100
+++ tagsoup-0.12.6/tagsoup.cabal	2012-08-17 15:52:39.957588113 +1000
@@ -58,7 +58,7 @@
 executable tagsoup
     if flag(testprog)
         buildable: True
-        build-depends: QuickCheck == 2.4.*, time, deepseq == 1.1.*, directory, process
+        build-depends: QuickCheck >= 2.4 && < 2.6, time, deepseq, directory, process
     else
         buildable: False
 
--- tagsoup-0.12.6-orig/TagSoup/Benchmark.hs	2011-12-12 02:21:42.000000000 +1100
+++ tagsoup-0.12.6/TagSoup/Benchmark.hs	2012-08-17 15:58:58.090370173 +1000
@@ -1,4 +1,4 @@
-
+{-# LANGUAGE CPP #-}
 module TagSoup.Benchmark where
 
 import Text.HTML.TagSoup
@@ -156,12 +156,13 @@
     rnf (TagWarning x) = rnf x
     rnf (TagPosition x y) = () -- both are already ! bound
 
+#if !MIN_VERSION_bytestring(0,10,0)
 instance NFData LBS.ByteString where
     rnf x = LBS.length x `seq` ()
 
 instance NFData BS.ByteString where
     rnf x = BS.length x `seq` ()
-
+#endif
 
 ---------------------------------------------------------------------
 -- STATISTICS
