--- tagsoup.cabal.orig	2010-01-08 15:52:55.963484267 +1000
+++ tagsoup.cabal	2010-01-08 16:04:43.138483937 +1000
@@ -49,9 +49,9 @@
     ghc-options: -O2
     if flag(testprog)
         buildable: True
+        build-depends: QuickCheck == 2.1.*, time, deepseq == 1.1.0.0, HTTP
     else
         buildable: False
-    build-depends: QuickCheck == 2.1.*, time, deepseq == 1.1.0.0, HTTP
 
     main-is: Main.hs
     other-modules:
