diff --git a/Text/HTML/TagSoup.hs b/Text/HTML/TagSoup.hs
index 06e2739..9820dce 100644
--- a/Text/HTML/TagSoup.hs
+++ b/Text/HTML/TagSoup.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TypeSynonymInstances, PatternGuards #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 {-|
     This module is for working with HTML/XML. It deals with both well-formed XML and
diff --git a/Text/StringLike.hs b/Text/StringLike.hs
index d9457fa..60bfc41 100644
--- a/Text/StringLike.hs
+++ b/Text/StringLike.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 -- | /WARNING/: This module is /not/ intended for use outside the TagSoup library.
 --
