--- lifted-base-0.1.0.1-orig/lifted-base.cabal	2011-12-03 22:38:23.000000000 +1100
+++ lifted-base-0.1.0.1/lifted-base.cabal	2011-12-20 12:00:44.848034768 +1100
@@ -42,7 +42,7 @@
                    Control.Concurrent.Lifted
                    System.Timeout.Lifted
 
-  Build-depends: base                 >= 3     && < 4.5
+  Build-depends: base                 >= 3     && < 4.6
                , base-unicode-symbols >= 0.1.1 && < 0.3
                , transformers-base    >= 0.4   && < 0.5
                , monad-control        >= 0.3   && < 0.4
@@ -58,7 +58,7 @@
   type:    exitcode-stdio-1.0
   main-is: test.hs
 
-  build-depends: base                 >= 3     && < 4.5
+  build-depends: base                 >= 3     && < 4.6
                , base-unicode-symbols >= 0.1.1 && < 0.3
                , transformers         >= 0.2   && < 0.3
                , transformers-base    >= 0.4   && < 0.5
