--- lifted-base-0.1.1.1-orig/Control/Concurrent/Lifted.hs	2012-06-30 21:36:37.000000000 +1000
+++ lifted-base-0.1.1.1/Control/Concurrent/Lifted.hs	2012-08-10 20:58:01.426044266 +1000
@@ -54,8 +54,10 @@
     , module Control.Concurrent.SampleVar.Lifted
 
       -- * Merging of streams
+#if __GLASGOW_HASKELL__ < 707
     , merge
     , nmerge
+#endif
 
       -- * Bound Threads
     , forkOS
@@ -197,6 +199,8 @@
 threadWaitWrite = liftBase ∘ C.threadWaitWrite
 {-# INLINABLE threadWaitWrite #-}
 
+#if __GLASGOW_HASKELL__ < 707
+
 -- | Generalized version of 'C.mergeIO'.
 merge ∷ MonadBase IO m ⇒ [a] → [a] → m [a]
 merge xs ys = liftBase $ C.mergeIO xs ys
@@ -207,6 +211,8 @@
 nmerge = liftBase ∘ C.nmergeIO
 {-# INLINABLE nmerge #-}
 
+#endif
+
 -- | Generalized version of 'C.forkOS'.
 --
 -- Note that, while the forked computation @m ()@ has access to the captured
--- lifted-base-0.1.1.1-orig/Control/Concurrent/QSemN/Lifted.hs	2012-06-30 21:36:37.000000000 +1000
+++ lifted-base-0.1.1.1/Control/Concurrent/QSemN/Lifted.hs	2012-08-10 20:56:54.365887435 +1000
@@ -20,12 +20,17 @@
 -}
 
 module Control.Concurrent.QSemN.Lifted
-    ( QSemN
+    ( 
+#if __GLASGOW_HASKELL__ < 707
+      QSemN
     , newQSemN
     , waitQSemN
     , signalQSemN
+#endif
     ) where
 
+#if __GLASGOW_HASKELL__ < 707
+
 --------------------------------------------------------------------------------
 -- Imports
 --------------------------------------------------------------------------------
@@ -63,3 +68,4 @@
 signalQSemN sem = liftBase ∘ QSemN.signalQSemN sem
 {-# INLINABLE signalQSemN #-}
 
+#endif
--- lifted-base-0.1.1.1-orig/Control/Concurrent/QSem/Lifted.hs	2012-06-30 21:36:37.000000000 +1000
+++ lifted-base-0.1.1.1/Control/Concurrent/QSem/Lifted.hs	2012-08-10 20:55:47.793745444 +1000
@@ -20,12 +20,17 @@
 -}
 
 module Control.Concurrent.QSem.Lifted
-    ( QSem
+    ( 
+#if __GLASGOW_HASKELL__ < 707
+      QSem
     , newQSem
     , waitQSem
     , signalQSem
+#endif
     ) where
 
+#if __GLASGOW_HASKELL__ < 707
+
 --------------------------------------------------------------------------------
 -- Imports
 --------------------------------------------------------------------------------
@@ -63,3 +68,4 @@
 signalQSem = liftBase ∘ QSem.signalQSem
 {-# INLINABLE signalQSem #-}
 
+#endif
--- lifted-base-0.1.1.1-orig/Control/Concurrent/SampleVar/Lifted.hs	2012-06-30 21:36:37.000000000 +1000
+++ lifted-base-0.1.1.1/Control/Concurrent/SampleVar/Lifted.hs	2012-08-10 20:57:30.537050901 +1000
@@ -20,15 +20,20 @@
 -}
 
 module Control.Concurrent.SampleVar.Lifted
-    ( SampleVar
+    ( 
+#if __GLASGOW_HASKELL__ < 707
+      SampleVar
     , newEmptySampleVar
     , newSampleVar
     , emptySampleVar
     , readSampleVar
     , writeSampleVar
     , isEmptySampleVar
+#endif
     ) where
 
+#if __GLASGOW_HASKELL__ < 707
+
 --------------------------------------------------------------------------------
 -- Imports
 --------------------------------------------------------------------------------
@@ -81,3 +86,4 @@
 isEmptySampleVar = liftBase ∘ SampleVar.isEmptySampleVar
 {-# INLINABLE isEmptySampleVar #-}
 
+#endif
