--- lifted-base-0.1.1-orig/test/test.hs	2012-04-28 23:47:19.000000000 +1000
+++ lifted-base-0.1.1/test/test.hs	2012-06-29 01:56:15.155311373 +1000
@@ -1,7 +1,9 @@
-{-# LANGUAGE DeriveDataTypeable, FlexibleContexts #-}
+{-# LANGUAGE CPP, DeriveDataTypeable, FlexibleContexts, ScopedTypeVariables #-}
 
 -- from base:
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding (catch)
+#endif
 import Data.IORef
 import Data.Maybe
 import Data.Typeable (Typeable)
--- lifted-base-0.1.1-orig/lifted-base.cabal	2012-04-28 23:47:19.000000000 +1000
+++ lifted-base-0.1.1/lifted-base.cabal	2012-06-29 01:33:29.498444028 +1000
@@ -43,7 +43,7 @@
                    Control.Concurrent.Lifted
                    System.Timeout.Lifted
 
-  Build-depends: base                 >= 3     && < 4.6
+  Build-depends: base                 >= 3     && < 4.7
                , base-unicode-symbols >= 0.1.1 && < 0.3
                , transformers-base    >= 0.4   && < 0.5
                , monad-control        >= 0.3   && < 0.4
@@ -61,7 +61,7 @@
   hs-source-dirs: test
 
   build-depends: lifted-base
-               , base                 >= 3     && < 4.6
+               , base                 >= 3     && < 4.7
                , transformers         >= 0.2   && < 0.4
                , transformers-base    >= 0.4   && < 0.5
                , monad-control        >= 0.3   && < 0.4
@@ -84,7 +84,7 @@
   ghc-options:    -O2
 
   build-depends: lifted-base
-               , base          >= 3   && < 4.6
+               , base          >= 3   && < 4.7
                , transformers  >= 0.2 && < 0.4
                , criterion     >= 0.5 && < 0.7
                , monad-control >= 0.3 && < 0.4
