diff --git a/src/Network/HaskellNet/SSL.hs b/src/Network/HaskellNet/SSL.hs
index acb9b84..cb5e613 100644
--- a/src/Network/HaskellNet/SSL.hs
+++ b/src/Network/HaskellNet/SSL.hs
@@ -1,9 +1,14 @@
+{-# LANGUAGE CPP #-}
 module Network.HaskellNet.SSL
   ( Settings (..)
   , defaultSettingsWithPort
   ) where
 
+#if MIN_VERSION_network(3,0,0)
+import Network.Socket (PortNumber)
+#else
 import Network.Socket.Internal (PortNumber)
+#endif
 
 data Settings = Settings
               { sslPort                        :: PortNumber
