--- a/Sound/Tidal/Parse.hs
+++ b/Sound/Tidal/Parse.hs
@@ -21,2 +21,3 @@ import Sound.Tidal.Pattern
 import Sound.Tidal.Time (Arc, Time)
+import qualified Data.Semigroup as S
 
@@ -42,2 +43,5 @@ data TPat a where
 
+instance Parseable a => S.Semigroup (TPat a) where
+   (<>) = TPat_Overlay
+
 instance Parseable a => Monoid (TPat a) where
--- a/Sound/Tidal/Pattern.hs
+++ b/Sound/Tidal/Pattern.hs
@@ -24,2 +24,3 @@ import Sound.Tidal.Bjorklund
 import Text.Show.Functions ()
+import qualified Data.Semigroup as S
 
@@ -162,2 +163,5 @@ instance Applicative Pattern where
 
+instance S.Semigroup (Pattern a) where
+    (<>) = overlay
+
 -- | @mempty@ is a synonym for @silence@.
