diff --git a/src/Data/Validator/Draft4/Array.hs b/src/Data/Validator/Draft4/Array.hs
index 48bf84a..b96e23b 100644
--- a/src/Data/Validator/Draft4/Array.hs
+++ b/src/Data/Validator/Draft4/Array.hs
@@ -7,3 +7,3 @@ import           Prelude
 import           Control.Monad
-import qualified Data.Aeson.Pointer     as AP
+import qualified JSONPointer     as AP
 import qualified Data.Text              as T
diff --git a/src/Data/Validator/Draft4/Object/Properties.hs b/src/Data/Validator/Draft4/Object/Properties.hs
index 83a1f3c..a25fbfa 100644
--- a/src/Data/Validator/Draft4/Object/Properties.hs
+++ b/src/Data/Validator/Draft4/Object/Properties.hs
@@ -7,3 +7,3 @@ import           Prelude
 import           Control.Monad
-import qualified Data.Aeson.Pointer     as AP
+import qualified JSONPointer     as AP
 import           Data.Functor           (($>))
diff --git a/src/Data/Validator/Failure.hs b/src/Data/Validator/Failure.hs
index 47b60a6..f4f8a2e 100644
--- a/src/Data/Validator/Failure.hs
+++ b/src/Data/Validator/Failure.hs
@@ -8,3 +8,3 @@ import           Prelude            hiding (concat, sequence)
 
-import qualified Data.Aeson.Pointer as AP
+import qualified JSONPointer as AP
 
diff --git a/src/Data/Validator/Reference.hs b/src/Data/Validator/Reference.hs
index f2cc518..f2d8e79 100644
--- a/src/Data/Validator/Reference.hs
+++ b/src/Data/Validator/Reference.hs
@@ -11,3 +11,3 @@ import           Prelude
 
-import qualified Data.Aeson.Pointer     as AP
+import qualified JSONPointer            as AP
 import qualified Data.Text              as T
diff --git a/test/Local/Failure.hs b/test/Local/Failure.hs
index 7a3a047..be17925 100644
--- a/test/Local/Failure.hs
+++ b/test/Local/Failure.hs
@@ -6,3 +6,3 @@ import           Prelude
 import           Data.Aeson
-import qualified Data.Aeson.Pointer          as P
+import qualified JSONPointer          as P
 import           Data.Monoid
diff --git a/test/Shared.hs b/test/Shared.hs
index cbc0df5..94c2b69 100644
--- a/test/Shared.hs
+++ b/test/Shared.hs
@@ -7,3 +7,3 @@ import           Control.Monad
 import           Data.Aeson
-import qualified Data.Aeson.Pointer     as AP
+import qualified JSONPointer     as AP
 import           Data.Aeson.TH
