diff --git a/safecopy.cabal b/safecopy.cabal
index 0fdf380..34e221f 100644
--- a/safecopy.cabal
+++ b/safecopy.cabal
@@ -89,2 +89,2 @@ Test-suite instances
                        containers, time, array, vector, lens >= 4.7 && < 5.0,
-                       lens-action, tasty, tasty-quickcheck, quickcheck-instances
+                       lens-action, tasty, tasty-quickcheck, quickcheck-instances, QuickCheck
diff --git a/test/instances.hs b/test/instances.hs
index 0636814..d3b67f3 100644
--- a/test/instances.hs
+++ b/test/instances.hs
@@ -44,2 +44,3 @@ instance (Arbitrary a, Arbitrary b, Arbitrary c, Arbitrary d, Arbitrary e, Arbit
 
+#if ! MIN_VERSION_QuickCheck(2,8,2)
 instance (Arbitrary a) => Arbitrary (V.Vector a) where
@@ -55,2 +56,3 @@ instance (Arbitrary a, VU.Unbox a) => Arbitrary (VU.Vector a) where
    arbitrary = VU.fromList <$> arbitrary
+#endif
 
