diff -rN -u old-cabal/Distribution/Simple.hs new-cabal/Distribution/Simple.hs
--- old-cabal/Distribution/Simple.hs	2006-10-04 13:47:26.000000000 +0100
+++ new-cabal/Distribution/Simple.hs	2006-10-04 13:47:26.000000000 +0100
@@ -625,6 +625,7 @@
        sDistHook = \p _ h f -> sdist p f srcPref distPref (allSuffixHandlers h),
        pfeHook   = pfe,
        cleanHook = clean,
+       preHaddock  = readHook haddockVerbose,
        haddockHook = haddock,
        preReg    = readHook regVerbose,
        regHook   = defaultRegHook,

