diff -Naur cabal-1.2.0.orig/Cabal.cabal cabal-1.2.0/Cabal.cabal
--- cabal-1.2.0.orig/Cabal.cabal	2007-09-19 14:34:24.000000000 +0200
+++ cabal-1.2.0/Cabal.cabal	2007-09-19 14:40:11.000000000 +0200
@@ -46,6 +46,7 @@
         Distribution.PackageDescription,
         Distribution.Configuration,
         Distribution.ParseUtils,
+        Distribution.Setup,
         Distribution.Simple,
         Distribution.Simple.Build,
         Distribution.Simple.Compiler,
diff -Naur cabal-1.2.0.orig/Distribution/Setup.hs cabal-1.2.0/Distribution/Setup.hs
--- cabal-1.2.0.orig/Distribution/Setup.hs	1970-01-01 01:00:00.000000000 +0100
+++ cabal-1.2.0/Distribution/Setup.hs	2007-09-21 20:00:00.000000000 +0200
@@ -0,0 +1,6 @@
+module Distribution.Setup {-# DEPRECATED "Use module Distribution.Simple instead" #-}
+    ( module Distribution.Simple.Setup
+    ) where
+
+import Distribution.Simple.Setup
+
