diff -urwpN cabal-1.1.3.orig/Distribution/Simple/GHCPackageConfig.hs cabal-1.1.3/Distribution/Simple/GHCPackageConfig.hs
--- cabal-1.1.3.orig/Distribution/Simple/GHCPackageConfig.hs	2006-02-21 13:12:48.000000000 +0000
+++ cabal-1.1.3/Distribution/Simple/GHCPackageConfig.hs	2006-02-21 13:16:30.000000000 +0000
@@ -81,7 +81,7 @@ mkGHCPackageConfig :: PackageDescription
 mkGHCPackageConfig pkg_descr lbi
   = defaultGHCPackageConfig {
 	name	        = pkg_name,
-	auto	        = False,
+	auto		= True,
 	import_dirs     = [mkLibDir pkg_descr lbi Nothing],
 	library_dirs    = (mkLibDir pkg_descr lbi Nothing : 
 			   maybe [] (extraLibDirs . libBuildInfo) (library pkg_descr)),
