--- protocol-buffers-2.0.6-orig/Text/ProtocolBuffers/WireMessage.hs	2011-08-29 23:26:33.000000000 +1000
+++ protocol-buffers-2.0.6/Text/ProtocolBuffers/WireMessage.hs	2011-11-26 17:09:03.658824808 +1100
@@ -448,11 +448,8 @@
 -- "Text.ProtocolBuffers.WireMessage" and exported to use user by
 -- "Text.ProtocolBuffers".  These are less likely to change.
 class Wire b where
-  {-# INLINE wireSize #-}
   wireSize :: FieldType -> b -> WireSize
-  {-# INLINE wirePut #-}
   wirePut :: FieldType -> b -> Put
-  {-# INLINE wireGet #-}
   wireGet :: FieldType -> Get b
   {-# INLINE wireGetPacked #-}
   wireGetPacked :: FieldType -> Get (Seq b)
@@ -849,4 +846,4 @@
 -- OPTIMIZE try inlinining getMessageWith and getBareMessageWith: bench-005, slower
 
 
--- OPTIMIZE try NO-inlining getMessageWith and getBareMessageWith
\ No newline at end of file
+-- OPTIMIZE try NO-inlining getMessageWith and getBareMessageWith
