--- AttoJson-0.5.10-orig/Text/JSON/AttoJSON.hs	2011-02-03 16:12:08.000000000 +1100
+++ AttoJson-0.5.10/Text/JSON/AttoJSON.hs	2011-11-29 21:49:05.148201117 +1100
@@ -12,7 +12,7 @@
     updateField, updateFields
   ) where
 
-import Control.Applicative hiding (many)
+import Control.Applicative
 import qualified Data.ByteString.Lazy.Char8 as L (toChunks)
 import Data.ByteString.Char8 
                        ( append, ByteString, unfoldr, take, reverse
@@ -22,7 +22,7 @@
 import Data.Attoparsec ( Parser, maybeResult, eitherResult)
 import Data.Traversable (mapM)
 import Data.Attoparsec.Char8 ( char8, parse, string, decimal, skipSpace, satisfy
-                             , inClass, sepBy, option, many, endOfInput, try, feed
+                             , inClass, sepBy, option, endOfInput, try, feed
                              , takeWhile1, isDigit
                              )
 import Prelude hiding (concatMap, reverse, replicate, take
