--- simple-sendfile-0.2.3-orig/Network/Sendfile/Linux.hsc	2012-04-16 12:31:28.000000000 +1000
+++ simple-sendfile-0.2.3/Network/Sendfile/Linux.hsc	2012-05-14 00:27:22.861600277 +1000
@@ -11,6 +11,9 @@
 import Foreign.Marshal (alloca)
 import Foreign.Ptr (Ptr)
 import Foreign.Storable (poke)
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
 import Network.Sendfile.Types
 import Network.Socket
 import System.Posix.Files
