diff --git a/src/Text/ParserCombinators/HuttonMeijer.hs b/src/Text/ParserCombinators/HuttonMeijer.hs
index 7cb4b75..94a3f17 100644
--- a/src/Text/ParserCombinators/HuttonMeijer.hs
+++ b/src/Text/ParserCombinators/HuttonMeijer.hs
@@ -37,6 +37,7 @@ module Text.ParserCombinators.HuttonMeijer
 import Data.Char
 import Control.Applicative ( Applicative(pure,(<*>)), Alternative(empty,(<|>)) )
 import Control.Monad
+import qualified Control.Monad.Fail as Fail
 
 infixr 5 +++
 
@@ -62,6 +63,9 @@ instance Monad Parser where
    -- >>=         :: Parser a -> (a -> Parser b) -> Parser b
    (P p) >>= f     = P (\inp -> concat [papply (f v) out | (v,out) <- p inp])
 
+   fail            = Fail.fail
+
+instance Fail.MonadFail Parser where
    -- fail        :: String -> Parser a
    fail _          = P (\_ -> [])
 
diff --git a/src/Text/ParserCombinators/HuttonMeijerWallace.hs b/src/Text/ParserCombinators/HuttonMeijerWallace.hs
index 40cca1a..6d304e6 100644
--- a/src/Text/ParserCombinators/HuttonMeijerWallace.hs
+++ b/src/Text/ParserCombinators/HuttonMeijerWallace.hs
@@ -56,6 +56,7 @@ module Text.ParserCombinators.HuttonMeijerWallace
 import Data.Char
 import Control.Applicative ( Applicative(pure,(<*>)), Alternative(empty,(<|>)) )
 import Control.Monad
+import qualified Control.Monad.Fail as Fail
 
 infixr 5 +++
 
@@ -88,6 +89,9 @@ instance Monad (Parser s t e) where
                             [ papply' (f v) s out | (v,s,out) <- res ]
                         Left err  -> Left err
                        )
+   fail            = Fail.fail
+
+instance Fail.MonadFail (Parser s t e) where
    -- fail        :: String -> Parser s t e a
    fail err        = P (\st inp -> Right [])
   -- I know it's counterintuitive, but we want no-parse, not an error.
diff --git a/src/Text/ParserCombinators/Poly/ByteString.hs b/src/Text/ParserCombinators/Poly/ByteString.hs
index 28f57b4..aab1a62 100644
--- a/src/Text/ParserCombinators/Poly/ByteString.hs
+++ b/src/Text/ParserCombinators/Poly/ByteString.hs
@@ -23,6 +23,7 @@ import Text.ParserCombinators.Poly.Result
 import qualified Data.ByteString.Lazy as BS
 import Data.ByteString.Lazy (ByteString)
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 import Data.Word
 
 -- | This @Parser@ datatype is a specialised parsing monad with error
@@ -39,13 +40,16 @@ instance Functor Parser where
 
 instance Monad Parser where
     return x     = P (\ts-> Success ts x)
-    fail e       = P (\ts-> Failure ts e)
+    fail         = Fail.fail
     (P f) >>= g  = P (continue . f)
       where
         continue (Success ts x)             = let (P g') = g x in g' ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail Parser where
+    fail e       = P (\ts-> Failure ts e)
+
 instance Commitment Parser where
     commit (P p)         = P (Committed . squash . p)
       where
diff --git a/src/Text/ParserCombinators/Poly/ByteStringChar.hs b/src/Text/ParserCombinators/Poly/ByteStringChar.hs
index 27b238a..162e32a 100644
--- a/src/Text/ParserCombinators/Poly/ByteStringChar.hs
+++ b/src/Text/ParserCombinators/Poly/ByteStringChar.hs
@@ -23,6 +23,7 @@ import Text.ParserCombinators.Poly.Result
 import qualified Data.ByteString.Lazy.Char8 as BS
 import Data.ByteString.Lazy.Char8 (ByteString)
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | This @Parser@ datatype is a specialised parsing monad with error
 --   reporting.  Whereas the standard version can be used for arbitrary
@@ -38,13 +39,16 @@ instance Functor Parser where
 
 instance Monad Parser where
     return x     = P (\ts-> Success ts x)
-    fail e       = P (\ts-> Failure ts e)
+    fail         = Fail.fail
     (P f) >>= g  = P (continue . f)
       where
         continue (Success ts x)             = let (P g') = g x in g' ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail Parser where
+    fail e       = P (\ts-> Failure ts e)
+
 instance Commitment Parser where
     commit (P p)         = P (Committed . squash . p)
       where
diff --git a/src/Text/ParserCombinators/Poly/Lazy.hs b/src/Text/ParserCombinators/Poly/Lazy.hs
index 5bdf712..0e471e9 100644
--- a/src/Text/ParserCombinators/Poly/Lazy.hs
+++ b/src/Text/ParserCombinators/Poly/Lazy.hs
@@ -22,6 +22,7 @@ import Text.ParserCombinators.Poly.Base
 import Text.ParserCombinators.Poly.Result
 import qualified Text.ParserCombinators.Poly.Parser as P
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 #if __GLASGOW_HASKELL__
 import Control.Exception hiding (bracket)
@@ -44,8 +45,10 @@ instance Functor (Parser t) where
     fmap f (P p) = P (fmap f p)
 instance Monad (Parser t) where
     return x  = P (return x)
-    fail e    = P (fail e)
+    fail      = Fail.fail
     (P f) >>= g = P (f >>= (\(P g')->g') . g)
+instance Fail.MonadFail (Parser t) where
+    fail e    = P (fail e)
 instance Commitment (Parser t) where
     commit (P p)   = P (commit p)
     (P p) `adjustErr` f  = P (p `adjustErr` f)
diff --git a/src/Text/ParserCombinators/Poly/Lex.hs b/src/Text/ParserCombinators/Poly/Lex.hs
index 65e237d..4f52694 100644
--- a/src/Text/ParserCombinators/Poly/Lex.hs
+++ b/src/Text/ParserCombinators/Poly/Lex.hs
@@ -30,6 +30,7 @@ module Text.ParserCombinators.Poly.Lex
 import Text.ParserCombinators.Poly.Base
 import Text.ParserCombinators.Poly.Result
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | In a strict language, where creating the entire input list of tokens
 --   in one shot may be infeasible, we can use a lazy "callback" kind of
@@ -55,13 +56,16 @@ instance Functor (Parser t) where
 
 instance Monad (Parser t) where
     return x     = P (\ts-> Success ts x)
-    fail e       = P (\ts-> Failure ts e)
+    fail         = Fail.fail
     (P f) >>= g  = P (continue . f)
       where
         continue (Success ts x)             = let (P g') = g x in g' ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail  (Parser t) where
+    fail e       = P (\ts-> Failure ts e)
+
 instance Commitment (Parser t) where
     commit (P p)         = P (Committed . squash . p)
       where
diff --git a/src/Text/ParserCombinators/Poly/Parser.hs b/src/Text/ParserCombinators/Poly/Parser.hs
index 66a320c..a453db0 100644
--- a/src/Text/ParserCombinators/Poly/Parser.hs
+++ b/src/Text/ParserCombinators/Poly/Parser.hs
@@ -20,6 +20,7 @@ module Text.ParserCombinators.Poly.Parser
 import Text.ParserCombinators.Poly.Base
 import Text.ParserCombinators.Poly.Result
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | This @Parser@ datatype is a fairly generic parsing monad with error
 --   reporting.  It can be used for arbitrary token types, not just
@@ -39,13 +40,16 @@ instance Applicative (Parser t) where
 
 instance Monad (Parser t) where
     return x     = P (\ts-> Success ts x)
-    fail e       = P (\ts-> Failure ts e)
+    fail         = Fail.fail
     (P f) >>= g  = P (continue . f)
       where
         continue (Success ts x)             = let (P g') = g x in g' ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail (Parser t) where
+    fail e       = P (\ts-> Failure ts e)
+
 instance Alternative (Parser t) where
     empty     = fail "no parse"
     p <|> q   = p `onFail` q
diff --git a/src/Text/ParserCombinators/Poly/StateLazy.hs b/src/Text/ParserCombinators/Poly/StateLazy.hs
index 1714d17..f1d3a1a 100644
--- a/src/Text/ParserCombinators/Poly/StateLazy.hs
+++ b/src/Text/ParserCombinators/Poly/StateLazy.hs
@@ -26,6 +26,7 @@ import Text.ParserCombinators.Poly.Base hiding (manyFinally)
 import Text.ParserCombinators.Poly.Result
 import qualified Text.ParserCombinators.Poly.StateParser as P
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 #if __GLASGOW_HASKELL__
 import Control.Exception hiding (bracket)
@@ -48,8 +49,10 @@ instance Functor (Parser s t) where
     fmap f (P p) = P (fmap f p)
 instance Monad (Parser s t) where
     return x  = P (return x)
-    fail e    = P (fail e)
+    fail      = Fail.fail
     (P f) >>= g = P (f >>= (\(P g')->g') . g)
+instance Fail.MonadFail (Parser s t) where
+    fail e    = P (fail e)
 instance Commitment (Parser s t) where
     commit (P p)   = P (commit p)
     (P p) `adjustErr` f  = P (p `adjustErr` f)
diff --git a/src/Text/ParserCombinators/Poly/StateParser.hs b/src/Text/ParserCombinators/Poly/StateParser.hs
index f21ee0a..05236bc 100644
--- a/src/Text/ParserCombinators/Poly/StateParser.hs
+++ b/src/Text/ParserCombinators/Poly/StateParser.hs
@@ -23,6 +23,7 @@ module Text.ParserCombinators.Poly.StateParser
 import Text.ParserCombinators.Poly.Base
 import Text.ParserCombinators.Poly.Result
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | This @Parser@ datatype is a fairly generic parsing monad with error
 --   reporting, and running state.
@@ -42,13 +43,16 @@ instance Applicative (Parser s t) where
 
 instance Monad (Parser s t) where
     return x     = P (\s ts-> Success (ts,s) x)
-    fail e       = P (\s ts-> Failure (ts,s) e)
+    fail         = Fail.fail
     (P f) >>= g  = P (\s-> continue . f s)
       where
         continue (Success (ts,s) x)        = let (P g') = g x in g' s ts
         continue (Committed r)             = Committed (continue r)
         continue (Failure tss e)           = Failure tss e
 
+instance Fail.MonadFail (Parser s t) where
+    fail e       = P (\s ts-> Failure (ts,s) e)
+
 instance Alternative (Parser s t) where
     empty     = fail "no parse"
     p <|> q   = p `onFail` q
diff --git a/src/Text/ParserCombinators/Poly/StateText.hs b/src/Text/ParserCombinators/Poly/StateText.hs
index 1823890..97b576f 100644
--- a/src/Text/ParserCombinators/Poly/StateText.hs
+++ b/src/Text/ParserCombinators/Poly/StateText.hs
@@ -28,6 +28,7 @@ import Text.ParserCombinators.Poly.Result
 import qualified Data.Text.Lazy as T
 import Data.Text.Lazy (Text)
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | This @Parser@ datatype is a specialised parsing monad with error
 --   reporting.  Whereas the standard version can be used for arbitrary
@@ -45,13 +46,16 @@ instance Functor (Parser s) where
 
 instance Monad (Parser s) where
     return x     = P (\s ts-> Success (ts,s) x)
-    fail e       = P (\s ts-> Failure (ts,s) e)
+    fail         = Fail.fail
     (P f) >>= g  = P (\s-> continue . f s)
       where
         continue (Success (ts,s) x)         = let (P g') = g x in g' s ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail (Parser s) where
+    fail e       = P (\s ts-> Failure (ts,s) e)
+
 instance Commitment (Parser s) where
     commit (P p)         = P (\s-> Committed . squash . p s)
       where
diff --git a/src/Text/ParserCombinators/Poly/Text.hs b/src/Text/ParserCombinators/Poly/Text.hs
index 2708e88..44e9ae5 100644
--- a/src/Text/ParserCombinators/Poly/Text.hs
+++ b/src/Text/ParserCombinators/Poly/Text.hs
@@ -24,6 +24,7 @@ import Text.ParserCombinators.Poly.Result
 import qualified Data.Text.Lazy as T
 import Data.Text.Lazy (Text)
 import Control.Applicative
+import qualified Control.Monad.Fail as Fail
 
 -- | This @Parser@ datatype is a specialised parsing monad with error
 --   reporting.  Whereas the standard version can be used for arbitrary
@@ -39,13 +40,16 @@ instance Functor Parser where
 
 instance Monad Parser where
     return x     = P (\ts-> Success ts x)
-    fail e       = P (\ts-> Failure ts e)
+    fail         = Fail.fail
     (P f) >>= g  = P (continue . f)
       where
         continue (Success ts x)             = let (P g') = g x in g' ts
         continue (Committed r)              = Committed (continue r)
         continue (Failure ts e)             = Failure ts e
 
+instance Fail.MonadFail Parser where
+    fail e       = P (\ts-> Failure ts e)
+
 instance Commitment Parser where
     commit (P p)         = P (Committed . squash . p)
       where
