diff -urwpN hmake-3.11.orig/src/hmake/Makefile hmake-3.11/src/hmake/Makefile
--- hmake-3.11.orig/src/hmake/Makefile	2006-05-03 14:38:29.000000000 +0100
+++ hmake-3.11/src/hmake/Makefile	2006-09-12 10:34:28.000000000 +0100
@@ -46,7 +46,7 @@ ifeq "nhc" "$(BUILDCOMP)"
 HEAP = -H4M
 endif
 ifeq "ghc" "$(BUILDCOMP)"
-HFLAGS += $(shell ${LOCAL}fixghc ${GHCSYM} -package lang)
+HFLAGS += $(shell ${LOCAL}fixghc ${GHCSYM})
 endif
 ifeq "hbc" "$(BUILDCOMP)"
 all: hbc
diff -urwpN hmake-3.11.orig/src/interpreter/Makefile hmake-3.11/src/interpreter/Makefile
--- hmake-3.11.orig/src/interpreter/Makefile	2005-10-07 14:34:32.000000000 +0100
+++ hmake-3.11/src/interpreter/Makefile	2006-09-12 10:37:23.000000000 +0100
@@ -11,7 +11,7 @@ CHFILES = $(patsubst %.hs, ../hmake/%.$C
 
 ifeq "${BUILDCOMP}" "ghc"
 HFLAGS = $(shell $(LOCAL)fixghc $(GHCSYM) \
-      -package lang -package util -package base $(READLINE) )
+      -package base $(READLINE) )
 export HFLAGS
 endif
 ifeq "${BUILDCOMP}" "hbc"
diff -urwpN hmake-3.11.orig/src/interpreter/SimpleLineEditor.hs hmake-3.11/src/interpreter/SimpleLineEditor.hs
--- hmake-3.11.orig/src/interpreter/SimpleLineEditor.hs	2004-06-25 20:58:31.000000000 +0100
+++ hmake-3.11/src/interpreter/SimpleLineEditor.hs	2006-09-12 10:39:43.000000000 +0100
@@ -26,7 +26,7 @@ import IO
 import Monad (when)
 import Char (isSpace)
 #if USE_READLINE
-import Readline
+import System.Console.Readline
 #else
 #if defined(__HBC__)
 import IOExts (IORef, newIORef, readIORef, writeIORef, unsafePerformIO)
