diff --git a/pandoc-citeproc.hs b/pandoc-citeproc.hs
index f72fa41..6954343 100644
--- a/pandoc-citeproc.hs
+++ b/pandoc-citeproc.hs
@@ -14,3 +14,3 @@ import qualified Data.Text as T
 import Data.Text.Encoding (encodeUtf8)
-import Data.Aeson.Encode.Pretty (encodePretty', Config(..))
+import Data.Aeson.Encode.Pretty (encodePretty', Config(..), defConfig, Indent(..))
 import System.Console.GetOpt
@@ -64,4 +64,4 @@ main = do
             else B8.putStrLn . unescapeUnicode . B.concat . BL.toChunks .
-              encodePretty' Config{ confIndent = 2
-                                  , confCompare = compare }
+              encodePretty' defConfig{ confIndent = Spaces 2
+                                     , confCompare = compare }
      else toJSONFilter doCites
