commit 86726dcf8428156367ec2b5cbbc60e368b2e1199
Author: John MacFarlane <jgm@berkeley.edu>
Date:   Sat Aug 20 23:24:36 2016 +0200

    Allow aeson 1.0.*.

diff --git a/src/Text/CSL/Style.hs b/src/Text/CSL/Style.hs
index 6a9d14a..1a44469 100644
--- a/src/Text/CSL/Style.hs
+++ b/src/Text/CSL/Style.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE OverloadedStrings, PatternGuards, DeriveDataTypeable,
     ScopedTypeVariables, FlexibleInstances, DeriveGeneric,
+    OverlappingInstances,
     GeneralizedNewtypeDeriving, CPP, MultiParamTypeClasses #-}
 -----------------------------------------------------------------------------
 -- |
