--- XMonad/Config.hs.old	2012-03-12 22:41:08.140535653 +0200
+++ XMonad/Config.hs	2012-03-12 22:41:36.318134114 +0200
@@ -27,11 +27,11 @@
 import XMonad.Core as XMonad hiding
     (workspaces,manageHook,keys,logHook,startupHook,borderWidth,mouseBindings
     ,layoutHook,modMask,terminal,normalBorderColor,focusedBorderColor,focusFollowsMouse
-    ,handleEventHook)
+    ,handleEventHook,clickJustFocuses)
 import qualified XMonad.Core as XMonad
     (workspaces,manageHook,keys,logHook,startupHook,borderWidth,mouseBindings
     ,layoutHook,modMask,terminal,normalBorderColor,focusedBorderColor,focusFollowsMouse
-    ,handleEventHook)
+    ,handleEventHook,clickJustFocuses)
 
 import XMonad.Layout
 import XMonad.Operations
@@ -157,6 +157,11 @@
 focusFollowsMouse :: Bool
 focusFollowsMouse = True
 
+-- | Whether a mouse click select the focus or is just passed to the window
+clickJustFocuses :: Bool
+clickJustFocuses = True
+
+
 -- | The xmonad key bindings. Add, modify or remove key bindings here.
 --
 -- (The comment formatting character is used when generating the manpage)
@@ -248,4 +253,5 @@
     , XMonad.manageHook         = manageHook
     , XMonad.handleEventHook    = handleEventHook
     , XMonad.focusFollowsMouse  = focusFollowsMouse
+    , XMonad.clickJustFocuses       = clickJustFocuses
     }
