--- XMonad/Main.hsc.old	2012-03-12 22:40:44.918164287 +0200
+++ XMonad/Main.hsc	2012-03-12 22:41:36.320133802 +0200
@@ -247,12 +247,16 @@
     | t == buttonPress = do
     -- If it's the root window, then it's something we
     -- grabbed in grabButtons. Otherwise, it's click-to-focus.
+    dpy <- asks display
     isr <- isRoot w
     m <- cleanMask $ ev_state e
     mact <- asks (M.lookup (m, b) . buttonActions)
     case mact of
-        (Just act) | isr -> act $ ev_subwindow e
-        _                -> focus w
+        Just act | isr -> act $ ev_subwindow e
+        _              -> do
+            focus w
+            ctf <- asks (clickJustFocuses . config)
+            unless ctf $ io (allowEvents dpy replayPointer currentTime)
     broadcastMessage e -- Always send button events.
 
 -- entered a normal window: focus it if focusFollowsMouse is set to
