diff -rN -u old-xmonad-0.6/Main.hs new-xmonad-0.6/Main.hs
--- old-xmonad-0.6/Main.hs	2008-02-17 18:31:59.000000000 +0100
+++ new-xmonad-0.6/Main.hs	2008-02-17 18:31:59.000000000 +0100
@@ -37,7 +37,7 @@
         ["--resume", _]       -> launch
         ["--recompile"]       -> recompile False >> return ()
         ["--recompile-force"] -> recompile True >> return ()
-        ["--version"]         -> putStrLn "xmonad 0.5"
+        ["--version"]         -> putStrLn "xmonad 0.6"
 #ifdef TESTING
         ("--run-tests":_)     -> Properties.main
 #endif

