diff --git a/src/Manager.hs b/src/Manager.hs
index a9665e0..7bea2c1 100644
--- a/src/Manager.hs
+++ b/src/Manager.hs
@@ -139,7 +139,7 @@ files. -}
 makeProfileChooserDialog :: IO FileChooserDialog
 makeProfileChooserDialog = do
   dialog <- fileChooserDialogNew (Just "Load profile") Nothing FileChooserActionOpen
-            [(stockOpen,ResponseOk),(stockCancel,ResponseCancel)]
+            [("Ok",ResponseOk),("Cancel",ResponseCancel)]
   mapM_ (fileChooserAddFilter dialog) =<< makeFileFilters [("Heap profiles","*.hp"),("All files","*")]
   fileChooserSetSelectMultiple dialog True
 
@@ -383,7 +383,7 @@ makeGraphCanvas selectRgb prof = do
   -- Repaint handler, called after every resize for instance.
   onExpose glCanvas $ const $ repaint >> return True
 
-  coordLabel <- labelNew Nothing
+  coordLabel <- labelNew (Nothing :: Maybe String)
   boxPackStart mainBox coordLabel PackNatural 0
 
   -- Highlighting cost centre names on hover and displaying
@@ -397,7 +397,7 @@ makeGraphCanvas selectRgb prof = do
 
     (t1,t2) <- getInterval
     c <- getMaxCost
-    labelSetText coordLabel $ printf " time=%0.2f, cost=%s "
+    labelSetText coordLabel $ (printf " time=%0.2f, cost=%s " :: Double -> String -> String)
         (t1+eventX evt*(t2-t1)/fromIntegral w)
         (showBigInteger ((fromIntegral h-fromIntegral y)*fromIntegral c `div` fromIntegral h :: Integer))
 
