diff --git a/HSH/ShellEquivs.hs b/HSH/ShellEquivs.hs
index 1c5ee4e..58933c9 100644
--- a/HSH/ShellEquivs.hs
+++ b/HSH/ShellEquivs.hs
@@ -93,3 +93,3 @@ import qualified Control.Exception as E
 #ifdef __HSH_POSIX__
-import System.Posix.Files (getFileStatus, isSymbolicLink, readSymbolicLink)
+import System.Posix.Files as SPF (getFileStatus, isSymbolicLink, readSymbolicLink)
 import System.Posix.User (getEffectiveUserName, getUserEntryForName, homeDirectory)
@@ -384,3 +384,3 @@ readlink :: FilePath -> IO FilePath
 readlink fp =
-    do issym <- (getFileStatus fp >>= return . isSymbolicLink)
+    do issym <- (getFileStatus fp >>= return . SPF.isSymbolicLink)
        if issym
@@ -394,3 +394,3 @@ readlinkabs :: FilePath -> IO FilePath
 readlinkabs inp =
-       do issym <- (getFileStatus inp >>= return . isSymbolicLink)
+       do issym <- (getFileStatus inp >>= return . SPF.isSymbolicLink)
           if issym
