diff -urN floskell-0.10.5/floskell.cabal floskell-0.10.5-new/floskell.cabal
--- floskell-0.10.5/floskell.cabal	2001-09-08 19:46:40.000000000 -0600
+++ floskell-0.10.5-new/floskell.cabal	2021-01-22 16:46:36.540200774 -0700
@@ -71,7 +71,13 @@
         unordered-containers >=0.2.8.0 && <0.3,
         utf8-string >=1.0.1.1 && <1.1
 
+flag executable
+    description: Build the floskell executable
+    default: False
+
 executable floskell
+    if !flag(executable)
+        buildable: False
     main-is:          Main.hs
     hs-source-dirs:   src/main
     default-language: Haskell2010
