--- testpack-2.0.3-orig/src/Test/HUnit/Tools.hs	2011-03-25 07:39:52.000000000 +1100
+++ testpack-2.0.3/src/Test/HUnit/Tools.hs	2011-03-25 20:10:21.198749936 +1100
@@ -61,7 +61,7 @@
     HU.TestLabel lbl $ HU.TestCase $
       do result <- localquickCheckWithResult config property
          case result of
-           Success _ -> return ()
+           Success _ _ _ -> return ()
            _ -> HU.assertFailure (show result)
 
 -- Modified from HUnit
@@ -159,7 +159,7 @@
 -- | Tests a property, using test arguments, produces a test result, and prints the results to 'stdout'.
 localquickCheckWithResult :: Testable prop => Args -> prop -> IO Result
 localquickCheckWithResult args p =
-  do tm  <- newTerminal
+  do tm  <- newStdioTerminal
      rnd <- case replay args of
               Nothing      -> newStdGen
               Just (rnd,_) -> return rnd
@@ -176,7 +176,6 @@
                  , collected         = []
                  , expectedFailure   = False
                  , randomSeed        = rnd
-                 , isShrinking       = False
                  , numSuccessShrinks = 0
                  , numTryShrinks     = 0
                  } (unGen (property p))
@@ -207,8 +206,7 @@
     runATest st f =
       do
         let size = computeSize st (numSuccessTests st) (numDiscardedTests st)
-        MkRose mres ts <- protectRose (unProp (f rnd1 size))
-        res <- mres
+        MkRose res ts <- protectRose (reduceRose (unProp (f rnd1 size)))
         callbackPostTest st res
      
         case ok res of
