diff --git a/src/Test/HUnit/Tools.hs b/src/Test/HUnit/Tools.hs
index 6b8a25e..036f713 100644
--- a/src/Test/HUnit/Tools.hs
+++ b/src/Test/HUnit/Tools.hs
@@ -201,2 +201,6 @@ localquickCheckWithResult args p =
                  , maxSuccessTests   = maxSuccess args
+#if MIN_VERSION_QuickCheck(2,10,0)
+                 , numTotMaxShrinks  = maxBound
+                 , maxDiscardedRatio = 10
+#else
                  , maxDiscardedTests =
@@ -206,2 +210,3 @@ localquickCheckWithResult args p =
                                        maxDiscard args
+#endif
 #endif
@@ -238,3 +243,5 @@ localquickCheckWithResult args p =
       | numSuccessTests st   >= maxSuccessTests st   = doneTesting st f
+#if ! MIN_VERSION_QuickCheck(2,10,0)
       | numDiscardedTests st >= maxDiscardedTests st = giveUp st f
+#endif
       | otherwise                                    = runATest st f
@@ -308,3 +315,9 @@ localquickCheckWithResult args p =
 #if MIN_VERSION_QuickCheck(2,3,0)
+
+#if MIN_VERSION_QuickCheck(2,10,0)
+               (numShrinks, totFailed, lastFailed, _result) <- foundFailure st res ts
+#else
                (numShrinks, totFailed, lastFailed) <- foundFailure st res ts
+#endif
+
 #else
