diff --git a/Codec/Compression/BZip/Stream.hsc b/Codec/Compression/BZip/Stream.hsc
index 1642811..5a8c785 100644
--- a/Codec/Compression/BZip/Stream.hsc
+++ b/Codec/Compression/BZip/Stream.hsc
@@ -69,7 +69,8 @@ import Foreign.C
 import Data.ByteString.Internal (nullForeignPtr)
 import System.IO.Unsafe (unsafeInterleaveIO)
 import System.IO (hPutStrLn, stderr)
-import Control.Monad (liftM)
+import Control.Applicative (Applicative(..))
+import Control.Monad (liftM, ap)
 import Control.Exception (assert)
 
 import Prelude hiding (length)
@@ -220,6 +221,13 @@ newtype Stream a = BZ {
               ,Int, Int, a)
   }
 
+instance Functor Stream where
+    fmap = liftM
+
+instance Applicative Stream where
+    pure  = return
+    (<*>) = ap
+
 instance Monad Stream where
   (>>=)  = thenZ
 --  m >>= f = (m `thenZ` \a -> consistencyCheck `thenZ_` returnZ a) `thenZ` f
