--- 1/fquery.cabal	2006-05-21 03:01:41.000000000 +0200
+++ 2/fquery.cabal	2007-09-21 18:51:10.000000000 +0200
@@ -6,7 +6,7 @@
 License-File:	LICENCE.txt
 License:	OtherLicense
 Synopsis:	Installed package query tool for Gentoo Linux
-Build-depends:	base, haskell98, parsec, unix
+Build-depends:	base, haskell98, parsec, unix, regex-compat
 Extra-Source-Files:
                 configure
 		Adelie/Config.hs.in
