diff --git a/convertible.cabal b/convertible.cabal
index b143136..c4e7d66 100644
--- a/convertible.cabal
+++ b/convertible.cabal
@@ -50,7 +50,7 @@ flag time_gte_113
 
 library
   if flag(splitBase)
-    Build-Depends: base>=3 && <5, old-time, time>=1.1.2.4 && <=1.2.0.5,
+    Build-Depends: base>=3 && <5, old-time, time>=1.1.2.4 && <1.5,
      bytestring, containers, old-locale
     if flag(time_gte_113)
       Build-Depends: time>=1.1.3
