diff --git a/Projectile/Pellet.hs b/Projectile/Pellet.hs
index ec6e760..fdd813e 100644
--- a/Projectile/Pellet.hs
+++ b/Projectile/Pellet.hs
@@ -15,3 +15,3 @@ import Prelude ((||), (.), max, (-), Maybe(..), Bool(..), Double)
 import Graphics.Gloss.Data.Picture ( Picture(Circle, Color) )
-import Graphics.Gloss.Data.Color ( makeColor8 )
+import Graphics.Gloss.Rendering  ( makeColor )
 
@@ -49,3 +49,3 @@ instance Animation Pellet where
 
-  image _ _ = Color (makeColor8 172 172 172 255) (Circle 1)
+  image _ _ = Color (makeColor 172 172 172 255) (Circle 1)
 
diff --git a/Resources.hs b/Resources.hs
index ffc6283..df1cb11 100644
--- a/Resources.hs
+++ b/Resources.hs
@@ -11,3 +11,3 @@ module Resources where
 import Prelude (map, return, (==), mapM, (++), pi, (+), (/), (*), zip, (.), foldr, ($))
-import Graphics.Gloss.Data.Picture ( loadBMP )
+import Graphics.Gloss.Rendering ( loadBMP )
 import Graphics.Gloss.Data.Color ( yellow, white, violet, rose, orange, magenta, green, cyan, chartreuse, blue, azure, aquamarine )
