--- a/Data/AttoLisp.hs
+++ b/Data/AttoLisp.hs
@@ -56,2 +56,3 @@ import qualified Blaze.ByteString.Builder.Char.Utf8 as Blaze
 import qualified Data.Map as M
+import qualified Data.Semigroup as S
 
@@ -156,2 +157,6 @@ instance MonadPlus Parser where
 
+instance S.Semigroup (Parser a) where
+    (<>) = mplus
+    {-# INLINE (<>) #-}
+
 instance Monoid (Parser a) where
@@ -209,2 +214,6 @@ instance Alternative Result where
 
+instance S.Semigroup (Result a) where
+    (<>) = mplus
+    {-# INLINE (<>) #-}
+
 instance Monoid (Result a) where
