--- pwstore-fast-2.2-orig/Crypto/PasswordStore.hs	2011-07-14 05:52:33.000000000 +1000
+++ pwstore-fast-2.2/Crypto/PasswordStore.hs	2012-07-07 23:25:34.308015379 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE OverloadedStrings, BangPatterns #-}
+{-# LANGUAGE OverloadedStrings, BangPatterns, CPP, ScopedTypeVariables #-}
 -- |
 -- Module      : Crypto.PasswordStore
 -- Copyright   : (c) Peter Scott, 2011
@@ -93,6 +93,10 @@
 import System.IO
 import System.Random
 import Data.Maybe
+#if !MIN_VERSION_base(4,6,0)
+import Prelude hiding(catch)
+#endif
+import Control.Exception (catch, SomeException)
 
 ---------------------
 -- Cryptographic base
@@ -120,7 +124,7 @@
 -- system RNG as a fallback. This is the function used to generate salts by
 -- 'makePassword'.
 genSaltIO :: IO Salt
-genSaltIO = catch genSaltDevURandom (\_ -> genSaltSysRandom)
+genSaltIO = catch genSaltDevURandom (\(_::SomeException) -> genSaltSysRandom)
 
 -- | Generate a 'Salt' from @\/dev\/urandom@.
 genSaltDevURandom :: IO Salt
