diff -Naur helium-1.2.orig/heliumNT/src/Makefile.in helium-1.2/heliumNT/src/Makefile.in
--- helium-1.2.orig/heliumNT/src/Makefile.in	2004-11-18 12:37:28.025725408 +0100
+++ helium-1.2/heliumNT/src/Makefile.in	2004-11-18 12:39:54.000000000 +0100
@@ -32,6 +32,7 @@
 exec_prefix = @exec_prefix@
 bindir      = @bindir@
 libdir      = @prefix@/helium/lib
+tclibdir    = @prefix@/helium/tclib
 demodir     = @prefix@/helium/demo
 
 status :
@@ -40,6 +41,8 @@
 	@echo "Type \"make install\" to install."
 	@echo "  Binaries will go into  $(bindir)"
 	@echo "  Libraries will go into $(libdir)"
+	@echo "  Libraries supporting type classes will go into"
+	@echo "                         $(tclibdir)"
 	@echo "  Demos will go into     $(demodir)"
 	@echo "*******************************************************************"
 
@@ -432,15 +435,18 @@
 install : smaller
 	# *** Copy sources to library directory
-	$(MKDIR) $(bindir) $(libdir) $(demodir)
-	$(INSTALL) -m 644 ../lib/Prelude.hs ../lib/List.hs ../lib/Maybe.hs $(libdir)
+	$(MKDIR) $(bindir) $(libdir) $(tclibdir) $(demodir)
+	$(INSTALL) -m 644 ../lib/simple/Prelude.hs ../lib/simple/List.hs ../lib/simple/Maybe.hs $(libdir)
+	$(INSTALL) -m 644 ../lib/Prelude.hs ../lib/List.hs ../lib/Maybe.hs $(tclibdir)
 	$(INSTALL) -m 755 $(HELIUMBINDIR)/$(MAIN)$(EXE) $(bindir)
 	$(INSTALL) -m 755 $(LVMRUNTIMEDIR)/lvmrun$(EXE) $(bindir)
 	$(INSTALL) -m 755 $(HELIUMBINDIR)/texthint$(EXE) $(bindir)
 	# *** Copying demos to demo directory
 	$(INSTALL) -m 644 ../demo/Calendar.hs ../demo/ParserLibrary.hs ../demo/ParserDemo.hs ../demo/TicTacToe.hs $(demodir)
 	# *** Copy generated lvms and touch them
-	$(INSTALL) -m 644 ../lib/*.lvm $(libdir)
+	$(INSTALL) -m 644 ../lib/simple/*.lvm $(libdir)
+	$(INSTALL) -m 644 ../lib/*.lvm $(tclibdir)
 	touch $(libdir)/*.lvm
+	touch $(tclibdir)/*.lvm
 
 movetohome: smaller
 	# *** Copying library sources to projects directory
