This makes both --with-xinerama and --without-xinerama work.

--- old-X11-1.4.2/configure.ac	2008-05-24 13:25:52.000000000 +0200
+++ new-X11-1.4.2/configure.ac	2008-05-24 13:25:52.000000000 +0200
@@ -56,11 +56,11 @@
 AC_MSG_CHECKING([whether to build Xinerama])
 AC_ARG_WITH(xinerama,
             AS_HELP_STRING([--without-xinerama], [do not build Xinerama support]),
-            want_xinerama=no,
-            want_xinerama=yes)
-AC_MSG_RESULT([$want_xinerama])
+            [],
+            [with_xinerama=yes])
+AC_MSG_RESULT([$with_xinerama])
 
-if test "$want_xinerama" = yes; then
+if test "$with_xinerama" = yes; then
     AC_CHECK_HEADERS([X11/extensions/Xinerama.h], [have_xinerama=yes])
     if test "$have_xinerama" = yes; then
         EXTRA_LIBRARIES="extra-libraries: Xinerama Xext"

