diff --git a/Stream.cabal b/Stream.cabal
index 361b45d..55a7f83 100644
--- a/Stream.cabal
+++ b/Stream.cabal
@@ -17,7 +17,7 @@ Description:            This package implements functions, analogous
 			nothing to do with the work on /Stream Fusion/ by 
 			Duncan Coutts, Roman Leshchinskiy, and Don Stewart.
 Category:               Data
-Build-Depends:          base < 4.4, QuickCheck >= 2.0, lazysmallcheck >= 0.3
+Build-Depends:          base < 4.5, QuickCheck >= 2.0, lazysmallcheck >= 0.3
 Build-Type:		Simple
 Exposed-modules:        Data.Stream
 
