diff -ur ghc-6.4/distrib/configure-bin.ac ghc-6.4/distrib/configure-bin.ac
--- ghc-6.4/distrib/configure-bin.ac	2005-03-10 09:10:09.000000000 -0500
+++ ghc-6.4/distrib/configure-bin.ac	2005-03-14 21:37:20.356380744 -0500
@@ -78,6 +78,10 @@
 	TargetPlatform=rs6000-ibm-aix;;
 powerpc-apple-darwin*)
 	TargetPlatform=powerpc-apple-darwin;;
+powerpc-*-linux*)
+	TargetPlatform=powerpc-unknown-linux;;
+powerpc64-*-linux*)
+	TargetPlatform=powerpc64-unknown-linux;;
 sparc-sun-sunos4*)
 	TargetPlatform=sparc-sun-sunos4;;
 sparc-sun-solaris2*)
diff -ur ghc-6.4/ghc/includes/MachRegs.h ghc-6.4/ghc/includes/MachRegs.h
--- ghc-6.4/ghc/includes/MachRegs.h	2005-01-28 07:55:51.000000000 -0500
+++ ghc-6.4/ghc/includes/MachRegs.h	2005-03-14 21:37:31.825368128 -0500
@@ -457,7 +457,7 @@
 #define REG_R7    	r20
 #define REG_R8    	r21
 
-#ifdef darwin_REGS
+#if darwin_REGS
 
 #define REG_F1		f14
 #define REG_F2		f15
