diff -Naur ghc-6.4.1.20050819.orig/ghc/compiler/Makefile ghc-6.4.1.20050819/ghc/compiler/Makefile
--- ghc-6.4.1.20050819.orig/ghc/compiler/Makefile	2005-08-23 10:22:25.000000000 +0200
+++ ghc-6.4.1.20050819/ghc/compiler/Makefile	2005-08-23 10:23:08.000000000 +0200
@@ -475,7 +475,7 @@
 endif
 endif
 
-ifneq "$(findstring YES, $(bootstrapped) $(ghc_ge_603))" ""
+ifneq "$(findstring YES, $(bootstrapped))" ""
 SRC_HC_OPTS += -package Cabal
 endif
 
diff -Naur ghc-6.4.1.20050819.orig/ghc/lib/compat/Makefile ghc-6.4.1.20050819/ghc/lib/compat/Makefile
--- ghc-6.4.1.20050819.orig/ghc/lib/compat/Makefile	2005-08-23 10:22:24.000000000 +0200
+++ ghc-6.4.1.20050819/ghc/lib/compat/Makefile	2005-08-23 10:23:56.000000000 +0200
@@ -42,28 +42,28 @@
 ifeq "$(ghc_ge_603)" "YES"
 # These modules are all provided in GHC 6.3+
 EXCLUDED_SRCS += \
-	System/Directory/Internals.hs \
-	Distribution/Compat/ReadP.hs \
-	Distribution/Extension.hs \
-	Distribution/GetOpt.hs \
-	Distribution/InstalledPackageInfo.hs \
-	Distribution/License.hs \
-	Distribution/Package.hs \
-	Distribution/ParseUtils.hs \
-	Distribution/Setup.hs \
-	Distribution/Version.hs
+	System/Directory/Internals.hs
+#	Distribution/Compat/ReadP.hs \
+#	Distribution/Extension.hs \
+#	Distribution/GetOpt.hs \
+#	Distribution/InstalledPackageInfo.hs \
+#	Distribution/License.hs \
+#	Distribution/Package.hs \
+#	Distribution/ParseUtils.hs \
+#	Distribution/Setup.hs \
+#	Distribution/Version.hs
 
 SRC_MKDEPENDHS_OPTS += \
-	-optdep--exclude-module=System.Directory.Internals \
-	-optdep--exclude-module=Distribution.Compat.ReadP \
-	-optdep--exclude-module=Distribution.Extension \
-	-optdep--exclude-module=Distribution.GetOpt \
-	-optdep--exclude-module=Distribution.InstalledPackageInfo \
-	-optdep--exclude-module=Distribution.License \
-	-optdep--exclude-module=Distribution.Package \
-	-optdep--exclude-module=Distribution.ParseUtils \
-	-optdep--exclude-module=Distribution.Compiler \
-	-optdep--exclude-module=Distribution.Version
+	-optdep--exclude-module=System.Directory.Internals
+#	-optdep--exclude-module=Distribution.Compat.ReadP \
+#	-optdep--exclude-module=Distribution.Extension \
+#	-optdep--exclude-module=Distribution.GetOpt \
+#	-optdep--exclude-module=Distribution.InstalledPackageInfo \
+#	-optdep--exclude-module=Distribution.License \
+#	-optdep--exclude-module=Distribution.Package \
+#	-optdep--exclude-module=Distribution.ParseUtils \
+#	-optdep--exclude-module=Distribution.Compiler \
+#	-optdep--exclude-module=Distribution.Version
 endif
 
 # Some explicit dependencies
diff -Naur ghc-6.4.1.20050819.orig/ghc/utils/ghc-pkg/Makefile ghc-6.4.1.20050819/ghc/utils/ghc-pkg/Makefile
--- ghc-6.4.1.20050819.orig/ghc/utils/ghc-pkg/Makefile	2005-08-23 10:22:25.000000000 +0200
+++ ghc-6.4.1.20050819/ghc/utils/ghc-pkg/Makefile	2005-08-23 10:24:17.000000000 +0200
@@ -30,7 +30,7 @@
 endif
 
 ifeq "$(ghc_ge_603)" "YES"
-SRC_HC_OPTS +=  -package Cabal
+# SRC_HC_OPTS +=  -package Cabal
 endif
 
 # On Windows, ghc-pkg is a standalone program
