diff --git a/src/ParseMonad.lhs b/src/ParseMonad.lhs
index 4b91c07..bb4c621 100644
--- a/src/ParseMonad.lhs
+++ b/src/ParseMonad.lhs
@@ -6,6 +6,9 @@ The parser monad.
 
 > module ParseMonad where
 
+> import Control.Applicative (Applicative(..))
+> import Control.Monad (liftM, ap)
+
 > data ParseResult a = OkP a | FailP String
 > newtype P a = P (String -> Int -> ParseResult a)
 > runP :: P a -> String -> Int -> ParseResult a
@@ -20,3 +23,10 @@ The parser monad.
 >               OkP a -> runP (k a) s l
 >               FailP err -> FailP err
 >       fail s = P $ \ _ _ -> FailP s
+
+> instance Functor P where
+>       fmap = liftM
+
+> instance Applicative P where
+>       pure = return
+>       (<*>) = ap
