diff -urwpN happy-1.15.orig/happy/src/LALR.lhs happy-1.15/happy/src/LALR.lhs
--- happy-1.15.orig/happy/src/LALR.lhs	2005-01-14 14:57:54.000000000 +0000
+++ happy-1.15/happy/src/LALR.lhs	2006-09-12 10:02:19.000000000 +0100
@@ -21,14 +21,14 @@ Generation of LALR parsing tables.
 
 > import Control.Monad.ST
 > import Data.Array.ST
-> import Data.Array hiding (bounds)
+> import Data.Array as Array
 > import Data.List (nub)
 
 #elif defined(__GLASGOW_HASKELL__)
 
 > import ST
 > import MArray
-> import Array 	hiding (bounds)
+> import Array as Array
 > import List (nub)
 
 #endif
@@ -623,7 +623,7 @@ Count the conflicts
 >   
 >   where
 >	   
->	conflictArray = listArray (bounds action) conflictList
+>	conflictArray = listArray (Array.bounds action) conflictList
 >	conflictList  = map countConflictsState (assocs action)
 >
 >	countConflictsState (state, actions) 
diff -urwpN happy-1.15.orig/happy/src/ProduceCode.lhs happy-1.15/happy/src/ProduceCode.lhs
--- happy-1.15.orig/happy/src/ProduceCode.lhs	2005-01-18 10:18:19.000000000 +0000
+++ happy-1.15/happy/src/ProduceCode.lhs	2006-09-12 09:51:32.000000000 +0100
@@ -27,8 +27,6 @@ The code generator.
 > import Data.Array.Unboxed ( UArray )
 > import Data.Array.MArray
 > import Data.Array.IArray 
-> 
-> marray_indices a = Data.Array.MArray.indices a
 
 #elif __GLASGOW_HASKELL__ > 408
 
