diff --git a/src/Stack/Types/BuildPlan.hs b/src/Stack/Types/BuildPlan.hs
index a815e88..60365c9 100644
--- a/src/Stack/Types/BuildPlan.hs
+++ b/src/Stack/Types/BuildPlan.hs
@@ -6,2 +6,3 @@
 {-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE OverlappingInstances       #-}
 -- | Shared types for various stackage packages.
diff --git a/src/Stack/Types/Compiler.hs b/src/Stack/Types/Compiler.hs
index 63bc6e2..61a1bf8 100644
--- a/src/Stack/Types/Compiler.hs
+++ b/src/Stack/Types/Compiler.hs
@@ -3,3 +3,3 @@
 {-# LANGUAGE FlexibleInstances #-}
-
+{-# LANGUAGE OverlappingInstances       #-}
 module Stack.Types.Compiler where
diff --git a/src/Stack/Types/FlagName.hs b/src/Stack/Types/FlagName.hs
index dce9daa..fcd1ce0 100644
--- a/src/Stack/Types/FlagName.hs
+++ b/src/Stack/Types/FlagName.hs
@@ -7,3 +7,3 @@
 {-# LANGUAGE TupleSections #-}
-
+{-# LANGUAGE OverlappingInstances       #-}
 -- | Names for flags.
diff --git a/src/Stack/Types/PackageName.hs b/src/Stack/Types/PackageName.hs
index 929d25e..b5c8321 100644
--- a/src/Stack/Types/PackageName.hs
+++ b/src/Stack/Types/PackageName.hs
@@ -7,3 +7,3 @@
 {-# LANGUAGE TupleSections #-}
-
+{-# LANGUAGE OverlappingInstances       #-}
 -- | Names for packages.
diff --git a/src/Stack/Types/Version.hs b/src/Stack/Types/Version.hs
index 50f0f10..6c45892 100644
--- a/src/Stack/Types/Version.hs
+++ b/src/Stack/Types/Version.hs
@@ -7,3 +7,3 @@
 {-# LANGUAGE DeriveDataTypeable #-}
-
+{-# LANGUAGE OverlappingInstances       #-}
 -- | Versions for packages.
