diff --git a/src/Options/Applicative/Builder/Extra.hs b/src/Options/Applicative/Builder/Extra.hs
index 93d378c..07d20df 100644
--- a/src/Options/Applicative/Builder/Extra.hs
+++ b/src/Options/Applicative/Builder/Extra.hs
@@ -129,3 +129,3 @@ execExtraHelp args helpOpt parser pd =
                                 parser <*>
-                                some (strArgument (metavar "OTHER ARGUMENTS"))))
+                                some (strArgument (metavar "OTHER ARGUMENTS") :: Parser String)))
                         (fullDesc <> progDesc pd))
diff --git a/src/main/Main.hs b/src/main/Main.hs
index 273c21f..a8a1381 100644
--- a/src/main/Main.hs
+++ b/src/main/Main.hs
@@ -190,4 +190,9 @@ main = do
 vcatErrorHelp :: ParserHelp -> ParserHelp -> ParserHelp
+#if MIN_VERSION_optparse_applicative(0,14,0)
+vcatErrorHelp (ParserHelp e1 _ _ _ _ _) (ParserHelp e2 s2 h2 u2 b2 f2) =
+  ParserHelp (vcatChunks [e2, e1]) s2 h2 u2 b2 f2
+#else
 vcatErrorHelp (ParserHelp e1 _ _ _ _) (ParserHelp e2 h2 u2 b2 f2) =
   ParserHelp (vcatChunks [e2, e1]) h2 u2 b2 f2
+#endif
 
@@ -540,4 +545,9 @@ interpreterHandler args f = do
 
+#if MIN_VERSION_optparse_applicative(0,14,0)
+    overrideErrorHelp (ParserHelp e1 _ _ _ _ _) (ParserHelp _ s2 h2 u2 b2 f2) =
+      ParserHelp e1 s2 h2 u2 b2 f2
+#else
     overrideErrorHelp (ParserHelp e1 _ _ _ _) (ParserHelp _ h2 u2 b2 f2) =
       ParserHelp e1 h2 u2 b2 f2
+#endif
 
