diff --git a/src/Network/HTTP/Download.hs b/src/Network/HTTP/Download.hs
index c12f713..ad23a77 100644
--- a/src/Network/HTTP/Download.hs
+++ b/src/Network/HTTP/Download.hs
@@ -5,2 +5,3 @@
 {-# LANGUAGE TemplateHaskell  #-}
+{-# LANGUAGE CPP  #-}
 module Network.HTTP.Download
@@ -50,3 +51,6 @@ import           Network.HTTP.Client         (path)
 import           Network.HTTP.Client.Conduit (HasHttpManager, Manager, Request,
-                                              Response, checkStatus,
+                                              Response,
+#if ! MIN_VERSION_http_client(0,5,0)
+                                              checkStatus,
+#endif
                                               getHttpManager, parseUrl,
@@ -110,3 +114,7 @@ redownload req0 dest = do
                     }
+#if ! MIN_VERSION_http_client(0,5,0)
         req2 = req1 { checkStatus = \_ _ _ -> Nothing }
+#else
+        req2 = req1
+#endif
     env <- ask
diff --git a/src/Stack/BuildPlan.hs b/src/Stack/BuildPlan.hs
index 2456d16..50a381b 100644
--- a/src/Stack/BuildPlan.hs
+++ b/src/Stack/BuildPlan.hs
@@ -8,2 +8,3 @@
 {-# LANGUAGE TupleSections      #-}
+{-# LANGUAGE CPP                #-}
 
@@ -78,3 +79,5 @@ import           Distribution.Text (display)
 import qualified Distribution.Version as C
+#if ! MIN_VERSION_http_client(0,5,0)
 import           Network.HTTP.Client (checkStatus)
+#endif
 import           Network.HTTP.Download
@@ -503,3 +506,7 @@ loadBuildPlan name = do
             $logDebug $ "Downloading build plan from: " <> url
+#if MIN_VERSION_http_client(0,5,0)
+            _ <- redownload req fp
+#else
             _ <- redownload req { checkStatus = handle404 } fp
+#endif
             $logStickyDone $ "Downloaded " <> renderSnapName name <> " build plan."
diff --git a/src/Stack/Upload.hs b/src/Stack/Upload.hs
index 8643018..80f3a2f 100644
--- a/src/Stack/Upload.hs
+++ b/src/Stack/Upload.hs
@@ -3,2 +3,3 @@
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE CPP #-}
 -- | Provide ability to upload tarballs to Hackage.
@@ -50,3 +51,6 @@ import           Network.HTTP.Client                   (BodyReader, Manager,
                                                         applyBasicAuth, brRead,
-                                                        checkStatus, newManager,
+#if ! MIN_VERSION_http_client(0,5,0)
+                                                        checkStatus,
+#endif
+                                                        newManager,
                                                         parseUrl,
@@ -200,3 +204,5 @@ mkUploader config us = do
             { requestHeaders = [("Accept", "text/plain")]
+#if ! MIN_VERSION_http_client(0,5,0)
             , checkStatus = \_ _ _ -> Nothing
+#endif
             }
diff --git a/stack.cabal b/stack.cabal
index 3cdf327..29763a3 100644
--- a/stack.cabal
+++ b/stack.cabal
@@ -181,5 +181,5 @@ library
         hpc >=0.6.0.2 && <0.7,
-        http-client >=0.4.17 && <0.5,
-        http-client-tls >=0.2.2 && <0.3,
-        http-conduit >=2.1.7 && <2.2,
+        http-client >=0.4.17,
+        http-client-tls >=0.2.2,
+        http-conduit >=2.1.7,
         http-types >=0.8.6 && <0.10,
@@ -256,3 +256,3 @@ executable stack
         filepath >=1.4.0.0 && <1.5,
-        http-client >=0.4.28 && <0.5,
+        http-client >=0.4.28,
         lifted-base >=0.2.3.6 && <0.3,
@@ -289,3 +289,3 @@ test-suite stack-test
         hspec >=2.2.3 && <2.3,
-        http-conduit >=2.1.10.1 && <2.2,
+        http-conduit >=2.1.10.1,
         monad-logger >=0.3.18 && <0.4,
