--- distributed-process-0.3.0-orig/distributed-process.cabal	2012-08-11 02:01:55.000000000 +1000
+++ distributed-process-0.3.0/distributed-process.cabal	2012-08-11 23:39:30.858791322 +1000
@@ -41,9 +41,9 @@
                      containers >= 0.4 && < 0.6,
                      old-locale >= 1.0 && < 1.1,
                      time >= 1.2 && < 1.5,
-                     template-haskell >= 2.6 && < 2.8,
+                     template-haskell >= 2.6 && < 2.9,
                      random >= 1.0 && < 1.1,
-                     ghc-prim >= 0.2 && < 0.3,
+                     ghc-prim >= 0.2 && < 0.4,
                      distributed-static >= 0.1 && < 0.2,
                      rank1dynamic >= 0.1 && < 0.2,
                      syb >= 0.3 && < 0.4
@@ -84,9 +84,9 @@
                      containers >= 0.4 && < 0.6,
                      old-locale >= 1.0 && < 1.1,
                      time >= 1.2 && < 1.5,
-                     template-haskell >= 2.6 && < 2.8,
+                     template-haskell >= 2.6 && < 2.9,
                      random >= 1.0 && < 1.1,
-                     ghc-prim >= 0.2 && < 0.3,
+                     ghc-prim >= 0.2 && < 0.4,
                      ansi-terminal >= 0.5 && < 0.6,
                      distributed-static >= 0.1 && < 0.2,
                      rank1dynamic >= 0.1 && < 0.2
@@ -117,9 +117,9 @@
                      containers >= 0.4 && < 0.6,
                      old-locale >= 1.0 && < 1.1,
                      time >= 1.2 && < 1.5,
-                     template-haskell >= 2.6 && < 2.8,
+                     template-haskell >= 2.6 && < 2.9,
                      random >= 1.0 && < 1.1,
-                     ghc-prim >= 0.2 && < 0.3,
+                     ghc-prim >= 0.2 && < 0.4,
                      ansi-terminal >= 0.5 && < 0.6,
                      distributed-static >= 0.1 && < 0.2,
                      rank1dynamic >= 0.1 && < 0.2,
