diff --git a/pango.cabal b/pango.cabal
index 9f607b0..3bf3181 100644
--- a/pango.cabal
+++ b/pango.cabal
@@ -41,7 +41,7 @@ Flag new-exception
 
 Library
         build-depends:  base >= 3 && < 5,
-                        process, directory, array, containers, pretty, mtl, haskell98,
+                        process, directory, array, containers, pretty, mtl,
                         glib  >= 0.12 && < 0.13,
                         cairo >= 0.12 && < 0.13
 
