--- authenticate-0.9.3.1-orig/authenticate.cabal	2011-08-19 15:35:11.000000000 +1000
+++ authenticate-0.9.3.1/authenticate.cabal	2011-08-20 19:55:33.120795045 +1000
@@ -36,7 +36,7 @@
                      attoparsec >= 0.9   && < 0.10,
                      tls >= 0.7 && < 0.8,
                      containers,
-                     process >= 1.0.1.1 && < 1.1
+                     process >= 1.0.1.1 && < 1.2
     exposed-modules: Web.Authenticate.Rpxnow,
                      Web.Authenticate.OpenId,
                      Web.Authenticate.BrowserId,
