--- xdg-basedir-0.2.1-orig/System/Environment/XDG/BaseDir.hs	2011-06-21 09:13:40.000000000 +1000
+++ xdg-basedir-0.2.1/System/Environment/XDG/BaseDir.hs	2012-08-11 09:37:10.694718722 +1000
@@ -20,7 +20,6 @@
 import Data.Maybe         ( fromMaybe )
 import System.FilePath    ( (</>), splitSearchPath )
 import System.Environment ( getEnvironment, getEnv )
-import System.IO.Error    ( try )
 import System.Directory   ( getHomeDirectory )
 import Control.Monad      ( liftM2 )
 
