diff --git a/Yesod/Paginator.hs b/Yesod/Paginator.hs
index b07ea09..097bb34 100644
--- a/Yesod/Paginator.hs
+++ b/Yesod/Paginator.hs
@@ -81,2 +81,3 @@ selectPaginated :: ( PersistEntity val
                    , Yesod m
+                   , BaseBackend (YesodPersistBackend m) ~ YesodPersistBackend m
                    )
@@ -92,2 +93,3 @@ selectPaginatedWith :: ( PersistEntity val
                        , Yesod m
+                       , BaseBackend (YesodPersistBackend m) ~ YesodPersistBackend m
                        )
