diff --git a/Test/Framework/TH/Prime/Parser.hs b/Test/Framework/TH/Prime/Parser.hs
index 996b4f6..7fe8998 100644
--- a/Test/Framework/TH/Prime/Parser.hs
+++ b/Test/Framework/TH/Prime/Parser.hs
@@ -70,7 +70,7 @@ parseTest file = do
         toStr (Ident str) = str
         toStr (Symbol str) = str
     opt raw = defaultParseMode {
-        extensions = nub $ TemplateHaskell : exts raw
+        extensions = nub $ map EnableExtension $ TemplateHaskell : exts raw
       -- to prevent "Ambiguous infix expression"
       , fixities = Nothing
       }
