diff --git a/src/System/Plugins/Make.hs b/src/System/Plugins/Make.hs
index 40e7a2f..b5a01b4 100644
--- a/src/System/Plugins/Make.hs
+++ b/src/System/Plugins/Make.hs
@@ -426,4 +426,4 @@ rawMerge src stb out always_merge = do
 makeClean :: FilePath -> IO ()
-makeClean f = let f_hi = dropSuffix  f <> hiSuf
-                  f_o  = dropSuffix  f <> objSuf
+makeClean f = let f_hi = dropSuffix  f U.<> hiSuf
+                  f_o  = dropSuffix  f U.<> objSuf
               in mapM_ rm_f [f_hi, f_o]
@@ -431,3 +431,3 @@ makeClean f = let f_hi = dropSuffix  f <> hiSuf
 makeCleaner :: FilePath -> IO ()
-makeCleaner f = makeClean f >> rm_f (dropSuffix f <> hsSuf)
+makeCleaner f = makeClean f >> rm_f (dropSuffix f U.<> hsSuf)
 
