diff --git a/src/System/Plugins/Load.hs b/src/System/Plugins/Load.hs
index 58c3e77..6d911b6 100644
--- a/src/System/Plugins/Load.hs
+++ b/src/System/Plugins/Load.hs
@@ -74,3 +74,9 @@ import BinIface
 import HscTypes
-import Module (moduleName, moduleNameString, packageIdString)
+import Module (moduleName, moduleNameString)
+#if MIN_VERSION_ghc(7,10,1)
+import Module (packageKeyString)
+#else
+import Module (packageIdString)
+#endif
+
 import HscMain (newHscEnv)
@@ -703,3 +709,5 @@ loadDepends obj incpaths = do
                 let ps = dep_pkgs ds
-#if MIN_VERSION_ghc(7,2,0)
+#if MIN_VERSION_ghc(7,10,0)
+                ps' <- filterM loaded . map packageKeyString . nub $ map fst ps
+#elif MIN_VERSION_ghc(7,2,0)
                 ps' <- filterM loaded . map packageIdString . nub $ map fst ps
