diff --git a/src/System/Plugins/Utils.hs b/src/System/Plugins/Utils.hs
index f754157..9029c38 100644
--- a/src/System/Plugins/Utils.hs
+++ b/src/System/Plugins/Utils.hs
@@ -70,8 +70,11 @@ import Prelude hiding                  (catch)
 
 import System.IO
 import System.Environment           ( getEnv )
-import System.Directory             ( doesFileExist, getModificationTime, removeFile )
-
+import System.Directory             ( doesFileExist, getModificationTime, removeFile
+#if MIN_VERSION_directory(1,2,0)
+                                    , findFile
+#endif
+                                    )
 -- ---------------------------------------------------------------------
 -- some misc types we use
 
@@ -163,6 +166,7 @@ hMkUniqueIn dir = do (t,h) <- mkTempIn dir
                         then hClose h >> removeFile t >> hMkUniqueIn dir
                         else return (t,h)
 
+#if ! MIN_VERSION_directory(1,2,0)
 findFile :: [String] -> FilePath -> IO (Maybe FilePath)
 findFile [] _  = return Nothing
 findFile (ext:exts) file
@@ -170,6 +174,7 @@ findFile (ext:exts) file
          b <- doesFileExist l
          if b then return $ Just l
               else findFile exts file
+#endif
 
 -- ---------------------------------------------------------------------
 -- some filename manipulation stuff
