--- plugins-1.5.7-orig/src/System/Plugins/Env.hs	2018-04-25 21:28:44.810882904 +1000
+++ plugins-1.5.7/src/System/Plugins/Env.hs	2018-04-25 21:18:59.559026720 +1000
@@ -467,6 +467,9 @@
                 -- real packages.
                 settings <- initSysTools (Just libdir)
                 dflags <- initDynFlags $ defaultDynFlags settings
+#if MIN_VERSION_ghc(8,4,0)
+                         [] -- TODO: I do not know how to provide the LlvmTargets parameter
+#endif
                 libs <- mapM (findHSlib
 #if MIN_VERSION_ghc(7,8,0)
                               (WayDyn `elem` ways dflags || dynamicGhc)
--- plugins-1.5.7-orig/src/System/Plugins/Load.hs	2018-04-25 21:28:44.818882916 +1000
+++ plugins-1.5.7/src/System/Plugins/Load.hs	2018-04-25 21:25:05.596561865 +1000
@@ -127,7 +127,11 @@
     -- kludgy as hell
 #if MIN_VERSION_ghc(7,2,0)
     mySettings <- initSysTools (Just libdir) -- how should we really set the top dir?
-    dflags <- initDynFlags (defaultDynFlags mySettings)
+    dflags <- initDynFlags (defaultDynFlags mySettings
+#if MIN_VERSION_ghc(8,4,0)
+                           [] -- TODO: I do not know how to provide the LlvmTargets parameter
+#endif
+                          )
     e <- newHscEnv dflags
 #else
     e <- newHscEnv defaultCallbacks undefined
--- plugins-1.5.7-orig/src/System/Plugins/Make.hs	2018-01-24 01:41:00.000000000 +1100
+++ plugins-1.5.7/src/System/Plugins/Make.hs	2018-04-25 21:24:18.872493490 +1000
@@ -58,7 +58,8 @@
 
   ) where
 
-import System.Plugins.Utils
+import System.Plugins.Utils hiding     ((<>))
+import qualified System.Plugins.Utils as U ((<>))
 import System.Plugins.Parser
 import System.Plugins.LoadTypes        ( Module (Module, path) )
 import System.Plugins.Consts           ( ghc, hiSuf, objSuf, hsSuf )
@@ -343,7 +344,7 @@
                 Nothing -> do out <- mkUnique
                               addMerge src stb (dropSuffix out)
                               return (out, True) -- definitely out of date
-                Just nm -> return $ (nm <> hsSuf, False)
+                Just nm -> return $ (nm U.<> hsSuf, False)
     rawMerge src stb out domerge
 
 -- | 'mergeTo' behaves like 'merge', but we can specify the file in
--- plugins-1.5.7-orig/src/System/Plugins/Utils.hs	2018-04-25 21:28:44.824882924 +1000
+++ plugins-1.5.7/src/System/Plugins/Utils.hs	2018-04-25 21:20:16.559139195 +1000
@@ -45,7 +45,7 @@
     dirname,
     basename,
 
-    (</>), (<.>), (<+>), (<>),
+    (</>), (<.>), (<+>), (System.Plugins.Utils.<>),
 
     newer,
 
