--- a/src/System/Plugins/Env.hs
+++ b/src/System/Plugins/Env.hs
@@ -80,3 +80,9 @@ import DynFlags (
   defaultDynFlags, initDynFlags)
+
+
+#if MIN_VERSION_ghc(8,10,1)
+import SysTools (initSysTools, lazyInitLlvmConfig)
+#else
 import SysTools (initSysTools, initLlvmConfig)
+#endif
 
@@ -493,3 +499,6 @@ lookupPkg' p = withPkgEnvs env $ \fms -> go fms p
                 -- real packages.
-#if MIN_VERSION_ghc(8,8,1)
+#if MIN_VERSION_ghc(8,10,1)
+                settings <- initSysTools (libdir)
+                llvmConfig <- lazyInitLlvmConfig (libdir)
+#elif MIN_VERSION_ghc(8,8,1)
                 settings <- initSysTools (libdir)
--- a/src/System/Plugins/Load.hs
+++ b/src/System/Plugins/Load.hs
@@ -106,4 +106,9 @@ import DynFlags                 (defaultDynFlags, initDynFlags)
 import GHC.Paths                (libdir)
+
+#if MIN_VERSION_ghc(8,10,1)
+import SysTools                 (initSysTools, lazyInitLlvmConfig)
+#else
 import SysTools                 (initSysTools, initLlvmConfig)
 #endif
+#endif
 import GHC.Ptr                  ( Ptr(..), nullPtr )
@@ -128,3 +133,6 @@ readBinIface' hi_path = do
 #if MIN_VERSION_ghc(7,2,0)
-#if MIN_VERSION_ghc(8,8,1)
+#if MIN_VERSION_ghc(8,10,1)
+    mySettings <- initSysTools (libdir) -- how should we really set the top dir?
+    llvmConfig <- lazyInitLlvmConfig (libdir)
+#elif MIN_VERSION_ghc(8,8,1)
     mySettings <- initSysTools (libdir) -- how should we really set the top dir?
