--- gitit-0.9.0.1-orig/gitit.cabal	2012-04-15 15:47:00.000000000 +1000
+++ gitit-0.9.0.1/gitit.cabal	2012-04-23 21:54:50.153944198 +1000
@@ -101,6 +101,9 @@
                      the executable and the memory it uses, so those who will not need
                      plugins should disable this flag.
   default:           True
+Flag blaze_html_0_5
+  Description:   Use blaze-html 0.5 and blaze-markup 0.5
+  Default:       False
 
 Library
   hs-source-dirs:    .
@@ -165,8 +168,14 @@
                      feed >= 0.3.6 && < 0.4,
                      xss-sanitize >= 0.3 && < 0.4,
                      tagsoup >= 0.12 && < 0.13,
-                     blaze-html >= 0.4 && < 0.5,
                      json >= 0.4 && < 0.6
+  if flag(blaze_html_0_5)
+    build-depends:
+                     blaze-html >= 0.5 && < 0.6,
+                     blaze-markup >= 0.5.1 && < 0.6
+  else
+    build-depends:
+                     blaze-html >= 0.4 && < 0.5
   if impl(ghc >= 6.10)
     build-depends:   base >= 4, syb
   if impl(ghc >= 7.0.3)
