diff --git a/src/Network/Gitit/Feed.hs b/src/Network/Gitit/Feed.hs
index 912f7b25..e66a659f 100644
--- a/src/Network/Gitit/Feed.hs
+++ b/src/Network/Gitit/Feed.hs
@@ -36,7 +36,7 @@ import Data.Maybe (fromMaybe)
 import Data.Ord (comparing)
 import Network.URI (isUnescapedInURI, escapeURIString)
 import System.FilePath (dropExtension, takeExtension, (<.>))
-import Data.FileStore.Generic (Diff(..), diff)
+import Data.FileStore.Generic (Diff, PolyDiff(..), diff)
 import Data.FileStore.Types (history, retrieve, Author(authorName), Change(..),
          FileStore, Revision(..), TimeRange(..), RevisionId)
 import Text.Atom.Feed (nullEntry, nullFeed, nullLink, nullPerson,
