--- gitit-0.8.0.1-orig/gitit.cabal	2011-06-07 05:59:50.000000000 +1000
+++ gitit-0.8.0.1/gitit.cabal	2011-08-30 11:41:00.057312082 +1000
@@ -143,14 +143,15 @@
                      filestore >= 0.4.0.2 && < 0.5,
                      zlib >= 0.5 && < 0.6,
                      url >= 2.1 && < 2.2,
-                     happstack-server >= 6.0 && < 6.2,
+                     happstack-server >= 6.0 && < 6.3,
                      happstack-util >= 6.0 && < 6.2,
                      xml >= 1.3.5,
                      hslogger >= 1 && < 1.2,
                      ConfigFile >= 1 && < 1.2,
                      feed >= 0.3.6 && < 0.4,
-                     xss-sanitize >= 0.2 && < 0.3,
-                     json >= 0.4 && < 0.5
+                     xss-sanitize >= 0.2 && < 0.4,
+                     json >= 0.4 && < 0.5,
+                     text >= 0.11 && < 0.12
   if impl(ghc >= 6.10)
     build-depends:   base >= 4, syb
   if impl(ghc >= 7.0.3)
--- gitit-0.8.0.1-orig/Network/Gitit/ContentTransformer.hs	2011-06-07 05:59:50.000000000 +1000
+++ gitit-0.8.0.1/Network/Gitit/ContentTransformer.hs	2011-08-30 12:05:14.175292486 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-
 Copyright (C) 2009 John MacFarlane <jgm@berkeley.edu>,
 Anton van Straaten <anton@appsolutions.com>
@@ -71,6 +72,7 @@
 import Control.Exception (throwIO, catch)
 import Control.Monad.State
 import Data.Maybe (isNothing, mapMaybe)
+import Data.Text (pack, unpack)
 import Network.Gitit.Cache (lookupCache, cacheContents)
 import Network.Gitit.Export (exportFormats)
 import Network.Gitit.Framework
@@ -336,7 +338,12 @@
   toc <- liftM ctxTOC get
   bird <- liftM ctxBirdTracks get
   cfg <- lift getConfig
-  return $ primHtml $ sanitizeBalance $
+  return $ primHtml $ 
+#if MIN_VERSION_xss_sanitize(0,3,0)
+           unpack $ sanitizeBalance $ pack $
+#else
+           sanitizeBalance $
+#endif
            writeHtmlString defaultWriterOptions{
                         writerStandalone = True
                       , writerTemplate = "$if(toc)$\n$toc$\n$endif$\n$body$"
