diff --git a/src/Network/Gitit/Initialize.hs b/src/Network/Gitit/Initialize.hs
index 64a7f6e..00c1eef 100644
--- a/src/Network/Gitit/Initialize.hs
+++ b/src/Network/Gitit/Initialize.hs
@@ -138,7 +138,7 @@ createDefaultPages conf = do
                        RST        -> writeRST defOpts <=< toPandoc
                        Textile    -> writeTextile defOpts <=< toPandoc
                        Org        -> writeOrg defOpts <=< toPandoc
-                       DocBook    -> writeDocbook5 defOpts <=< toPandoc
+                       DocBook    -> writeDocBook5 defOpts <=< toPandoc
                        MediaWiki  -> writeMediaWiki defOpts <=< toPandoc
                        CommonMark -> writeCommonMark defOpts <=< toPandoc
 
