diff --git a/include/libssh2_local.h b/include/libssh2_local.h
index 09db01d..5165069 100644
--- a/include/libssh2_local.h
+++ b/include/libssh2_local.h
@@ -1,7 +1 @@
-
 #include <sys/stat.h>
-
-// This definition is used to determine sizeof(struct stat),
-// because c2hs lacks support for such case.
-typedef struct stat stat_t;
-
diff --git a/src/Network/SSH/Client/LibSSH2/Foreign.chs b/src/Network/SSH/Client/LibSSH2/Foreign.chs
index e02b140..61d08e9 100644
--- a/src/Network/SSH/Client/LibSSH2/Foreign.chs
+++ b/src/Network/SSH/Client/LibSSH2/Foreign.chs
@@ -479,3 +479,3 @@ type Offset = {# type off_t #}
 
--- {# pointer *stat_t as Stat newtype #}
+-- {# pointer *stat as Stat newtype #}
 
@@ -486,5 +486,5 @@ scpReceiveChannel s path = do
   withCString path $ \pathptr ->
-     allocaBytes {# sizeof stat_t #} $ \statptr -> do
+     allocaBytes {# sizeof stat #} $ \statptr -> do
        channel <- handleNullPtr (Just s) (channelFromPointer s) $ {# call scp_recv #} (toPointer s) pathptr statptr
-       size <- {# get stat_t->st_size #} statptr
+       size <- {# get stat->st_size #} statptr
        return (channel, size)
