diff --git a/src-exec/y0l0bot.hs b/src-exec/y0l0bot.hs
index 147e7ec..cca0295 100644
--- a/src-exec/y0l0bot.hs
+++ b/src-exec/y0l0bot.hs
@@ -71,4 +71 @@ import System.IO.Error
-import System.Locale
-  (
-  defaultTimeLocale,
-  )
+import Data.Time.Locale.Compat (defaultTimeLocale)
diff --git a/y0l0bot.cabal b/y0l0bot.cabal
index 9cab247..9690da8 100644
--- a/y0l0bot.cabal
+++ b/y0l0bot.cabal
@@ -16 +16 @@ executable yolobot
-  build-depends:    base       >=4.7  && <4.8,
+  build-depends:    base       >=4.7,
@@ -26 +26,2 @@ executable yolobot
-                    time       >=1.4  && <1.5
+                    time       >=1.4,
+                    time-locale-compat
