--- lens-3.5.1-orig/src/Control/Lens/IndexedTraversal.hs	2012-11-23 05:43:43.000000000 +1100
+++ lens-3.5.1/src/Control/Lens/IndexedTraversal.hs	2012-11-24 14:25:41.734475664 +1100
@@ -221,7 +221,7 @@
 --
 -- @'ignored' :: 'IndexedTraversal' i s s a b@
 --
--- @'ignored' ≡ 'const' 'pure'@
+-- @'ignored' = 'const' 'pure'@
 ignored :: forall k f i s a b. (Indexed i k, Applicative f) => Overloaded k f s s a b
 ignored = index $ \ (_ :: i -> a -> f b) s -> pure s :: f s
 {-# INLINE ignored #-}
