diff --git a/src/Control/Lens/Iso.hs b/src/Control/Lens/Iso.hs
index 3137f49..8ebc828 100644
--- a/src/Control/Lens/Iso.hs
+++ b/src/Control/Lens/Iso.hs
@@ -230,7 +230,7 @@ anon a p = iso (fromMaybe a) go where
 
 -- | The canonical isomorphism for currying and uncurrying a function.
 --
--- >>> :t fst^.curried
+-- :t fst^.curried
 -- fst^.curried :: a -> b -> a
 --
 -- @'curried' = 'iso' 'curry' 'uncurry'@
@@ -240,7 +240,7 @@ curried = iso curry uncurry
 
 -- | The canonical isomorphism for uncurrying and currying a function.
 --
--- >>> :t flip (,)^.uncurried
+-- :t flip (,)^.uncurried
 -- flip (,)^.uncurried :: (b, a) -> (a, b)
 --
 -- @'uncurried' = 'iso' 'uncurry' 'curry'@
