diff --git a/mwc-random.cabal b/mwc-random.cabal
index 13c045c..1a36ae1 100644
--- a/mwc-random.cabal
+++ b/mwc-random.cabal
@@ -55,18 +55,18 @@ library
     ghc-options: -fwarn-tabs
 
 test-suite tests
-  buildable:      False
   type:           exitcode-stdio-1.0
   hs-source-dirs: test
   main-is:        tests.hs
   other-modules:  KS
                   QC
-                  Uniform
+                  
 
   ghc-options:
     -Wall -threaded -rtsopts
 
   build-depends:
+    vector >= 0.7,
     HUnit,
     QuickCheck,
     base,
