diff --git a/lib/Hakyll/Web/Pandoc/Biblio.hs b/lib/Hakyll/Web/Pandoc/Biblio.hs
index dfe6d93..c93813b 100644
--- a/lib/Hakyll/Web/Pandoc/Biblio.hs
+++ b/lib/Hakyll/Web/Pandoc/Biblio.hs
@@ -13,4 +13,5 @@
 {-# LANGUAGE DeriveDataTypeable         #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE CPP #-}
 module Hakyll.Web.Pandoc.Biblio
     ( CSL
@@ -85,6 +86,9 @@ biblioCompiler :: Compiler (Item Biblio)
 biblioCompiler = do
     filePath <- toFilePath <$> getUnderlying
+#if MIN_VERSION_pandoc_citeproc(0,14,0)
+    makeItem =<< unsafeCompiler (Biblio <$> CSL.readBiblioFile (const True) filePath)
+#else
     makeItem =<< unsafeCompiler (Biblio <$> CSL.readBiblioFile filePath)
-
+#endif
 
 --------------------------------------------------------------------------------
diff --git a/lib/Hakyll/Web/Pandoc/Binary.hs b/lib/Hakyll/Web/Pandoc/Binary.hs
index 3c5b5a3..4e1a0b9 100644
--- a/lib/Hakyll/Web/Pandoc/Binary.hs
+++ b/lib/Hakyll/Web/Pandoc/Binary.hs
@@ -1,4 +1,4 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
-{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE CPP, DeriveGeneric #-}
 module Hakyll.Web.Pandoc.Binary where
 
@@ -31,2 +31,5 @@ instance Binary REF.RefType
 instance Binary STY.Agent
 instance Binary STY.Formatted
+#if MIN_VERSION_pandoc_citeproc(0,14,0)
+instance Binary REF.Season
+#endif
