diff '--color=auto' -Naur a/hakyll.cabal b/hakyll.cabal
--- a/hakyll.cabal	2001-09-09 05:46:40.000000000 +0400
+++ b/hakyll.cabal	2026-06-24 11:37:33.145970458 +0300
@@ -254,7 +254,7 @@
     Other-Modules:
       Hakyll.Web.Pandoc.Binary
     Build-Depends:
-      pandoc >= 2.11 && < 2.20 || >= 3.0 && < 3.10,
+      pandoc >= 2.11 && < 2.20 || >= 3.0 && < 3.11,
       pandoc-types >= 1.22 && < 1.24
     Cpp-options:
       -DUSE_PANDOC
@@ -324,7 +324,7 @@
     Cpp-options:
       -DUSE_PANDOC
     Build-Depends:
-      pandoc >= 2.11 && < 2.20 || >= 3.0 && < 3.10,
+      pandoc >= 2.11 && < 2.20 || >= 3.0 && < 3.11,
       pandoc-types >= 1.22 && < 1.24
 
 
@@ -359,5 +359,5 @@
     base      >= 4.12  && < 5,
     directory >= 1.0   && < 1.4,
     filepath  >= 1.0   && < 1.6,
-    pandoc    >= 2.11  && < 2.20 || >= 3.0 && < 3.10,
+    pandoc    >= 2.11  && < 2.20 || >= 3.0 && < 3.11,
     pandoc-types >= 1.22 && < 1.24
