commit 889e6f59f761795af3f2cb83c2a551b9c8478d66
Author: Jasper Van der Jeugt <m@jaspervdj.be>
Date:   Mon Apr 7 19:49:17 2014 +0200

    Fix compilation when disabling preview server

diff --git a/src/Hakyll/Commands.hs b/src/Hakyll/Commands.hs
index e221dc5..8db889c 100644
--- a/src/Hakyll/Commands.hs
+++ b/src/Hakyll/Commands.hs
@@ -102,7 +102,7 @@ watch conf verbosity host port runServer rules = do
     loop = threadDelay 100000 >> loop
     server' = if runServer then server conf host port else loop
 #else
-watch _ _ _ _ = watchServerDisabled
+watch _ _ _ _ _ _ = watchServerDisabled
 #endif
 
 --------------------------------------------------------------------------------
