--- hakyll-3.2.7.2-orig/hakyll.cabal	2012-04-16 03:48:06.000000000 +1000
+++ hakyll-3.2.7.2/hakyll.cabal	2012-04-23 22:53:26.277858418 +1000
@@ -61,7 +61,8 @@
   Build-Depends:
     base        >= 4      && < 5,
     binary      >= 0.5    && < 0.6,
-    blaze-html  >= 0.4    && < 0.6,
+    blaze-html  >= 0.5    && < 0.6,
+    blaze-markup >= 0.5.1  && < 0.6,
     bytestring  >= 0.9    && < 0.10,
     citeproc-hs >= 0.3.2  && < 0.4,
     containers  >= 0.3    && < 0.5,
--- hakyll-3.2.7.2-orig/src/Hakyll/Web/Util/Html.hs	2012-04-16 03:48:05.000000000 +1000
+++ hakyll-3.2.7.2/src/Hakyll/Web/Util/Html.hs	2012-04-23 23:05:30.077184565 +1000
@@ -5,8 +5,8 @@
     , escapeHtml
     ) where
 
-import Text.Blaze (toHtml)
-import Text.Blaze.Renderer.String (renderHtml)
+import Text.Blaze.Html (toHtml)
+import Text.Blaze.Html.Renderer.String (renderHtml)
 
 -- | Strip all HTML tags from a string
 --
--- hakyll-3.2.7.2-orig/src/Hakyll/Core/Writable.hs	2012-04-16 03:48:05.000000000 +1000
+++ hakyll-3.2.7.2/src/Hakyll/Core/Writable.hs	2012-04-23 23:06:07.217380492 +1000
@@ -9,8 +9,8 @@
 
 import qualified Data.ByteString as SB
 import qualified Data.ByteString.Lazy as LB
-import Text.Blaze (Html)
-import Text.Blaze.Renderer.String (renderHtml)
+import Text.Blaze.Html (Html)
+import Text.Blaze.Html.Renderer.String (renderHtml)
 
 import Hakyll.Core.Identifier
 
--- hakyll-3.2.7.2-orig/src/Hakyll/Web/Blaze.hs	2012-04-16 03:48:05.000000000 +1000
+++ hakyll-3.2.7.2/src/Hakyll/Web/Blaze.hs	2012-04-23 23:02:00.559436401 +1000
@@ -7,7 +7,8 @@
     , getBodyHtml'
     ) where
 
-import Text.Blaze (Html, toHtml, preEscapedString)
+import Text.Blaze.Html (Html, toHtml)
+import Text.Blaze.Internal (preEscapedString)
 
 import Hakyll.Web.Page
 import Hakyll.Web.Page.Metadata
--- hakyll-3.2.7.2-orig/src/Hakyll/Web/Tags.hs	2012-04-16 03:48:05.000000000 +1000
+++ hakyll-3.2.7.2/src/Hakyll/Web/Tags.hs	2012-04-23 23:10:02.206945147 +1000
@@ -52,8 +52,8 @@
 
 import Data.Typeable (Typeable)
 import Data.Binary (Binary, get, put)
-import Text.Blaze.Renderer.String (renderHtml)
-import Text.Blaze ((!), toHtml, toValue)
+import Text.Blaze.Html.Renderer.String (renderHtml)
+import Text.Blaze.Html ((!), toHtml, toValue)
 import qualified Text.Blaze.Html5 as H
 import qualified Text.Blaze.Html5.Attributes as A
 
