diff --git a/FeedCLI.hs b/FeedCLI.hs
index 30393e1..2bae7b6 100644
--- a/FeedCLI.hs
+++ b/FeedCLI.hs
@@ -32,3 +32,3 @@ import Text.XML.Light.Input (parseXMLDoc)
 import System.Environment(getArgs)
-import System.Locale (defaultTimeLocale)
+import Data.Time.Locale.Compat (defaultTimeLocale)
 import System.IO (openTempFile, openFile, hGetContents
diff --git a/feed-cli.cabal b/feed-cli.cabal
index fb07cb0..a03f1b1 100644
--- a/feed-cli.cabal
+++ b/feed-cli.cabal
@@ -9,3 +9,3 @@ License: BSD3
 License-File: LICENSE
-build-depends: feed, xml,  base < 4, time, old-locale, old-time, directory
+build-depends: feed, xml,  base, time, time-locale-compat, old-locale, old-time, directory
 data-files: README, examples/getURL.sh,  examples/piper2.sh,  examples/piper.sh
