diff --git a/src/library/Yi/Buffer/Misc.hs b/src/library/Yi/Buffer/Misc.hs
index d8ae2ed..2f60af8 100644
--- a/src/library/Yi/Buffer/Misc.hs
+++ b/src/library/Yi/Buffer/Misc.hs
@@ -391,9 +391,9 @@ getPercent a b = T.justifyRight 3 ' ' (T.pack $ show p) `T.snoc` '%'
 queryBuffer :: (forall syntax. BufferImpl syntax -> x) -> BufferM x
-queryBuffer = gets . queryRawbuf
+queryBuffer x = gets (queryRawbuf x)
 
 modifyBuffer :: (forall syntax. BufferImpl syntax -> BufferImpl syntax) -> BufferM ()
-modifyBuffer = modify . modifyRawbuf
+modifyBuffer x = modify (modifyRawbuf x)
 
 queryAndModify :: (forall syntax. BufferImpl syntax -> (BufferImpl syntax,x)) -> BufferM x
-queryAndModify = getsAndModify . queryAndModifyRawbuf
+queryAndModify x = getsAndModify (queryAndModifyRawbuf x)
 
@@ -814,3 +814,3 @@ withMode0 f FBuffer {bmode = m} = f m
 withModeB :: (forall syntax. Mode syntax -> BufferM a) -> BufferM a
-withModeB = join . gets . withMode0
+withModeB x = join (gets (withMode0 x))
 
@@ -828,3 +828,3 @@ focusSyntax r = modifyRawbuf (focusAst r)
 withSyntaxB' :: (forall syntax. Mode syntax -> syntax -> BufferM a) -> BufferM a
-withSyntaxB' = join . withSyntaxB
+withSyntaxB' x = join (withSyntaxB x)
 
