diff --git a/src/library/Yi/UI/Vte.hs b/src/library/Yi/UI/Vte.hs
index 2580b3c..3476b83 100644
--- a/src/library/Yi/UI/Vte.hs
+++ b/src/library/Yi/UI/Vte.hs
@@ -27,7 +27,7 @@ initUI cfg _ch _outCh _editor = do
     discard $ win `onDestroy` mainQuit
 
     -- Setup vte
-    exe  <- getExecutablePath
+    exe  <- System.Environment.Executable.getExecutablePath
     term <- terminalNew
     discard $ Graphics.UI.Gtk.on term childExited $ end False
 
