--- yi-0.6.6.0-orig/yi.cabal	2012-10-13 19:40:12.923138129 +1100
+++ yi-0.6.6.0/yi.cabal	2012-10-13 18:45:00.446156795 +1100
@@ -80,7 +80,7 @@
 
 library
   hs-source-dirs: src/library
-  default-language: Haskell2010
+  default-language: Haskell98
 
   if flag(hacking)
     buildable: False
@@ -231,12 +231,12 @@
     regex-base ==0.93.*,
     regex-tdfa == 1.1.*,
     rosezipper >= 0.1 && < 0.3,
-    split >= 0.1 && < 0.3,
+    split == 0.1.*,
     template-haskell >= 2.4 && < 2.9,
     time >= 1.1 && < 1.5,
     utf8-string >= 0.3.1,
     uniplate,
-    unix-compat >=0.1 && <0.5,
+    unix-compat >=0.1 && <0.4,
     unordered-containers >= 0.1.3 && < 0.3
 
   build-tools: alex >= 3
@@ -312,7 +312,7 @@
     exposed-modules:
       Yi.UI.Vty
     build-depends:
-      unix-compat >=0.1 && <0.5,
+      unix-compat >=0.1 && <0.4,
       vty >= 4.7.0.0 && <5
     cpp-options: -DFRONTEND_VTY
 
@@ -360,7 +360,7 @@
   buildable: False
 
   hs-source-dirs: src/parsertest
-  default-language: Haskell2010
+  default-language: Haskell98
 
   if !flag(hacking) || flag(dochack)
     buildable: False
@@ -378,7 +378,7 @@
 
 executable yi
   hs-source-dirs: src/executable
-  default-language: Haskell2010
+  default-language: Haskell98
 
   if flag(dochack)
     buildable: False
--- yi-0.6.6.0-orig/src/library/Yi/UI/Pango.hs	2012-10-13 18:08:47.021706305 +1100
+++ yi-0.6.6.0/src/library/Yi/UI/Pango.hs	2012-10-13 18:45:00.447156819 +1100
@@ -478,9 +478,9 @@
                                 Gtk.lineWidth = if wideCursor then 2 else 1 })
   -- tell the renderer
   if im 
-  then -- if we are inserting, we just want a line
+   then -- if we are inserting, we just want a line
     drawLine drawWindow gc (round curX, round curY) (round $ curX + curW, round $ curY + curH)
-  else do -- if we aren't inserting, we want a rectangle around the current character
+   else do -- if we aren't inserting, we want a rectangle around the current character
     PangoRectangle (succ -> chx) chy chw chh <- layoutIndexToPos layout (rel cur)
     drawRectangle drawWindow gc False (round chx) (round chy) (if chw > 0 then round chw else 8) (round chh)
 
--- yi-0.6.6.0-orig/src/library/Yi/Buffer/HighLevel.hs	2012-07-21 16:15:17.000000000 +1000
+++ yi-0.6.6.0/src/library/Yi/Buffer/HighLevel.hs	2012-10-13 18:45:00.447156819 +1100
@@ -627,8 +627,8 @@
 revertB s now = do
     r <- regionOfB Document
     if R.length s <= smallBufferSize -- for large buffers, we must avoid building strings, because we'll end up using huge amounts of memory
-    then replaceRegionClever r (R.toString s)
-    else replaceRegionB' r s
+     then replaceRegionClever r (R.toString s)
+     else replaceRegionB' r s
     markSavedB now
 
 smallBufferSize :: Int
--- yi-0.6.6.0-orig/src/library/Yi/UI/Pango/Layouts.hs	2012-07-21 16:15:17.000000000 +1000
+++ yi-0.6.6.0/src/library/Yi/UI/Pango/Layouts.hs	2012-10-13 18:45:00.448156844 +1100
@@ -28,7 +28,7 @@
  ) where
 
 
-import Control.Monad(void)
+import Foreign.Marshal.Error(void)
 import qualified Data.List.PointedList as PL
 import Data.Maybe
 import qualified Prelude
