diff --git a/src/library/Yi/Config/Default.hs b/src/library/Yi/Config/Default.hs
index 4374a82..f3f1d16 100644
--- a/src/library/Yi/Config/Default.hs
+++ b/src/library/Yi/Config/Default.hs
@@ -138,4 +138,8 @@ defaultConfig =
 #ifdef FRONTEND_VTY
+#if MIN_VERSION_vty(5,15,0)
+           , configVty = mempty
+#else
            , configVty = def
 #endif
+#endif
            }
@@ -195,4 +199,8 @@ toEmacsStyleConfig cfg
                        -- ESC is terrible for me. Maybe that's just how it is under emacs...
+#if MIN_VERSION_vty(5,15,0)
+                       , configVty = mempty { Vty.vtime = Just 100, Vty.vmin = Just 2 }
+#else
                        , configVty = def { Vty.vtime = Just 100, Vty.vmin = Just 2 }
 #endif
+#endif
                        },
