diff --git a/Network/Protocol/HTTP/DAV.hs b/Network/Protocol/HTTP/DAV.hs
index 02e5d15..07c9c72 100644
--- a/Network/Protocol/HTTP/DAV.hs
+++ b/Network/Protocol/HTTP/DAV.hs
@@ -16,7 +16,7 @@
 -- You should have received a copy of the GNU General Public License
 -- along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-{-# LANGUAGE OverloadedStrings, ConstraintKinds, FlexibleContexts, QuasiQuotes #-}
+{-# LANGUAGE OverloadedStrings, ConstraintKinds, FlexibleContexts, QuasiQuotes, CPP #-}
 
 module Network.Protocol.HTTP.DAV (
     DAVState
@@ -83,7 +83,11 @@ davRequest meth addlhdrs rbody = do
     return resp
 
 matchStatusCodeException :: Status -> HttpException -> Maybe ()
+#ifdef MIN_VERSION_http_conduit(1,9,0)
+matchStatusCodeException want (StatusCodeException s _ _)
+#else
 matchStatusCodeException want (StatusCodeException s _)
+#endif
     | s == want = Just ()
     | otherwise = Nothing
 matchStatusCodeException _ _ = Nothing
