--- leksah-server-0.13.1.1-orig/src/IDE/Utils/GHCUtils.hs	2014-06-15 05:00:28.000000000 +1000
+++ leksah-server-0.13.1.1/src/IDE/Utils/GHCUtils.hs	2014-06-26 22:18:31.592278128 +1000
@@ -56,7 +56,7 @@
 import System.Log.Logger(debugM)
 import Control.Monad.IO.Class (MonadIO(..))
 
-#if !MIN_VERSION_ghc(7,7,0)
+#if !MIN_VERSION_ghc(7,7,0) || MIN_VERSION_transformers(0,4,0)
 -- this should not be repeated here, why is it necessary?
 instance MonadIO Ghc where
   liftIO ioA = Ghc $ \_ -> ioA
