--- leksah-server-9999-orig/leksah-server.cabal	2012-02-04 11:58:01.108060180 +1100
+++ leksah-server-9999/leksah-server.cabal	2012-02-04 13:35:50.472002658 +1100
@@ -32,25 +32,28 @@
 
 library
     default-language: Haskell98
-    build-depends: Cabal >=1.6.0.1 && <1.14, base >= 4.0.0.0 && <4.5, binary >=0.5.0.0 && <0.6,
+    build-depends: Cabal >=1.6.0.1 && <1.15, base >= 4.0.0.0 && <4.6, binary >=0.5.0.0 && <0.6,
                binary-shared >=0.8 && <0.9, bytestring >=0.9.0.1 && <0.10,
                containers >=0.2.0.0 && <0.5,
-               directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.3, ghc >=6.10.1 && <7.4,
+               directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.4, ghc >=6.10.1 && <7.5,
                ltk >=0.12 && <0.13, parsec >=2.1.0.1 && <3.2,
-               pretty >=1.0.1.0 && <1.2, time >=1.1 && <1.5, deepseq >=1.1 && <1.3,
+               pretty >=1.0.1.0 && <1.2, time >=1.1 && <1.5, deepseq >=1.1 && <1.4,
                hslogger >= 1.0.7 && <1.2, network >=2.2 && <3.0, enumerator >=0.4.14 && < 0.5,
                attoparsec-enumerator >=0.3 && <0.4, attoparsec >=0.10.0.3 && <0.11,
                transformers >=0.2.2.0 && <0.3
-    if (impl(ghc >= 7.2))
-       build-depends: haddock >= 2.7.2 && <2.10
+    if (impl(ghc >= 7.4))
+       build-depends: haddock >= 2.7.2 && <2.11
     else
-       if (impl(ghc >= 7.0))
-          build-depends: haddock >= 2.7.2 && <2.9.3
+       if (impl(ghc >= 7.2))
+          build-depends: haddock >= 2.7.2 && <2.10
        else
-          if (impl(ghc >= 6.12))
-             build-depends: haddock >= 2.7.2 && <2.9
+          if (impl(ghc >= 7.0))
+             build-depends: haddock >= 2.7.2 && <2.9.3
           else
-             build-depends: haddock-leksah == 2.5.0
+             if (impl(ghc >= 6.12))
+                build-depends: haddock >= 2.7.2 && <2.9
+             else
+                build-depends: haddock-leksah == 2.5.0
 
     if (impl(ghc >= 7.2))
        build-depends: process >= 1.1 && <1.2
@@ -89,25 +92,28 @@
 
 executable leksah-server
     default-language: Haskell98
-    build-depends: Cabal >=1.6.0.1 && <1.14, base >= 4.0.0.0 && <4.5, binary >=0.5.0.0 && <0.6,
+    build-depends: Cabal >=1.6.0.1 && <1.15, base >= 4.0.0.0 && <4.6, binary >=0.5.0.0 && <0.6,
                binary-shared >=0.8 && <0.9, bytestring >=0.9.0.1 && <0.10,
                containers >=0.2.0.0 && <0.5,
-               directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.3, ghc >=6.10.1 && <7.4,
+               directory >=1.0.0.2 && <1.2, filepath >=1.1.0.1 && <1.6, ghc >=6.10.1 && <7.5,
                ltk >=0.12 && <0.13, parsec >=2.1.0.1 && <3.2,
-               pretty >=1.0.1.0 && <1.2, time >=1.1 && <1.5, deepseq >=1.1 && <1.3,
+               pretty >=1.0.1.0 && <1.2, time >=1.1 && <1.5, deepseq >=1.1 && <1.4,
                hslogger >= 1.0.7 && <1.2, network >=2.2 && <3.0, enumerator >= 0.4.14 && <0.5,
                attoparsec-enumerator >=0.3 && <0.4, attoparsec >=0.10.0.3 && <0.11,
                transformers >=0.2.2.0 && <0.3
-    if (impl(ghc >= 7.2))
-       build-depends: haddock >= 2.7.2 && <2.10
+    if (impl(ghc >= 7.4))
+       build-depends: haddock >= 2.7.2 && <2.11
     else
-       if (impl(ghc >= 7.0))
-          build-depends: haddock >= 2.7.2 && <2.9.3
+       if (impl(ghc >= 7.2))
+          build-depends: haddock >= 2.7.2 && <2.10
        else
-          if (impl(ghc >= 6.12))
-             build-depends: haddock >= 2.7.2 && <2.9
+          if (impl(ghc >= 7.0))
+             build-depends: haddock >= 2.7.2 && <2.9.3
           else
-             build-depends: haddock-leksah == 2.5.0
+             if (impl(ghc >= 6.12))
+                build-depends: haddock >= 2.7.2 && <2.9
+             else
+                build-depends: haddock-leksah == 2.5.0
 
     if (impl(ghc >= 7.2))
        build-depends: process >= 1.1 && <1.2
@@ -157,7 +163,7 @@
     hs-source-dirs: src
     ghc-prof-options: -auto-all -prof
 --    ghc-shared-options: -auto-all -prof
-    build-depends:  base >= 4.0.0.0 && <4.5, hslogger >= 1.0.7 && <1.2, deepseq >=1.1 && <1.3,
+    build-depends:  base >= 4.0.0.0 && <4.6, hslogger >= 1.0.7 && <1.2, deepseq >=1.1 && <1.4,
                bytestring >=0.9.0.1 && <0.10, enumerator >= 0.4.14 && <0.5,
                attoparsec-enumerator >=0.3 && <0.4, attoparsec >=0.10.0.3 && <0.11,
                transformers >=0.2.2.0 && <0.3
@@ -180,7 +186,7 @@
     type:       exitcode-stdio-1.0
     hs-source-dirs: tests
     main-is:    TestTool.hs
-    build-depends: base >= 4.0.0.0 && <4.5,  hslogger >= 1.0.7 && <1.3,
+    build-depends: base >= 4.0.0.0 && <4.6,  hslogger >= 1.0.7 && <1.3,
                leksah-server,
                HUnit >=1.2 && <1.3, transformers >=0.2.2.0 && <0.3, enumerator >=0.4.14 && <0.5
 
--- leksah-server-9999-orig/src/IDE/HeaderParser.hs	2012-02-04 11:56:20.139495609 +1100
+++ leksah-server-9999/src/IDE/HeaderParser.hs	2012-02-04 11:44:13.509280940 +1100
@@ -24,7 +24,11 @@
 import RdrName(showRdrName)
 import IDE.Utils.GHCUtils
 import Data.Maybe (mapMaybe)
+#if MIN_VERSION_ghc(7,4,1)
+import Outputable(pprPrefixOcc,showSDoc)
+#else
 import Outputable(pprHsVar,showSDoc)
+#endif
 import IDE.Utils.FileUtils (figureOutHaddockOpts, figureOutGhcOpts)
 import Control.Monad.IO.Class (MonadIO(..))
 
@@ -72,7 +76,11 @@
                         Just (hide, list) -> Just (ImportSpecList hide (mapMaybe transformEntity list))
 
 transformEntity :: LIE RdrName -> Maybe ImportSpec
+#if MIN_VERSION_ghc(7,2,0)
+transformEntity (L _ (IEVar name))              = Just (IVar (showSDoc (pprPrefixOcc name)))
+#else
 transformEntity (L _ (IEVar name))              = Just (IVar (showSDoc (pprHsVar name)))
+#endif
 transformEntity (L _ (IEThingAbs name))         = Just (IAbs (showRdrName name))
 transformEntity (L _ (IEThingAll name))         = Just (IThingAll (showRdrName name))	
 transformEntity (L _ (IEThingWith name list))   = Just (IThingWith (showRdrName name)
--- leksah-server-9999-orig/src/IDE/Metainfo/SourceCollectorH.hs	2012-02-04 11:56:48.271209526 +1100
+++ leksah-server-9999/src/IDE/Metainfo/SourceCollectorH.hs	2012-02-04 11:31:30.721948220 +1100
@@ -185,6 +185,9 @@
                         (ifaceInstances interface) [] --(ifaceLocals interface)
         imports    = Map.empty --TODO
 
+#if MIN_VERSION_ghc(7,4,1)
+type DeclInfo = [LHsDecl Name]
+#endif
 #if MIN_VERSION_ghc(6,12,1)
 extractDescrs :: PackModule -> Map Name DeclInfo -> [ExportItem Name] -> [Instance] -> [Name] -> [Descr]
 extractDescrs pm _ifaceDeclMap ifaceExportItems' ifaceInstances' _ifaceLocals =
