diff --git a/AttoBencode.cabal b/AttoBencode.cabal
index 33f3102..2b7e354 100644
--- a/AttoBencode.cabal
+++ b/AttoBencode.cabal
@@ -45,7 +45,7 @@ Test-suite tests
                   containers,
                   test-framework,
                   test-framework-quickcheck2,
-                  QuickCheck >= 2.4 && < 2.5
+                  QuickCheck >= 2.4 && < 2.6
 
 source-repository head
   Type: mercurial
