--- a/App/Behaviours/Exception.hs
+++ b/App/Behaviours/Exception.hs
@@ -13,2 +13,3 @@ import Data.Time.Locale.Compat as DTLC
 import System.Locale
+import Prelude hiding ((<>))
 
--- a/App/Behaviours/PrintEvents.hs
+++ b/App/Behaviours/PrintEvents.hs
@@ -9,2 +9,3 @@ import App.EventBus
 import System.IO
+import Prelude hiding ((<>))
 
--- a/App/DebugEventBus.hs
+++ b/App/DebugEventBus.hs
@@ -34,2 +34,3 @@ import System.IO
 import Debug.Trace
+import qualified Data.Semigroup as S
 
@@ -215,2 +216,5 @@ topEvent = head . Set.toList
 
+instance S.Semigroup (Bus a) where
+    (<>) (Bus n0 s0 g0 f0 cpc h) (Bus n1 s1 g1 f1 _ _) = Bus (Map.union n0 n1) (Map.union s0 s1) (Map.union g0 g1) (Map.union f0 f1) cpc h
+
 instance Monoid (Bus a) where
@@ -254,2 +258,5 @@ type Behaviour a = Bus a -> Future a
 
+instance S.Semigroup (Behaviour a) where
+    (<>) = (>~>) -- x behind y
+
 instance Monoid (Behaviour a) where
--- a/App/EventBus.hs
+++ b/App/EventBus.hs
@@ -29,2 +29,3 @@ import System.IO.Unsafe
 import Debug.Trace
+import qualified Data.Semigroup as S
 
@@ -203,2 +204,5 @@ topEvent = head . Set.toList
 
+instance S.Semigroup (Bus a) where
+    (<>) (Bus n0 s0 g0 f0) (Bus n1 s1 g1 f1) = Bus (Map.union n0 n1) (Map.union s0 s1) (Map.union g0 g1) (Map.union f0 f1)
+
 instance Monoid (Bus a) where
@@ -242,2 +246,5 @@ type Behaviour a = Bus a -> Future a
 
+instance S.Semigroup (Behaviour a) where
+    (<>) = (>~>) -- x behind y
+
 instance Monoid (Behaviour a) where
