diff --git a/App/DebugEventBus.hs b/App/DebugEventBus.hs
index 9c3dc3a..430560f 100644
--- a/App/DebugEventBus.hs
+++ b/App/DebugEventBus.hs
@@ -1,2 +1,2 @@
-{-# LANGUAGE TypeSynonymInstances, ScopedTypeVariables #-}
+{-# LANGUAGE TypeSynonymInstances, ScopedTypeVariables, OverlappingInstances #-}
 {-# OPTIONS_GHC -O0 #-}
diff --git a/App/EventBus.hs b/App/EventBus.hs
index 1f3ccb0..5a06987 100644
--- a/App/EventBus.hs
+++ b/App/EventBus.hs
@@ -1,2 +1,2 @@
-{-# LANGUAGE TypeSynonymInstances, ScopedTypeVariables #-}
+{-# LANGUAGE TypeSynonymInstances, ScopedTypeVariables, OverlappingInstances #-}
 -- | Not exactly the FRP model, but rather a model of a large application with
