diff --git a/App/Behaviours/Exception.hs b/App/Behaviours/Exception.hs
index ddbca97..123c082 100644
--- a/App/Behaviours/Exception.hs
+++ b/App/Behaviours/Exception.hs
@@ -11,2 +11,3 @@ import System.IO
 import Data.Time
+import Data.Time.Locale.Compat as DTLC
 import System.Locale
@@ -14,3 +15,3 @@ import System.Locale
 renderException (Event nm _ _ edata source tm) =
-            brackets (text (formatTime defaultTimeLocale "%T" tm)) <+> text "Exception thrown from" <+> text source <> colon <+> text nm $+$
+            brackets (text (formatTime DTLC.defaultTimeLocale "%T" tm)) <+> text "Exception thrown from" <+> text source <> colon <+> text nm $+$
             (nest 4 . vcat . map text . map (safeShow (Just 80)) $ edata)
diff --git a/App/Behaviours/PrintEvents.hs b/App/Behaviours/PrintEvents.hs
index 1864f68..68cbdbf 100644
--- a/App/Behaviours/PrintEvents.hs
+++ b/App/Behaviours/PrintEvents.hs
@@ -5,2 +5,3 @@ import Text.PrettyPrint
 import Data.Time
+import Data.Time.Locale.Compat as DTLC
 import System.Locale
@@ -15,3 +16,3 @@ printEventsBehaviour b = pollAllEventsWith b $ (\(Event n g lifetime edata sourc
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -42,3 +43,3 @@ printEventGroupBehaviour grp b = pollEventGroupWith b grp $ (\(Event n g lifetim
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -53,3 +54,3 @@ printEventNameBehaviour grp b = pollNamedEventsWith b grp $ (\(Event n g lifetim
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -64,3 +65,3 @@ printEventSourceBehaviour grp b = pollEventsFromSourceWith b grp $ (\(Event n g
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -84,3 +85,3 @@ logEventsBehaviour handle  b = pollAllEventsWith b $ (\(Event n g lifetime edata
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -94,3 +95,3 @@ logEventGroupBehaviour handle grp b = pollEventGroupWith b grp $ (\(Event n g li
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -104,3 +105,3 @@ logEventNameBehaviour handle grp b = pollNamedEventsWith b grp $ (\(Event n g li
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
@@ -114,3 +115,3 @@ logEventSourceBehaviour handle grp b = pollEventsFromSourceWith b grp $ (\(Event
 	                  text "ttl:     " <+> text (show lifetime) $+$
-	                  text "emitTime:" <+> text (formatTime defaultTimeLocale "%T" t) $+$
+	                  text "emitTime:" <+> text (formatTime DTLC.defaultTimeLocale "%T" t) $+$
 	                  (vcat.map showIfPossible $ edata)
diff --git a/App/DebugEventBus.hs b/App/DebugEventBus.hs
index 63030fd..9c3dc3a 100644
--- a/App/DebugEventBus.hs
+++ b/App/DebugEventBus.hs
@@ -26,2 +26,3 @@ import Data.Time.Clock
 import Data.Time.Format
+import Data.Time.Locale.Compat as DTLC
 import System.Locale
@@ -284,3 +285,3 @@ applyDiff (m,ds) = do
     ds' <- immediate ds 
-    now <- formatTime defaultTimeLocale "%H:%M:%S%Q" <$> getCurrentTime
+    now <- formatTime DTLC.defaultTimeLocale "%H:%M:%S%Q" <$> getCurrentTime
     (foldM (busDiff now) m $ ds') >>= \b -> return b{ currentProducerConsumer = Nothing }
diff --git a/buster.cabal b/buster.cabal
index d1a720f..95d77c4 100644
--- a/buster.cabal
+++ b/buster.cabal
@@ -10,3 +10,3 @@ build-depends: base >= 4, binary -any, bytestring -any,
                containers -any, dataenc -any, mtl -any, old-locale -any,
-               parsec >=3.0.0, pretty -any, time -any
+               parsec >=3.0.0, pretty -any, time -any, time-locale-compat
 stability: Experimental
