diff -urN mfsolve-0.3.2.1/tests/test.hs mfsolve-0.3.2.1-r1/tests/test.hs
--- mfsolve-0.3.2.1/tests/test.hs	2001-09-08 19:46:40.000000000 -0600
+++ mfsolve-0.3.2.1-r1/tests/test.hs	2022-03-22 20:47:30.167140245 -0600
@@ -49,9 +49,6 @@
 zero (toSimple -> Const 0) = True
 zero _ = False
 
-instance (Floating n, Eq n, Ord n, Ord v) => Eq (Expr v n) where
-  a == b = zero $ a-b
-
 TestExpr a b ?= TestExpr c d =
   (\s -> testEq (a s) (c s) (show b) (show d),
    (b,d))
