--- a/src/StatusNotifier/Util.hs
+++ b/src/StatusNotifier/Util.hs
@@ -23,2 +23,4 @@ import           System.Log.Handler.Simple
 import           System.Log.Logger
+import qualified Data.Text.IO as T
+import qualified Data.Text as T
 
@@ -26,3 +28,3 @@ getIntrospectionObjectFromFile :: FilePath -> T.ObjectPath -> Q I.Object
 getIntrospectionObjectFromFile filepath nodePath = runIO $
-  head . maybeToList . I.parseXML nodePath <$> readFile filepath
+  head . maybeToList . I.parseXML nodePath <$> T.readFile filepath
 
@@ -124,3 +126,3 @@ getInterfaceAt
 getInterfaceAt client bus path =
-  right (I.parseXML "/") <$> introspect client bus path
+  right (I.parseXML "/") . fmap T.pack <$> introspect client bus path
 
--- a/status-notifier-item.cabal
+++ b/status-notifier-item.cabal
@@ -55,2 +55,3 @@ library
     , template-haskell
+    , text
     , transformers
