--- IOSpec-0.2.1-orig/IOSpec.cabal	2009-10-27 00:51:07.000000000 +1100
+++ IOSpec-0.2.1/IOSpec.cabal	2011-01-04 21:29:36.500510089 +1100
@@ -42,7 +42,7 @@
 			with the source distribution.
 Category:       	Testing
 Build-Type:		Simple
-Build-Depends:  	base >= 2 && < 4, mtl, QuickCheck >= 2 && < 3, Stream
+Build-Depends:  	base >= 2 && < 4.4, mtl, QuickCheck >= 2 && < 3, Stream
 Extensions:		MultiParamTypeClasses, OverlappingInstances
 Ghc-options:		-Wall
 Hs-source-dirs:		src
--- IOSpec-0.2.1-orig/src/Test/IOSpec/Surrogate.hs	2009-10-27 00:51:07.000000000 +1100
+++ IOSpec-0.2.1/src/Test/IOSpec/Surrogate.hs	2011-01-04 13:56:49.339636944 +1100
@@ -1,4 +1,5 @@
 {-# LANGUAGE EmptyDataDecls #-}
+{-# LANGUAGE TypeOperators #-}
 -- | This module contains a few type signatures to help replace pure
 -- specifications by their effectful counterparts.
 module Test.IOSpec.Surrogate
