diff --git a/IOSpec.cabal b/IOSpec.cabal
index edcef49..9593d30 100644
--- a/IOSpec.cabal
+++ b/IOSpec.cabal
@@ -42,5 +42,5 @@ Description:		This package consists of several modules, that give a
 Category:       	Testing
 Build-Type:		Simple
-Build-Depends:  	base >= 4.8.0.0 && < 5, mtl, QuickCheck >= 2 && < 3, Stream
+Build-Depends:  	base >= 4.7.0.0 && < 5, mtl, QuickCheck >= 2 && < 3, Stream
 Extensions:		MultiParamTypeClasses, OverlappingInstances
 Ghc-options:		-Wall
diff --git a/src/Test/IOSpec/STM.hs b/src/Test/IOSpec/STM.hs
index 357c646..21b5929 100644
--- a/src/Test/IOSpec/STM.hs
+++ b/src/Test/IOSpec/STM.hs
@@ -25,4 +25,5 @@ import Data.Maybe (fromJust)
 import Control.Monad.State
 import Control.Monad (ap)
+import Control.Applicative
 
 -- The 'STMS' data type and its instances.
diff --git a/src/Test/IOSpec/Types.hs b/src/Test/IOSpec/Types.hs
index 8b85fe8..4514dc7 100644
--- a/src/Test/IOSpec/Types.hs
+++ b/src/Test/IOSpec/Types.hs
@@ -17,4 +17,5 @@ module Test.IOSpec.Types
 
 import Control.Monad (ap)
+import Control.Applicative
 
 -- | A value of type 'IOSpec' @f@ @a@ is either a pure value of type @a@
diff --git a/src/Test/IOSpec/VirtualMachine.hs b/src/Test/IOSpec/VirtualMachine.hs
index 093db46..99dc97c 100644
--- a/src/Test/IOSpec/VirtualMachine.hs
+++ b/src/Test/IOSpec/VirtualMachine.hs
@@ -45,4 +45,5 @@ import Test.IOSpec.Types
 import Test.QuickCheck
 import Control.Monad (ap)
+import Control.Applicative
 
 type Data         = Dynamic
