diff --git a/IOSpec.cabal b/IOSpec.cabal
index c923b55..0e05fd2 100644
--- a/IOSpec.cabal
+++ b/IOSpec.cabal
@@ -42,7 +42,7 @@ Description:		This package consists of several modules, that give a
 			with the source distribution.
 Category:       	Testing
 Build-Type:		Simple
-Build-Depends:  	base >= 2 && < 4.4, mtl, QuickCheck >= 2 && < 3, Stream
+Build-Depends:  	base >= 2 && < 4.5, mtl, QuickCheck >= 2 && < 3, Stream
 Extensions:		MultiParamTypeClasses, OverlappingInstances
 Ghc-options:		-Wall
 Hs-source-dirs:		src
