--- IOSpec-0.2.3-orig/IOSpec.cabal	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/IOSpec.cabal	2012-01-14 23:04:55.659635020 +1100
@@ -42,7 +42,7 @@
 			with the source distribution.
 Category:       	Testing
 Build-Type:		Simple
-Build-Depends:  	base >= 2 && < 4.5, mtl, QuickCheck >= 2 && < 3, Stream
+Build-Depends:  	base >= 2 && < 4.6, mtl, QuickCheck >= 2 && < 3, Stream
 Extensions:		MultiParamTypeClasses, OverlappingInstances
 Ghc-options:		-Wall
 Hs-source-dirs:		src
--- IOSpec-0.2.3-orig/src/Test/IOSpec/Fork.hs	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/src/Test/IOSpec/Fork.hs	2012-01-14 23:13:22.214249307 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE ExistentialQuantification, FlexibleContexts #-}
+{-# LANGUAGE ExistentialQuantification, FlexibleContexts, TypeOperators #-}
 -- | A pure specification of 'forkIO'.
 module Test.IOSpec.Fork
    (
--- IOSpec-0.2.3-orig/src/Test/IOSpec/IORef.hs	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/src/Test/IOSpec/IORef.hs	2012-01-14 23:10:27.482892279 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleContexts, TypeOperators #-}
 -- | A pure specification of mutable variables.
 module Test.IOSpec.IORef
    (
--- IOSpec-0.2.3-orig/src/Test/IOSpec/MVar.hs	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/src/Test/IOSpec/MVar.hs	2012-01-14 23:12:37.539134744 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveDataTypeable, FlexibleContexts #-}
+{-# LANGUAGE DeriveDataTypeable, FlexibleContexts, TypeOperators #-}
 -- | A pure specification of basic operations on MVars.
 
 module Test.IOSpec.MVar
--- IOSpec-0.2.3-orig/src/Test/IOSpec/STM.hs	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/src/Test/IOSpec/STM.hs	2012-01-14 23:09:05.311845381 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE ExistentialQuantification, FlexibleContexts #-}
+{-# LANGUAGE ExistentialQuantification, FlexibleContexts, TypeOperators #-}
 
 module Test.IOSpec.STM
    (
--- IOSpec-0.2.3-orig/src/Test/IOSpec/Teletype.hs	2012-01-13 22:14:30.000000000 +1100
+++ IOSpec-0.2.3/src/Test/IOSpec/Teletype.hs	2012-01-14 23:06:15.085609403 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleContexts, TypeOperators #-}
 -- | A pure specification of getChar and putChar.
 module Test.IOSpec.Teletype
    (
