--- http-enumerator-0.6.0.2-orig/http-enumerator.cabal	2011-04-12 20:11:46.000000000 +1000
+++ http-enumerator-0.6.0.2/http-enumerator.cabal	2011-04-20 21:15:26.104008419 +1000
@@ -22,6 +22,7 @@
 library
     build-depends: base                  >= 4       && < 5
                  , bytestring            >= 0.9.1.4 && < 0.10
+                 , cprng-aes             >= 0.2     && < 0.3
                  , transformers          >= 0.2     && < 0.3
                  , failure               >= 0.1     && < 0.2
                  , enumerator            >= 0.4.7   && < 0.5
@@ -32,8 +33,8 @@
                  , zlib-enum             >= 0.2     && < 0.3
                  , http-types            >= 0.6     && < 0.7
                  , blaze-builder-enumerator >= 0.2  && < 0.3
-                 , tls                   >= 0.5.1   && < 0.6
-                 , tls-extra             >= 0.1.6   && < 0.2
+                 , tls                   >= 0.6     && < 0.7
+                 , tls-extra             >= 0.2     && < 0.3
                  , monad-control         >= 0.2     && < 0.3
                  , containers            >= 0.2     && < 0.5
                  , certificate           >= 0.7     && < 0.8
--- http-enumerator-0.6.0.2-orig/Network/TLS/Client/Enumerator.hs	2011-04-12 20:11:46.000000000 +1000
+++ http-enumerator-0.6.0.2/Network/TLS/Client/Enumerator.hs	2011-04-20 21:47:23.477869502 +1000
@@ -21,6 +21,7 @@
     )
 import Data.Certificate.X509 (X509)
 import Network.TLS.Extra (ciphersuite_all)
+import Crypto.Random.AESCtr (makeSystem)
 
 data ConnInfo = ConnInfo
     { connRead :: IO [ByteString]
@@ -65,8 +66,7 @@
             , pCiphers = ciphersuite_all
             , onCertificatesRecv = onCerts
             }
-    esrand <- liftIO makeSRandomGen
-    let srg = either (error . show) id esrand
+    srg <- liftIO makeSystem
     istate <- liftIO $ client tcp srg h
     liftIO $ handshake istate
     return ConnInfo
