--- comonad-extras-3.0.2-orig/src/Control/Comonad/Store/Pointer.hs	2013-10-14 00:46:49.000000000 +1100
+++ comonad-extras-3.0.2/src/Control/Comonad/Store/Pointer.hs	2013-10-18 19:50:57.373912731 +1100
@@ -103,7 +103,7 @@
   lower (PointerT g i) = fmap (! i) g
 
 instance Ix i => ComonadHoist (PointerT i) where
-  cohoist (PointerT g i) = PointerT (Identity (extract g)) i
+  cohoist l (PointerT g i) = PointerT (l g) i
 
 instance (Comonad w, Ix i) => ComonadStore i (PointerT i w) where
   pos (PointerT _ i) = i
