diff -urN fay-0.24.2.0/src/Fay/Convert.hs fay-0.24.2.0-r1/src/Fay/Convert.hs
--- fay-0.24.2.0/src/Fay/Convert.hs	2001-09-08 19:46:40.000000000 -0600
+++ fay-0.24.2.0-r1/src/Fay/Convert.hs	2022-03-04 09:01:04.347715420 -0700
@@ -38,7 +38,7 @@
 --   values aren't handled by explicit cases.  'encodeFay' can be used to
 --   resolve this issue.
 showToFay :: Data a => a -> Maybe Value
-showToFay = spoon . encodeFay id
+showToFay = spoon . encodeFay (\gq -> gq)
 
 -- | Convert a Haskell value to a Fay json value.  This can fail when primitive
 --   values aren't handled by explicit cases.  When this happens, you can add
