--- poppler-0.12.2-orig/poppler.cabal	2011-05-07 19:48:56.000000000 +1000
+++ poppler-0.12.2/poppler.cabal	2011-09-17 15:42:06.544985839 +1000
@@ -37,7 +37,7 @@
   location:     http://patch-tag.com/r/AndyStewart/poppler
 
 Library
-        build-depends:  base >= 4 && < 5, array, containers, haskell98, mtl, bytestring,
+        build-depends:  base >= 4 && < 5, array, containers, mtl, bytestring,
                         glib >= 0.12 && < 0.13,
                         cairo >= 0.12 && < 0.13,
                         gtk >= 0.12 && < 0.13
