--- HDBC-postgresql-2.3.2.0-orig/Database/HDBC/PostgreSQL/Parser.hs	2011-08-10 07:20:56.000000000 +1000
+++ HDBC-postgresql-2.3.2.0/Database/HDBC/PostgreSQL/Parser.hs	2012-02-09 18:46:07.108176997 +1100
@@ -42,13 +42,13 @@
 
 -- FIXME: handle pgsql dollar-quoted constants
 
-qmark :: Num st => GenParser Char st [Char]
+qmark :: (Eq st, Num st, Show st) => GenParser Char st [Char]
 qmark = do char '?'
            n <- getState
            updateState (+1)
            return $ "$" ++ show n
 
-statement :: Num st => GenParser Char st [Char]
+statement :: (Eq st, Num st, Show st) => GenParser Char st [Char]
 statement = 
     do s <- many ((try qmark) <|>
                   (try comment) <|>
