diff --git a/src/Elm/Docs.hs b/src/Elm/Docs.hs
index 10f9b2e..9b6b4f1 100644
--- a/src/Elm/Docs.hs
+++ b/src/Elm/Docs.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE OverloadedStrings, DeriveDataTypeable, CPP #-}
 module Elm.Docs where
 
 import Control.Applicative ((<|>))
@@ -98,8 +98,12 @@ prettyJson value =
 
 config :: Json.Config
 config =
-    Json.Config
+    Json.defConfig
+#if MIN_VERSION_aeson_pretty(0,8,1)
+    { Json.confIndent = Json.Spaces 2
+#else
     { Json.confIndent = 2
+#endif
     , Json.confCompare = Json.keyOrder keys
     }
   where
