diff --git a/src/Docs.hs b/src/Docs.hs
index 7a99e1d..683dfa3 100644
--- a/src/Docs.hs
+++ b/src/Docs.hs
@@ -126,3 +126,3 @@ docComment =
       let reversed =
-              dropWhile (`elem` " \n\r") . drop 2 $ reverse contents
+              dropWhile (`elem` (" \n\r" :: String)) . drop 2 $ reverse contents
 
diff --git a/src/Parse/Helpers.hs b/src/Parse/Helpers.hs
index 6c9652c..3c74196 100644
--- a/src/Parse/Helpers.hs
+++ b/src/Parse/Helpers.hs
@@ -1 +1,2 @@
+{-# LANGUAGE FlexibleContexts #-}
 module Parse.Helpers where
diff --git a/src/Transform/Canonicalize.hs b/src/Transform/Canonicalize.hs
index e896506..4104e32 100644
--- a/src/Transform/Canonicalize.hs
+++ b/src/Transform/Canonicalize.hs
@@ -1 +1,2 @@
+{-# LANGUAGE FlexibleContexts #-}
 module Transform.Canonicalize (module') where
diff --git a/src/Transform/Canonicalize/Setup.hs b/src/Transform/Canonicalize/Setup.hs
index 5552dfe..55d1577 100644
--- a/src/Transform/Canonicalize/Setup.hs
+++ b/src/Transform/Canonicalize/Setup.hs
@@ -1 +1,2 @@
+{-# LANGUAGE FlexibleContexts #-}
 {-# OPTIONS_GHC -Wall #-}
diff --git a/src/Transform/Canonicalize/Variable.hs b/src/Transform/Canonicalize/Variable.hs
index c239b4d..4ca8ec6 100644
--- a/src/Transform/Canonicalize/Variable.hs
+++ b/src/Transform/Canonicalize/Variable.hs
@@ -1 +1,2 @@
+{-# LANGUAGE FlexibleContexts #-}
 {-# OPTIONS_GHC -Wall #-}
