diff --git a/src/Docs.hs b/src/Docs.hs
index 683dfa3..670c62c 100644
--- a/src/Docs.hs
+++ b/src/Docs.hs
@@ -1,2 +1,2 @@
-{-# LANGUAGE OverloadedStrings, DeriveDataTypeable #-}
+{-# LANGUAGE OverloadedStrings, DeriveDataTypeable, CPP #-}
 module Main where
@@ -84,4 +84,8 @@ config :: Json.Config
 config =
-    Json.Config
+    Json.defConfig
+#if MIN_VERSION_aeson_pretty(0,8,1)
+    { Json.confIndent = Json.Spaces 2
+#else
     { Json.confIndent = 2
+#endif
     , Json.confCompare = Json.keyOrder keys
