--- conduit-0.5.2.3-orig/Data/Conduit/Binary.hs	2012-08-03 22:42:14.000000000 +1000
+++ conduit-0.5.2.3/Data/Conduit/Binary.hs	2012-08-10 21:16:42.801058391 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE CPP, Rank2Types #-}
 -- | Functions for interacting with bytes.
 module Data.Conduit.Binary
     ( -- * Files and @Handle@s
--- conduit-0.5.2.3-orig/Data/Conduit/Text.hs	2012-08-03 22:42:14.000000000 +1000
+++ conduit-0.5.2.3/Data/Conduit/Text.hs	2012-08-10 21:16:55.967482072 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveDataTypeable, Rank2Types #-}
 -- |
 -- Copyright: 2011 Michael Snoyman, 2010-2011 John Millikin
 -- License: MIT
