commit c005f5322ec5324e53b4fe9558e7670ed9e792bc
Author: RyanGlScott <ryan.gl.scott@gmail.com>
Date:   Mon Feb 8 09:28:09 2016 -0500

    Fix conduit and resourcet on GHC 8.0

diff --git a/conduit/Data/Conduit/Internal/Pipe.hs b/conduit/Data/Conduit/Internal/Pipe.hs
index 8c5a4cb..aa9de68 100644
--- a/conduit/Data/Conduit/Internal/Pipe.hs
+++ b/conduit/Data/Conduit/Internal/Pipe.hs
@@ -6,7 +6,6 @@
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE TupleSections #-}
-{-# LANGUAGE ImpredicativeTypes #-}
 {-# LANGUAGE Trustworthy #-}
 module Data.Conduit.Internal.Pipe
     ( -- ** Types
diff --git a/conduit/conduit.cabal b/conduit/conduit.cabal
index 7490ab0..178e3c7 100644
--- a/conduit/conduit.cabal
+++ b/conduit/conduit.cabal
@@ -32,7 +32,7 @@ Library
                      , exceptions               >= 0.6
                      , lifted-base              >= 0.1
                      , transformers-base        >= 0.4.1        && < 0.5
-                     , transformers             >= 0.2.2        && < 0.5
+                     , transformers             >= 0.2.2        && < 0.6
                      , mtl
                      , mmorph
   if !impl(ghc>=7.9)
