--- acid-state-0.4.2-orig/src/Data/Acid/Local.hs	2011-07-07 23:39:48.000000000 +1000
+++ acid-state-0.4.2/src/Data/Acid/Local.hs	2011-07-14 18:53:39.651371766 +1000
@@ -1,5 +1,5 @@
 {-# LANGUAGE GADTs, OverloadedStrings, DeriveDataTypeable, TypeFamilies,
-             GeneralizedNewtypeDeriving, BangPatterns #-}
+             GeneralizedNewtypeDeriving, BangPatterns, CPP #-}
 -----------------------------------------------------------------------------
 -- |
 -- Module      :  Data.Acid.Local
@@ -57,6 +57,18 @@
 
 import Control.Applicative            (Applicative(..))
 
+#if !MIN_VERSION_mtl(2,0,0)
+import Control.Monad                  ( ap )
+
+instance Applicative (State st) where
+    pure = return
+    (<*>) = ap
+
+instance Applicative (Reader st) where
+    pure = return
+    (<*>) = ap
+#endif
+
 -- | Events return the same thing as Methods. The exact type of 'EventResult'
 --   depends on the event.
 type EventResult ev = MethodResult ev
