diff --git a/src/Data/Acid/Archive.hs b/src/Data/Acid/Archive.hs
index cbc8967..5c87800 100644
--- a/src/Data/Acid/Archive.hs
+++ b/src/Data/Acid/Archive.hs
@@ -20,4 +20,4 @@ import qualified Data.ByteString        as Strict
 import qualified Data.ByteString.Lazy   as Lazy
+import           Data.ByteString.Lazy.Builder
 import           Data.Monoid
-import           Data.Serialize.Builder
 import           Data.Serialize.Get     hiding (Result (..))
@@ -41,5 +41,5 @@ putEntry :: Entry -> Builder
 putEntry content
-    = putWord64le contentLength `mappend`
-      putWord16le contentHash `mappend`
-      fromLazyByteString content
+    = word64LE contentLength `mappend`
+      word16LE contentHash `mappend`
+      lazyByteString content
     where contentLength = fromIntegral $ Lazy.length content
