--- heap-1.0.0-orig/Test.hs	2009-06-03 23:08:50.000000000 +1000
+++ heap-1.0.0/Test.hs	2011-08-14 21:47:59.176389728 +1000
@@ -9,10 +9,10 @@
 main :: IO ()
 main = do
     putStrLn "Ensuring assertions are not ignored:"
-    result <- quickCheckWithResult (Args Nothing 1 1 1) $ expectFailure (assert False True)
+    result <- quickCheckWithResult (Args Nothing 1 1 1 True) $ expectFailure (assert False True)
     putStrLn ""
     case result of
-        (Success _) -> do
+        (Success _ _ _) -> do
             putStrLn "Tests for Data.Heap.Internal:" >> Internal.runTests >> putStrLn ""
             putStrLn "Tests for Data.Heap.Item:"     >> Item.runTests     >> putStrLn ""
             putStrLn "Tests for Data.Heap:"          >> Heap.runTests
