diff -urN ghcide-0.7.3.0/ghcide.cabal ghcide-0.7.3.0-new/ghcide.cabal
--- ghcide-0.7.3.0/ghcide.cabal	2021-02-10 16:08:17.390191536 -0700
+++ ghcide-0.7.3.0-new/ghcide.cabal	2021-02-10 16:16:49.503241208 -0700
@@ -204,7 +204,13 @@
         Development.IDE.Types.Action
     ghc-options: -Wall -Wno-name-shadowing -Wincomplete-uni-patterns
 
+flag test-exe
+    description: Build the ghcide-test-preprocessor executable
+    default: False
+
 executable ghcide-test-preprocessor
+    if !flag(test-exe)
+        buildable: False    
     default-language: Haskell2010
     hs-source-dirs: test/preprocessor
     ghc-options: -Wall -Wno-name-shadowing
@@ -247,9 +253,15 @@
         text,
         yaml
 
+flag executable
+    description: Build the ghcide executable
+    default: False
+
 executable ghcide
     if flag(ghc-lib)
       buildable: False
+    if !flag(executable)
+      buildable: False      
     default-language:   Haskell2010
     hs-source-dirs:     exe
     ghc-options:
@@ -381,7 +393,13 @@
         TypeApplications
         ViewPatterns
 
+flag bench-exe
+    description: Build the ghcide-bench executable
+    default: False
+
 executable ghcide-bench
+    if !flag(bench-exe)
+        buildable: False
     default-language: Haskell2010
     build-tool-depends:
         ghcide:ghcide
