From https://github.com/haskell/haskell-language-server/pull/4444
From: Fendor <fendor@posteo.de>
Date: Mon, 4 Nov 2024 09:12:00 +0100
Subject: [PATCH] Build HLS with GHC 9.8.3

--- a/src/Development/IDE/GHC/Compat/Core.hs
+++ b/src/Development/IDE/GHC/Compat/Core.hs
@@ -674,7 +674,7 @@ initObjLinker env =
 loadDLL :: HscEnv -> String -> IO (Maybe String)
 loadDLL env str = do
     res <- GHCi.loadDLL (GHCi.hscInterp env) str
-#if MIN_VERSION_ghc(9,11,0)
+#if MIN_VERSION_ghc(9,11,0) || (MIN_VERSION_ghc(9, 8, 3) && !MIN_VERSION_ghc(9, 9, 0))
     pure $
       case res of
         Left err_msg -> Just err_msg
