diff --git a/Graph.hs b/Graph.hs
index ecc0493..fbe377c 100644
--- a/Graph.hs
+++ b/Graph.hs
@@ -19,7 +19,7 @@ import Data.ByteString.Char8 (ByteString,pack,unpack)
 import qualified Data.ByteString.Char8 as C
 import qualified Data.ByteString as B
 
-import Control.Exception        (bracket,handle)
+import qualified Control.Exception as E
 import Control.Monad            (when)
 
 import System.Console.GetOpt
@@ -95,7 +95,7 @@ main = do
         mymax   = getYFlag flags
 
     -- aquire some temp files, make sure to clean them up afterwards
-    bracket
+    E.bracket
         (do x <- openTempFile tmp "darcs-graph.data"
             y <- openTempFile tmp "darcs-graph-average.data"
             return (x,y))
@@ -137,7 +137,7 @@ main = do
                     out
 
 
-    (ih,_,eh,pid) <- handle
+    (ih,_,eh,pid) <- (E.handle :: (E.IOException -> IO a) -> IO a -> IO a)
         (\e -> error $ "Couldn't fork gnuplot: " ++ show e)
         (do mgnu <- findExecutable "gnuplot"
             case mgnu of
