diff --git a/Graph.hs b/Graph.hs
index fbe377c..24abf84 100644
--- a/Graph.hs
+++ b/Graph.hs
@@ -31,5 +31,5 @@ import System.Process
 import System.Exit
 import Data.Time
-import System.Locale
+import Data.Time.Locale.Compat
 
 tmp :: FilePath
diff --git a/darcs-graph.cabal b/darcs-graph.cabal
index b67c1cc..b57758f 100644
--- a/darcs-graph.cabal
+++ b/darcs-graph.cabal
@@ -26,4 +26,5 @@ Executable darcs-graph
                        old-time,
                        old-locale,
+                       time-locale-compat,
                        bytestring,
                        containers
