diff --git a/Graph.hs b/Graph.hs
index 541ec99..ecc0493 100644
--- a/Graph.hs
+++ b/Graph.hs
@@ -19,7 +19,7 @@ import Data.ByteString.Char8 (ByteString,pack,unpack)
 import qualified Data.ByteString.Char8 as C
 import qualified Data.ByteString as B
 
-import Control.Exception        (bracket,handle)
+import Control.OldException     (bracket,handle)
 import Control.Monad            (when)
 
 import System.Console.GetOpt
diff --git a/darcs-graph.cabal b/darcs-graph.cabal
index 7867520..b67c1cc 100644
--- a/darcs-graph.cabal
+++ b/darcs-graph.cabal
@@ -20,7 +20,7 @@ Executable darcs-graph
 
     build-depends:     time
     if flag(small_base)
-        build-depends: base >= 3 && < 4,
+        build-depends: base >= 4 && < 5,
                        process,
                        directory,
                        old-time,
