diff --git a/Text/Shakespeare.hs b/Text/Shakespeare.hs
index 45149f0..20a3581 100644
--- a/Text/Shakespeare.hs
+++ b/Text/Shakespeare.hs
@@ -26,7 +26,9 @@ module Text.Shakespeare
 import Text.ParserCombinators.Parsec hiding (Line)
 import Language.Haskell.TH.Quote (QuasiQuoter (..))
 import Language.Haskell.TH.Syntax
-import Language.Haskell.TH.Syntax.Internals
+#if !MIN_VERSION_template_haskell(2,8,0)
+import Language.Haskell.TH.Syntax.Internal
+#endif
 import Data.Text.Lazy.Builder (Builder, fromText)
 import Data.Monoid
 import System.IO.Unsafe (unsafePerformIO)
