diff --git a/lib/Test/Chell/QuickCheck.hs b/lib/Test/Chell/QuickCheck.hs
index c630320..eaf882c 100644
--- a/lib/Test/Chell/QuickCheck.hs
+++ b/lib/Test/Chell/QuickCheck.hs
@@ -46,3 +46,9 @@ property name prop = Chell.test name $ \opts -> do
 		, State.maxSuccessTests = QuickCheck.maxSuccess args
+
+#if MIN_VERSION_QuickCheck(2,10,0)
+		, State.numTotMaxShrinks = maxBound
+		, State.maxDiscardedRatio = 10
+#else
 		, State.maxDiscardedTests = maxDiscardedTests args prop
+#endif
 
