From 36fa1c39787529fc8e16e0575709adf7bd8153f5 Mon Sep 17 00:00:00 2001
From: Xavier Dectot <xavier@wheredoibegin.fr>
Date: Sat, 22 Mar 2025 16:46:56 +0100
Subject: [PATCH] mtl

--- a/src/Text/Parsec/Extra.hs
+++ b/src/Text/Parsec/Extra.hs
@@ -12,7 +12,7 @@ module Text.Parsec.Extra
 
 
 import           Control.Applicative    ((<$>),(<*>),(<*),(*>),pure)
-import           Control.Monad.Error    (ErrorType,MonadError,throwError)
+import           Control.Monad.Except    (ErrorType,MonadError,throwError)
 import           Data.Char              (toLower,toUpper)
 import           Data.List              (foldl')
 import           Text.Parsec.Prim       ((<|>),(<?>),parse)
-- 
2.48.1

