--- yesod-core-0.9.1.1-orig/Yesod/Widget.hs	2011-09-02 01:53:25.000000000 +1000
+++ yesod-core-0.9.1.1/Yesod/Widget.hs	2011-09-11 00:14:57.406926907 +1000
@@ -4,6 +4,7 @@
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
 -- | Widgets combine HTML with JS and CSS dependencies with a unique identifier
 -- generator, allowing you to create truly modular HTML components.
 module Yesod.Widget
