diff -Naur hugs98-Mar2005/libraries/tools/make_bootlib hugs98-Mar2005.new/libraries/tools/make_bootlib
--- hugs98-Mar2005/libraries/tools/make_bootlib	2005-03-10 01:03:39.000000000 +0100
+++ hugs98-Mar2005.new/libraries/tools/make_bootlib	2006-02-28 18:07:37.000000000 +0100
@@ -78,8 +78,9 @@
 
 	(
 		cd $package_dir
-		$FIND . \( -name '[a-z]*' -o -name '[GN]HC' \) -prune -o \
+		$FIND [A-Z]* -name '[GN]HC' -prune -o \
 			\( -name \*.hs -o -name \*.lhs -o -name \*.hsc \) -print |
+			grep -ve '^[GN]HC$' |
 			sed '	s:^\./::
 				s/\..*//' |
 			grep -v '^Text/Regex' |
