diff --git a/Base/Font.hs b/Base/Font.hs
index 7c37b52..0cadd1c 100644
--- a/Base/Font.hs
+++ b/Base/Font.hs
@@ -45,2 +45,3 @@ import Base.Prose
 import Base.Font.ColorVariant
+import Prelude hiding (Word)
 
diff --git a/Control/Monad/CatchState.hs b/Control/Monad/CatchState.hs
index b1ee2c9..42368af 100644
--- a/Control/Monad/CatchState.hs
+++ b/Control/Monad/CatchState.hs
@@ -9,3 +9,3 @@ module Control.Monad.CatchState (
 
-
+import Control.Applicative
 import Control.Monad.State.Class
@@ -13,2 +13,3 @@ import Control.Monad.IO.Class
 import Control.Monad.CatchIO as CatchIO
+import qualified Control.Monad as CM
 
@@ -26,2 +27,6 @@ data CatchState state m a =
 
+instance Monad m => Applicative (CatchState state m) where
+    pure  = return
+    (<*>) = CM.ap
+
 instance Monad m => Monad (CatchState state m) where
diff --git a/Data/Indexable.hs b/Data/Indexable.hs
index 3eee6e3..abde735 100644
--- a/Data/Indexable.hs
+++ b/Data/Indexable.hs
@@ -55,3 +55,3 @@ import Data.Initial
 import Data.Traversable (Traversable)
-import Data.Vector as Vector
+import Data.Vector as Vector hiding (null, elem, maximum)
 import Data.Data (Typeable, Data)
diff --git a/Data/StrictList.hs b/Data/StrictList.hs
index bbea067..c0db2c4 100644
--- a/Data/StrictList.hs
+++ b/Data/StrictList.hs
@@ -7,3 +7,3 @@ import Data.Monoid
 import Data.Foldable as F
-
+import qualified Control.Monad as CM
 
@@ -21,2 +21,7 @@ instance Monoid (StrictList a) where
 
+instance Applicative StrictList where
+    pure = return
+    (<*>) = CM.ap
+    (*>) = (>>)
+
 instance Monad StrictList where
diff --git a/Network/Client.hs b/Network/Client.hs
index 4d9e8e3..bf7809e 100644
--- a/Network/Client.hs
+++ b/Network/Client.hs
@@ -1,2 +1,3 @@
 {-# language MultiParamTypeClasses, DeriveDataTypeable, ScopedTypeVariables #-}
+{-# language CPP #-}
 
@@ -37,2 +38,3 @@ class (Binary p, NFData p) => Protocol p where
 
+#if ! MIN_VERSION_base(4,9,0)
 instance Binary Version where
@@ -42,3 +44,3 @@ instance Binary Version where
         Version <$> get <*> get
-
+#endif
 
diff --git a/Sorts/Terminal.hs b/Sorts/Terminal.hs
index 7e1922d..1c707ed 100644
--- a/Sorts/Terminal.hs
+++ b/Sorts/Terminal.hs
@@ -144,3 +144,3 @@ mkBatteryTerminalSort = do
     case mPngDir of
-        Nothing -> return Nothing
+        Nothing -> return (Nothing :: Maybe TSort)
         Just pngDir -> do
diff --git a/Sorts/Tiles/Baking.hs b/Sorts/Tiles/Baking.hs
index 143759b..72c435a 100644
--- a/Sorts/Tiles/Baking.hs
+++ b/Sorts/Tiles/Baking.hs
@@ -1,2 +1,3 @@
 {-# language ViewPatterns #-}
+{-# language FlexibleContexts #-}
 
