diff --git a/Network/Client.hs b/Network/Client.hs
index f9175aa..3377606 100644
--- a/Network/Client.hs
+++ b/Network/Client.hs
@@ -28,2 +28,3 @@ import           Text.Logging
 
+newtype NikkiVersion = NV Version
 
@@ -34,8 +35,9 @@ class (Binary p, NFData p) => Protocol p where
 
-instance Binary Version where
-    put (Version a b) = putWord8 143 >> put a >> put b
+instance Binary NikkiVersion where
+    put (NV (Version a b)) = putWord8 143 >> put a >> put b
     get = do
         143 <- getWord8
-        Version <$> get <*> get
-
+        a <- get
+        b <- get
+        return $ NV $ Version a b
 
