commit ebdbbd4829b155a0ff739596cb12f1d1356c9591
Author: jao <jao@gnu.org>
Date:   Thu Jun 2 03:48:44 2011 -0700

    Compatibility with libmpd 0.6 (fixes #50)

diff --git a/src/Plugins/Monitors/MPD.hs b/src/Plugins/Monitors/MPD.hs
index 06d0fa7..60c3e48 100644
--- a/src/Plugins/Monitors/MPD.hs
+++ b/src/Plugins/Monitors/MPD.hs
@@ -91,7 +91,7 @@ parseSong (Right Nothing) = return $ repeat ""
 parseSong (Right (Just s)) =
   let join [] = ""
       join (x:xs) = foldl (\a o -> a ++ ", " ++ o) x xs
-      str sel = maybe "" join (M.sgGet sel s)
+      str sel = maybe "" join (M.sgGetTag sel s)
       sels = [ M.Name, M.Artist, M.Composer, M.Performer
              , M.Album, M.Title, M.Track, M.Genre ]
       fields = M.sgFilePath s : map str sels
