--- xmobar-0.26-orig/src/Plugins/Mail.hs	2018-04-07 04:09:54.000000000 +1000
+++ xmobar-0.26/src/Plugins/Mail.hs	2018-04-24 19:45:28.892658697 +1000
@@ -29,6 +29,19 @@
 import Data.List (isPrefixOf)
 import Data.Set (Set)
 import qualified Data.Set as S
+
+#if MIN_VERSION_hinotify(0,3,10)
+import qualified Data.ByteString.Char8 as BS (ByteString, pack, unpack)
+unpack :: BS.ByteString -> String
+unpack = BS.unpack
+pack :: String -> BS.ByteString
+pack = BS.pack
+#else
+unpack :: String -> String
+unpack = id
+pack :: String -> String
+pack = id
+#endif
 #else
 import System.IO
 #endif
@@ -54,7 +67,7 @@
 
         ds <- mapM expandHome rs
         i <- initINotify
-        zipWithM_ (\d v -> addWatch i ev d (handle v)) ds vs
+        zipWithM_ (\d v -> addWatch i ev d (handle v)) (map pack ds) vs
 
         forM_ (zip ds vs) $ \(d, v) -> do
             s <- fmap (S.fromList . filter (not . isPrefixOf "."))
@@ -74,6 +87,6 @@
     MovedOut {} -> delete
     _           -> id
  where
-    delete = S.delete (filePath e)
-    create = S.insert (filePath e)
+    delete = S.delete ((unpack . filePath) e)
+    create = S.insert ((unpack . filePath) e)
 #endif
--- xmobar-0.26-orig/src/Plugins/MBox.hs	2018-04-07 04:09:54.000000000 +1000
+++ xmobar-0.26/src/Plugins/MBox.hs	2018-04-24 19:47:17.491836739 +1000
@@ -31,6 +31,15 @@
 
 import qualified Data.ByteString.Lazy.Char8 as B
 
+#if MIN_VERSION_hinotify(0,3,10)
+import qualified Data.ByteString.Char8 as BS (ByteString, pack)
+pack :: String -> BS.ByteString
+pack = BS.pack
+#else
+pack :: String -> String
+pack = id
+#endif
+
 data Options = Options
                { oAll :: Bool
                , oUniq :: Bool
@@ -93,7 +102,7 @@
                    n <- if exists then countMails f else return (-1)
                    v <- newTVarIO (f, n)
                    when exists $
-                     addWatch i events f (handleNotification v) >> return ()
+                     addWatch i events (pack f) (handleNotification v) >> return ()
                    return v)
                 boxes
 
