diff --git a/aeson-extra.cabal b/aeson-extra.cabal
index 1814692..a05f1f0 100644
--- a/aeson-extra.cabal
+++ b/aeson-extra.cabal
@@ -42,3 +42,3 @@ library
     , parsec                   >=3.1.9 && <3.2
-    , recursion-schemes        >=4.1.2 && <4.2
+    , recursion-schemes        >=5
     , scientific               >=0.3  && <0.4
diff --git a/src/Data/Aeson/Extra/Foldable.hs b/src/Data/Aeson/Extra/Foldable.hs
index 1075051..f3fbf44 100644
--- a/src/Data/Aeson/Extra/Foldable.hs
+++ b/src/Data/Aeson/Extra/Foldable.hs
@@ -62,3 +62,3 @@ type instance Base Value = ValueF
 
-instance F.Foldable Value where
+instance F.Recursive Value where
     project (Object o) = ObjectF o
@@ -70,3 +70,3 @@ instance F.Foldable Value where
 
-instance F.Unfoldable Value where
+instance F.Corecursive Value where
     embed (ObjectF o) = Object o
