diff --git a/src/Data/Aeson/Extra/Map.hs b/src/Data/Aeson/Extra/Map.hs
index 2915c7c..f405be5 100644
--- a/src/Data/Aeson/Extra/Map.hs
+++ b/src/Data/Aeson/Extra/Map.hs
@@ -26 +26 @@ import Prelude        ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Data/Aeson/Extra/SingObject.hs b/src/Data/Aeson/Extra/SingObject.hs
index 5f50da9..4c5569c 100644
--- a/src/Data/Aeson/Extra/SingObject.hs
+++ b/src/Data/Aeson/Extra/SingObject.hs
@@ -23 +23 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
