--- ChasingBottoms-1.3.0.7-orig/ChasingBottoms.cabal	2013-10-15 15:50:33.000000000 +1100
+++ ChasingBottoms-1.3.0.7/ChasingBottoms.cabal	2014-07-02 15:23:40.070800449 +1000
@@ -121,9 +121,9 @@
 
     other-modules: Test.ChasingBottoms.IsType
 
-    build-depends: QuickCheck >= 2.1 && < 2.7,
-                   mtl >= 1.1 && < 2.2,
-                   base >= 4.0 && < 4.8,
+    build-depends: QuickCheck >= 2.1 && < 2.8,
+                   mtl >= 1.1 && < 2.3,
+                   base >= 4.0,
                    containers >= 0.3 && < 0.6,
                    random == 1.0.*,
                    syb >= 0.1.0.2 && < 0.5
@@ -150,9 +150,9 @@
                    Test.ChasingBottoms.TestUtilities.Generators,
                    Test.ChasingBottoms.TimeOut.Tests
 
-    build-depends: QuickCheck >= 2.1 && < 2.7,
-                   mtl >= 1.1 && < 2.2,
-                   base >= 4.0 && < 4.8,
+    build-depends: QuickCheck >= 2.1 && < 2.8,
+                   mtl >= 1.1 && < 2.3,
+                   base >= 4.0,
                    containers >= 0.3 && < 0.6,
                    random == 1.0.*,
                    syb >= 0.1.0.2 && < 0.5,
--- ChasingBottoms-1.3.0.7-orig/Test/ChasingBottoms/ContinuousFunctions.hs	2013-10-15 15:50:33.000000000 +1100
+++ ChasingBottoms-1.3.0.7/Test/ChasingBottoms/ContinuousFunctions.hs	2014-07-02 15:32:00.952887830 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE RankNTypes, ScopedTypeVariables,
+{-# LANGUAGE CPP, RankNTypes, ScopedTypeVariables,
              GeneralizedNewtypeDeriving, DeriveDataTypeable #-}
 
 -- TODO: Can we pattern match on functions?
@@ -143,7 +143,12 @@
   , listOf
   ) where
 
+#if MIN_VERSION_QuickCheck(2,7,0)
+import Test.QuickCheck hiding ((><), listOf, infiniteListOf)
+import Test.QuickCheck.Gen.Unsafe (promote)
+#else
 import Test.QuickCheck hiding ((><), listOf)
+#endif
 import Data.Sequence as Seq
 import Data.Foldable as Seq (foldr)
 import Prelude as P hiding (concat)
--- ChasingBottoms-1.3.0.7-orig/Test/ChasingBottoms/TestUtilities/Generators.hs	2013-10-15 15:50:33.000000000 +1100
+++ ChasingBottoms-1.3.0.7/Test/ChasingBottoms/TestUtilities/Generators.hs	2014-07-02 15:33:26.677545051 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE RankNTypes, DeriveDataTypeable #-}
+{-# LANGUAGE CPP, RankNTypes, DeriveDataTypeable #-}
 
 -- | Generators that are part of the testing framework.
 
@@ -46,7 +46,12 @@
 import Test.ChasingBottoms.IsBottom
 import Test.ChasingBottoms.SemanticOrd
 import Test.ChasingBottoms.TestUtilities
+#if MIN_VERSION_QuickCheck(2,7,0)
+import Test.QuickCheck hiding (infiniteListOf)
+import Test.QuickCheck.Gen.Unsafe (promote)
+#else
 import Test.QuickCheck
+#endif
 import Data.Generics
 import Control.Monad
 import Data.Maybe
