https://github.com/hpdeifel/hledger-iadd/issues/53
--- a/tests/ModelSpec.hs
+++ b/tests/ModelSpec.hs
@@ -188,7 +188,7 @@ isDuplicateTransactionSpec = do
     isDuplicateTransaction (mkJournal [t1]) (mkTransaction t4) `shouldBe` False
     isDuplicateTransaction (mkJournal [t1]) t5 `shouldBe` False
 
-  it "ignores amount presentation" $ do
+  xit "ignores amount presentation" $ do
     let a1 = (HL.eur 0.5) { HL.astyle = HL.amountstyle}
         -- We use 'read' in the following because hledger-lib 1.19 changed the
         -- type of 'asprecision' from Int to 'AmountPrecision'. 'read' works in
