--- a/Graphics/Gloss/Internals/Data/Picture.hs
+++ b/Graphics/Gloss/Internals/Data/Picture.hs
@@ -31,2 +31,3 @@ import qualified Data.ByteString.Unsafe as BSU
 import Prelude hiding (map)
+import qualified Data.Semigroup as S
 
@@ -126,2 +127,5 @@ data Picture
 
+instance S.Semigroup Picture where
+        (<>) a b     = Pictures [a, b]
+
 -- Instances ------------------------------------------------------------------
