diff --git a/src/Data/Hoodle/Zipper.hs b/src/Data/Hoodle/Zipper.hs
index 7ec3ab1..f10cd3e 100644
--- a/src/Data/Hoodle/Zipper.hs
+++ b/src/Data/Hoodle/Zipper.hs
@@ -23,3 +23,3 @@ import Data.Foldable
 import Data.Monoid
-import Data.Sequence hiding (fromList)
+import Data.Sequence as S hiding (fromList)
 import Data.Traversable
@@ -61,3 +61,3 @@ singletonSZ x = SZ (x, (empty,empty))
 lengthSZ :: SeqZipper a -> Int 
-lengthSZ (SZ (_x, (x1s,x2s))) = length x1s + length x2s + 1 
+lengthSZ (SZ (_x, (x1s,x2s))) = S.length x1s + S.length x2s + 1 
 
@@ -65,3 +65,3 @@ lengthSZ (SZ (_x, (x1s,x2s))) = length x1s + length x2s + 1
 currIndex :: SeqZipper a -> Int
-currIndex (SZ (_x, (x1s,_x2s))) = length x1s 
+currIndex (SZ (_x, (x1s,_x2s))) = S.length x1s 
 
@@ -97,4 +97,4 @@ moveTo :: Int -> SeqZipper a -> Maybe (SeqZipper a)
 moveTo n orig@(SZ (x,(x1s,x2s))) = 
-  let n_x1s = length x1s 
-      n_x2s = length x2s 
+  let n_x1s = S.length x1s 
+      n_x2s = S.length x2s 
       res | n < 0 || n > n_x1s + n_x2s = Nothing 
