--- a/src/Data/Hoodle/BBox.hs
+++ b/src/Data/Hoodle/BBox.hs
@@ -53,2 +53,3 @@ import Prelude hiding (fst,snd)
 import qualified Prelude as Prelude (fst,snd)
+import qualified Data.Semigroup as S
 
@@ -211,2 +212,10 @@ newtype UnionBBox = Union { unUnion :: ULMaybe BBox }
                     deriving (Show,Eq)
+
+instance S.Semigroup (IntersectBBox) where 
+  (Intersect Bottom) <> _ = Intersect Bottom
+  _ <> (Intersect Bottom) = Intersect Bottom 
+  (Intersect Top) <> x = x 
+  x <> (Intersect Top) = x 
+  (Intersect (Middle x)) <> (Intersect (Middle y)) = 
+    maybe (Intersect Bottom) (Intersect . Middle) (x `intersectBBox` y)
      
@@ -220,3 +229,10 @@ instance Monoid (IntersectBBox) where
   mempty = Intersect Top 
-  
+
+instance S.Semigroup (UnionBBox) where
+  (Union Bottom) <> x = x 
+  x <> (Union Bottom) = x
+  (Union Top) <> _ = Union Top
+  _ <> (Union Top) = Union Top 
+  (Union (Middle x)) <> (Union (Middle y)) = Union (Middle (x `unionBBox` y))
+
 instance Monoid (UnionBBox) where
