--- haskell-src-meta-0.5.1.2-orig/haskell-src-meta.cabal	2012-03-31 00:21:12.000000000 +1100
+++ haskell-src-meta-0.5.1.2/haskell-src-meta.cabal	2012-05-18 15:10:11.208875261 +1000
@@ -24,7 +24,7 @@
                    syb >= 0.1 && < 0.4,
                    th-lift == 0.5.*
 
-  if impl(ghc >= 7.4)
+  if impl(ghc >= 7.4) && impl(ghc < 7.5)
     Build-depends: template-haskell >= 2.7 && < 2.9
   else
     Build-depends: template-haskell >= 2.4 && < 2.7,
--- haskell-src-meta-0.5.1.2-orig/src/Language/Haskell/TH/Instances/Lift.hs	2012-03-31 00:21:12.000000000 +1100
+++ haskell-src-meta-0.5.1.2/src/Language/Haskell/TH/Instances/Lift.hs	2012-05-18 15:19:06.320535774 +1000
@@ -44,6 +44,12 @@
 deriving instance Ord Pragma
 #endif /* MIN_VERSION_template_haskell(2,4,0) */
 
+#if (__GLASGOW_HASKELL__==705)
+deriving instance Ord Fixity
+deriving instance Ord FixityDirection
+deriving instance Ord TyLit
+#endif
+
 deriving instance Show Loc
 deriving instance Eq Loc
 
@@ -81,9 +87,12 @@
                  , ''Pragma
                  , ''Pred
                  , ''TyVarBndr
-#if MIN_VERSION_template_haskell(2,5,0) && !(MIN_VERSION_template_haskell(2,7,0))
+#if MIN_VERSION_template_haskell(2,5,0) && !(MIN_VERSION_template_haskell(2,7,0)) && !(__GLASGOW_HASKELL__==705)
                  , ''ClassInstance
-#endif /* MIN_VERSION_template_haskell(2,5,0) && !(MIN_VERSION_template_haskell(2,7,0)) */
+#endif /* MIN_VERSION_template_haskell(2,5,0) && !(MIN_VERSION_template_haskell(2,7,0)) && !(__GLASGOW_HASKELL__==705) */
 #endif /* MIN_VERSION_template_haskell(2,4,0) */
+#if (__GLASGOW_HASKELL__==705)
+                 , ''TyLit
+#endif
                  ])
 
