--- haskell-src-meta-0.4.0.1-orig/haskell-src-meta.cabal	2011-06-28 10:25:53.000000000 +1000
+++ haskell-src-meta-0.4.0.1/haskell-src-meta.cabal	2011-08-01 18:49:54.559388509 +1000
@@ -17,10 +17,10 @@
 extra-source-files: examples/*.hs README
 
 library
-  build-depends:   base >= 4.1 && < 4.4,
+  build-depends:   base >= 4.1 && < 4.5,
                    haskell-src-exts >= 1.6 && < 1.12,
-                   template-haskell >= 2.3 && < 2.6,
-                   pretty == 1.0.*,
+                   template-haskell >= 2.3 && < 2.7,
+                   pretty >= 1.0 && < 1.2,
                    syb >= 0.1 && < 0.4,
                    th-lift == 0.5.*
   extensions:      CPP,
--- haskell-src-meta-0.4.0.1-orig/src/Language/Haskell/Meta/Syntax/Translate.hs	2011-06-28 10:25:53.000000000 +1000
+++ haskell-src-meta-0.4.0.1/src/Language/Haskell/Meta/Syntax/Translate.hs	2011-08-01 18:51:31.255853936 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE CPP, TemplateHaskell, TypeSynonymInstances #-}
+{-# LANGUAGE CPP, TemplateHaskell, TypeSynonymInstances, FlexibleInstances #-}
 
 {- |
   Module      :  Language.Haskell.Meta.Syntax.Translate
