diff --git a/src/Game/H2048/UI/Vty.hs b/src/Game/H2048/UI/Vty.hs
index 875e92b..e4abb79 100644
--- a/src/Game/H2048/UI/Vty.hs
+++ b/src/Game/H2048/UI/Vty.hs
@@ -11,2 +11,3 @@ A CLI version of Game 2048 implemented using vty-ui
 -}
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE OverloadedStrings #-}
@@ -53,3 +54,7 @@ colorize i = [(s,attr)]
         s = if i /= 0 then (T.pack . show) i else " "
+#if MIN_VERSION_vty(5,18,0)
+        attr = Attr (SetTo colorSty) (SetTo colorNum) Default Default
+#else
         attr = Attr (SetTo colorSty) (SetTo colorNum) Default
+#endif
         (colorSty, colorNum) = fromMaybe (bold,ISOColor 3) (lookup i colorDict)
