diff --git a/src/GHC/Vacuum.hs b/src/GHC/Vacuum.hs
index 33dc219..0c2cd46 100644
--- a/src/GHC/Vacuum.hs
+++ b/src/GHC/Vacuum.hs
@@ -108,22 +108,22 @@ import System.Mem.StableName
 
 -- | Vacuums the entire reachable heap subgraph rooted at the @a@.
 vacuum :: a -> IntMap HNode
-vacuum a = unsafePerformIO (dump a)
+vacuum a = System.IO.Unsafe.unsafePerformIO (dump a)
 
 -- | Returns nodes as it encounters them.
 vacuumStream :: a -> [(HNodeId, HNode)]
-vacuumStream a = unsafePerformIO (dumpStream a)
+vacuumStream a = System.IO.Unsafe.unsafePerformIO (dumpStream a)
 
 vacuumDebug :: a -> IntMap [(StableName HValue, HNodeId)]
-vacuumDebug a = unsafePerformIO (dumpDebug a)
+vacuumDebug a = System.IO.Unsafe.unsafePerformIO (dumpDebug a)
 
 -- | Stop after a given depth.
 vacuumTo :: Int -> a -> IntMap HNode
-vacuumTo n a = unsafePerformIO (dumpTo n a)
+vacuumTo n a = System.IO.Unsafe.unsafePerformIO (dumpTo n a)
 
 -- | Doesn't force anything.
 vacuumLazy :: a -> IntMap HNode
-vacuumLazy a = unsafePerformIO (dumpLazy a)
+vacuumLazy a = System.IO.Unsafe.unsafePerformIO (dumpLazy a)
 
 dump :: a -> IO (IntMap HNode)
 dump a = execH (dumpH a)
