diff --git a/Control/Applicative/QQ/ADo.hs b/Control/Applicative/QQ/ADo.hs
index f912a0b..9b4b324 100644
--- a/Control/Applicative/QQ/ADo.hs
+++ b/Control/Applicative/QQ/ADo.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE TemplateHaskell #-}
@@ -163,3 +164,7 @@ singleCon n = do
     Just vn <- lookupValueName (show n)
+#if MIN_VERSION_template_haskell(2,11,0)
+    DataConI _ _ tn <- reify vn
+#else
     DataConI _ _ tn _ <- reify vn
+#endif
     TyConI dec <- reify tn
@@ -167,2 +172,7 @@ singleCon n = do
   case dec of
+#if MIN_VERSION_template_haskell(2,11,0)
+    DataD _ _ _ _ [_] _ -> return True
+    NewtypeD {} -> return True
+    DataD _ _ _ _ (_:_) _ -> return False
+#else
     DataD _ _ _ [_] _ -> return True
@@ -170,2 +180,3 @@ singleCon n = do
     DataD _ _ _ (_:_) _ -> return False
+#endif
     _ -> fail $ "ado singleCon: not a data declaration: " ++ show dec
