--- a/Setup.hs
+++ b/Setup.hs
@@ -94,3 +94,3 @@ lhs2texPostConf a cf pd lbi =
                                                               let (sv,nec) = case vp of
-                                                                               Just n  -> (showVersion n,versionBranch n < minPolytableVersion)
+                                                                               Just n  -> (Data.Version.showVersion n,versionBranch n < minPolytableVersion)
                                                                                Nothing -> ("unknown",True)
--- a/src/Document.lhs
+++ b/src/Document.lhs
@@ -8,2 +8,3 @@
 > where
+> import Prelude hiding ((<>))
 
--- a/src/Main.lhs
+++ b/src/Main.lhs
@@ -23,3 +23,3 @@
 > import Control.Monad.Trans
-> import Prelude hiding ( getContents )
+> import Prelude hiding ( getContents, (<>) )
 >
--- a/src/Math.lhs
+++ b/src/Math.lhs
@@ -9,3 +9,3 @@
 >
-> import Prelude hiding         (  lines )
+> import Prelude hiding         (  lines, (<>) )
 > import Data.List              (  partition )
--- a/src/MathPoly.lhs
+++ b/src/MathPoly.lhs
@@ -15,3 +15,3 @@ are subtle differences, and they will grow over time \dots
 >
-> import Prelude hiding         (  lines )
+> import Prelude hiding         (  lines, (<>) )
 > import Data.List              (  partition, nub, insert, sort, transpose )
--- a/src/NewCode.lhs
+++ b/src/NewCode.lhs
@@ -26,2 +26,3 @@ way that is compatible with the @poly@ or @math@ formatters.
 > import TeXCommands            (  Lang(..) )
+> import Prelude hiding ((<>))
 
--- a/src/Typewriter.lhs
+++ b/src/Typewriter.lhs
@@ -18,2 +18,3 @@
 > import TeXCommands ( Lang (..) )
+> import Prelude hiding ((<>))
 
--- a/src/Verbatim.lhs
+++ b/src/Verbatim.lhs
@@ -15,2 +15,3 @@
 > import Auxiliaries
+> import Prelude hiding ((<>))
 
