--- old/lhs2tex.cabal	(revision 200)
+++ lhs2tex-1.12/lhs2tex.cabal	(revision 201)
@@ -9,7 +9,7 @@
 homepage:	http://www.andres-loeh.de/lhs2tex/
 package-url:	http://www.andres-loeh.de/lhs2tex/lhs2tex-1.11.tar.gz
 synopsis:	Preprocessor for typesetting Haskell sources with LaTeX
-build-depends:	base, regex-compat
+build-depends:	base, regex-compat, mtl
 
 executable:	lhs2TeX
 main-is:	Main.lhs
--- old/Auxiliaries.lhs	(revision 200)
+++ lhs2tex-1.12/Auxiliaries.lhs	(revision 201)
@@ -9,6 +9,7 @@
 >
 > import Data.Char              (  isSpace  )
 > import Control.Monad          (  MonadPlus(..)  )
+> import Control.Monad.Error
 
 %endif
 
@@ -109,6 +110,7 @@
 
 |Either| as an exception monad.
 
+> {-
 > instance Functor (Either a) where
 >     fmap f (Left  a)          =  Left  a
 >     fmap f (Right b)          =  Right (f b)
@@ -117,6 +119,8 @@
 >     Left  a >>= k             =  Left a
 >     Right b >>= k             =  k b
 >     return                    =  Right
+> -}
+> instance (Error a, Error b) => Error (a,b) where
 >
 > fromRight                     :: Either a b -> b
 > fromRight (Left _)            =  error "fromRight"
--- old/HsLexer.lhs	(revision 200)
+++ lhs2tex-1.12/HsLexer.lhs	(revision 201)
@@ -8,6 +8,7 @@
 > where
 > import Data.Char 	(  isSpace, isUpper, isLower, isDigit, isAlphaNum  )
 > import Control.Monad
+> import Control.Monad.Error ()
 > import Document
 > import Auxiliaries
 
