diff --git a/src/Cryptol/ModuleSystem/Base.hs b/src/Cryptol/ModuleSystem/Base.hs
index 91eb95e..6002348 100644
--- a/src/Cryptol/ModuleSystem/Base.hs
+++ b/src/Cryptol/ModuleSystem/Base.hs
@@ -63 +63 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/ModuleSystem/Env.hs b/src/Cryptol/ModuleSystem/Env.hs
index fc4219a..b071865 100644
--- a/src/Cryptol/ModuleSystem/Env.hs
+++ b/src/Cryptol/ModuleSystem/Env.hs
@@ -44 +44 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/ModuleSystem/Monad.hs b/src/Cryptol/ModuleSystem/Monad.hs
index 789efe7..60951db 100644
--- a/src/Cryptol/ModuleSystem/Monad.hs
+++ b/src/Cryptol/ModuleSystem/Monad.hs
@@ -41 +41 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/ModuleSystem/Name.hs b/src/Cryptol/ModuleSystem/Name.hs
index ed3ae64..93f8c8d 100644
--- a/src/Cryptol/ModuleSystem/Name.hs
+++ b/src/Cryptol/ModuleSystem/Name.hs
@@ -62 +62 @@ import           Prelude ()
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/ModuleSystem/Renamer.hs b/src/Cryptol/ModuleSystem/Renamer.hs
index 1c3e367..575de82 100644
--- a/src/Cryptol/ModuleSystem/Renamer.hs
+++ b/src/Cryptol/ModuleSystem/Renamer.hs
@@ -47 +47 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/Parser/AST.hs b/src/Cryptol/Parser/AST.hs
index 51bf030..3cc812f 100644
--- a/src/Cryptol/Parser/AST.hs
+++ b/src/Cryptol/Parser/AST.hs
@@ -90 +90 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/Parser/NoPat.hs b/src/Cryptol/Parser/NoPat.hs
index 2ae6969..7f1de88 100644
--- a/src/Cryptol/Parser/NoPat.hs
+++ b/src/Cryptol/Parser/NoPat.hs
@@ -37 +37 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/Parser/ParserUtils.hs b/src/Cryptol/Parser/ParserUtils.hs
index ed7a743..9fd1039 100644
--- a/src/Cryptol/Parser/ParserUtils.hs
+++ b/src/Cryptol/Parser/ParserUtils.hs
@@ -36 +36 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Cryptol/Utils/PP.hs b/src/Cryptol/Utils/PP.hs
index 236ce65..38599d1 100644
--- a/src/Cryptol/Utils/PP.hs
+++ b/src/Cryptol/Utils/PP.hs
@@ -26 +26 @@ import Prelude ()
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
