diff --git a/src/Cryptol/Eval/Env.hs b/src/Cryptol/Eval/Env.hs
index 04c1a52..4f44030 100644
--- a/src/Cryptol/Eval/Env.hs
+++ b/src/Cryptol/Eval/Env.hs
@@ -22,2 +22,3 @@ import qualified Data.Map as Map
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Eval/Value.hs b/src/Cryptol/Eval/Value.hs
index 0211e99..5b7a863 100644
--- a/src/Cryptol/Eval/Value.hs
+++ b/src/Cryptol/Eval/Value.hs
@@ -31,2 +31,3 @@ import Numeric (showIntAtBase)
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/Env.hs b/src/Cryptol/ModuleSystem/Env.hs
index 00b1776..da741c8 100644
--- a/src/Cryptol/ModuleSystem/Env.hs
+++ b/src/Cryptol/ModuleSystem/Env.hs
@@ -40,2 +40,3 @@ import qualified Data.List as List
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/Interface.hs b/src/Cryptol/ModuleSystem/Interface.hs
index 290acbb..1cc42eb 100644
--- a/src/Cryptol/ModuleSystem/Interface.hs
+++ b/src/Cryptol/ModuleSystem/Interface.hs
@@ -30,2 +30,3 @@ import qualified Data.Map as Map
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/Monad.hs b/src/Cryptol/ModuleSystem/Monad.hs
index dcb62c1..d1c675e 100644
--- a/src/Cryptol/ModuleSystem/Monad.hs
+++ b/src/Cryptol/ModuleSystem/Monad.hs
@@ -36,2 +36,3 @@ import MonadLib
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/Name.hs b/src/Cryptol/ModuleSystem/Name.hs
index b4864ae..9a67c94 100644
--- a/src/Cryptol/ModuleSystem/Name.hs
+++ b/src/Cryptol/ModuleSystem/Name.hs
@@ -50,2 +50,3 @@ import           Cryptol.Utils.PP
 
+import           Control.DeepSeq
 import           Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/NamingEnv.hs b/src/Cryptol/ModuleSystem/NamingEnv.hs
index 86684ea..0f9f822 100644
--- a/src/Cryptol/ModuleSystem/NamingEnv.hs
+++ b/src/Cryptol/ModuleSystem/NamingEnv.hs
@@ -32,2 +32,3 @@ import qualified Data.Set as Set
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/ModuleSystem/Renamer.hs b/src/Cryptol/ModuleSystem/Renamer.hs
index d9880fa..3e8b54f 100644
--- a/src/Cryptol/ModuleSystem/Renamer.hs
+++ b/src/Cryptol/ModuleSystem/Renamer.hs
@@ -42,2 +42,3 @@ import qualified Data.Map as Map
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/AST.hs b/src/Cryptol/Parser/AST.hs
index deceede..e9b9e63 100644
--- a/src/Cryptol/Parser/AST.hs
+++ b/src/Cryptol/Parser/AST.hs
@@ -83,2 +83,3 @@ import           Numeric(showIntAtBase)
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/LexerUtils.hs b/src/Cryptol/Parser/LexerUtils.hs
index c009204..9afe5d7 100644
--- a/src/Cryptol/Parser/LexerUtils.hs
+++ b/src/Cryptol/Parser/LexerUtils.hs
@@ -26,2 +26,3 @@ import           Data.Word(Word8)
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/NoInclude.hs b/src/Cryptol/Parser/NoInclude.hs
index 29a58b1..ba84b77 100644
--- a/src/Cryptol/Parser/NoInclude.hs
+++ b/src/Cryptol/Parser/NoInclude.hs
@@ -37,2 +37,3 @@ import           System.FilePath (takeDirectory,(</>),isAbsolute)
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/NoPat.hs b/src/Cryptol/Parser/NoPat.hs
index e2ea1d6..c18826c 100644
--- a/src/Cryptol/Parser/NoPat.hs
+++ b/src/Cryptol/Parser/NoPat.hs
@@ -32,2 +32,3 @@ import qualified Data.Map as Map
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/ParserUtils.hs b/src/Cryptol/Parser/ParserUtils.hs
index 053de7e..4263db2 100644
--- a/src/Cryptol/Parser/ParserUtils.hs
+++ b/src/Cryptol/Parser/ParserUtils.hs
@@ -30,2 +30,3 @@ import qualified Data.Text.Lazy as T
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Parser/Position.hs b/src/Cryptol/Parser/Position.hs
index f18e6ca..e150901 100644
--- a/src/Cryptol/Parser/Position.hs
+++ b/src/Cryptol/Parser/Position.hs
@@ -18,2 +18,3 @@ import qualified Data.Text.Lazy as T
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Prims/Syntax.hs b/src/Cryptol/Prims/Syntax.hs
index a386bb9..28b96c1 100644
--- a/src/Cryptol/Prims/Syntax.hs
+++ b/src/Cryptol/Prims/Syntax.hs
@@ -20,2 +20,3 @@ import qualified Data.Map as Map
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/TypeCheck/AST.hs b/src/Cryptol/TypeCheck/AST.hs
index 2ab895a..0c300af 100644
--- a/src/Cryptol/TypeCheck/AST.hs
+++ b/src/Cryptol/TypeCheck/AST.hs
@@ -39,2 +39,3 @@ import Cryptol.TypeCheck.Solver.InfNat
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/TypeCheck/InferTypes.hs b/src/Cryptol/TypeCheck/InferTypes.hs
index 2d579f9..1fc679c 100644
--- a/src/Cryptol/TypeCheck/InferTypes.hs
+++ b/src/Cryptol/TypeCheck/InferTypes.hs
@@ -33,2 +33,3 @@ import qualified Data.IntMap as IntMap
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/TypeCheck/Monad.hs b/src/Cryptol/TypeCheck/Monad.hs
index 5eeb6eb..0209aed 100644
--- a/src/Cryptol/TypeCheck/Monad.hs
+++ b/src/Cryptol/TypeCheck/Monad.hs
@@ -38,2 +38,3 @@ import           MonadLib hiding (mapM)
 import GHC.Generics (Generic)
+import Control.DeepSeq
 import Control.DeepSeq.Generics
diff --git a/src/Cryptol/Utils/PP.hs b/src/Cryptol/Utils/PP.hs
index b63b173..56a9cd6 100644
--- a/src/Cryptol/Utils/PP.hs
+++ b/src/Cryptol/Utils/PP.hs
@@ -14,3 +14,3 @@ module Cryptol.Utils.PP where
 import           Cryptol.Utils.Ident
-
+import           Control.DeepSeq
 import           Control.DeepSeq.Generics
