diff --git a/ghc-tcplugins-extra.cabal b/ghc-tcplugins-extra.cabal
index 95e7226..8edf351 100644
--- a/ghc-tcplugins-extra.cabal
+++ b/ghc-tcplugins-extra.cabal
@@ -33,7 +33,7 @@ flag deverror
 library
   exposed-modules:     GHC.TcPluginM.Extra
   build-depends:       base >=4.8  && <5,
-                       ghc  >=7.10 && <8.6
+                       ghc  >=7.10 && <8.9
   hs-source-dirs:      src
   default-language:    Haskell2010
   other-extensions:    CPP
