--- system-fileio-0.3.9-orig/system-fileio.cabal	2012-07-13 10:27:00.000000000 +1000
+++ system-fileio-0.3.9/system-fileio.cabal	2012-09-15 15:18:59.597411287 +1000
@@ -50,7 +50,7 @@
 
   build-depends:
       base >= 4.0 && < 5.0
-    , bytestring >= 0.9 && < 0.10
+    , bytestring >= 0.9 && < 0.11
     , system-filepath >= 0.3.1 && < 0.5
     , text >= 0.7.1 && < 0.12
     , time >= 1.0 && < 1.5
@@ -63,7 +63,7 @@
     c-sources: lib/hssystemfileio-win32.c
   else
     build-depends:
-        unix >= 2.3 && < 2.6
+        unix >= 2.3 && < 2.7
     c-sources: lib/hssystemfileio-unix.c
 
   exposed-modules:
--- system-fileio-0.3.9-orig/lib/Filesystem.hs	2012-07-13 10:27:00.000000000 +1000
+++ system-fileio-0.3.9/lib/Filesystem.hs	2012-09-15 15:18:30.579666735 +1000
@@ -129,6 +129,10 @@
 import qualified System.Posix.Internals
 #endif
 
+#if (__GLASGOW_HASKELL__>=705)
+import Foreign.C.Types(CInt(..))
+#endif
+
 -- | Check if a file exists at the given path.
 --
 -- Any non&#8208;directory object, including devices and pipes, are
