--- ltk-0.10.0.4-orig/ltk.cabal	2011-04-21 18:45:19.000000000 +1000
+++ ltk-0.10.0.4/ltk.cabal	2012-02-04 00:52:47.321153256 +1100
@@ -16,13 +16,13 @@
 tested-with: GHC ==6.10 || ==6.12 || ==7.0
 
 Library
-    build-depends: Cabal >=1.6.0 && <1.11, base >=4.0.0.0 && <4.4,
-               containers >=0.2 && <0.5, filepath >=1.1.0 && <1.3,
+    build-depends: Cabal >=1.6.0 && <1.15, base >=4.0.0.0 && <4.6,
+               containers >=0.2 && <0.5, filepath >=1.1.0 && <1.4,
                glib >=0.10.0 && <0.13, gtk >=0.10.0 && <0.13,
                mtl >=1.1.0.2 && <2.1, parsec >=2.1.0.1 && <3.2,
-               pretty >=1.0.1.0 && <1.1,
+               pretty >=1.0.1.0 && <1.2,
                -- Make sure we get the Cabal compaitble with ghc and haddock
-               haddock <2.10, ghc <7.3
+               haddock <2.11, ghc <7.5
     exposed-modules: Default MyMissing Control.Event
                  Graphics.UI.Editor.Basics Graphics.UI.Editor.Composite
                  Graphics.UI.Editor.DescriptionPP Graphics.UI.Editor.MakeEditor
--- ltk-0.10.0.4-orig/src/Graphics/UI/Frame/ViewFrame.hs	2011-04-21 18:45:19.000000000 +1000
+++ ltk-0.10.0.4/src/Graphics/UI/Frame/ViewFrame.hs	2012-02-04 00:53:34.300575831 +1100
@@ -4,7 +4,8 @@
     -XFlexibleInstances
     -XMultiParamTypeClasses
     -XUndecidableInstances
-    -XDeriveDataTypeable #-}
+    -XDeriveDataTypeable
+    -XBangPatterns #-}
 -----------------------------------------------------------------------------
 --
 -- Module      :  IDE.Core.ViewFrame
--- ltk-0.10.0.4-orig/src/Text/PrinterParser.hs	2011-04-21 18:45:19.000000000 +1000
+++ ltk-0.10.0.4/src/Text/PrinterParser.hs	2012-02-04 00:53:12.892927667 +1100
@@ -1,4 +1,4 @@
-{-# OPTIONS_GHC -XTypeSynonymInstances #-}
+{-# OPTIONS_GHC -XTypeSynonymInstances -XFlexibleInstances #-}
 --
 -- | Module for saving and restoring preferences and settings
 --
