diff -Naur /home/jkarlson/monads-fd-0.1.0.4/monads-fd.cabal /home/jkarlson/monads-fd-0.1.0.4-r1/monads-fd.cabal
--- /home/jkarlson/monads-fd-0.1.0.4/monads-fd.cabal	2010-11-07 17:10:17.000000000 +0200
+++ /home/jkarlson/monads-fd-0.1.0.4-r1/monads-fd.cabal	2012-04-14 16:33:13.435242768 +0300
@@ -40,7 +40,7 @@
     Control.Monad.Writer.Class
     Control.Monad.Writer.Lazy
     Control.Monad.Writer.Strict
-  build-depends: base < 6, transformers == 0.2.*
+  build-depends: base < 6, transformers >=0.2 && < 0.4
   extensions:
     MultiParamTypeClasses
     FunctionalDependencies
