--- remote-0.1.1-orig/Remote/Process.hs	2011-12-16 23:13:04.000000000 +1100
+++ remote-0.1.1/Remote/Process.hs	2012-06-30 00:48:00.909676425 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE CPP, DeriveDataTypeable #-}
 
 -- | This module is the core of Cloud Haskell. It provides 
 -- processes, messages, monitoring, and configuration.
@@ -65,7 +65,11 @@
                        where
 
 import qualified Prelude as Prelude
-import Prelude hiding (catch, id, init, last, lookup, pi)
+import Prelude hiding (
+#if !MIN_VERSION_base(4,6,0)
+  catch,
+#endif
+  id, init, last, lookup, pi)
 
 import Control.Concurrent (forkIO,ThreadId,threadDelay)
 import Control.Concurrent.MVar (MVar,newMVar, newEmptyMVar,takeMVar,putMVar,modifyMVar,modifyMVar_,readMVar)
