From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Tests require the "output" directory to exist

diff -urN brittany-0.14.0.2/cabal.project.local brittany-0.14.0.2-r1/cabal.project.local
--- brittany-0.14.0.2/cabal.project.local	1969-12-31 17:00:00.000000000 -0700
+++ brittany-0.14.0.2-r1/cabal.project.local	2022-08-25 22:48:47.318726494 -0600
@@ -0,0 +1,2 @@
+ignore-project: False
+tests: True
diff -urN brittany-0.14.0.2/source/test-suite/Main.hs brittany-0.14.0.2-r1/source/test-suite/Main.hs
--- brittany-0.14.0.2/source/test-suite/Main.hs	2001-09-08 19:46:40.000000000 -0600
+++ brittany-0.14.0.2-r1/source/test-suite/Main.hs	2022-08-25 23:03:30.888726922 -0600
@@ -9,6 +9,7 @@
 main = Hspec.hspec . Hspec.parallel $ do
   let directory = "data"
   entries <- Hspec.runIO $ Directory.listDirectory directory
+  Hspec.runIO $ Directory.createDirectory "output"
   Monad.forM_ (List.sort entries) $ \entry ->
     case FilePath.stripExtension "hs" entry of
       Nothing -> pure ()
