diff -uraN encoding-0.6.5//Data/Encoding/ByteSource.hs encoding-0.6.6//Data/Encoding/ByteSource.hs
--- encoding-0.6.5//Data/Encoding/ByteSource.hs	2012-04-05 14:21:19.951389002 +0300
+++ encoding-0.6.6//Data/Encoding/ByteSource.hs	2012-04-05 14:21:21.895389002 +0300
@@ -177,11 +177,11 @@
       pos <- liftIO $ hGetPosn h
       res <- act
       liftIO $ hSetPosn pos
-      return res{-
+      return res
     sourcePos = do
       h <- ask
       p <- liftIO $ hTell h
-      return $ Just p-}
+      return $ Just p
 
 {-
 instance Throws DecodingException (State st) => Throws DecodingException (State (Integer,st)) where
