diff --git a/Bundled/SHA256.hs b/Bundled/SHA256.hs
index d1fec17..ad24800 100644
--- a/Bundled/SHA256.hs
+++ b/Bundled/SHA256.hs
@@ -8,11 +8,12 @@ module Bundled.SHA256 ( sha256 ) where
 
 import Foreign
 import Foreign.C.Types
+import qualified System.IO.Unsafe as U
 import Data.ByteString.Unsafe (unsafeUseAsCStringLen)
 import qualified Data.ByteString.Internal as BSI
 
 sha256 :: BSI.ByteString -> BSI.ByteString
-sha256 p = unsafePerformIO $ do
+sha256 p = U.unsafePerformIO $ do
              digest <- BSI.create 32 $ \digest ->
                        unsafeUseAsCStringLen p $ \(ptr,n) ->
                            c_sha256 ptr (fromIntegral n) digest
