diff -urN gtk-sni-tray-0.1.8.0/gtk-sni-tray.cabal gtk-sni-tray-0.1.8.0-r1/gtk-sni-tray.cabal
--- gtk-sni-tray-0.1.8.0/gtk-sni-tray.cabal	2022-02-22 16:34:07.486608237 -0700
+++ gtk-sni-tray-0.1.8.0-r1/gtk-sni-tray.cabal	2022-02-22 16:38:20.286608359 -0700
@@ -62,7 +62,13 @@
     , unix
   default-language: Haskell2010
 
+flag executable
+  description: Build the gtk-sni-tray-standalone executable
+  default: False
+
 executable gtk-sni-tray-standalone
+  if !flag(executable)
+      buildable: False
   main-is: Main.hs
   other-modules:
       Paths_gtk_sni_tray
