--- GLUT-2.3.0.0-orig/GLUT.cabal	2012-02-06 09:36:33.000000000 +1100
+++ GLUT-2.3.0.0/GLUT.cabal	2012-09-16 11:19:04.044485991 +1000
@@ -150,7 +150,7 @@
    c-sources:
       cbits/HsGLUT.c
    if flag(split-base)
-      build-depends: base >= 3 && < 5, array >= 0.3 && < 0.5, containers >= 0.3 && < 0.5
+      build-depends: base >= 3 && < 5, array >= 0.3 && < 0.5, containers >= 0.3 && < 0.6
    else
       build-depends: base < 3
    build-depends: OpenGL == 2.5.*, OpenGLRaw == 1.2.*, StateVar == 1.0.*, Tensor == 1.0.*
--- GLUT-2.3.0.0-orig/Graphics/UI/GLUT/Initialization.hs	2012-02-06 09:36:33.000000000 +1100
+++ GLUT-2.3.0.0/Graphics/UI/GLUT/Initialization.hs	2012-09-16 11:22:23.593849280 +1000
@@ -331,7 +331,7 @@
 handleMultisampling spps (WithSamplesPerPixel spp) = (spp : spps, Multisampling)
 handleMultisampling spps mode                      = (spps, mode)
 
-toBitfield :: (Bits b) => (a -> b) -> [a] -> b
+toBitfield :: (Bits b, Num b) => (a -> b) -> [a] -> b
 toBitfield marshal = foldl (.|.) 0 . map marshal
 
 -- | Contains 'True' if the /current display mode/ is supported, 'False'
