diff --git a/lib/Network/Protocol/TLS/GNU.hs b/lib/Network/Protocol/TLS/GNU.hs
index 3d4eab3..5751506 100644
--- a/lib/Network/Protocol/TLS/GNU.hs
+++ b/lib/Network/Protocol/TLS/GNU.hs
@@ -63,0 +64,3 @@ import qualified Network.Protocol.TLS.GNU.Foreign as F
+import           Control.Monad
+import           Control.Applicative
+
@@ -97,0 +101,4 @@ instance Functor TLS where
+instance Applicative TLS where
+    pure  = return
+    (<*>) = ap  -- defined in Control.Monad
+
diff --git a/lib/Network/Protocol/TLS/GNU/ErrorT.hs b/lib/Network/Protocol/TLS/GNU/ErrorT.hs
index 4f90510..7bf98ff 100644
--- a/lib/Network/Protocol/TLS/GNU/ErrorT.hs
+++ b/lib/Network/Protocol/TLS/GNU/ErrorT.hs
@@ -29,0 +30,2 @@ import           Control.Monad.Reader (EnvType)
+import           Control.Monad
+import           Control.Applicative
@@ -37,0 +40,4 @@ instance Functor m => Functor (ErrorT e m) where
+instance Monad m => Applicative (ErrorT e m) where
+    pure  = return
+    (<*>) = ap  -- defined in Control.Monad
+
