diff --git a/hspec-expectations-lens.cabal b/hspec-expectations-lens.cabal
index e9cc52a..6a542b8 100644
--- a/hspec-expectations-lens.cabal
+++ b/hspec-expectations-lens.cabal
@@ -32,7 +32,8 @@ library
       base                    == 4.*
     , hspec-expectations      >= 0.5
     , hspec                   >= 1.8
-    , lens                    >= 4.0
+    , lens                    >= 4.7
+    , lens-action
     , HUnit
   exposed-modules:
     Test.Hspec.Expectations.Lens
diff --git a/src/Test/Hspec/Expectations/Lens.hs b/src/Test/Hspec/Expectations/Lens.hs
index a48c73e..52839a7 100644
--- a/src/Test/Hspec/Expectations/Lens.hs
+++ b/src/Test/Hspec/Expectations/Lens.hs
@@ -12,6 +12,7 @@ module Test.Hspec.Expectations.Lens
   ) where
 
 import Control.Lens
+import Control.Lens.Action
 import Control.Monad
 import Control.Exception (SomeException)
 import Control.Exception.Lens
