--- a/src/Text/Pandoc/Readers/Haddock.hs
+++ b/src/Text/Pandoc/Readers/Haddock.hs
@@ -129,7 +129,11 @@ docHToInlines isCode d' =
           DocIdentifier s -> B.codeWith ("",["haskell","identifier"],[]) $ T.pack s
           _               -> mempty
     DocIdentifierUnchecked s -> B.codeWith ("",["haskell","identifier"],[]) $ T.pack s
+#if MIN_VERSION_haddock_library(1,10,0)
+    DocModule (ModLink s _) -> B.codeWith ("",["haskell","module"],[]) $ T.pack s
+#else
     DocModule s -> B.codeWith ("",["haskell","module"],[]) $ T.pack s
+#endif
     DocWarning _ -> mempty -- TODO
     DocEmphasis d -> B.emph (docHToInlines isCode d)
     DocMonospaced (DocString s) -> B.code $ T.pack s
