From ca583a340fdb721baeadb2657c184490c73363e4 Mon Sep 17 00:00:00 2001
From: John MacFarlane <fiddlosopher@gmail.com>
Date: Sat, 20 Oct 2012 23:13:49 -0700
Subject: [PATCH] Updated lhs-test.latex for new highlighting-kate treatment
 of `.

---
 tests/lhs-test.latex | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/lhs-test.latex b/tests/lhs-test.latex
index d2d1f9a..acfb866 100644
--- a/tests/lhs-test.latex
+++ b/tests/lhs-test.latex
@@ -70,7 +70,7 @@ return a single value:
 \begin{Highlighting}[]
 \OtherTok{unsplit ::} \NormalTok{(}\DataTypeTok{Arrow} \NormalTok{a) }\OtherTok{=>} \NormalTok{(b }\OtherTok{->} \NormalTok{c }\OtherTok{->} \NormalTok{d) }\OtherTok{->} \NormalTok{a (b, c) d}
 \NormalTok{unsplit }\FunctionTok{=} \NormalTok{arr }\FunctionTok{.} \FunctionTok{uncurry}       
-          \CommentTok{-- arr (\textbackslash{}op (x,y) -> x `op` y) }
+          \CommentTok{-- arr (\textbackslash{}op (x,y) -> x {\char18}op{\char18} y) }
 \end{Highlighting}
 \end{Shaded}
 
-- 
1.7.12.4

