diff --git a/tests/Tests/Helpers.hs b/tests/Tests/Helpers.hs
index 66879ef..c589e59 100644
--- a/tests/Tests/Helpers.hs
+++ b/tests/Tests/Helpers.hs
@@ -62,15 +62,22 @@ test fn name (input, expected) =
            actual' = toString $ fn input
            expected' = toString expected
            diff = getDiff (lines expected') (lines actual')
-           expected'' = unlines $ map vividize $ filter (\(d,_) -> d /= S) diff
-           actual''   = unlines $ map vividize $ filter (\(d,_) -> d /= F) diff
+           expected'' = unlines $ map vividize $ filter (not . isS) diff
+           actual''   = unlines $ map vividize $ filter (not . isF) diff
            dashes "" = replicate 72 '-'
            dashes x  = replicate (72 - length x - 5) '-' ++ " " ++ x ++ " ---"
 
-vividize :: (DI,String) -> String
-vividize (B,s) = s
-vividize (F,s) = s
-vividize (S,s) = setSGRCode [SetColor Background Dull Red
+isS :: Diff String -> Bool
+isS (Second s) = True
+isS _          = False
+isF :: Diff String -> Bool
+isF (First s)  = True
+isF _          = False
+
+vividize :: (Diff String) -> String
+vividize (Both s _s) = s
+vividize (First s) = s
+vividize (Second s) = setSGRCode [SetColor Background Dull Red
                          , SetColor Foreground Vivid White] ++ s
                  ++ setSGRCode [Reset]
 
diff --git a/tests/Tests/Old.hs b/tests/Tests/Old.hs
index 196fa45..803091e 100644
--- a/tests/Tests/Old.hs
+++ b/tests/Tests/Old.hs
@@ -26,7 +26,7 @@ pandocPath = ".." </> "dist" </> "build" </> "pandoc" </> "pandoc"
 
 data TestResult = TestPassed
                 | TestError ExitCode
-                | TestFailed String FilePath [(DI, String)]
+                | TestFailed String FilePath [(Diff String)]
      deriving (Eq)
 
 instance Show TestResult where
@@ -38,13 +38,13 @@ instance Show TestResult where
                                  dash
     where dash = replicate 72 '-'
 
-showDiff :: (Int,Int) -> [(DI, String)] -> String
+showDiff :: (Int,Int) -> [(Diff String)] -> String
 showDiff _ []             = ""
-showDiff (l,r) ((F, ln) : ds) =
+showDiff (l,r) ((First ln) : ds) =
   printf "+%4d " l ++ ln ++ "\n" ++ showDiff (l+1,r) ds
-showDiff (l,r) ((S, ln) : ds) =
+showDiff (l,r) ((Second ln) : ds) =
   printf "-%4d " r ++ ln ++ "\n" ++ showDiff (l,r+1) ds
-showDiff (l,r) ((B, _ ) : ds) =
+showDiff (l,r) ((Both _ _) : ds) =
   showDiff (l+1,r+1) ds
 
 tests :: [Test]
