diff --git a/src/Install/Fetch.hs b/src/Install/Fetch.hs
index 6d83ec0..158c78a 100644
--- a/src/Install/Fetch.hs
+++ b/src/Install/Fetch.hs
@@ -98,10 +98,10 @@ toDoc (Result name version either) =
     bullet =
       case either of
         Right _ ->
-          green (text "●")
+          green (text "[ok]")
 
         Left _ ->
-          red (text "✗")
+          red (text "[fail]")
   in
     text "  " <> bullet <+> nameDoc <+> versionDoc <> text "\n"
 
