commit bacb59e09a45c7d565f259ea9bad9b6ca9d0069e
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 09:55:04 2019 +0100

    Update Gtk2HsSetup.hs

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index c3c2109a..5901bb3c 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -10,6 +10,8 @@ module Gtk2HsSetup (
   c2hsLocal
   ) where
 
+import Data.Maybe (mapMaybe)
+import Distribution.Pretty (prettyShow)
 import Distribution.Simple
 import Distribution.Simple.PreProcess
 import Distribution.InstalledPackageInfo ( importDirs,
@@ -55,7 +57,7 @@ import Distribution.Simple.Install ( install )
 import Distribution.Simple.Register ( generateRegistrationInfo, registerPackage )
 import Distribution.Text ( simpleParse, display )
 import System.FilePath
-import System.Exit (exitFailure)
+import System.Exit (die, exitFailure)
 import System.Directory ( doesFileExist, getDirectoryContents, doesDirectoryExist )
 import Distribution.Version (Version(..))
 import Distribution.Verbosity
@@ -90,13 +92,13 @@ onDefaultSearchPath f a b = f a b defaultProgramSearchPath
 #if MIN_VERSION_Cabal(2,5,0)
 componentsConfigs :: LocalBuildInfo -> [(LBI.ComponentName, ComponentLocalBuildInfo, [LBI.ComponentName])]
 componentsConfigs lbi =
-    [ (componentLocalName clbi,
+    [ (LBI.componentLocalName clbi,
        clbi,
-       mapMaybe (fmap componentLocalName . flip Graph.lookup g)
-                (componentInternalDeps clbi))
+       mapMaybe (fmap LBI.componentLocalName . flip Graph.lookup g)
+                (LBI.componentInternalDeps clbi))
     | clbi <- Graph.toList g ]
   where
-    g = componentGraph lbi
+    g = LBI.componentGraph lbi
 
 libraryConfig lbi = case [clbi | (LBI.CLibName _, clbi, _) <- componentsConfigs lbi] of
 #else
@@ -190,7 +192,7 @@ registerHook pkg_descr localbuildinfo _ flags =
 
 getComponentLocalBuildInfo :: LocalBuildInfo -> LBI.ComponentName -> ComponentLocalBuildInfo
 getComponentLocalBuildInfo lbi cname =
-    case componentNameCLBIs lbi cname of
+    case LBI.componentNameCLBIs lbi cname of
       [clbi] -> clbi
       [] ->
           error $ "internal error: there is no configuration data "
@@ -198,7 +200,7 @@ getComponentLocalBuildInfo lbi cname =
       clbis ->
           error $ "internal error: the component name " ++ show cname
                ++ "is ambiguous.  Refers to: "
-               ++ intercalate ", " (map (prettyShow . componentUnitId) clbis)
+               ++ intercalate ", " (map (prettyShow . LBI.componentUnitId) clbis)
 
 register :: PackageDescription -> LocalBuildInfo
          -> RegisterFlags -- ^Install in the user's database?; verbose
@@ -381,7 +383,7 @@ genSynthezisedFiles verb pd lbi = do
       genFile :: ([String] -> IO String) -> [ProgArg] -> FilePath -> IO ()
       genFile prog args outFile = do
          res <- prog args
-         rewriteFile outFile res
+         rewriteFileEx verb outFile res
 
   forM_ (filter (\(tag,_) -> "x-types-" `isPrefixOf` tag && "file" `isSuffixOf` tag) xList) $
     \(fileTag, f) -> do
