commit 811f7e5e9d501d015560117d4661c5428273e1c7
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 09:49:41 2019 +0100

    Update Gtk2HsSetup.hs

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index b43d4c9d..c3c2109a 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -88,7 +88,7 @@ versionNumbers = versionBranch
 
 onDefaultSearchPath f a b = f a b defaultProgramSearchPath
 #if MIN_VERSION_Cabal(2,5,0)
-componentsConfigs :: LocalBuildInfo -> [(ComponentName, ComponentLocalBuildInfo, [ComponentName])]
+componentsConfigs :: LocalBuildInfo -> [(LBI.ComponentName, ComponentLocalBuildInfo, [LBI.ComponentName])]
 componentsConfigs lbi =
     [ (componentLocalName clbi,
        clbi,
@@ -188,12 +188,24 @@ registerHook pkg_descr localbuildinfo _ flags =
            "Package contains no library to register:" (packageId pkg_descr)
   where verbosity = fromFlag (regVerbosity flags)
 
+getComponentLocalBuildInfo :: LocalBuildInfo -> LBI.ComponentName -> ComponentLocalBuildInfo
+getComponentLocalBuildInfo lbi cname =
+    case componentNameCLBIs lbi cname of
+      [clbi] -> clbi
+      [] ->
+          error $ "internal error: there is no configuration data "
+               ++ "for component " ++ show cname
+      clbis ->
+          error $ "internal error: the component name " ++ show cname
+               ++ "is ambiguous.  Refers to: "
+               ++ intercalate ", " (map (prettyShow . componentUnitId) clbis)
+
 register :: PackageDescription -> LocalBuildInfo
          -> RegisterFlags -- ^Install in the user's database?; verbose
          -> IO ()
 register pkg@PackageDescription { library       = Just lib  } lbi regFlags
   = do
-    let clbi = LBI.getComponentLocalBuildInfo lbi
+    let clbi = getComponentLocalBuildInfo lbi
 #if MIN_VERSION_Cabal(2,5,0)
                    (LBI.CLibName $ PD.libName lib)
 #else
