diff -ru gtk2hs-buildtools-0.13.3.1.orig/c2hs/c/CPretty.hs gtk2hs-buildtools-0.13.3.1/c2hs/c/CPretty.hs
--- gtk2hs-buildtools-0.13.3.1.orig/c2hs/c/CPretty.hs	2017-11-25 18:58:05.000000000 -0500
+++ gtk2hs-buildtools-0.13.3.1/c2hs/c/CPretty.hs	2018-02-26 12:06:21.956401282 -0500
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 --  C->Haskell Compiler: pretty printing of C abstract syntax
 --
 --  Author : Manuel M T Chakravarty
@@ -34,6 +35,9 @@
   -- we are just providing instances to the class `Pretty'
 ) where
 
+#if MIN_VERSION_base(4,11,0)
+import Prelude hiding ((<>))
+#endif
 import Idents (Ident, identToLexeme)
 import Text.PrettyPrint.HughesPJ
 
diff -ru gtk2hs-buildtools-0.13.3.1.orig/gtk2hs-buildtools.cabal gtk2hs-buildtools-0.13.3.1/gtk2hs-buildtools.cabal
--- gtk2hs-buildtools-0.13.3.1.orig/gtk2hs-buildtools.cabal	2017-11-25 18:58:05.000000000 -0500
+++ gtk2hs-buildtools-0.13.3.1/gtk2hs-buildtools.cabal	2018-02-26 12:06:21.956401282 -0500
@@ -40,10 +40,10 @@
         build-depends:   base >= 4 && < 5,
                          process, array, pretty,
                          filepath, random,
-                         Cabal >= 1.24.0.0 && < 2.1,
-                         filepath >= 1.3.0.0 && < 1.5,
-                         directory >= 1.2.0.0 && < 1.4,
-                         containers >= 0.5.5.1 && < 0.6
+                         Cabal >= 1.24.0.0,
+                         filepath >= 1.3.0.0,
+                         directory >= 1.2.0.0,
+                         containers >= 0.5.5.1
         if impl(ghc >= 7.7)
           build-depends: hashtables
         build-tools:     alex >= 3.0.1, happy >= 1.18.9
