commit 521d0ae377a2196dbcf2c4e17102cf3b458e025e
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 09:36:01 2019 +0100

    Support Cabal 3

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index 8fb19de5..4e61594c 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -23,7 +23,7 @@ import Distribution.PackageDescription as PD ( PackageDescription(..),
                                                BuildInfo(..),
                                                emptyBuildInfo, allBuildInfo,
                                                Library(..),
-                                               libModules, hasLibs)
+                                               explicitLibModules, hasLibs)
 import Distribution.Simple.LocalBuildInfo (LocalBuildInfo(withPackageDB, buildDir, localPkgDescr, installedPkgs, withPrograms),
                                            InstallDirs(..),
                                            ComponentLocalBuildInfo,
@@ -34,7 +34,7 @@ import Distribution.Simple.LocalBuildInfo (LocalBuildInfo(withPackageDB, buildDi
 import Distribution.Simple.Compiler  ( Compiler(..) )
 import Distribution.Simple.Program (
   Program(..), ConfiguredProgram(..),
-  rawSystemProgramConf, rawSystemProgramStdoutConf, programName, programPath,
+  runDbProgram, runDbProgramOutput, programName, programPath,
   c2hsProgram, pkgConfigProgram, gccProgram, requireProgram, ghcPkgProgram,
   simpleProgram, lookupProgram, rawSystemProgramStdout, ProgArg)
 #if MIN_VERSION_Cabal(2,0,0)
@@ -49,7 +49,9 @@ import Distribution.Simple.Setup (CopyFlags(..), InstallFlags(..), CopyDest(..),
                                   defaultCopyFlags, ConfigFlags(configVerbosity),
                                   fromFlag, toFlag, RegisterFlags(..), flagToMaybe,
                                   fromFlagOrDefault, defaultRegisterFlags)
+#if MIN_VERSION_Cabal(2,0,0)
 import Distribution.Simple.BuildPaths ( autogenModulesDir )
+#endif
 import Distribution.Simple.Install ( install )
 import Distribution.Simple.Register ( generateRegistrationInfo, registerPackage )
 import Distribution.Text ( simpleParse, display )
@@ -239,12 +241,17 @@ register _ _ regFlags = notice verbosity "No package to register"
 -- This is a hack for Cabal-1.8, It is not needed in Cabal-1.9.1 or later
 ------------------------------------------------------------------------------
 
+#if MIN_VERSION_Cabal(2,0,0)
+adjustLocalBuildInfo :: LocalBuildInfo -> LocalBuildInfo
+adjustLocalBuildInfo = id
+#else
 adjustLocalBuildInfo :: LocalBuildInfo -> LocalBuildInfo
 adjustLocalBuildInfo lbi =
   let extra = (Just libBi, [])
       libBi = emptyBuildInfo { includeDirs = [ autogenModulesDir lbi
                                              , buildDir lbi ] }
    in lbi { localPkgDescr = updatePackageDescription extra (localPkgDescr lbi) }
+#endif
 
 ------------------------------------------------------------------------------
 -- Processing .chs files with our local c2hs.
@@ -306,7 +313,7 @@ installCHI pkg@PD.PackageDescription { library = Just lib } lbi verbosity copyde
   -- cannot use the recommended 'findModuleFiles' since it fails if there exists
   -- a modules that does not have a .chi file
   mFiles <- mapM (findFileWithExtension' ["chi"] [buildDir lbi] . toFilePath)
-                   (PD.libModules lib)
+                   (PD.explicitLibModules lib)
 
   let files = [ f | Just f <- mFiles ]
   installOrdinaryFiles verbosity libPref files
@@ -408,7 +415,7 @@ getPkgConfigPackages verbosity lbi pkg =
 #endif
     <- concatMap pkgconfigDepends (allBuildInfo pkg) ]
   where
-    pkgconfig = rawSystemProgramStdoutConf verbosity
+    pkgconfig = runDbProgramOutput verbosity
                   pkgConfigProgram (withPrograms lbi)
 
 ------------------------------------------------------------------------------
