commit 266214a0a2e529785401420b7c03f3bcfc8aa8a5
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 09:45:50 2019 +0100

    Update Gtk2HsSetup.hs

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index 78afcd14..b43d4c9d 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -41,7 +41,6 @@ import Distribution.Simple.Program (
 import Distribution.Simple.Program.HcPkg ( defaultRegisterOptions )
 import Distribution.Types.PkgconfigDependency ( PkgconfigDependency(..) )
 import Distribution.Types.PkgconfigName
-import qualified Distribution.Types.LocalBuildInfo as LBI (componentsConfigs)  -- TODO will be removed in Cabal 2.2
 #endif
 import Distribution.ModuleName ( ModuleName, components, toFilePath )
 import Distribution.Simple.Utils
@@ -72,6 +71,7 @@ import qualified Distribution.Simple.LocalBuildInfo as LBI
 import qualified Distribution.InstalledPackageInfo as IPI
        (installedUnitId)
 import Distribution.Simple.Compiler (compilerVersion)
+import qualified Distribution.Compat.Graph as Graph
 
 import Control.Applicative ((<$>))
 
