commit 1cccb8544e0cac9a794609a520f392ca90e77999
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 09:39:16 2019 +0100

    Update Gtk2HsSetup.hs

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index 1d35863d..3a79b116 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -36,7 +36,7 @@ import Distribution.Simple.Program (
   Program(..), ConfiguredProgram(..),
   runDbProgram, getDbProgramOutput, programName, programPath,
   c2hsProgram, pkgConfigProgram, gccProgram, requireProgram, ghcPkgProgram,
-  simpleProgram, lookupProgram, rawSystemProgramStdout, ProgArg)
+  simpleProgram, lookupProgram, getProgramOutput, ProgArg)
 #if MIN_VERSION_Cabal(2,0,0)
 import Distribution.Simple.Program.HcPkg ( defaultRegisterOptions )
 import Distribution.Types.PkgconfigDependency ( PkgconfigDependency(..) )
