commit 46cb4f40cad47bb4aa94c063a9f216c2e4b033c2
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Thu Sep 19 10:01:45 2019 +0100

    Update Gtk2HsSetup.hs

diff --git a/src/Gtk2HsSetup.hs b/src/Gtk2HsSetup.hs
index 5901bb3c..797b9554 100644
--- a/src/Gtk2HsSetup.hs
+++ b/src/Gtk2HsSetup.hs
@@ -33,6 +33,7 @@ import Distribution.Simple.LocalBuildInfo (LocalBuildInfo(withPackageDB, buildDi
                                            absoluteInstallDirs,
                                            relocatable,
                                            compiler)
+import Distribution.Types.LocalBuildInfo as LBI (componentNameCLBIs)
 import Distribution.Simple.Compiler  ( Compiler(..) )
 import Distribution.Simple.Program (
   Program(..), ConfiguredProgram(..),
