From b9c56b3aa6ce716a25dc893f62d47535464d787b Mon Sep 17 00:00:00 2001
From: Jack Todaro <solpeth@posteo.org>
Date: Fri, 12 Mar 2021 18:41:05 +1100
Subject: [PATCH] Main.hs: adapt to hledger-lib-1.21

Signed-off-by: Jack Todaro <solpeth@posteo.org>
---
 Main.hs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Main.hs b/Main.hs
index 042da78..3ea1ab9 100644
--- a/Main.hs
+++ b/Main.hs
@@ -97,7 +97,7 @@ main = bracket (return ()) (\() -> hFlush stdout >> hFlush stderr) $ \() -> do
                    []    -> commandLineError "required argument ACCOUNT is missing\n"
                    [acc] -> return acc
                    _     -> commandLineError "only one interest ACCOUNT may be specified\n"
-  let jnl = filterJournalTransactions (Acct (toRegex' interestAcc)) jnl'
+  let jnl = filterJournalTransactions (Acct (toRegex' $ T.pack interestAcc)) jnl'
       ts  = sortOn tdate (jtxns jnl)
       cfg = Config
             { interestAccount = T.pack interestAcc
@@ -114,4 +114,4 @@ main = bracket (return ()) (\() -> hFlush stdout >> hFlush stderr) $ \() -> do
       result
         | optVerbose opts = ts' ++ ts
         | otherwise       = ts'
-  mapM_ (putStr . showTransactionUnelided) (sortOn tdate result)
+  mapM_ (putStr . T.unpack . showTransaction) (sortOn tdate result)
-- 
2.30.2

