diff --git a/Network/Wreq/Internal.hs b/Network/Wreq/Internal.hs
index 7e37e6c..828342e 100644
--- a/Network/Wreq/Internal.hs
+++ b/Network/Wreq/Internal.hs
@@ -114,3 +114,5 @@ prepare modify opts url = do
                    & setProxy opts
+#if ! MIN_VERSION_http_client(0,5,0)
                    & setCheckStatus opts
+#endif
                    & setRedirects opts
@@ -148,2 +150,3 @@ setProxy = maybe id f . proxy
 
+#if ! MIN_VERSION_http_client(0,5,0)
 setCheckStatus :: Options -> Request -> Request
@@ -151,2 +154,3 @@ setCheckStatus = maybe id f . checkStatus
   where f cs = ( & Lens.checkStatus .~ cs)
+#endif
 
diff --git a/Network/Wreq/Internal/Lens.hs b/Network/Wreq/Internal/Lens.hs
index f292826..ea09960 100644
--- a/Network/Wreq/Internal/Lens.hs
+++ b/Network/Wreq/Internal/Lens.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE RankNTypes, TemplateHaskell #-}
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fno-warn-missing-signatures #-}
@@ -22,4 +23,6 @@ module Network.Wreq.Internal.Lens
     , responseTimeout
+#if ! MIN_VERSION_http_client(0,5,0)
     , checkStatus
     , getConnectionWrapper
+#endif
     , cookieJar
