diff -ru data-dword-0.3.1.2.orig/src/Data/DoubleWord/TH.hs data-dword-0.3.1.2/src/Data/DoubleWord/TH.hs
--- data-dword-0.3.1.2.orig/src/Data/DoubleWord/TH.hs	2018-01-13 13:49:48.000000000 -0500
+++ data-dword-0.3.1.2/src/Data/DoubleWord/TH.hs	2019-05-02 16:32:16.395011833 -0400
@@ -25,6 +25,9 @@
 import Language.Haskell.TH hiding (unpacked, match)
 import Data.BinaryWord (BinaryWord(..))
 import Data.DoubleWord.Base
+#if MIN_VERSION_template_haskell(2,15,0)
+import Data.List (foldl')
+#endif
 
 -- | Declare signed and unsigned binary word types built from
 --   the specified low and high halves. The high halves /must/ have
@@ -1378,8 +1381,12 @@
     hi'  = mkName "hi'"
     lo'  = mkName "lo'"
     tpT  = ConT tp
+
+    tySynInst :: Name -> [Type] -> Type -> Dec
     tySynInst n ps t =
-#if MIN_VERSION_template_haskell(2,9,0)
+#if MIN_VERSION_template_haskell(2,15,0)
+      TySynInstD (TySynEqn Nothing (foldl' AppT (ConT n) ps) t)
+#elif MIN_VERSION_template_haskell(2,9,0)
       TySynInstD n (TySynEqn ps t)
 #else
       TySynInstD n ps t
@@ -1448,11 +1455,19 @@
 #endif
     singE e = appC '(:) [e, ConE '[]]
     mkRules = do
-      let idRule = RuleP ("fromIntegral/" ++ show tp ++ "->" ++ show tp) []
+      let idRule = RuleP ("fromIntegral/" ++ show tp ++ "->" ++ show tp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                         Nothing
+#endif
+                         []
                          (VarE 'fromIntegral)
                          (SigE (VarE 'id) (AppT (AppT ArrowT tpT) tpT))
                          AllPhases
-          signRule = RuleP ("fromIntegral/" ++ show tp ++ "->" ++ show otp) []
+          signRule = RuleP ("fromIntegral/" ++ show tp ++ "->" ++ show otp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                           Nothing
+#endif
+                           []
                            (VarE 'fromIntegral)
                            (SigE (VarE (if signed then 'unsignedWord
                                                   else 'signedWord))
@@ -1464,11 +1479,17 @@
                (VarE 'signExtendLo)
     mkRules' rules t narrowE extE signExtE = do
       let narrowRule = RuleP ("fromIntegral/" ++ show tp ++ "->" ++ showT t)
+#if MIN_VERSION_template_haskell(2,15,0)
+                             Nothing
+#endif
                              []
                              (VarE 'fromIntegral)
                              (SigE narrowE (AppT (AppT ArrowT tpT) t))
                              AllPhases
           extRule = RuleP ("fromIntegral/" ++ showT t ++ "->" ++ show tp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                          Nothing
+#endif
                           []
                           (VarE 'fromIntegral)
                           (SigE extE (AppT (AppT ArrowT t) tpT))
@@ -1476,18 +1497,26 @@
       signedRules ← do
         insts ← reifyInstances ''SignedWord [t]
         case insts of
-#if MIN_VERSION_template_haskell(2,9,0)
+#if MIN_VERSION_template_haskell(2,15,0)
+          [TySynInstD (TySynEqn _ _ signT)] -> return $
+#elif MIN_VERSION_template_haskell(2,9,0)
           [TySynInstD _ (TySynEqn _ signT)] → return $
 #else
           [TySynInstD _ _ signT] → return $
 #endif
             [ RuleP ("fromIntegral/" ++ show tp ++ "->" ++ showT signT)
+#if MIN_VERSION_template_haskell(2,15,0)
+                    Nothing
+#endif
                     []
                     (VarE 'fromIntegral)
                     (SigE (AppE (appVN '(.) ['signedWord]) narrowE)
                           (AppT (AppT ArrowT tpT) signT))
                     AllPhases
             , RuleP ("fromIntegral/" ++ showT signT ++ "->" ++ show tp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                    Nothing
+#endif
                     []
                     (VarE 'fromIntegral)
                     (SigE signExtE (AppT (AppT ArrowT signT) tpT))
@@ -1501,6 +1530,9 @@
                     sSmallT = ConT sSmallName in
                 [ RuleP ("fromIntegral/" ++
                          show tp ++ "->" ++ show uSmallName)
+#if MIN_VERSION_template_haskell(2,15,0)
+                        Nothing
+#endif
                         []
                         (VarE 'fromIntegral)
                         (SigE (appV '(.) [VarE 'fromIntegral, narrowE])
@@ -1508,6 +1540,9 @@
                         AllPhases
                 , RuleP ("fromIntegral/" ++
                          show uSmallName ++ "->" ++ show tp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                        Nothing
+#endif
                         []
                         (VarE 'fromIntegral)
                         (SigE (appV '(.) [extE, VarE 'fromIntegral])
@@ -1515,6 +1550,9 @@
                         AllPhases
                 , RuleP ("fromIntegral/" ++
                          show tp ++ "->" ++ show sSmallName)
+#if MIN_VERSION_template_haskell(2,15,0)
+                        Nothing
+#endif
                         []
                         (VarE 'fromIntegral)
                         (SigE (appV '(.) [VarE 'fromIntegral, narrowE])
@@ -1522,6 +1560,9 @@
                         AllPhases
                 , RuleP ("fromIntegral/" ++
                          show sSmallName ++ "->" ++ show tp)
+#if MIN_VERSION_template_haskell(2,15,0)
+                        Nothing
+#endif
                         []
                         (VarE 'fromIntegral)
                         (SigE (appV '(.) [signExtE, VarE 'fromIntegral])
@@ -1532,7 +1573,9 @@
         _ → do
           insts ← reifyInstances ''LoWord [t]
           case insts of
-#if MIN_VERSION_template_haskell(2,9,0)
+#if MIN_VERSION_template_haskell(2,15,0)
+            [TySynInstD (TySynEqn _ _ t')] ->
+#elif MIN_VERSION_template_haskell(2,9,0)
             [TySynInstD _ (TySynEqn _ t')] →
 #else
             [TySynInstD _ _ t'] →
