commit 25b8d00a149098d9b7842600dbb93f40836b4546
Author: Ben Gamari <ben@smart-cactus.org>
Date:   Mon Feb 8 13:27:06 2016 +0100

    Fix GHC and haddock-library dependency bounds

diff --git a/haddock.cabal b/haddock.cabal
index 1d7a9fe..11bf386 100644
--- a/haddock.cabal
+++ b/haddock.cabal
@@ -123,7 +123,7 @@ executable haddock
       Haddock.Syb
       Haddock.Convert
   else
-    build-depends:  haddock-api == 2.16.*
+    build-depends:  haddock-api == 2.17.*
 
 test-suite driver-test
   type:             exitcode-stdio-1.0

