--- Chart-diagrams-1.9-orig/Graphics/Rendering/Chart/Backend/Diagrams.hs	2018-06-04 08:44:46.000000000 +1000
+++ Chart-diagrams-1.9/Graphics/Rendering/Chart/Backend/Diagrams.hs	2018-10-12 21:33:51.392752954 +1100
@@ -2,6 +2,7 @@
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE CPP #-}
 
 -- | The backend to render charts with the diagrams library.
 module Graphics.Rendering.Chart.Backend.Diagrams
@@ -568,7 +569,11 @@
       , F.textHeight = scaledH -- _font_size fs
       }
 
+#if MIN_VERSION_SVGFonts(1,7,0)
+fontFromName :: (Read n, RealFloat n) => String -> IO (F.PreparedFont n)
+#else
 fontFromName :: (Read n, RealFloat n) => String -> F.PreparedFont n
+#endif
 fontFromName name = case name of
   "serif" -> F.lin
   "monospace" -> F.bit
