--- base16-bytestring-0.1.1.3-orig/Data/ByteString/Base16.hs	2011-11-18 08:02:45.000000000 +1100
+++ base16-bytestring-0.1.1.3/Data/ByteString/Base16.hs	2011-11-21 01:03:54.799339019 +1100
@@ -116,7 +116,9 @@
           poke d . fromIntegral $ lo + (hi `shiftL` 4)
           go (s `plusPtr` 2) (d `plusPtr` 1)
 
-    hex (I# index) = W8# $ indexWord8OffAddr# table index
+    -- for ghc 6.12.3 bos already fixed it upstream: 
+    -- https://github.com/bos/base16-bytestring/commit/83d42a09eb90c8a5c47d6c41b8d2845f16c239f7
+    hex (I# index) = W8# (indexWord8OffAddr# table index)
     !table =
         "\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\
         \\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\
