--- a/Database/ContentIdentifier.hs
+++ b/Database/ContentIdentifier.hs
@@ -15,6 +15,8 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE StandaloneDeriving #-}
 #endif
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Database.ContentIdentifier (
 	ContentIdentifierHandle,
--- a/Database/Export.hs
+++ b/Database/Export.hs
@@ -15,6 +15,8 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE StandaloneDeriving #-}
 #endif
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Database.Export (
 	ExportHandle,
--- a/Database/Fsck.hs
+++ b/Database/Fsck.hs
@@ -15,6 +15,8 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE StandaloneDeriving #-}
 #endif
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Database.Fsck (
 	FsckHandle,
--- a/Database/Keys/SQL.hs
+++ b/Database/Keys/SQL.hs
@@ -15,6 +15,8 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE StandaloneDeriving #-}
 #endif
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Database.Keys.SQL where
 
