--- git-annex-3.20120406-orig/git-annex.cabal	2012-04-08 05:59:20.000000000 +1000
+++ git-annex-3.20120406/git-annex.cabal	2012-04-12 14:25:16.951181338 +1000
@@ -38,6 +38,11 @@
 
 Executable git-annex-shell
   Main-Is: git-annex-shell.hs
+  Build-Depends: MissingH, hslogger, directory, filepath,
+   unix, containers, utf8-string, network, mtl, bytestring, old-locale, time,
+   pcre-light, extensible-exceptions, dataenc, SHA, process, hS3, json, HTTP,
+   base >= 4.5, base < 5, monad-control, transformers-base, lifted-base,
+   IfElse, text, QuickCheck >= 2.1, bloomfilter
   C-Sources: Utility/diskfree.c
 
 Test-Suite test
