From 26baae8967d55d0793d0609475a75d265bdd64a3 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Tue, 17 Sep 2013 11:54:09 -0400
Subject: [PATCH] fix build with haskell DNS 1.0.0

---
 Utility/SRV.hs | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/Utility/SRV.hs b/Utility/SRV.hs
index 0a77191..a2ee704 100644
--- a/Utility/SRV.hs
+++ b/Utility/SRV.hs
@@ -67,8 +67,14 @@ lookupSRV (SRV srv) = initResolver [] $ \resolver -> do
 lookupSRV (SRV srv) = do
 	seed <- makeResolvSeed defaultResolvConf
 	r <- withResolver seed $ flip DNS.lookupSRV $ B8.fromString srv
-	return $ maybe [] (orderHosts . map tohosts) r
+	return $
+#if MIN_VERSION_dns(1,0,0)
+		either (const []) use r
+#else
+		maybe [] use r
+#endif
   where
+  	use = orderHosts . map tohosts
 	tohosts (priority, weight, port, hostname) =
 		( (priority, weight)
 		, (B8.toString hostname, PortNumber $ fromIntegral port)
-- 
1.8.3.2

