--- a/Database/ContentIdentifier.hs
+++ b/Database/ContentIdentifier.hs
@@ -10,6 +10,8 @@
 {-# LANGUAGE MultiParamTypeClasses, GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 
 module Database.ContentIdentifier (
 	ContentIdentifierHandle,
--- a/Database/Export.hs
+++ b/Database/Export.hs
@@ -10,6 +10,8 @@
 {-# LANGUAGE MultiParamTypeClasses, GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 
 module Database.Export (
 	ExportHandle,
--- a/Database/Fsck.hs
+++ b/Database/Fsck.hs
@@ -10,6 +10,8 @@
 {-# LANGUAGE MultiParamTypeClasses, GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 
 module Database.Fsck (
 	FsckHandle,
--- a/Database/Keys/SQL.hs
+++ b/Database/Keys/SQL.hs
@@ -10,6 +10,8 @@
 {-# LANGUAGE MultiParamTypeClasses, GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE RankNTypes, ScopedTypeVariables #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 
 module Database.Keys.SQL where
 
