From 018b5b81736a321f3eb9762a2afb7124e19dbdf9 Mon Sep 17 00:00:00 2001

diff --git a/Utility/Tor.hs b/Utility/Tor.hs
index 427fb100b..094c91aac 100644
--- a/Utility/Tor.hs
+++ b/Utility/Tor.hs
@@ -5,6 +5,8 @@
  - Licensed under the GNU AGPL version 3 or higher.
  -}
 
+{-# LANGUAGE CPP #-}
+
 module Utility.Tor where
 
 import Common
@@ -37,7 +39,12 @@ connectHiddenService (OnionAddress address) port = do
 	return s
   where
 	torsocksport = 9050
+#if MIN_VERSION_socks(0,6,0)
+	torsockconf = defaultSocksConf $ SockAddrInet torsocksport $
+		tupleToHostAddress (127,0,0,1)
+#else
 	torsockconf = defaultSocksConf "127.0.0.1" torsocksport
+#endif
 	socksdomain = SocksAddrDomainName (BU8.fromString address)
 	socksaddr = SocksAddress socksdomain (fromIntegral port)
