diff --git a/src/Data/MessagePack/Get.hs b/src/Data/MessagePack/Get.hs
index 2e36b9b..2801070 100644
--- a/src/Data/MessagePack/Get.hs
+++ b/src/Data/MessagePack/Get.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE CPP #-}
 
@@ -121,2 +122,3 @@ getExt = do
 
+#if ! MIN_VERSION_binary(0,8,0)
 getInt8 :: Get Int8
@@ -132,2 +134,3 @@ getInt64be :: Get Int64
 getInt64be = fromIntegral <$> getWord64be
+#endif
 
