From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

lens-5 removes `itraversed` from class `TraversableWithIndex`

diff -urN patch-0.0.5.2/src/Data/Patch/IntMap.hs patch-0.0.5.2-r1/src/Data/Patch/IntMap.hs
--- patch-0.0.5.2/src/Data/Patch/IntMap.hs	2001-09-08 19:46:40.000000000 -0600
+++ patch-0.0.5.2-r1/src/Data/Patch/IntMap.hs	2022-02-17 21:59:41.207966780 -0700
@@ -52,7 +52,9 @@
 instance FoldableWithIndex Int PatchIntMap
 instance TraversableWithIndex Int PatchIntMap where
   itraverse = itraversed . Indexed
+#if !MIN_VERSION_lens(5,0,0)
   itraversed = _Wrapped .> itraversed <. traversed
+#endif
 
 -- | Map a function @Int -> a -> b@ over all @a@s in the given @'PatchIntMap' a@
 -- (that is, all inserts/updates), producing a @PatchIntMap b@.
diff -urN patch-0.0.5.2/src/Data/Patch/Map.hs patch-0.0.5.2-r1/src/Data/Patch/Map.hs
--- patch-0.0.5.2/src/Data/Patch/Map.hs	2001-09-08 19:46:40.000000000 -0600
+++ patch-0.0.5.2-r1/src/Data/Patch/Map.hs	2022-02-17 22:00:00.787966789 -0700
@@ -63,7 +63,9 @@
 instance FoldableWithIndex k (PatchMap k)
 instance TraversableWithIndex k (PatchMap k) where
   itraverse = itraversed . Indexed
+#if !MIN_VERSION_lens(5,0,0)
   itraversed = _Wrapped .> itraversed <. traversed
+#endif
 
 -- | Returns all the new elements that will be added to the 'Map'
 patchMapNewElements :: PatchMap k v -> [v]
diff -urN patch-0.0.5.2/src/Data/Patch/MapWithMove.hs patch-0.0.5.2-r1/src/Data/Patch/MapWithMove.hs
--- patch-0.0.5.2/src/Data/Patch/MapWithMove.hs	2001-09-08 19:46:40.000000000 -0600
+++ patch-0.0.5.2-r1/src/Data/Patch/MapWithMove.hs	2022-02-17 21:57:01.917966702 -0700
@@ -134,10 +134,12 @@
 instance FoldableWithIndex k (PatchMapWithMove k)
 instance TraversableWithIndex k (PatchMapWithMove k) where
   itraverse = itraversed . Indexed
+#if !MIN_VERSION_lens(5,0,0)
   itraversed =
     _PatchMapWithMove .>
     itraversed <.
     traverse
+#endif
 
 -- | Create a 'PatchMapWithMove', validating it
 patchMapWithMove :: Ord k => Map k (NodeInfo k v) -> Maybe (PatchMapWithMove k v)
