diff --git a/src/Drawing.hs b/src/Drawing.hs
index a695ce4..44d6014 100644
--- a/src/Drawing.hs
+++ b/src/Drawing.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE CPP #-}
 module Drawing(ImageInfo(..),
   Rectangle,
   Camera,
@@ -136,7 +137,13 @@ loadTexture mmodfunc mcuttop mtaketop fp = do
             in filterImage mf movedimg (fromIntegral imgw) totheight numbytes
       withStorableArray (imageData img) $ \imgdata -> do
         let movedimg = plusPtr imgdata (fromIntegral imgw * cuttop * numbytes)
-        texImage2D Nothing NoProxy 0 intform
+        texImage2D
+#if MIN_VERSION_OpenGL(2,9,0)
+          Texture2D
+#else
+          Nothing
+#endif
+          NoProxy 0 intform
           (TextureSize2D (fromIntegral imgw) (fromIntegral totheight)) 0 
           (PixelData pformat UnsignedByte movedimg)
       return texName
