diff --git a/src/Gen.hs b/src/Gen.hs
index 6329baa..ddab74e 100644
--- a/src/Gen.hs
+++ b/src/Gen.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TemplateHaskell, FlexibleContexts #-}
 module Gen
 where
 
diff --git a/src/Match/Ball.hs b/src/Match/Ball.hs
index ed57970..0e816b8 100644
--- a/src/Match/Ball.hs
+++ b/src/Match/Ball.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TemplateHaskell, FlexibleContexts #-}
 module Match.Ball
 where
 
diff --git a/src/Match/Player.hs b/src/Match/Player.hs
index 301d861..5d2285c 100644
--- a/src/Match/Player.hs
+++ b/src/Match/Player.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TemplateHaskell, FlexibleContexts #-}
 module Match.Player(TeamOwner(..),
   PlayerID,
   PlayerTextureSet(..),
diff --git a/src/Match/State/MatchState.hs b/src/Match/State/MatchState.hs
index 44c87d1..79a02da 100644
--- a/src/Match/State/MatchState.hs
+++ b/src/Match/State/MatchState.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TemplateHaskell, FlexibleContexts #-}
 module Match.State.MatchState
 where
 
