diff --git a/src/DeriveMod.hs b/src/DeriveMod.hs
index 23ef072..ee0f2b4 100644
--- a/src/DeriveMod.hs
+++ b/src/DeriveMod.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 module DeriveMod(deriveMod, deriveSMod, deriveMods)
@@ -28,3 +29,7 @@ mkModM d = do
 dToFs d = do
+#if MIN_VERSION_template_haskell(2,11,0)
+  TyConI (DataD _ _ _ _ cons _) <- reify d
+#else
   TyConI (DataD _ _ _ cons _) <- reify d
+#endif
   return $ concatMap getF cons
