diff --git a/tests/Properties.hs b/tests/Properties.hs
index b2f17f6..9168982 100644
--- a/tests/Properties.hs
+++ b/tests/Properties.hs
@@ -18,2 +18,7 @@ instance Arbitrary ByteString where
 
+-- downgrade 'Traversable a => a'
+-- to just [a].
+l :: [a] -> [a]
+l = id
+
 -- dummy test
@@ -38,3 +43,3 @@ prop_PubSub = monadicIO $ do
         threadDelay 1000
-        r <- mapM (sendMsg pub sub1 sub2) msgs
+        r <- mapM (sendMsg pub sub1 sub2) (l msgs)
         unsubscribe sub2 $ C.pack ""
@@ -72,3 +77,3 @@ prop_Pair = monadicIO $ do
         -- Send message from s1 to s2, then back from s2 to s1, then make sure it hasn't changed
-        r <- mapM (\m -> send s1 m >> recv s2 >>= send s2 >> recv s1 >>= return . (== m)) msgs
+        r <- mapM (\m -> send s1 m >> recv s2 >>= send s2 >> recv s1 >>= return . (== m)) (l msgs)
         close s1
@@ -92,3 +97,3 @@ prop_Pipeline = monadicIO $ do
         threadDelay 1000
-        r <- mapM (testSockets push pull1 pull2) msgs
+        r <- mapM (testSockets push pull1 pull2) (l msgs)
         close push
@@ -125,3 +130,3 @@ prop_ReqRep = monadicIO $ do
         threadDelay 1000
-        r <- mapM (\m -> send req m >> recv rep >>= send rep >> recv req >>= return . (== m)) msgs
+        r <- mapM (\m -> send req m >> recv rep >>= send rep >> recv req >>= return . (== m)) (l msgs)
         close req
@@ -149,3 +154,3 @@ prop_Bus = monadicIO $ do
         threadDelay 1000
-        r <- mapM (testSockets b1 b2 b3) msgs
+        r <- mapM (testSockets b1 b2 b3) (l msgs)
         close b1
