diff --git a/src/Byline/Shell.hs b/src/Byline/Shell.hs
index 6c1f00d..2046887 100644
--- a/src/Byline/Shell.hs
+++ b/src/Byline/Shell.hs
@@ -14,6 +14,7 @@
 -- License: BSD-2-Clause
 --
 -- Interactive shells.
+{-# LANGUAGE CPP #-}
 module Byline.Shell
   ( -- * Shell-like Interfaces
     Shell (..),
@@ -82,7 +83,11 @@ shellHelp ::
   Shell a ->
   m ()
 shellHelp Shell {..} = do
+#if MIN_VERSION_optparse_applicative(0,16,0)
   let h = O.parserFailure shellPrefs shellInfo (O.ShowHelpText Nothing) mempty
+#else
+  let h = O.parserFailure shellPrefs shellInfo O.ShowHelpText mempty
+#endif
       s = fst (O.renderFailure h "")
   sayLn (text $ toText s)
 
