--- instant-generics-0.3.6-orig/instant-generics.cabal	2012-09-12 23:20:19.000000000 +1000
+++ instant-generics-0.3.6/instant-generics.cabal	2012-09-18 14:35:31.243340335 +1000
@@ -14,7 +14,7 @@
   <http://www.cse.unsw.edu.au/~chak/project/generics/>). The current release
   implements the extensions to support indexed datatypes described in:
   .
-  *  José Pedro Magalhães and Johan Jeuring.
+  *  Jose Pedro Magalhaes and Johan Jeuring.
      /Generic Programming for Indexed Datatypes./
      Draft version, 2011.
      <http://dreixel.net/research/pdf/gpid_draft.pdf>
@@ -33,7 +33,7 @@
 library
   hs-source-dirs:         src
   build-depends:          base >= 3.0 && < 5, template-haskell >= 2.4 && < 3,
-                          containers < 0.5, syb < 0.4
+                          containers < 0.6, syb < 0.4
   exposed-modules:        Generics.Instant,
                           Generics.Instant.Base,
                           Generics.Instant.TH,
