--- instant-generics-0.3.2-orig/instant-generics.cabal	2011-06-16 16:01:45.000000000 +1000
+++ instant-generics-0.3.2/instant-generics.cabal	2011-06-18 15:30:17.921700494 +1000
@@ -14,7 +14,7 @@
   <http://www.cse.unsw.edu.au/~chak/project/generics/>). The current release
   implements the extensions to support indexed datatypes described in:
   .
-  *  José Pedro Magalhães and Johan Jeuring.
+  *  Jose Pedro Magalhaes and Johan Jeuring.
      /Generic Programming for Indexed Datatypes./
      Draft version, 2011.
      <http://dreixel.net/research/pdf/gpid_draft.pdf>
