--- instant-generics-0.2.1-orig/instant-generics.cabal	2010-09-29 23:34:44.000000000 +1000
+++ instant-generics-0.2.1/instant-generics.cabal	2011-04-10 23:33:41.940586554 +1000
@@ -17,7 +17,7 @@
   in the paper:
   .
   *  Thomas van Noort, Alexey Rodriguez Yakushev, Stefan Holdermans, 
-     Johan Jeuring, Bastiaan Heeren, José Pedro Magalhães.
+     Johan Jeuring, Bastiaan Heeren, Jose Pedro Magalhaes.
      /A Lightweight Approach to Datatype-Generic Rewriting./
      Journal of Functional Programming, Special Issue on Generic Programming, 
      2010.
