diff --git a/Annotations.cabal b/Annotations.cabal
index fe134f0..528ba69 100644
--- a/Annotations.cabal
+++ b/Annotations.cabal
@@ -20,7 +20,7 @@ Build-type:     Simple
 
 Library
   Build-Depends:    base >= 4.1 && < 4.4, mtl >= 1.1 && < 2.1,
-                    parsec >= 3.0 && < 3.2, multirec >= 0.4 && < 0.6
+                    parsec >= 3.0 && < 3.2, multirec >= 0.4 && < 0.8
   Exposed-Modules:  Annotations.Bounds,
                     Annotations.BoundsParser,
                     Annotations.Except,
