diff --git a/dataenc.cabal b/dataenc.cabal
index 4b0ef1d..ea1a3a0 100644
--- a/dataenc.cabal
+++ b/dataenc.cabal
@@ -21,7 +21,7 @@ flag BuildTests
 
 library
     hs-source-dirs: src
-    build-depends: array >= 0.1.0 && < 0.4, base >= 3.0.0 && < 4.4, containers >= 0.1.0 && < 0.5
+    build-depends: array >= 0.1.0 && < 0.4, base >= 3.0.0 && < 4.5, containers >= 0.1.0 && < 0.5
     exposed-modules:
             Codec.Binary.Base16
             Codec.Binary.Base32
