--- hslua-0.3.4-orig/hslua.cabal	2011-07-18 19:15:44.000000000 +1000
+++ hslua-0.3.4/hslua.cabal	2011-11-06 10:48:05.795826113 +1100
@@ -16,26 +16,9 @@
         The package is standalone: full Lua interpreter version 5.1.4 
         is distributed in this package as well.
 Category: Scripting
-Extra-source-files: src/*.h
 
 Library
   Build-depends: base==4.*
   Exposed-modules: Scripting.Lua, Scripting.Lua.ConfigFile
   Hs-source-dirs: src
-  C-sources: src/lapi.c, src/lauxlib.c, src/lbaselib.c, src/lcode.c,
-             src/ldblib.c, src/ldebug.c, src/ldo.c, src/ldump.c, src/lfunc.c,
-             src/lgc.c, src/linit.c, src/liolib.c, src/llex.c, src/lmathlib.c,
-             src/lmem.c, src/loadlib.c, src/lobject.c, src/lopcodes.c,
-             src/loslib.c, src/lparser.c, src/lstate.c, src/lstring.c,
-             src/lstrlib.c, src/ltable.c, src/ltablib.c, src/ltm.c,
-             src/lundump.c, src/lvm.c, src/lzio.c, src/ntrljmp.c
-  Include-dirs: src
-
-  if os(linux)
-    CC-Options: "-DLUA_USE_LINUX"
-
-  if os(darwin)
-    CC-Options: "-DLUA_USE_MACOSX"
-
-  if os(freebsd)
-    CC-Options: "-DLUA_USE_POSIX"
+  Extra-libraries:   lua
