--- hslua-0.3.4-orig/hslua.cabal	2011-07-18 19:15:44.000000000 +1000
+++ hslua-0.3.4/hslua.cabal	2011-11-06 23:10:03.747115482 +1100
@@ -16,19 +16,14 @@
         The package is standalone: full Lua interpreter version 5.1.4 
         is distributed in this package as well.
 Category: Scripting
-Extra-source-files: src/*.h
+Extra-source-files: src/ntrljmp.h
 
 Library
   Build-depends: base==4.*
   Exposed-modules: Scripting.Lua, Scripting.Lua.ConfigFile
   Hs-source-dirs: src
-  C-sources: src/lapi.c, src/lauxlib.c, src/lbaselib.c, src/lcode.c,
-             src/ldblib.c, src/ldebug.c, src/ldo.c, src/ldump.c, src/lfunc.c,
-             src/lgc.c, src/linit.c, src/liolib.c, src/llex.c, src/lmathlib.c,
-             src/lmem.c, src/loadlib.c, src/lobject.c, src/lopcodes.c,
-             src/loslib.c, src/lparser.c, src/lstate.c, src/lstring.c,
-             src/lstrlib.c, src/ltable.c, src/ltablib.c, src/ltm.c,
-             src/lundump.c, src/lvm.c, src/lzio.c, src/ntrljmp.c
+  Extra-libraries:   lua
+  C-sources: src/ntrljmp.c
   Include-dirs: src
 
   if os(linux)
--- hslua-0.3.4-orig/src/ntrljmp.h	2011-07-18 19:15:44.000000000 +1000
+++ hslua-0.3.4/src/ntrljmp.h	2011-11-06 23:19:38.028134287 +1100
@@ -10,9 +10,7 @@
 #define luac_c
 #define LUA_CORE
 
-#include "lua.h"
-#include "lobject.h"
-#include "ltm.h"
+#include <lualib.h>
 
 LUAI_FUNC int lua_neutralize_longjmp(lua_State *state);
 
--- hslua-0.3.4-orig/src/ntrljmp.c	2011-07-18 19:15:44.000000000 +1000
+++ hslua-0.3.4/src/ntrljmp.c	2011-11-06 23:19:28.462885469 +1100
@@ -8,10 +8,7 @@
 #define luac_c
 #define LUA_CORE
 
-#include "ldebug.h"
-#include "lobject.h"
-#include "lopcodes.h"
-#include "lundump.h"
+#include <lualib.h>
 
 extern int hsmethod__call( lua_State *state );
 
