diff --git a/src/Scripting/Lua.hs b/src/Scripting/Lua.hs
index 4a40a86..c94082b 100644
--- a/src/Scripting/Lua.hs
+++ b/src/Scripting/Lua.hs
@@ -180,6 +180,7 @@ import Prelude hiding (concat)
 import Foreign.C
 import Foreign.Ptr
 import Foreign.StablePtr
+import qualified Control.Exception as E
 import Control.Monad
 import Foreign.Marshal.Alloc
 import Data.IORef
@@ -961,7 +962,7 @@ newcfunction = mkWrapper . luaimport
 -- Any Haskell exception will be converted to a string and returned
 -- as Lua error.
 luaimport :: LuaImport a => a -> LuaCFunction
-luaimport a l = luaimport' 1 a l `catch` (\e -> push l (show e) >> return (-1))
+luaimport a l = (E.catch :: IO a -> (IOError -> IO a) -> IO a) (luaimport' 1 a l) (\e -> push l (show e) >> return (-1))
 
 -- | Free function pointer created with @newcfunction@.
 freecfunction :: FunPtr LuaCFunction -> IO ()
