diff -urN streamly-0.7.0/streamly.cabal streamly-0.7.0-new/streamly.cabal
--- streamly-0.7.0/streamly.cabal	2001-09-08 19:46:40.000000000 -0600
+++ streamly-0.7.0-new/streamly.cabal	2020-01-16 10:20:35.716911046 -0700
@@ -725,7 +725,7 @@
   else
     buildable: False
 
-executable nano-bench
+executable streamly-nano-bench
   import: bench-options
   hs-source-dirs: benchmark, src
   include-dirs:     src/Streamly/Internal/Data/Time
@@ -767,7 +767,7 @@
   else
     buildable: False
 
-executable adaptive
+executable streamly-adaptive
   import: bench-options
   hs-source-dirs: benchmark
   main-is: Adaptive.hs
@@ -782,7 +782,7 @@
   else
     buildable: False
 
-executable chart
+executable streamly-chart
   default-language: Haskell2010
   ghc-options: -Wall
   hs-source-dirs: benchmark
@@ -801,7 +801,7 @@
 -- Examples
 -------------------------------------------------------------------------------
 
-executable SearchQuery
+executable streamly-example-SearchQuery
   import: exe-options
   main-is: SearchQuery.hs
   hs-source-dirs:  examples
@@ -814,7 +814,7 @@
   else
     buildable: False
 
-executable ListDir
+executable streamly-example-ListDir
   import: exe-options
   main-is: ListDir.hs
   hs-source-dirs:  examples
@@ -830,7 +830,7 @@
   else
     buildable: False
 
-executable MergeSort
+executable streamly-example-MergeSort
   import: exe-options
   main-is: MergeSort.hs
   hs-source-dirs:  examples
@@ -843,7 +843,7 @@
   else
     buildable: False
 
-executable AcidRain
+executable streamly-example-AcidRain
   import: exe-options
   main-is: AcidRain.hs
   hs-source-dirs:  examples
@@ -860,7 +860,7 @@
   else
     buildable: False
 
-executable CirclingSquare
+executable streamly-example-CirclingSquare
   import: exe-options
   main-is: CirclingSquare.hs
   hs-source-dirs:  examples
@@ -873,7 +873,7 @@
   else
     buildable: False
 
-executable ControlFlow
+executable streamly-example-ControlFlow
   import: exe-options
   main-is: ControlFlow.hs
   hs-source-dirs:  examples
@@ -891,7 +891,7 @@
   else
     buildable: False
 
-executable HandleIO
+executable streamly-example-HandleIO
   import: exe-options
   main-is: HandleIO.hs
   hs-source-dirs:  examples
@@ -903,7 +903,7 @@
   else
     buildable: False
 
-executable FileIOExamples
+executable streamly-example-FileIOExamples
   import: exe-options
   main-is: FileIOExamples.hs
   hs-source-dirs:  examples
@@ -915,7 +915,7 @@
   else
     buildable: False
 
-executable EchoServer
+executable streamly-example-EchoServer
   import: exe-options
   main-is: EchoServer.hs
   hs-source-dirs:  examples
@@ -931,7 +931,7 @@
   else
     buildable: False
 
-executable FileSinkServer
+executable streamly-example-FileSinkServer
   import: exe-options
   main-is: FileSinkServer.hs
   hs-source-dirs:  examples
@@ -947,7 +947,7 @@
   else
     buildable: False
 
-executable FromFileClient
+executable streamly-example-FromFileClient
   import: exe-options
   main-is: FromFileClient.hs
   hs-source-dirs:  examples
@@ -959,7 +959,7 @@
   else
     buildable: False
 
-executable WordClassifier
+executable streamly-example-WordClassifier
   import: exe-options
   main-is: WordClassifier.hs
   hs-source-dirs:  examples
@@ -973,7 +973,7 @@
   else
     buildable: False
 
-executable WordCount
+executable streamly-example-WordCount
   import: exe-options
   main-is: WordCount.hs
   hs-source-dirs:  examples
@@ -986,7 +986,7 @@
   else
     buildable: False
 
-executable CamelCase
+executable streamly-example-CamelCase
   import: exe-options
   main-is: CamelCase.hs
   hs-source-dirs:  examples
