diff -ur a/gen/Amazonka/DataPipeline/Types.hs b/gen/Amazonka/DataPipeline/Types.hs
--- a/gen/Amazonka/DataPipeline/Types.hs	2001-09-09 03:46:40.000000000 +0200
+++ b/gen/Amazonka/DataPipeline/Types.hs	2025-02-20 10:32:03.532059216 +0100
@@ -1,4 +1,4 @@
-{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE StrictData #-}
 {-# LANGUAGE NoImplicitPrelude #-}
