From 5b8e1667a7d2ebad6b6f79d2fd8d5f8b4ac3a37f Mon Sep 17 00:00:00 2001
From: Jack Todaro <jack.todaro@posteo.net>
Date: Sun, 12 Jan 2020 19:03:26 +1100
Subject: [PATCH] Storage.hs: add GHC extensions required by
 persistent-template-2.8

---
 src/Pantry/Storage.hs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Pantry/Storage.hs b/src/Pantry/Storage.hs
index 98d9c2fa..dd8159b8 100644
--- a/src/Pantry/Storage.hs
+++ b/src/Pantry/Storage.hs
@@ -13,6 +13,8 @@
 {-# LANGUAGE ViewPatterns #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE StandaloneDeriving #-}
 module Pantry.Storage
   ( SqlBackend
   , initStorage
-- 
2.24.1

