diff -Naur /home/jkarlson/data-accessor-monads-fd-0.2.0.3/data-accessor-monads-fd.cabal /home/jkarlson/data-accessor-monads-fd-0.2.0.3-r1/data-accessor-monads-fd.cabal
--- /home/jkarlson/data-accessor-monads-fd-0.2.0.3/data-accessor-monads-fd.cabal	2011-02-27 00:30:15.000000000 +0200
+++ /home/jkarlson/data-accessor-monads-fd-0.2.0.3-r1/data-accessor-monads-fd.cabal	2012-04-14 16:42:43.179999699 +0300
@@ -11,7 +11,7 @@
 Build-Depends:
   data-accessor >=0.2 && <0.4,
   monads-fd >=0.1 && <0.3,
-  transformers >=0.2 && <0.3,
+  transformers >=0.2 && <0.4,
   base >=1.0 && <6
 Synopsis:         Use Accessor to access state in monads-fd State monad class
 Description:      Use Accessor to access state in monads-fd State monad class
