diff --git a/alex.cabal b/alex.cabal
index d04467b..b262af7 100644
--- a/alex.cabal
+++ b/alex.cabal
@@ -72,7 +72,7 @@ executable alex
 
   build-depends: base < 5
 
-  extensions: CPP
+  extensions: CPP, BangPatterns
   ghc-options: -Wall
   other-modules:
         AbsSyn
diff --git a/doc/configure.ac b/doc/configure.ac
index 51fa85d..6e772c0 100644
--- a/doc/configure.ac
+++ b/doc/configure.ac
@@ -9,5 +9,5 @@ FP_DIR_DOCBOOK_XSL([/usr/share/xml/docbook/stylesheet/nwalsh/current /usr/share/
 
 AC_PATH_PROG(DbLatexCmd,dblatex)
 
-AC_CONFIG_FILES([config.mk])
+AC_CONFIG_FILES([config.mk alex.1])
 AC_OUTPUT
