--- a/goatee.cabal
+++ b/goatee.cabal
@@ -90,3 +90,2 @@ test-suite test-goatee
         Game.Goatee.Test.Common
-        Test
     type: exitcode-stdio-1.0
--- a/src/Game/Goatee/Lib/Property/Parser.hs
+++ b/src/Game/Goatee/Lib/Property/Parser.hs
@@ -64,2 +64,3 @@ import Text.ParserCombinators.Parsec (
   )
+import qualified Data.Semigroup as S
 
@@ -75,2 +76,7 @@ newtype CoordListMonoid = CoordListMonoid { runCoordListMonoid :: CoordList }
 
+instance S.Semigroup CoordListMonoid where
+  (<>) (CoordListMonoid x) (CoordListMonoid y) =
+    CoordListMonoid $ coords' (coordListSingles x ++ coordListSingles y)
+                              (coordListRects x ++ coordListRects y)
+
 instance Monoid CoordListMonoid where
