diff --git a/src/Game/Goatee/Lib/Property/Base.hs b/src/Game/Goatee/Lib/Property/Base.hs
index 69d4d9d..ebf0d88 100644
--- a/src/Game/Goatee/Lib/Property/Base.hs
+++ b/src/Game/Goatee/Lib/Property/Base.hs
@@ -336,3 +336,7 @@ defValuedProperty name propType inherited valueType = do
   bar <- newName "bar"
+#if MIN_VERSION_template_haskell(2,11,0)
+  DataConI _ (AppT (AppT _ haskellValueType) _) _ <- reify propName
+#else
   DataConI _ (AppT (AppT _ haskellValueType) _) _ _ <- reify propName
+#endif
   sequence [
