diff --git a/Text/Dot.hs b/Text/Dot.hs
index 3bc610a..ebc805c 100644
--- a/Text/Dot.hs
+++ b/Text/Dot.hs
@@ -40,2 +40,4 @@ import qualified Data.Map as M
 import qualified Data.Set as S
+import Control.Monad
+import Control.Applicative
 
@@ -61,2 +63,9 @@ data Dot a = Dot { unDot :: Int -> ([GraphElement],Int,a) }
 
+instance Functor Dot where
+    fmap  = liftM
+
+instance Applicative Dot where
+    pure  = return
+    (<*>) = ap  -- defined in Control.Monad
+
 instance Monad Dot where
