--- snap-server-0.6.0.1-orig/src/Snap/Internal/Http/Parser.hs	2011-10-29 06:03:32.000000000 +1100
+++ snap-server-0.6.0.1/src/Snap/Internal/Http/Parser.hs	2011-11-29 23:27:42.579398643 +1100
@@ -21,7 +21,7 @@
 import           Control.Exception
 import           Control.Monad (liftM)
 import           Control.Monad.Trans
-import           Data.Attoparsec hiding (many, Result(..))
+import           Data.Attoparsec hiding (Result(..))
 import           Data.Attoparsec.Enumerator
 import           Data.ByteString.Char8 (ByteString)
 import qualified Data.ByteString.Char8 as S
