--- a/src/Network/Simple/TCP.hs
+++ b/src/Network/Simple/TCP.hs
@@ -414,3 +414,7 @@ connectSockSOCKS5 psock dhn dsn = liftIO $ do
    let dsa = NS5.SocksAddress (NS5.SocksAddrDomainName (B8.pack dhn)) dpn
-   NS5.establish psock [NS5.SocksMethodNone] >>= \case
+   NS5.establish
+#if MIN_VERSION_socks(0,6,0)
+                 NS5.SocksVer5
+#endif
+                 psock [NS5.SocksMethodNone] >>= \case
      NS5.SocksMethodNone -> do
