--- hprotoc-2.0.10-orig/Text/ProtocolBuffers/ProtoCompile/Resolve.hs	2012-07-01 21:07:19.000000000 +1000
+++ hprotoc-2.0.10/Text/ProtocolBuffers/ProtoCompile/Resolve.hs	2012-07-14 15:07:54.275661376 +1000
@@ -159,7 +159,7 @@
 import Data.Maybe(mapMaybe)
 import Data.Typeable
 -- import Data.Monoid()
-import System.Directory
+import System.Directory (doesFileExist, canonicalizePath)
 import qualified System.FilePath as Local(pathSeparator,splitDirectories,joinPath,combine,makeRelative)
 import qualified System.FilePath.Posix as Canon(pathSeparator,splitDirectories,joinPath,takeBaseName)
 import qualified Data.ByteString.Lazy.Char8 as LC
