diff -rN -u old-protocol-buffers-0.2.9/hprotoc/Text/ProtocolBuffers/ProtoCompile/Parser.hs new-protocol-buffers-0.2.9/hprotoc/Text/ProtocolBuffers/ProtoCompile/Parser.hs
--- old-protocol-buffers-0.2.9/hprotoc/Text/ProtocolBuffers/ProtoCompile/Parser.hs	2008-09-21 15:16:08.000000000 +0200
+++ new-protocol-buffers-0.2.9/hprotoc/Text/ProtocolBuffers/ProtoCompile/Parser.hs	2008-09-21 15:16:08.000000000 +0200
@@ -45,7 +45,7 @@
 
 default ()
 
-type P = GenParser Lexed
+type P a = GenParser Lexed a
 
 pbParse :: String -> IO (Either ParseError D.FileDescriptorProto)
 pbParse filename = fmap (parseProto filename) (LC.readFile filename)

