--- geniplate-mirror-0.7.2-orig/Data/Generics/Geniplate.hs	2016-01-15 02:32:09.000000000 +1100
+++ geniplate-mirror-0.7.2/Data/Generics/Geniplate.hs	2016-04-24 20:31:58.657901802 +1000
@@ -80,7 +80,11 @@
 funDef f e = [FunD f [Clause [] (NormalB e) []]]
 
 instDef :: Name -> [Type] -> Name -> Exp -> [Dec]
-instDef cls ts met e = [InstanceD [] (foldl AppT (ConT cls) ts) (funDef met e)]
+instDef cls ts met e = [InstanceD
+#if __GLASGOW_HASKELL__ >= 800
+                        Nothing
+#endif
+                        [] (foldl AppT (ConT cls) ts) (funDef met e)]
 
 -- | Create a 'TransformBi' instance.
 -- The 'TypeQ' argument should be a pair; the /inner/ and /outer/ types for 'transformBi'.
