From dcd6d7de069144166103f8da3946a0178fe62782 Mon Sep 17 00:00:00 2001
From: Tao He <sighingnow@gmail.com>
Date: Sun, 19 Jul 2020 18:11:35 +0800
Subject: [PATCH] Fixes for newer version of GHC.

Signed-off-by: Tao He <sighingnow@gmail.com>
---
 JVM/Assembler.hs | 6 +++---
 stack.yaml       | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/JVM/Assembler.hs b/JVM/Assembler.hs
index 9e49168..3d3de65 100644
--- a/JVM/Assembler.hs
+++ b/JVM/Assembler.hs
@@ -473,14 +473,14 @@ instance BinaryState Integer Instruction where
   put  LCMP           = putByte 148
   put (FCMP C_LT)     = putByte 149
   put (FCMP C_GT)     = putByte 150
-  put (FCMP c)        = fail $ "No such instruction: FCMP " ++ show c
+  put (FCMP c)        = error $ "No such instruction: FCMP " ++ show c
   put (DCMP C_LT)     = putByte 151
   put (DCMP C_GT)     = putByte 152
-  put (DCMP c)        = fail $ "No such instruction: DCMP " ++ show c
+  put (DCMP c)        = error $ "No such instruction: DCMP " ++ show c
   put (IF c x)        = putByte (fromIntegral $ 153 + fromEnum c) >> put x
   put (IF_ACMP C_EQ x) = put1 165 x
   put (IF_ACMP C_NE x) = put1 166 x
-  put (IF_ACMP c _)   = fail $ "No such instruction: IF_ACMP " ++ show c
+  put (IF_ACMP c _)   = error $ "No such instruction: IF_ACMP " ++ show c
   put (IF_ICMP c x)   = putByte (fromIntegral $ 159 + fromEnum c) >> put x
   put (GOTO x)        = put1 167 x
   put (JSR x)         = put1 168 x
