diff --git a/JVM/Builder/Monad.hs b/JVM/Builder/Monad.hs
index cddb055..a1434f5 100644
--- a/JVM/Builder/Monad.hs
+++ b/JVM/Builder/Monad.hs
@@ -71,3 +71,3 @@ newtype Generate e a = Generate {
   runGenerate :: EMT e (State GState) a }
-  deriving (Monad, MonadState GState)
+  deriving (Functor, Applicative, Monad, MonadState GState)
 
@@ -84,3 +84,3 @@ newtype GenerateIO e a = GenerateIO {
   runGenerateIO :: EMT e (StateT GState IO) a }
-  deriving (Monad, MonadState GState, MonadIO)
+  deriving (Functor, Applicative, Monad, MonadState GState, MonadIO)
 
