diff --git a/Data/Tree/AVL/Join.hs b/Data/Tree/AVL/Join.hs
index 6520b58..13bb02e 100644
--- a/Data/Tree/AVL/Join.hs
+++ b/Data/Tree/AVL/Join.hs
@@ -26,3 +26,3 @@ import Data.List(foldl')
 #ifdef __GLASGOW_HASKELL__
-import GHC.Base
+import GHC.Base hiding (join)
 #include "ghcdefs.h"
@@ -32,2 +32,4 @@ import GHC.Base
 
+import Prelude hiding (foldr)
+
 -- | Join two AVL trees. This is the AVL equivalent of (++).
