diff --git a/src/Client/State/Network.hs b/src/Client/State/Network.hs
index fdd0ef7..fec58a2 100644
--- a/src/Client/State/Network.hs
+++ b/src/Client/State/Network.hs
@@ -441,7 +441,7 @@ doRandomNick cs = reply [ircNick candidate] cs'
     primaryNick = NonEmpty.head (view (csSettings . ssNicks) cs)
     candidate   = Text.take (limit-length suffix) primaryNick <> Text.pack suffix
 
-    (n, cs')    = cs & csSeed %%~ Random.uniformR range
+    (n, cs')    = cs & csSeed %%~ Random.randomR range
 
 doTopic :: ZonedTime -> UserInfo -> Identifier -> Text -> NetworkState -> NetworkState
 doTopic when user chan topic =
