--- a/src/Client/State/Network.hs
+++ b/src/Client/State/Network.hs
@@ -1,2 +1,2 @@
-{-# Language TemplateHaskell, OverloadedStrings, BangPatterns #-}
+{-# Language TemplateHaskell, OverloadedStrings, BangPatterns, CPP #-}
 
@@ -712,4 +712,9 @@ doAuthenticate param cs =
       , Just pass <- view ssSaslPassword ss
+#if MIN_VERSION_irc_core(2,7,0)
+      -> (ircAuthenticates (encodePlainAuthentication user pass),
+          set csAuthenticationState AS_None cs)
+#else
       -> ([ircAuthenticate (encodePlainAuthentication user pass)],
           set csAuthenticationState AS_None cs)
+#endif
 
@@ -718,4 +723,9 @@ doAuthenticate param cs =
       , Just user <- view ssSaslUsername ss
+#if MIN_VERSION_irc_core(2,7,0)
+      -> (ircAuthenticates (encodeExternalAuthentication user),
+          set csAuthenticationState AS_None cs)
+#else
       -> ([ircAuthenticate (encodeExternalAuthentication user)],
           set csAuthenticationState AS_None cs)
+#endif
 
