--- reactive-banana-0.7.0.1-orig/src/Reactive/Banana/Internal/PulseLatch0.hs	2012-08-27 21:33:33.000000000 +1000
+++ reactive-banana-0.7.0.1/src/Reactive/Banana/Internal/PulseLatch0.hs	2012-09-15 14:22:55.934735115 +1000
@@ -2,7 +2,7 @@
     reactive-banana
 ------------------------------------------------------------------------------}
 {-# LANGUAGE Rank2Types, RecursiveDo, ExistentialQuantification,
-    TypeSynonymInstances #-}
+    TypeSynonymInstances, FlexibleInstances #-}
 module Reactive.Banana.Internal.PulseLatch0 where
 
 import Control.Applicative
