--- reactive-banana-0.7.0.1-orig/reactive-banana.cabal	2012-08-27 21:33:33.000000000 +1000
+++ reactive-banana-0.7.0.1/reactive-banana.cabal	2012-09-15 14:06:22.164341734 +1000
@@ -90,6 +90,12 @@
     hs-source-dirs:     src
     main-is:            Reactive/Banana/Test.hs
     build-depends:      base >= 4.2 && < 5,
+                        containers >= 0.3 && < 0.6,
+                        transformers >= 0.2 && < 0.4,
+                        vault == 0.2.*,
+                        QuickCheck >= 1.2 && < 2.6,
+                        unordered-containers >= 0.2.1.0 && < 0.3,
+                        hashable == 1.1.*,
                         HUnit >= 1.2 && < 2,
                         test-framework == 0.6.*,
                         test-framework-hunit == 0.2.*,
