From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Quick fix to get working with hashable-1.4

diff -urN reactive-banana-1.2.2.0/src/Reactive/Banana/Prim/Types.hs reactive-banana-1.2.2.0-r1/src/Reactive/Banana/Prim/Types.hs
--- reactive-banana-1.2.2.0/src/Reactive/Banana/Prim/Types.hs	2001-09-08 19:46:40.000000000 -0600
+++ reactive-banana-1.2.2.0-r1/src/Reactive/Banana/Prim/Types.hs	2022-03-19 02:58:09.246983453 -0600
@@ -123,7 +123,6 @@
 data Output' = Output
     { _evalO     :: EvalP EvalO
     }
-instance Eq Output where (==) = equalRef
 
 data SomeNode
     = forall a. P (Pulse a)
diff -urN reactive-banana-1.2.2.0/src/Reactive/Banana/Prim/Util.hs reactive-banana-1.2.2.0-r1/src/Reactive/Banana/Prim/Util.hs
--- reactive-banana-1.2.2.0/src/Reactive/Banana/Prim/Util.hs	2001-09-08 19:46:40.000000000 -0600
+++ reactive-banana-1.2.2.0-r1/src/Reactive/Banana/Prim/Util.hs	2022-03-19 02:57:58.116983448 -0600
@@ -28,6 +28,7 @@
 ------------------------------------------------------------------------------}
 data Ref a = Ref !(IORef a) !Unique
 
+instance Eq (Ref a) where (==) = equalRef
 instance Hashable (Ref a) where hashWithSalt s (Ref _ u) = hashWithSalt s u 
 
 equalRef :: Ref a -> Ref b -> Bool
