--- a/Extra/Orphans.hs
+++ b/Extra/Orphans.hs
@@ -46,7 +46,9 @@ deriving instance Generic UTCTime
 #else
 #endif
 
+#if ! MIN_VERSION_network_uri(2,6,2)
 deriving instance Generic URIAuth
+#endif
 
 -- $(deriveLift ''UserId)
 instance Lift UserId where lift (UserId x0) = [|UserId $(lift x0)|]
@@ -127,6 +129,8 @@ instance Arbitrary URI where
 instance SafeCopy URI where version = 0
 instance SafeCopy URIAuth where version = 0
 
+#if ! MIN_VERSION_network_uri(2,6,2)
 $(concat <$>
   sequence
   [ deriveLiftMany [''URI, ''URIAuth] ])
+#endif
