diff --git a/Test/QUnit.hs b/Test/QUnit.hs
index 04b4e43..162c00c 100644
--- a/Test/QUnit.hs
+++ b/Test/QUnit.hs
@@ -60,7 +60,7 @@ testQuickCheck args prop =
     HU.TestCase $
     do result <- QC.quickCheckWithResult args prop
        case result of
-         (QC.Success _ _ _) -> return ()
-         (QC.GaveUp ntest _ _) -> HU.assertFailure $ "Arguments exhausted after" ++ show ntest ++ (if ntest == 1 then " test." else " tests.")
-         (QC.Failure _ _ _ _usedSize reason _ _) -> HU.assertFailure reason
-         (QC.NoExpectedFailure _ _ _) -> HU.assertFailure $ "No Expected Failure"
+         (QC.Success{}) -> return ()
+         (QC.GaveUp{}) -> let ntest = QC.numTests result in  HU.assertFailure $ "Arguments exhausted after" ++ show ntest ++ (if ntest == 1 then " test." else " tests.")
+         (QC.Failure{}) -> let reason = QC.reason result in HU.assertFailure reason
+         (QC.NoExpectedFailure{}) -> HU.assertFailure $ "No Expected Failure"
