--- a/src/Graphics/Gnuplot/Private/Display.hs
+++ b/src/Graphics/Gnuplot/Private/Display.hs
@@ -13,2 +13,3 @@ import Prelude (FilePath, String, Int, ($), (.), )
 
+import qualified Data.Semigroup as S
 
@@ -31,2 +32,6 @@ type OptionSet = Map.Map Option.T [String]
 
+instance S.Semigroup Script where
+   (<>) (Script b0) (Script b1) =
+      Script (liftM2 mappend b0 b1)
+
 {-
@@ -40,2 +45,6 @@ instance Monoid Script where
 
+instance S.Semigroup Body where
+   (<>) (Body f0 c0) (Body f1 c1) =
+      Body (mappend f0 f1) (mappend c0 c1)
+
 {-
--- a/src/Graphics/Gnuplot/Private/Plot.hs
+++ b/src/Graphics/Gnuplot/Private/Plot.hs
@@ -27,2 +27,3 @@ import Prelude (Functor, fmap, String, show, Int, succ, writeFile, undefined, )
 
+import qualified Data.Semigroup as S
 
@@ -37,2 +38,6 @@ pure = Cons . return
 
+instance S.Semigroup (T graph) where
+   (<>) (Cons s0) (Cons s1) =
+      Cons (liftM2 mappend s0 s1)
+
 {-
