--- Decimal-0.2.1-orig/src/Data/Decimal.hs	2012-01-16 03:10:28.000000000 +1100
+++ Decimal-0.2.1/src/Data/Decimal.hs	2012-01-19 22:00:06.174164699 +1100
@@ -120,7 +120,7 @@
       (Decimal _ n2) = roundTo e d2
 
 
-instance (Integral i) => Show (DecimalRaw i) where
+instance (Integral i, Show i) => Show (DecimalRaw i) where
    showsPrec _ (Decimal e n)
        | e == 0     = (concat [signStr, strN] ++)
        | otherwise  = (concat [signStr, intPart, ".", fracPart] ++)
