--- Decimal-0.2.2-orig/Decimal.cabal	2012-01-20 18:40:46.000000000 +1100
+++ Decimal-0.2.2/Decimal.cabal	2012-01-23 17:40:29.421093405 +1100
@@ -39,8 +39,7 @@
   ghc-options:     -Wall -rtsopts
   hs-source-dirs:  src, src/Data, tests
   -- The following lines trigger a bug in Hackage.  Uncommment for compilation on GHC 6.
-  -- if impl(ghc >= 7.0.0)
-  --      default-language: Haskell2010
-  default-language: Haskell2010
+  if impl(ghc >= 7.0.0)
+     default-language: Haskell2010
   main-is:         Main.hs
 
