--- Decimal-0.2.0-orig/Decimal.cabal	2011-03-19 04:57:56.000000000 +1100
+++ Decimal-0.2.0/Decimal.cabal	2011-12-19 19:01:51.554599950 +1100
@@ -7,7 +7,7 @@
 Maintainer:         paul@cogito.org.uk
 Stability:          beta
 Category:           Math
-Build-Depends:	    base==4, QuickCheck>2.4, HUnit
+Build-Depends:	    base>=4 && <=5, QuickCheck>2.4, HUnit
 Build-type:         Simple
 Synopsis:           Decimal numbers with variable precision
 Description:        A decimal number has an integer mantissa and a negative
