--- Decimal-0.2.1-orig/Decimal.cabal	2012-01-16 03:10:28.000000000 +1100
+++ Decimal-0.2.1/Decimal.cabal	2012-01-19 22:11:04.662500789 +1100
@@ -21,7 +21,8 @@
                     base >= 4 && < 5,
                     QuickCheck >= 2.4
   hs-source-dirs:   src
-  default-language: Haskell2010
+  if impl(ghc >= 7.0.0)
+    default-language: Haskell2010
   ghc-options:      -Wall
   exposed-modules:  Data.Decimal
 
@@ -37,6 +38,7 @@
                    test-framework-hunit
   ghc-options:     -Wall -rtsopts
   hs-source-dirs:  src, src/Data, tests
-  default-language: Haskell2010
+  if impl(ghc >= 7.0.0)
+    default-language: Haskell2010
   main-is:         Main.hs
 
