diff --git a/Data/Concurrent/Queue/MichaelScott.hs b/Data/Concurrent/Queue/MichaelScott.hs
index ddb7f39..ace4a95 100644
--- a/Data/Concurrent/Queue/MichaelScott.hs
+++ b/Data/Concurrent/Queue/MichaelScott.hs
@@ -25,6 +25,7 @@ import Data.ByteString.Char8 (hPutStrLn, pack)
 
 -- import GHC.Types (Word(W#))
 import GHC.Prim (sameMutVar#)
+import GHC.Types (isTrue#)
 import GHC.IORef(IORef(IORef))
 import GHC.STRef(STRef(STRef))
 
@@ -61,7 +62,7 @@ data Pair a = Null | Cons a {-# UNPACK #-}!(IORef (Pair a))
 pairEq :: Pair a -> Pair a -> Bool
 pairEq Null       Null        = True
 pairEq (Cons _ (IORef (STRef mv1)))
-       (Cons _ (IORef (STRef mv2))) = sameMutVar# mv1 mv2
+       (Cons _ (IORef (STRef mv2))) = isTrue# (sameMutVar# mv1 mv2)
 pairEq _          _           = False
 
 -- | Push a new element onto the queue.  Because the queue can grow,
