--- happstack-state-6.1.0-orig/src/Happstack/State/Monad.hs	2011-07-21 02:42:36.000000000 +1000
+++ happstack-state-6.1.0/src/Happstack/State/Monad.hs	2011-07-21 20:58:09.017560363 +1000
@@ -76,7 +76,11 @@
 #if __GLASGOW_HASKELL__ < 610
     catchEv :: Ev m a -> (Exception -> a) -> Ev m a
 #else
+#if MIN_VERSION_mtl(2,0,0)
     catchEv :: Exception e => Ev m a -> (e -> a) -> Ev m a
+#else
+    catchEv :: Ev m a -> (SomeException -> a) -> Ev m a
+#endif
 #endif
 instance CatchEv (ReaderT st STM) where
     catchEv (Ev cmd) fun = Ev $ \s -> ReaderT $ \r -> runReaderT (cmd s) r `catchSTM` (\a -> return (fun a))
