--- a/Distribution/PackageDescription/TH.hs
+++ b/Distribution/PackageDescription/TH.hs
@@ -3,2 +3,3 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE CPP #-}
 
@@ -25,3 +26,7 @@ import Distribution.Verbosity (silent)
 import Text.PrettyPrint
+#if MIN_VERSION_Cabal(2,2,0)
+import Distribution.PackageDescription.Parsec (readGenericPackageDescription)
+#else
 import Distribution.PackageDescription.Parse (readPackageDescription)
+#endif
 import System.Directory (getCurrentDirectory, getDirectoryContents)
@@ -51,3 +56,7 @@ packageVariable = renderField currentPackageDescription
 packageVariableFrom :: Text a => FilePath -> (PackageDescription -> a) -> Q Exp
+#if MIN_VERSION_Cabal(2,2,0)
+packageVariableFrom s = renderField $ fmap packageDescription (readGenericPackageDescription silent s)
+#else
 packageVariableFrom s = renderField $ fmap packageDescription (readPackageDescription silent s)
+#endif
 
@@ -65,3 +74,7 @@ currentPackageDescription = fmap packageDescription $ do
   case cs of
+#if MIN_VERSION_Cabal(2,2,0)
+    (c:_) -> readGenericPackageDescription silent c
+#else
     (c:_) -> readPackageDescription silent c
+#endif
     [] -> error $ "Couldn't find a cabal file in the current working directory (" ++ dir ++ ")"
