diff --git a/Hipmunk.cabal b/Hipmunk.cabal
index e352e0c..33c4d2f 100644
--- a/Hipmunk.cabal
+++ b/Hipmunk.cabal
@@ -28,7 +28,7 @@ Description:
         functionality available in Chipmunk 5.3.4 is here yet.  But
         the bug fixes and performance improvements are.
       .
-      * Fix linking bug present since Hipmunk 5.1.0.  Thanks, Sönke Hahn.
+      * Fix linking bug present since Hipmunk 5.1.0.  Thanks, Sonke Hahn.
       .
       * Correctly destroy the space in the presence of callbacks.
       .
