diff --git a/nclient/NClient/GUI.hs b/nclient/NClient/GUI.hs
index 8e27ffe..5cb0256 100644
--- a/nclient/NClient/GUI.hs
+++ b/nclient/NClient/GUI.hs
@@ -7,3 +7,3 @@ import Control.Monad (forever, forM_, void)
 import Data.IORef (newIORef, atomicModifyIORef)
-import Graphics.Vty
+import Graphics.Vty hiding (Input, Output)
 import Graphics.Vty.Widgets.All
@@ -18,3 +18,3 @@ gui (i,o) = do
   history <- newIORef $ H.empty 10
-  messages <- newList (getNormalAttr defaultContext)
+  messages <- newList 1
   newMessage <- editWidget
@@ -54,3 +54,3 @@ gui (i,o) = do
     schedule $
-      do a:as <- S.words (fromS2C m) . region_width <$> getCurrentSize messages
+      do a:as <- S.words (fromS2C m) . regionWidth <$> getCurrentSize messages
          addMessage (formatter Tail m) a messages
diff --git a/nclient/NClient/Message/Format.hs b/nclient/NClient/Message/Format.hs
index 6efc13c..fafc5de 100644
--- a/nclient/NClient/Message/Format.hs
+++ b/nclient/NClient/Message/Format.hs
@@ -33,2 +33,2 @@ formatter f s2c = case f of
           SSystem {} -> fgColor blue
-          _ -> def_attr
+          _ -> defAttr
