diff --git a/Hach.cabal b/Hach.cabal
index fae0183..9766cc4 100644
--- a/Hach.cabal
+++ b/Hach.cabal
@@ -22,2 +22,3 @@ Library
                         old-locale,
+                        time-locale-compat,
                         network
diff --git a/libhach/Hach/Types.hs b/libhach/Hach/Types.hs
index db05682..82a04e5 100644
--- a/libhach/Hach/Types.hs
+++ b/libhach/Hach/Types.hs
@@ -14,3 +14,3 @@ import Data.Time
 import System.Exit (exitSuccess)
-import System.Locale (defaultTimeLocale)
+import Data.Time.Locale.Compat (defaultTimeLocale)
 
