--- file-location-0.4.5-orig/file-location.cabal	2012-10-18 03:19:59.000000000 +1100
+++ file-location-0.4.5/file-location.cabal	2012-10-20 21:31:33.235778880 +1100
@@ -70,6 +70,7 @@
                 , template-haskell
                 , transformers     >= 0.2 && < 0.4
                 , containers
+                , lifted-base
 
 -- This just helps you get the packages for test/main.hs
 -- tests are ran with test/run.sh
@@ -85,6 +86,7 @@
                  , containers
                  , template-haskell
                  , transformers
+                 , lifted-base
                   -- shelltestrunner: need shelltest executable
 
 Source-Repository head
