--- vty-4.7.0.14-orig/src/Graphics/Vty/Terminal/TerminfoBased.hs	2012-05-19 18:34:42.000000000 +1000
+++ vty-4.7.0.14/src/Graphics/Vty/Terminal/TerminfoBased.hs	2012-07-07 22:38:43.400597792 +1000
@@ -23,7 +23,7 @@
 import Data.Maybe ( isJust, isNothing, fromJust )
 import Data.Word
 
-import Foreign.C.Types ( CLong )
+import Foreign.C.Types ( CLong(..) )
 
 import GHC.IO.Handle
 
--- vty-4.7.0.14-orig/src/Data/Terminfo/Eval.hs	2012-05-19 18:34:42.000000000 +1000
+++ vty-4.7.0.14/src/Data/Terminfo/Eval.hs	2012-07-10 22:00:11.364867008 +1000
@@ -167,7 +167,7 @@
     let ( !start_ptr, _ ) = cap_bytes cap
         !src_ptr = start_ptr `plusPtr` offset
         !out_ptr' = out_ptr `plusPtr` next_offset
-    liftIO $! memcpy out_ptr src_ptr byte_count
+    liftIO $! memcpy out_ptr src_ptr (fromIntegral byte_count)
     return $! out_ptr'
 serialize_cap_op out_ptr DecOut = do
     p <- pop
