--- vty-4.7.0.18-orig/vty.cabal	2012-10-21 13:36:47.000000000 +1100
+++ vty-4.7.0.18/vty.cabal	2012-10-21 15:12:28.744615896 +1100
@@ -91,8 +91,8 @@
   type:                detailed-0.9
   hs-source-dirs:      src test
   test-module:         VerifyAttributeOps
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -142,8 +142,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -193,8 +193,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -221,8 +221,8 @@
   other-modules:       Graphics.Vty.Picture
                        Verify
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -252,8 +252,8 @@
                        Codec.Binary.UTF8.Width
                        Verify
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -295,8 +295,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -332,8 +332,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -379,8 +379,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -410,8 +410,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -440,8 +440,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -491,8 +491,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -542,8 +542,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
@@ -576,8 +576,8 @@
 
   include-dirs:        cbits
 
-  build-depends:       Cabal == 1.17.*,
-                       QuickCheck == 2.4.*,
+  build-depends:       Cabal >= 1.10,
+                       QuickCheck >= 2.4 && < 2.6,
                        random == 1.0.*,
                        base >= 4 && < 5,
                        bytestring,
--- vty-4.7.0.18-orig/src/Data/Terminfo/Eval.hs	2012-10-21 13:36:47.000000000 +1100
+++ vty-4.7.0.18/src/Data/Terminfo/Eval.hs	2012-10-21 15:16:54.886812483 +1100
@@ -166,7 +166,7 @@
     let ( !start_ptr, _ ) = cap_bytes cap
         !src_ptr = start_ptr `plusPtr` offset
         !out_ptr' = out_ptr `plusPtr` next_offset
-    liftIO $! memcpy out_ptr src_ptr byte_count
+    liftIO $! memcpy out_ptr src_ptr ((toEnum . fromEnum) byte_count)
     return $! out_ptr'
 serialize_cap_op out_ptr DecOut = do
     p <- pop
--- vty-4.7.0.18-orig/test/VerifyDisplayAttributes.hs	2012-10-21 13:36:47.000000000 +1100
+++ vty-4.7.0.18/test/VerifyDisplayAttributes.hs	2012-10-21 15:21:34.721459636 +1100
@@ -1,6 +1,5 @@
 module VerifyDisplayAttributes where
 
-import Verify.Graphics.Vty.DisplayAttributes
 import Verify.Graphics.Vty.Attributes
 
 import Verify
--- vty-4.7.0.18-orig/test/VerifyPictureToSpan.hs	2012-10-21 13:36:47.000000000 +1100
+++ vty-4.7.0.18/test/VerifyPictureToSpan.hs	2012-10-21 15:21:23.172151773 +1100
@@ -2,7 +2,6 @@
 
 import Graphics.Vty.Picture
 import Graphics.Vty.Span
-import Graphics.Vty.PictureToSpans
 
 import Verify
 
