--- groundhog-sqlite-0.0.1.1-orig/groundhog-sqlite.cabal	2011-06-17 17:55:50.000000000 +1000
+++ groundhog-sqlite-0.0.1.1/groundhog-sqlite.cabal	2011-12-12 01:12:29.439316521 +1100
@@ -32,4 +32,6 @@
         extra-libraries: sqlite3
     else
         c-sources:   cbits/sqlite3.c
-    default-language: Haskell2010
\ No newline at end of file
+    default-language: Haskell2010
+    if impl(ghc >= 7.2)
+       extensions: NondecreasingIndentation
