diff --git a/src/Main.hs b/src/Main.hs
index 8ab0080..b64f9f1 100644
--- a/src/Main.hs
+++ b/src/Main.hs
@@ -6,2 +6,3 @@
 {-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE CPP #-}
 
@@ -164,3 +165,7 @@ pre x i =
            -- <deriving (X,Y,Z)> becomes <deriving (<X,Y,Z>)
+#if MIN_VERSION_haskell_src_exts(1,20,0)
+           Just (Deriving _ Nothing ds@(_:_)) ->
+#else
            Just (Deriving _ ds@(_:_)) ->
+#endif
              [spanHSE (show "InstHeads")
@@ -278,4 +283,3 @@ defaultExtensions :: [Extension]
 defaultExtensions =
-  [e | e@EnableExtension{} <- knownExtensions] \\
-  map EnableExtension badExtensions
+  [e | e@EnableExtension{} <- knownExtensions] \\ map EnableExtension badExtensions
 
