--- srcloc-0.2.1-orig/Data/Loc.hs	2012-07-31 22:50:22.000000000 +1000
+++ srcloc-0.2.1/Data/Loc.hs	2012-09-14 18:43:30.819991404 +1000
@@ -75,10 +75,10 @@
 import Data.List (foldl')
 
 -- | Position type.
-data Pos = Pos !String             -- ^ File
-               {-# UNPACK #-} !Int -- ^ Line
-               {-# UNPACK #-} !Int -- ^ Column
-               {-# UNPACK #-} !Int -- ^ Character offset
+data Pos = Pos !String             -- File
+               {-# UNPACK #-} !Int -- Line
+               {-# UNPACK #-} !Int -- Column
+               {-# UNPACK #-} !Int -- Character offset
 #ifdef __GLASGOW_HASKELL__
   deriving (Eq, Show, Data, Typeable)
 #else
