--- liblawless-0.26.0-orig/liblawless.cabal	2017-10-02 15:27:26.000000000 +1100
+++ liblawless-0.26.0/liblawless.cabal	2018-02-11 23:50:34.411619830 +1100
@@ -89,7 +89,7 @@
                      TypeOperators
                      TypeSynonymInstances
   build-depends:
-                aeson                      >= 0.11.2 && < 0.12,
+                aeson                      >= 1.0.0,
                 base                       >= 4.9 && < 5,
                 base-unicode-symbols       >= 0.2.2 && < 0.3,
                 binary                     >= 0.8.3 && < 0.9,
@@ -156,7 +156,7 @@
                 Paths_liblawless
   build-depends:
                 QuickCheck                 >= 2.8 && < 2.10,
-                aeson < 1.0,
+                aeson,
                 base >= 4.9 && < 5,
                 binary >= 0.7.5.0,
                 bytestring,
--- liblawless-0.26.0-orig/Source/Aeson.hs	2017-10-02 15:27:26.000000000 +1100
+++ liblawless-0.26.0/Source/Aeson.hs	2018-02-11 23:51:11.635608372 +1100
@@ -31,8 +31,8 @@
     omitNothingFields = False,
     sumEncoding = ObjectWithSingleField}
 
-lawlessToJSONEncoding ∷ ∀ a. (GToEncoding (Rep a), Generic a) ⇒ a → Encoding
+lawlessToJSONEncoding ∷ ∀ a. (GToEncoding Zero (Rep a), Generic a) ⇒ a → Encoding
 lawlessToJSONEncoding = genericToEncoding lawlessJSONOptions
 
-lawlessParseJSON ∷ ∀ a. (GFromJSON (Rep a), Generic a) ⇒ Value → Parser a
+lawlessParseJSON ∷ ∀ a. (GFromJSON Zero (Rep a), Generic a) ⇒ Value → Parser a
 lawlessParseJSON = genericParseJSON lawlessJSONOptions
