diff --git a/shell-conduit.cabal b/shell-conduit.cabal
index a60124e..b72d24f 100644
--- a/shell-conduit.cabal
+++ b/shell-conduit.cabal
@@ -30,7 +30,7 @@ library
                    , control-monad-loop
                    , directory
                    , filepath
-                   , monad-control
+                   , monad-control >= 1.0
                    , monads-tf
                    , process
                    , resourcet
diff --git a/src/Data/Conduit/Shell/Types.hs b/src/Data/Conduit/Shell/Types.hs
index 40d8d94..1e2c991 100644
--- a/src/Data/Conduit/Shell/Types.hs
+++ b/src/Data/Conduit/Shell/Types.hs
@@ -34,15 +34,14 @@ deriving instance (MonadResourceBase m) => MonadResource (ShellT m)
 
 -- | Dumb instance.
 instance (MonadThrow m,MonadIO m,MonadBaseControl IO m) => MonadBaseControl IO (ShellT m) where
-  newtype StM (ShellT m) a = StMShell{unStMGeoServer ::
-                                    StM (ResourceT m) a}
+  type StM (ShellT m) a = StM (ResourceT m) a
   liftBaseWith f =
     ShellT (liftBaseWith
               (\run ->
-                 f (liftM StMShell .
+                 f (liftM id .
                     run .
                     runShellT)))
-  restoreM = ShellT . restoreM . unStMGeoServer
+  restoreM = ShellT . restoreM
 
 -- | Intentionally only handles 'ShellException'. Use normal exception
 -- handling to handle usual exceptions.
