diff --git a/src/Data/Edison/Assoc/StandardMap.hs b/src/Data/Edison/Assoc/StandardMap.hs
index 54f9e73..7d15a80 100644
--- a/src/Data/Edison/Assoc/StandardMap.hs
+++ b/src/Data/Edison/Assoc/StandardMap.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 -- |
@@ -362,2 +363,3 @@ instance Ord k => A.OrdFiniteMap (FM k) k
 
+#if ! MIN_VERSION_QuickCheck(2,8,2)
 instance (Ord k,Arbitrary k,Arbitrary a) => Arbitrary (FM k a) where
@@ -368 +370,2 @@ instance (Ord k,CoArbitrary k,CoArbitrary a) => CoArbitrary (FM k a) where
    coarbitrary mp = coarbitrary (A.toList mp)
+#endif
diff --git a/src/Data/Edison/Coll/StandardSet.hs b/src/Data/Edison/Coll/StandardSet.hs
index ad14cc8..55f12f6 100644
--- a/src/Data/Edison/Coll/StandardSet.hs
+++ b/src/Data/Edison/Coll/StandardSet.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 -- |
@@ -265,2 +266,3 @@ instance Ord a => C.OrdSet (Set a) a
 
+#if ! MIN_VERSION_QuickCheck(2,8,2)
 instance (Ord a, Arbitrary a) => Arbitrary (Set a) where
@@ -271 +273,2 @@ instance (Ord a, CoArbitrary a) => CoArbitrary (Set a) where
   coarbitrary set = coarbitrary (C.toList set)
+#endif
