--- threadscope-0.2.2-orig/GUI/SaveAs.hs	2012-11-02 15:57:16.000000000 +1100
+++ threadscope-0.2.2/GUI/SaveAs.hs	2013-12-10 11:59:56.633876157 +1100
@@ -9,7 +9,9 @@
 
 -- Imports for GTK
 import Graphics.UI.Gtk
-import Graphics.Rendering.Cairo
+import Graphics.Rendering.Cairo (Format(..), Operator(..), Render, fill, getOperator,
+                                 liftIO, rectangle, renderWith, setOperator, surfaceWriteToPNG,
+                                 translate, withImageSurface, withPDFSurface)
 
 saveAs :: HECs -> ViewParameters -> Double -> DrawingArea
        -> (Int, Int, Render ())
--- threadscope-0.2.2-orig/GUI/Timeline.hs	2012-11-02 15:57:16.000000000 +1100
+++ threadscope-0.2.2/GUI/Timeline.hs	2013-12-10 11:57:16.580877683 +1100
@@ -34,7 +34,7 @@
 import Events.HECs
 
 import Graphics.UI.Gtk
-import Graphics.Rendering.Cairo
+import Graphics.Rendering.Cairo (liftIO)
 
 import Data.IORef
 import Control.Monad
--- threadscope-0.2.2-orig/GUI/Timeline/Render.hs	2012-11-02 15:57:16.000000000 +1100
+++ threadscope-0.2.2/GUI/Timeline/Render.hs	2013-12-10 12:23:31.130862667 +1100
@@ -21,8 +21,10 @@
 import GUI.Timeline.CairoDrawing
 
 import Graphics.UI.Gtk
-import Graphics.Rendering.Cairo
-
+import Graphics.Rendering.Cairo (Content(..), Operator(..), Render, Surface,
+                                 clip, createSimilarSurface, fill, liftIO, lineTo, moveTo, paint,
+                                 rectangle, renderWith, restore, save, scale, setSourceRGBA, setLineWidth, setOperator,
+                                 setSourceSurface, stroke, surfaceFinish, translate, withTargetSurface)
 import Data.IORef
 import Control.Monad
 
