diff --git a/src/System/Console/CmdTheLine/ArgVal.hs b/src/System/Console/CmdTheLine/ArgVal.hs
index 7ce8e51..41228e2 100644
--- a/src/System/Console/CmdTheLine/ArgVal.hs
+++ b/src/System/Console/CmdTheLine/ArgVal.hs
@@ -34,2 +34,3 @@ import Text.Parsec         hiding ( char )
 import Text.PrettyPrint
+import Prelude hiding ((<>))
 
diff --git a/src/System/Console/CmdTheLine/Err.hs b/src/System/Console/CmdTheLine/Err.hs
index f5473bd..86bd8e6 100644
--- a/src/System/Console/CmdTheLine/Err.hs
+++ b/src/System/Console/CmdTheLine/Err.hs
@@ -15,2 +15,3 @@ import Control.Monad.Trans.Error
 import System.IO
+import Prelude hiding ((<>))
 
