--- a/src/Data/IxSet/Typed.hs
+++ b/src/Data/IxSet/Typed.hs
@@ -393,7 +393,11 @@ instance (Indexable ixs a, Show a) => Show (IxSet ixs a) where
 instance (Indexable ixs a, Read a) => Read (IxSet ixs a) where
   readsPrec n = map (first fromSet) . readsPrec n
 
+#if MIN_VERSION_safecopy(0,10,0)
+instance (Indexable ixs a, SafeCopy a, Typeable ixs, Typeable a) => SafeCopy (IxSet ixs a) where
+#else
 instance (Indexable ixs a, SafeCopy a) => SafeCopy (IxSet ixs a) where
+#endif
   putCopy = contain . safePut . toList
   getCopy = contain $ fmap fromList safeGet
 
