diff --git a/library/Hbro.hs b/library/Hbro.hs
index 63a2888..bef4d05 100644
--- a/library/Hbro.hs
+++ b/library/Hbro.hs
@@ -21,3 +21,3 @@ import           Hbro.IPC          as X
 -- import           Hbro.Options
-import           Hbro.Prelude      as X
+import           Hbro.Prelude      as X hiding (Handler)
 -- import           Hbro.WebView.Signals
diff --git a/library/Hbro/Boot.hs b/library/Hbro/Boot.hs
index ff29150..3e6112d 100644
--- a/library/Hbro/Boot.hs
+++ b/library/Hbro/Boot.hs
@@ -21,3 +21,3 @@ import           Hbro.Logger
 import           Hbro.Options                    as Options
-import           Hbro.Prelude
+import           Hbro.Prelude hiding (withAsyncBound, withAsync)
 
diff --git a/library/Hbro/Defaults.hs b/library/Hbro/Defaults.hs
index 0d7bf43..9c1119f 100644
--- a/library/Hbro/Defaults.hs
+++ b/library/Hbro/Defaults.hs
@@ -23,4 +23,4 @@ import           Hbro.Keys                                as Keys
 import           Hbro.Keys.Model                          (singleKey, (.|))
-import           Hbro.Logger
-import           Hbro.Prelude
+import           Hbro.Logger hiding (Handler)
+import           Hbro.Prelude hiding (Handler)
 import           Hbro.WebView.Signals
diff --git a/library/Hbro/Event.hs b/library/Hbro/Event.hs
index 4e145df..a5a1ac9 100644
--- a/library/Hbro/Event.hs
+++ b/library/Hbro/Event.hs
@@ -25,3 +25,3 @@ module Hbro.Event (
 -- {{{ Imports
-import           Hbro.Prelude
+import           Hbro.Prelude hiding (Handler)
 
diff --git a/library/Hbro/Gui/PromptBar.hs b/library/Hbro/Gui/PromptBar.hs
index e394c60..fc44053 100644
--- a/library/Hbro/Gui/PromptBar.hs
+++ b/library/Hbro/Gui/PromptBar.hs
@@ -42,3 +42,3 @@ import           Hbro.Gui.Builder
 import           Hbro.Logger
-import           Hbro.Prelude                             hiding (on)
+import           Hbro.Prelude                             hiding (on, throwM)
 
diff --git a/library/Hbro/Prelude.hs b/library/Hbro/Prelude.hs
index 2d02e48..1bf02cb 100644
--- a/library/Hbro/Prelude.hs
+++ b/library/Hbro/Prelude.hs
@@ -32,3 +32,4 @@ import           ClassyPrelude                 as X hiding (Builder (..),
                                                      defaultTimeLocale, error,
-                                                     log, toList)
+                                                     log, toList, asks, tail,
+                                                     async, cancel)
 
diff --git a/library/Hbro/WebView/Signals.hs b/library/Hbro/WebView/Signals.hs
index ce9301f..af22c10 100644
--- a/library/Hbro/WebView/Signals.hs
+++ b/library/Hbro/WebView/Signals.hs
@@ -13,3 +13,3 @@ import           Hbro.Keys.Model                            ((.|))
 import           Hbro.Logger
-import           Hbro.Prelude                               hiding (on)
+import           Hbro.Prelude                               hiding (on, throwM)
 
