--- a/src/Text/Hoodle/Builder.hs
+++ b/src/Text/Hoodle/Builder.hs
@@ -36,2 +36,3 @@ import           Data.Hoodle.Simple
 -- 
+import Prelude hiding ((<>))
 
--- a/src/Text/Hoodle/Builder/V0_1_1.hs
+++ b/src/Text/Hoodle/Builder/V0_1_1.hs
@@ -32,2 +32,3 @@ import Data.Strict.Tuple
 import Data.Hoodle.Simple.V0_1_1
+import Prelude hiding ((<>))
 
--- a/src/Text/Hoodle/Builder/V0_2_2.hs
+++ b/src/Text/Hoodle/Builder/V0_2_2.hs
@@ -34,2 +34,3 @@ import           Data.Hoodle.Simple.V0_2_2
 -- 
+import Prelude hiding ((<>))
 
--- a/src/Text/Hoodle/Builder/V0_3.hs
+++ b/src/Text/Hoodle/Builder/V0_3.hs
@@ -36,2 +36,3 @@ import           Data.Hoodle.Simple
 -- 
+import Prelude hiding ((<>))
 
--- a/src/Text/Hoodlet/Builder.hs
+++ b/src/Text/Hoodlet/Builder.hs
@@ -22,2 +22,3 @@ import           Data.Hoodle.Simple
 import           Text.Hoodle.Builder
+import Prelude hiding ((<>))
 --
