--- text-0.11.2.2-orig/text.cabal	2012-06-16 09:52:25.000000000 +1000
+++ text-0.11.2.2/text.cabal	2012-08-07 19:25:36.100565576 +1000
@@ -147,7 +147,7 @@
       build-depends: integer-simple >= 0.1 && < 0.5
     else
       cpp-options: -DINTEGER_GMP
-      build-depends: integer-gmp >= 0.2 && < 0.5
+      build-depends: integer-gmp >= 0.2 && < 0.6
    
 
   if impl(ghc >= 6.9) && impl(ghc < 6.11)
@@ -174,7 +174,7 @@
     directory                  >= 1.0 && < 1.2,
     ghc-prim,
     random                     >= 1.0 && < 1.1,
-    QuickCheck                 >= 2.4 && < 2.5,
+    QuickCheck                 >= 2.4 && < 2.6,
     HUnit                      >= 1.2 && < 1.3,
     test-framework             >= 0.4 && < 0.7,
     test-framework-quickcheck2 >= 0.2 && < 0.3,
