--- text-0.11.2.2-orig/Data/Text/IO.hs	2012-06-16 09:52:25.000000000 +1000
+++ text-0.11.2.2/Data/Text/IO.hs	2012-06-27 12:37:27.796691182 +1000
@@ -38,7 +38,11 @@
     ) where
 
 import Data.Text (Text)
-import Prelude hiding (appendFile, catch, getContents, getLine, interact,
+import Prelude hiding (appendFile,
+#if __GLASGOW_HASKELL__ < 705
+                       catch,
+#endif
+                       getContents, getLine, interact,
                        putStr, putStrLn, readFile, writeFile)
 import System.IO (Handle, IOMode(..), hPutChar, openFile, stdin, stdout,
                   withFile)
--- text-0.11.2.2-orig/Data/Text/Lazy/IO.hs	2012-06-16 09:52:25.000000000 +1000
+++ text-0.11.2.2/Data/Text/Lazy/IO.hs	2012-06-27 12:36:36.131493664 +1000
@@ -38,7 +38,11 @@
     ) where
 
 import Data.Text.Lazy (Text)
-import Prelude hiding (appendFile, catch, getContents, getLine, interact,
+import Prelude hiding (appendFile,
+#if __GLASGOW_HASKELL__ < 705
+                       catch,
+#endif
+                       getContents, getLine, interact,
                        putStr, putStrLn, readFile, writeFile)
 import System.IO (Handle, IOMode(..), hPutChar, openFile, stdin, stdout,
                   withFile)
--- text-0.11.2.2-orig/Data/Text/IO/Internal.hs	2012-06-16 09:52:25.000000000 +1000
+++ text-0.11.2.2/Data/Text/IO/Internal.hs	2012-06-27 12:35:44.393294355 +1000
@@ -32,7 +32,9 @@
                       withRawBuffer, writeCharBuf)
 import GHC.IO.Handle.Internals (ioe_EOF, readTextDevice, wantReadableHandle_)
 import GHC.IO.Handle.Types (Handle__(..), Newline(..))
+#if __GLASGOW_HASKELL__ < 705
 import Prelude hiding (catch)
+#endif
 import System.IO (Handle)
 import System.IO.Error (isEOFError)
 import qualified Data.Text as T
--- text-0.11.2.2-orig/tests/tests/src/Data/Text/Tests/Properties.hs	2012-06-16 09:52:25.000000000 +1000
+++ text-0.11.2.2/tests/tests/src/Data/Text/Tests/Properties.hs	2012-06-27 16:51:51.500041772 +1000
@@ -26,7 +26,11 @@
 import Data.Text.Search (indices)
 import Data.Word (Word8, Word16, Word32)
 import Numeric (showHex)
-import Prelude hiding (catch, replicate)
+import Prelude hiding (
+#if __GLASGOW_HASKELL__ < 705
+  catch,
+#endif
+  replicate)
 import Test.Framework (Test, testGroup)
 import Test.Framework.Providers.QuickCheck2 (testProperty)
 import qualified Data.Bits as Bits (shiftL, shiftR)
