From ca34dec7c311b271211c94cbb9abe2c305fb4a67 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Mon, 10 Dec 2018 11:28:19 -0500
Subject: [PATCH 1/9] Bump megaparec versions in idris.cabal and stack.yaml

---
 idris.cabal | 3 ++-
 stack.yaml  | 5 ++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/idris.cabal b/idris.cabal
index a3ef4793c..7eca90c16 100644
--- a/idris.cabal
+++ b/idris.cabal
@@ -289,10 +289,11 @@ Library
                 , fingertree >= 0.1.4.1 && < 0.2
                 , haskeline >= 0.7 && < 0.8
                 , ieee754 >= 0.7 && < 0.9
-                , megaparsec >= 6.2 && < 7
+                , megaparsec >= 7.0.4 && < 8
                 , mtl >= 2.1 && < 2.3
                 , network < 2.8
                 , optparse-applicative >= 0.13 && < 0.15
+                , parser-combinators >= 1.0.0
                 , pretty < 1.2
                 , process < 1.7
                 , regex-tdfa >= 1.2
diff --git a/stack.yaml b/stack.yaml
index f19bcfe3b..e19218d46 100644
--- a/stack.yaml
+++ b/stack.yaml
@@ -4,7 +4,10 @@ resolver: lts-11.14
 packages:
   - location: .
 
-extra-deps: []
+extra-deps:
+  - megaparsec-7.0.4
+  - parser-combinators-1.0.0
+
 
 flags:
   idris:

From 8af277258e3ee2a3017c27798538c1f72b603149 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Tue, 11 Dec 2018 01:00:12 -0500
Subject: [PATCH 2/9] Change anyChar to anySingle

---
 src/Idris/Parser/Expr.hs |  4 ++--
 src/Idris/REPL/Parser.hs | 11 +++++------
 2 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/src/Idris/Parser/Expr.hs b/src/Idris/Parser/Expr.hs
index 8337e3f16..ea552af81 100644
--- a/src/Idris/Parser/Expr.hs
+++ b/src/Idris/Parser/Expr.hs
@@ -27,7 +27,7 @@ import Data.Maybe
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
 import qualified Text.Megaparsec.Char as P
-import qualified Text.Megaparsec.Expr as P
+import qualified Control.Monad.Combinators.Expr as P
 
 -- | Allow implicit type declarations
 allowImp :: SyntaxInfo -> SyntaxInfo
@@ -1406,7 +1406,7 @@ VerbatimString_t ::=
  -}
 verbatimStringLiteral :: Parsing m => m String
 verbatimStringLiteral = token $ do P.try $ string "\"\"\""
-                                   str <- P.manyTill P.anyChar $ P.try (string "\"\"\"")
+                                   str <- P.manyTill P.anySingle $ P.try (string "\"\"\"")
                                    moreQuotes <- P.many $ P.char '"'
                                    return $ str ++ moreQuotes
 
diff --git a/src/Idris/REPL/Parser.hs b/src/Idris/REPL/Parser.hs
index c3a3b61d2..1c146f8c8 100644
--- a/src/Idris/REPL/Parser.hs
+++ b/src/Idris/REPL/Parser.hs
@@ -33,7 +33,6 @@ import Data.List.Split (splitOn)
 import System.Console.ANSI (Color(..))
 import System.FilePath ((</>))
 import qualified Text.Megaparsec as P
-import qualified Text.Megaparsec.Char as P
 
 parseCmd :: IState -> String -> String -> Either IP.ParseError (Either String Command)
 parseCmd i inputname = IP.runparser pCmd i inputname . trim
@@ -68,7 +67,7 @@ parserCommandsForHelp =
   , nameArgCmd ["miss", "missing"] Missing "Show missing clauses"
   , (["doc"], NameArg, "Show internal documentation", cmd_doc)
   , (["mkdoc"], NamespaceArg, "Generate IdrisDoc for namespace(s) and dependencies"
-    , genArg "namespace" (P.many P.anyChar) MakeDoc)
+    , genArg "namespace" (P.many P.anySingle) MakeDoc)
   , (["apropos"], SeqArgs (OptionalArg PkgArgs) NameArg, " Search names, types, and documentation"
     , cmd_apropos)
   , (["s", "search"], SeqArgs (OptionalArg PkgArgs) ExprArg
@@ -198,7 +197,7 @@ pCmd = P.choice [ do c <- cmd names; parser c
     where nop = do P.eof; return (Right NOP)
           unrecognized = do
               IP.lchar ':'
-              cmd <- P.many P.anyChar
+              cmd <- P.many P.anySingle
               let cmd' = takeWhile (/=' ') cmd
               return (Left $ "Unrecognized command: " ++ cmd')
 
@@ -260,7 +259,7 @@ nameArg = genArg "name" IP.name
 fnNameArg = genArg "functionname" IP.fnName
 
 strArg :: (String -> Command) -> String -> IP.IdrisParser (Either String Command)
-strArg = genArg "string" (P.many P.anyChar)
+strArg = genArg "string" (P.many P.anySingle)
 
 moduleArg :: (FilePath -> Command) -> String -> IP.IdrisParser (Either String Command)
 moduleArg = genArg "module" (fmap toPath IP.identifier)
@@ -342,7 +341,7 @@ cmd_execute name = do
 
 cmd_dynamic :: String -> IP.IdrisParser (Either String Command)
 cmd_dynamic name = do
-    let optArg = do l <- P.many P.anyChar
+    let optArg = do l <- P.many P.anySingle
                     if (l /= "")
                         then return $ Right (DynamicLink l)
                         else return $ Right ListDynamic
@@ -440,7 +439,7 @@ cmd_unlet name = Right . Undefine <$> P.many IP.name
 cmd_loadto :: String -> IP.IdrisParser (Either String Command)
 cmd_loadto name = do
     toline <- fromInteger <$> IP.natural
-    f <- P.many P.anyChar
+    f <- P.many P.anySingle
     return (Right (Load f (Just toline)))
 
 cmd_colour :: String -> IP.IdrisParser (Either String Command)

From 80b60a66ecb55eacac4fa7c37d4bd96e741ff890 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Tue, 11 Dec 2018 01:00:42 -0500
Subject: [PATCH 3/9] Remove unnecesary imports and fix small hlint warning

---
 src/Idris/Package/Parser.hs | 3 +--
 src/Idris/Parser/Ops.hs     | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/Idris/Package/Parser.hs b/src/Idris/Package/Parser.hs
index 5a101c76f..e0dbd00ab 100644
--- a/src/Idris/Package/Parser.hs
+++ b/src/Idris/Package/Parser.hs
@@ -26,14 +26,13 @@ import System.Exit
 import System.FilePath (isValid, takeExtension, takeFileName)
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
-import qualified Text.Megaparsec.Char as P
 import qualified Text.PrettyPrint.ANSI.Leijen as PP
 
 type PParser = Parser PkgDesc
 
 parseDesc :: FilePath -> IO PkgDesc
 parseDesc fp = do
-    when (not $ takeExtension fp == ".ipkg") $ do
+    unless (takeExtension fp == ".ipkg") $ do
         putStrLn $ unwords ["The presented iPKG file does not have a '.ipkg' extension:", show fp]
         exitWith (ExitFailure 1)
     res <- doesFileExist fp
diff --git a/src/Idris/Parser/Ops.hs b/src/Idris/Parser/Ops.hs
index 7b368c467..6190cd866 100644
--- a/src/Idris/Parser/Ops.hs
+++ b/src/Idris/Parser/Ops.hs
@@ -23,8 +23,7 @@ import Data.List
 import Data.List.NonEmpty (fromList)
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
-import qualified Text.Megaparsec.Char as P
-import qualified Text.Megaparsec.Expr as P
+import qualified Control.Monad.Combinators.Expr as P
 
 -- | Creates table for fixity declarations to build expression parser
 -- using pre-build and user-defined operator/fixity declarations

From 5222ad9b5bee23b5ba662818e88d3231b52f13a2 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Tue, 11 Dec 2018 01:01:19 -0500
Subject: [PATCH 4/9] Replace getPosition with getSourcePos

---
 src/Idris/Parser/Helpers.hs | 2 +-
 src/Idris/Parser/Stack.hs   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Idris/Parser/Helpers.hs b/src/Idris/Parser/Helpers.hs
index fd0a4b60c..c10adc1aa 100644
--- a/src/Idris/Parser/Helpers.hs
+++ b/src/Idris/Parser/Helpers.hs
@@ -393,7 +393,7 @@ popIndent = do ist <- get
 
 -- | Gets current indentation
 indent :: Parsing m => m Int
-indent = P.unPos . P.sourceColumn <$> P.getPosition
+indent = P.unPos . P.sourceColumn <$> P.getSourcePos
 
 -- | Gets last indentation
 lastIndent :: (MonadState IState m) => m Int
diff --git a/src/Idris/Parser/Stack.hs b/src/Idris/Parser/Stack.hs
index 61b775db6..18e6fae94 100644
--- a/src/Idris/Parser/Stack.hs
+++ b/src/Idris/Parser/Stack.hs
@@ -104,7 +104,7 @@ sourcePositionFC (P.SourcePos name line column) =
 -- This is useful when the position is needed in a way unrelated to the
 -- heirarchy of parsers.  Prefer using @withExtent@ and friends.
 getFC :: Parsing m => m FC
-getFC = sourcePositionFC <$> P.getPosition
+getFC = sourcePositionFC <$> P.getSourcePos
 
 -- | Add an extent (widen) our current parsing context.
 addExtent :: MonadWriter FC m => FC -> m ()

From d3a619db58569afedd3cdb8ed13098950161df98 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Tue, 11 Dec 2018 01:01:36 -0500
Subject: [PATCH 5/9] Use megaparsec's new ParseErrorBundle type

The Idris ParseError is not just a newtype wrapper over megaparsec's
ParseErrorBundle, since that type now includes the source file name.
---
 src/Idris/Parser/Stack.hs | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/Idris/Parser/Stack.hs b/src/Idris/Parser/Stack.hs
index 18e6fae94..c3e4ef215 100644
--- a/src/Idris/Parser/Stack.hs
+++ b/src/Idris/Parser/Stack.hs
@@ -57,23 +57,24 @@ type Parsing m = (P.MonadParsec Void String m, MonadWriter FC m)
 runparser :: Parser st res -> st -> String -> String -> Either ParseError res
 runparser p i inputname s =
   case P.parse (runWriterT (evalStateT p i)) inputname s of
-    Left err -> Left $ ParseError s err
+    Left err -> Left $ ParseError err
     Right v  -> Right $ fst v
 
 {- * Parse errors -}
 
-data ParseError = ParseError String (P.ParseError (P.Token String) Void)
+newtype ParseError = ParseError (P.ParseErrorBundle String Void)
 
 instance Message ParseError where
-  messageExtent (ParseError _ err) = sourcePositionFC pos
+  messageExtent (ParseError err) = sourcePositionFC pos
     where
-      (pos NonEmpty.:| _) = P.errorPos err
-  messageText (ParseError _ err) = PP.text . init . P.parseErrorTextPretty $ err
-  messageSource (ParseError src _) = Just src
+      (pos, _) = P.reachOffsetNoLine (P.errorOffset err') (P.bundlePosState err)
+      err' = NonEmpty.head . P.bundleErrors $ err
+  messageText (ParseError err) = PP.text . init . P.errorBundlePretty $ err
+  messageSource (ParseError err) = Just . P.sourceName . P.pstateSourcePos . P.bundlePosState $ err
 
 -- | A fully formatted parse error, with caret and bar, etc.
 prettyError                    :: ParseError -> String
-prettyError (ParseError s err) = P.parseErrorPretty' s err
+prettyError (ParseError err) = P.errorBundlePretty err
 
 {- * Mark and restore -}
 

From 4994292dc8a05984c266fd506a1461197f652239 Mon Sep 17 00:00:00 2001
From: Joseph Morag <jm4157@columbia.edu>
Date: Tue, 11 Dec 2018 01:30:22 -0500
Subject: [PATCH 6/9] Run stylize.sh

---
 src/Idris/Parser/Expr.hs | 2 +-
 src/Idris/Parser/Ops.hs  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Idris/Parser/Expr.hs b/src/Idris/Parser/Expr.hs
index ea552af81..f73f4353a 100644
--- a/src/Idris/Parser/Expr.hs
+++ b/src/Idris/Parser/Expr.hs
@@ -20,6 +20,7 @@ import Prelude hiding (pi)
 import Control.Applicative
 import Control.Arrow (left)
 import Control.Monad
+import qualified Control.Monad.Combinators.Expr as P
 import Control.Monad.State.Strict
 import Data.Function (on)
 import Data.List
@@ -27,7 +28,6 @@ import Data.Maybe
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
 import qualified Text.Megaparsec.Char as P
-import qualified Control.Monad.Combinators.Expr as P
 
 -- | Allow implicit type declarations
 allowImp :: SyntaxInfo -> SyntaxInfo
diff --git a/src/Idris/Parser/Ops.hs b/src/Idris/Parser/Ops.hs
index 6190cd866..3bac4c91b 100644
--- a/src/Idris/Parser/Ops.hs
+++ b/src/Idris/Parser/Ops.hs
@@ -17,13 +17,13 @@ import Prelude hiding (pi)
 
 import Control.Applicative
 import Control.Monad
+import qualified Control.Monad.Combinators.Expr as P
 import Control.Monad.State.Strict
 import Data.Char (isAlpha)
 import Data.List
 import Data.List.NonEmpty (fromList)
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
-import qualified Control.Monad.Combinators.Expr as P
 
 -- | Creates table for fixity declarations to build expression parser
 -- using pre-build and user-defined operator/fixity declarations

From a20cb92d11036d5acf0769a1a4be4760d7bed4da Mon Sep 17 00:00:00 2001
From: Joseph Morag <joseph.morag@columbia.edu>
Date: Tue, 11 Dec 2018 11:53:24 -0500
Subject: [PATCH 7/9] Remove unnecessary import from parser

---
 src/Idris/Parser.hs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Idris/Parser.hs b/src/Idris/Parser.hs
index 00e21a124..4580e7b5d 100644
--- a/src/Idris/Parser.hs
+++ b/src/Idris/Parser.hs
@@ -60,7 +60,6 @@ import qualified System.Directory as Dir (doesFileExist, getModificationTime,
 import System.FilePath
 import Text.Megaparsec ((<?>))
 import qualified Text.Megaparsec as P
-import qualified Text.Megaparsec.Char as P
 import qualified Text.PrettyPrint.ANSI.Leijen as PP
 
 {-

From 4a8b9c22f034279dee40cf699ed876b50f681785 Mon Sep 17 00:00:00 2001
From: Jorge Acereda <jacereda@gmail.com>
Date: Fri, 11 Jan 2019 21:42:58 +0100
Subject: [PATCH 8/9] Relax Cabal and zip-archive bounds

---
 idris.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/idris.cabal b/idris.cabal
index ceb004ebd..3bedec2af 100644
--- a/idris.cabal
+++ b/idris.cabal
@@ -100,7 +100,7 @@ source-repository head
 
 custom-setup
   setup-depends:
-    Cabal >= 1.10 && <2.3,
+    Cabal >= 1.10 && <2.5,
     base  >= 4 && <5,
     directory,
     filepath,
@@ -308,7 +308,7 @@ Library
                 , utf8-string < 1.1
                 , vector < 0.13
                 , vector-binary-instances < 0.3
-                , zip-archive > 0.2.3.5 && < 0.4
+                , zip-archive > 0.2.3.5 && < 0.5
                 , fsnotify >= 0.2 && < 2.2
                 , async < 2.3
 

From 18c98cb902286e77378992294af91342975cebcc Mon Sep 17 00:00:00 2001
From: Jorge Acereda <jacereda@gmail.com>
Date: Sat, 12 Jan 2019 14:30:52 +0100
Subject: [PATCH 9/9] Position is no longer reported twice

---
 src/Idris/Output.hs       |  7 ++++---
 src/Idris/Parser/Stack.hs | 26 ++++++++++++++++++--------
 2 files changed, 22 insertions(+), 11 deletions(-)

diff --git a/src/Idris/Output.hs b/src/Idris/Output.hs
index 78a7da11c..70b4d48a3 100644
--- a/src/Idris/Output.hs
+++ b/src/Idris/Output.hs
@@ -75,9 +75,10 @@ instance Message SimpleWarning where
 formatMessage :: Message w => w -> Idris OutputDoc
 formatMessage w = do
     i <- getIState
-    maybeSource <- case messageSource w of
-                     Just src -> pure (Just src)
-                     Nothing  -> readSource fc
+    rs <- readSource fc
+    let maybeSource = case rs of
+          Just src -> Just src
+          Nothing -> messageSource w
     let maybeFormattedSource = maybeSource >>= layoutSource fc (regions (idris_highlightedRegions i))
     return $ layoutMessage (layoutFC fc) maybeFormattedSource (messageText w)
   where
diff --git a/src/Idris/Parser/Stack.hs b/src/Idris/Parser/Stack.hs
index c3e4ef215..f75789ab6 100644
--- a/src/Idris/Parser/Stack.hs
+++ b/src/Idris/Parser/Stack.hs
@@ -62,19 +62,29 @@ runparser p i inputname s =
 
 {- * Parse errors -}
 
-newtype ParseError = ParseError (P.ParseErrorBundle String Void)
+newtype ParseError = ParseError { unParseError :: P.ParseErrorBundle String Void }
+
+parseError :: ParseError -> P.ParseError String Void
+parseError = NonEmpty.head . P.bundleErrors . unParseError
+
+parseErrorPosState :: ParseError -> P.PosState String
+parseErrorPosState = P.bundlePosState . unParseError
+
+parseErrorOffset :: ParseError -> Int
+parseErrorOffset = P.errorOffset . parseError
 
 instance Message ParseError where
-  messageExtent (ParseError err) = sourcePositionFC pos
+  messageExtent err = sourcePositionFC pos
+    where
+      (pos, _) = P.reachOffsetNoLine (parseErrorOffset err) (parseErrorPosState err)
+  messageText = PP.text . init . P.parseErrorTextPretty . parseError
+  messageSource err = Just sline
     where
-      (pos, _) = P.reachOffsetNoLine (P.errorOffset err') (P.bundlePosState err)
-      err' = NonEmpty.head . P.bundleErrors $ err
-  messageText (ParseError err) = PP.text . init . P.errorBundlePretty $ err
-  messageSource (ParseError err) = Just . P.sourceName . P.pstateSourcePos . P.bundlePosState $ err
+      (_, sline, _) = P.reachOffset (parseErrorOffset err) (parseErrorPosState err)
 
 -- | A fully formatted parse error, with caret and bar, etc.
-prettyError                    :: ParseError -> String
-prettyError (ParseError err) = P.errorBundlePretty err
+prettyError :: ParseError -> String
+prettyError =  P.errorBundlePretty . unParseError
 
 {- * Mark and restore -}
 
