diff --git a/src/Idris/AbsSyntaxTree.hs b/src/Idris/AbsSyntaxTree.hs
index fa76245..bfb062c 100644
--- a/src/Idris/AbsSyntaxTree.hs
+++ b/src/Idris/AbsSyntaxTree.hs
@@ -2,5 +2,6 @@
              DeriveDataTypeable, TypeSynonymInstances, PatternGuards #-}
-
+{-# LANGUAGE CPP #-}
 module Idris.AbsSyntaxTree where
 
+import Prelude hiding ((<$>))
 import Idris.Core.TT
@@ -1750,4 +1751,6 @@ showDeclImp _ _ = text "..."
 
+#if ! MIN_VERSION_annotated_wl_pprint(0,6,0)
 instance Show (Doc OutputAnnotation) where
   show = flip (displayS . renderCompact) ""
+#endif
 
diff --git a/src/Idris/Delaborate.hs b/src/Idris/Delaborate.hs
index 051f4d5..80d1666 100644
--- a/src/Idris/Delaborate.hs
+++ b/src/Idris/Delaborate.hs
@@ -7,2 +7,3 @@ module Idris.Delaborate (annName, bugaddr, delab, delab', delabMV, delabTy, dela
 
+import Prelude hiding ((<$>))
 import Util.Pretty
diff --git a/src/Idris/Docs.hs b/src/Idris/Docs.hs
index 23dbad4..b6ae270 100644
--- a/src/Idris/Docs.hs
+++ b/src/Idris/Docs.hs
@@ -4,2 +4,3 @@ module Idris.Docs (pprintDocs, getDocs, pprintConstDocs, FunDoc, FunDoc'(..), Do
 
+import Prelude hiding ((<$>))
 import Idris.AbsSyntax
diff --git a/src/Idris/Docstrings.hs b/src/Idris/Docstrings.hs
index a9e7eed..1fd4e4b 100644
--- a/src/Idris/Docstrings.hs
+++ b/src/Idris/Docstrings.hs
@@ -9,2 +9,3 @@ module Idris.Docstrings (
 
+import Prelude hiding ((<$>))
 import qualified Cheapskate as C
diff --git a/src/Idris/Output.hs b/src/Idris/Output.hs
index 689ed47..9f66511 100644
--- a/src/Idris/Output.hs
+++ b/src/Idris/Output.hs
@@ -4,2 +4,3 @@ module Idris.Output where
 
+import Prelude hiding ((<$>))
 import Idris.Core.TT
diff --git a/src/Idris/REPL.hs b/src/Idris/REPL.hs
index 7807df5..be1930f 100644
--- a/src/Idris/REPL.hs
+++ b/src/Idris/REPL.hs
@@ -61,3 +61,3 @@ import Control.Category
 import qualified Control.Exception as X
-import Prelude hiding ((.), id)
+import Prelude hiding ((.), id, (<$>))
 import Data.List.Split (splitOn)
