diff --git a/lib/Prelude/Complex.idr b/lib/Prelude/Complex.idr
index 2d5b73f..b772eac 100644
--- a/lib/Prelude/Complex.idr
+++ b/lib/Prelude/Complex.idr
@@ -1,5 +1,5 @@
 {-
-  © 2012 Copyright Mekeor Melire
+  (c) 2012 Copyright Mekeor Melire
 -}
 
 
diff --git a/rts/libidris_rts.a b/rts/libidris_rts.a
deleted file mode 100644
index d1ab337..0000000
Binary files a/rts/libidris_rts.a and /dev/null differ
