commit ae55c561355224b7ab7aef28255142facfcc0356
Author: RyanGlScott <ryan.gl.scott@ku.edu>
Date:   Wed Sep 10 20:53:07 2014 -0500

    Fix for language-java 0.2.7

diff --git a/idris.cabal b/idris.cabal
index 0f959c7..6ce31d1 100644
--- a/idris.cabal
+++ b/idris.cabal
@@ -680,7 +680,7 @@ Library
                 , filepath
                 , fingertree >= 0.1
                 , haskeline >= 0.7
-                , language-java >= 0.2.6
+                , language-java >= 0.2.7
                 , lens >= 4.1.1
                 , mtl
                 , parsers >= 0.9 && < 0.13
diff --git a/src/IRTS/Java/ASTBuilding.hs b/src/IRTS/Java/ASTBuilding.hs
index cb7ace4..44c7fe6 100644
--- a/src/IRTS/Java/ASTBuilding.hs
+++ b/src/IRTS/Java/ASTBuilding.hs
@@ -60,7 +60,7 @@ localVar i = Ident $ "loc" ++ show i
 
 (@!) :: Exp -> Int -> ArrayIndex
 (@!) target pos =
-  ArrayIndex target (Lit $ Int (toInteger pos))
+  ArrayIndex target [Lit $ Int (toInteger pos)]
 
 (@:=) :: Either ArrayIndex Ident -> Exp -> BlockStmt
 (@:=) (Right lhs) rhs =
