diff --git a/src/Control/Monad/Layer.hs b/src/Control/Monad/Layer.hs
index d95232d..821e1b7 100644
--- a/src/Control/Monad/Layer.hs
+++ b/src/Control/Monad/Layer.hs
@@ -5,6 +5,7 @@
 {-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ImpredicativeTypes #-}
 
 {-|
 
@@ -270,3 +270,3 @@ class (MonadLayerControl m, MonadTransFunctor m) => MonadTransControl m where
     transControl
-        :: (forall n. (MonadTrans n, Outer n ~ Outer m)
+        :: (forall n. (MonadTrans n, Outer n ~ Outer m, n ~ m)
             => (forall b. n b -> Inner n (LayerState n b))
--- a/src/Control/Monad/Interface/Mask.hs
+++ b/src/Control/Monad/Interface/Mask.hs
@@ -10,2 +10,3 @@
 {-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ImpredicativeTypes #-}
 
