diff --git a/HFlags.hs b/HFlags.hs
index 3ac416b..bf77e18 100644
--- a/HFlags.hs
+++ b/HFlags.hs
@@ -408,3 +408,7 @@ getFlagsData = do
     where
+#if MIN_VERSION_template_haskell(2,11,0)
+      instanceToFlagData (InstanceD _ _ (AppT _ inst) _) = [| getFlagData (undefined :: $(return inst)) |]
+#else
       instanceToFlagData (InstanceD _ (AppT _ inst) _) = [| getFlagData (undefined :: $(return inst)) |]
+#endif
       instanceToFlagData _ = error "Shouldn't happen"
@@ -414,3 +418,7 @@ getFlagsData = do
       -- on the just reified instances.
+#if MIN_VERSION_template_haskell(2,11,0)
+      instanceToModuleNamePair (InstanceD _ _ (AppT _ (ConT inst)) _) =
+#else
       instanceToModuleNamePair (InstanceD _ (AppT _ (ConT inst)) _) =
+#endif
         let (flagrev, modrev) = span (/= '.') $ reverse $ show inst
