diff --git a/Test/LazySmallCheck.hs b/Test/LazySmallCheck.hs
index 0a66043..6d542d6 100644
--- a/Test/LazySmallCheck.hs
+++ b/Test/LazySmallCheck.hs
@@ -33,7 +33,7 @@ module Test.LazySmallCheck
   )
   where
 
-import Monad
+import Control.Monad
 import Control.Exception
 import System.Exit
 
diff --git a/lazysmallcheck.cabal b/lazysmallcheck.cabal
index 9fccaee..0a9d708 100644
--- a/lazysmallcheck.cabal
+++ b/lazysmallcheck.cabal
@@ -2,7 +2,7 @@ Name:               lazysmallcheck
 Version:            0.5
 Maintainer:         Matthew Naylor <mfn@cs.york.ac.uk>
 Homepage:           http://www.cs.york.ac.uk/~mfn/lazysmallcheck/
-Build-Depends:      base < 5, haskell98
+Build-Depends:      base < 5
 License:            BSD3
 License-File:       LICENSE
 Author:             Matthew Naylor and Fredrik Lindblad
@@ -16,7 +16,7 @@ Description:
   significantly fewer test-cases to verify a property for all inputs up 
   to a given depth.
 Category:           Testing
-Build-Depends:      base, haskell98
+Build-Depends:      base
 Build-Type:         Simple
 Extra-Source-Files:
   examples/Catch.hs
