--- a/test/EmbedTestMongo.hs
+++ b/test/EmbedTestMongo.hs
@@ -10,6 +10,7 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE DataKinds #-}
 {-# OPTIONS_GHC -Wno-unused-top-binds -Wno-orphans -O0 #-}
 module EmbedTestMongo (specs) where
 
--- a/test/EntityEmbedTestMongo.hs
+++ b/test/EntityEmbedTestMongo.hs
@@ -9,6 +9,7 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE DataKinds #-}
 module EntityEmbedTestMongo where
 
 -- because we are using a type alias we need to declare in a separate module
--- a/test/main.hs
+++ b/test/main.hs
@@ -9,6 +9,7 @@
 {-# LANGUAGE DerivingStrategies #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE DataKinds #-}
 {-# OPTIONS_GHC -Wno-unused-top-binds #-}
 
 import qualified Data.ByteString as BS
