--- a/src/Language/Bash/Expand.hs
+++ b/src/Language/Bash/Expand.hs
@@ -10,3 +10,3 @@ module Language.Bash.Expand
 #if __GLASGOW_HASKELL__ >= 710
-import Prelude hiding (Word)
+import Prelude hiding (Word, (<>))
 #else
--- a/src/Language/Bash/Syntax.hs
+++ b/src/Language/Bash/Syntax.hs
@@ -28,3 +28,3 @@ module Language.Bash.Syntax
 #if __GLASGOW_HASKELL__ >= 710
-import Prelude hiding (Word)
+import Prelude hiding (Word, (<>))
 #endif
--- a/src/Language/Bash/Word.hs
+++ b/src/Language/Bash/Word.hs
@@ -29,3 +29,3 @@ module Language.Bash.Word
 #if __GLASGOW_HASKELL__ >= 710
-import Prelude hiding (Word)
+import Prelude hiding (Word, (<>))
 #endif
