diff --git a/Text/HMarkup/XHtml.hs b/Text/HMarkup/XHtml.hs
index c8cfe42..a807094 100644
--- a/Text/HMarkup/XHtml.hs
+++ b/Text/HMarkup/XHtml.hs
@@ -122,10 +122,7 @@ textToHtml _ WhiteSpace = return $ toHtml " "
 markupURI :: Monad m => MarkupXHtmlPrefs m -> URI -> URI
 markupURI pref uri = case markupBaseURI pref of
                    Nothing   -> uri
-                   Just baseURI -> uri `relativeTo'` baseURI
-  where relativeTo' x y = 
-            fromMaybe (error $ "relativeTo " ++ show x ++ " " ++ show y) 
-                          (x `relativeTo` y)
+                   Just baseURI -> uri `relativeTo` baseURI
 
 markupURL :: Monad m => MarkupXHtmlPrefs m -> URI -> URL
 markupURL pref = show . markupURI pref
