--- hastache-0.6.1-orig/hastache.cabal	2014-12-23 21:40:08.000000000 +1100
+++ hastache-0.6.1/hastache.cabal	2018-04-15 16:21:22.024645708 +1000
@@ -27,7 +27,7 @@
 executable mkReadme
   main-is: mkReadme.hs
   build-depends: hastache, process,
-    base >=4 && <4.9
+    base >=4
     ,bytestring
     ,mtl
     ,transformers
@@ -46,7 +46,7 @@
     Text.Hastache.Context
 
   build-depends:
-    base >=4 && <4.9
+    base >=4
     ,bytestring
     ,mtl
     ,transformers
@@ -69,7 +69,7 @@
 
   build-depends:
     hastache
-   ,base >=4 && <4.9
+   ,base >=4
    ,directory
    ,mtl
    ,HUnit
--- hastache-0.6.1-orig/Text/Hastache.hs	2014-12-23 21:40:08.000000000 +1100
+++ hastache-0.6.1/Text/Hastache.hs	2018-04-15 21:22:04.987177215 +1000
@@ -1,5 +1,5 @@
 {-# LANGUAGE ExistentialQuantification, FlexibleInstances, IncoherentInstances,
-             OverloadedStrings #-}
+             OverloadedStrings, CPP #-}
 -- Module:      Text.Hastache
 -- Copyright:   Sergey S Lymar (c) 2011-2013 
 -- License:     BSD3
@@ -92,6 +92,9 @@
 import Data.Int
 import Data.Maybe (isJust)
 import Data.Monoid (Monoid, mappend, mempty)
+#if MIN_VERSION_base(4,9,0)
+import qualified Data.Semigroup as S (Semigroup, (<>))
+#endif
 import Data.Text hiding (map, foldl1)
 import Data.Text.IO
 import Data.Version (Version)
@@ -122,13 +125,27 @@
     Text            -- ^ Variable name
     -> m (MuType m) -- ^ Value
 
+#if MIN_VERSION_base(4,9,0)
+instance (Monad m) => S.Semigroup (MuContext m) where
+    a <> b = \v -> do
+        x <- a v
+        case x of
+            MuNothing -> b v
+            _ -> return x
+#endif
+
 instance (Monad m) => Monoid (MuContext m) where
     mempty = const $ return MuNothing
+#if MIN_VERSION_base(4,11,0)
+#elif MIN_VERSION_base(4,9,0)
+    mappend = (S.<>)
+#else
     a `mappend` b = \v -> do
         x <- a v
         case x of
             MuNothing -> b v
             _ -> return x
+#endif
 
 -- | Left-leaning compoistion of contexts. Given contexts @c1@ and
 -- @c2@, the behaviour of @(c1 <> c2) x@ is following: if @c1 x@
