--- a/src/Internal/Container.hs
+++ b/src/Internal/Container.hs
@@ -31,2 +31,3 @@ import Internal.Numeric
 import Internal.Algorithms(Field,linearSolveSVD)
+import Prelude hiding ((<>))
 
--- a/src/Internal/Convolution.hs
+++ b/src/Internal/Convolution.hs
@@ -25,3 +25,3 @@ import Internal.Conversion
 import Internal.Container
-
+import Prelude hiding ((<>))
 
--- a/src/Internal/Modular.hs
+++ b/src/Internal/Modular.hs
@@ -40,3 +40,3 @@ import Internal.Util(Normed(..),Indexable(..),
 import Internal.ST(mutable)
-import GHC.TypeLits
+import GHC.TypeLits hiding (Mod)
 import Data.Proxy(Proxy)
@@ -47,3 +47,3 @@ import Data.Complex
 import Control.DeepSeq ( NFData(..) )
-
+import Prelude hiding ((<>))
 
--- a/src/Internal/Util.hs
+++ b/src/Internal/Util.hs
@@ -81,2 +81,3 @@ import Data.Function(on)
 import Internal.ST
+import Prelude hiding ((<>))
 
--- a/src/Numeric/LinearAlgebra.hs
+++ b/src/Numeric/LinearAlgebra.hs
@@ -185,2 +185,3 @@ import Internal.CG
 import Internal.Conversion
+import Prelude hiding ((<>))
 
--- a/src/Numeric/LinearAlgebra/HMatrix.hs
+++ b/src/Numeric/LinearAlgebra/HMatrix.hs
@@ -21,2 +21,3 @@ import Internal.Util
 import Internal.Algorithms(cholSH, mbCholSH, eigSH', eigenvaluesSH', geigSH')
+import Prelude hiding ((<>))
 
--- a/src/Numeric/LinearAlgebra/Static.hs
+++ b/src/Numeric/LinearAlgebra/Static.hs
@@ -72,2 +72,3 @@ import Internal.Static
 import Control.Arrow((***))
+import Prelude hiding ((<>))
 
--- a/src/Numeric/Matrix.hs
+++ b/src/Numeric/Matrix.hs
@@ -35,2 +35,3 @@ import Data.List(partition)
 import Internal.Chain
+import qualified Data.Semigroup as S
 
@@ -86,2 +87,6 @@ adaptScalarM f1 f2 f3 x y
 
+instance (Container Vector t, Eq t, Num (Vector t), Product t) => S.Semigroup (Matrix t)
+  where
+    (<>) = adaptScalarM scale mXm (flip scale)
+
 instance (Container Vector t, Eq t, Num (Vector t), Product t) => M.Monoid (Matrix t)
