diff --git a/hmatrix.cabal b/hmatrix.cabal
index 7bed0d3..f6060da 100644
--- a/hmatrix.cabal
+++ b/hmatrix.cabal
@@ -28,7 +28,2 @@ extra-source-files: src/Internal/C/lapack-aux.h
 
-flag openblas
-    description:    Link with OpenBLAS (https://github.com/xianyi/OpenBLAS) optimized libraries.
-    default:        False
-    manual: True
-
 library
@@ -99,12 +94,2 @@ library
     if os(OSX)
-        if flag(openblas)
-            extra-lib-dirs:     /opt/local/lib/openblas/lib
-            extra-libraries:    openblas
-        else
-            extra-libraries:    blas lapack
-
-        extra-lib-dirs: /opt/local/lib/
-        include-dirs: /opt/local/include/
-        extra-lib-dirs: /usr/local/lib/
-        include-dirs: /usr/local/include/
         if arch(i386)
@@ -113,29 +98,3 @@ library
 
-    if os(freebsd)
-        if flag(openblas)
-            extra-lib-dirs:     /usr/local/lib/openblas/lib
-            extra-libraries:    openblas
-        else
-            extra-libraries:    blas lapack
-
-       extra-lib-dirs: /usr/local/lib
-       include-dirs: /usr/local/include
-       extra-libraries: gfortran
-
-    if os(windows)
-        if flag(openblas)
-            extra-libraries:    libopenblas
-        else
-            extra-libraries:    blas lapack
-
-    if os(linux)
-        if flag(openblas)
-            extra-lib-dirs:     /usr/lib/openblas/lib
-            extra-libraries:    openblas
-        else
-            extra-libraries:    blas lapack
-
-        if arch(x86_64)
-            cc-options: -fPIC
-
+    pkgconfig-depends: blas, lapack
 
