--- hmatrix-0.11.1.0-orig/lib/Numeric/Vector.hs	2011-05-31 17:27:03.000000000 +1000
+++ hmatrix-0.11.1.0/lib/Numeric/Vector.hs	2011-08-29 19:26:55.667024620 +1000
@@ -41,11 +41,13 @@
 
 #ifdef VECTOR
 
+#if !MIN_VERSION_vector(0,8,0)
 instance (Element a, Read a) => Read (Vector a) where
     readsPrec _ s = [(fromList . read $ listnums, rest)]
         where (thing,trest) = breakAt ']' s
               (dims,listnums) = breakAt ' ' (dropWhile (==' ') thing)
               rest = drop 31 trest
+#endif
 #else
 
 instance (Element a, Read a) => Read (Vector a) where
