diff --git a/data-inttrie.cabal b/data-inttrie.cabal
index 2bb5434..f64b0ac 100644
--- a/data-inttrie.cabal
+++ b/data-inttrie.cabal
@@ -10,5 +10,5 @@ Author: Luke Palmer <lrpalmer@gmail.com>
 Homepage: http://github.com/luqui/data-inttrie
 Maintainer: Luke Palmer <lrpalmer@gmail.com>
 Build-Type: Simple
-Build-Depends: base >= 3 && <= 4.4
+Build-Depends: base >= 3 && < 4.5
 Exposed-Modules: Data.IntTrie
