--- regex-posix-0.92-old/Text/Regex/Posix/ByteString.hs	2007-11-25 18:22:51.000000000 +0100
+++ regex-posix-0.92/Text/Regex/Posix/ByteString.hs	2007-11-25 18:20:03.000000000 +0100
@@ -54,7 +54,11 @@
 import Data.Array(Array,listArray)
 import Data.ByteString(ByteString)
 import qualified Data.ByteString as B(empty,useAsCString,last,take,drop,null)
+#if __GLASGOW_HASKELL__>=608
+import qualified Data.ByteString.Unsafe as B(unsafeUseAsCString)
+#else
 import qualified Data.ByteString.Base as B(unsafeUseAsCString)
+#endif
 import System.IO.Unsafe(unsafePerformIO)
 import Text.Regex.Base.RegexLike(RegexMaker(..),RegexContext(..),RegexLike(..),MatchOffset,MatchLength)
 import Text.Regex.Posix.Wrap -- all
--- regex-posix-0.92-old/Text/Regex/Posix/ByteString/Lazy.hs	2007-11-25 18:23:20.000000000 +0100
+++ regex-posix-0.92/Text/Regex/Posix/ByteString/Lazy.hs	2007-11-25 18:20:41.000000000 +0100
@@ -55,7 +55,11 @@
 import Data.Array(Array)
 import qualified Data.ByteString.Lazy as L (ByteString,null,toChunks,fromChunks,last,snoc)
 import qualified Data.ByteString as B(ByteString,concat)
+#if __GLASGOW_HASKELL__>=608
+import qualified Data.ByteString.Unsafe as B(unsafeUseAsCString)
+#else
 import qualified Data.ByteString.Base as B(unsafeUseAsCString)
+#endif
 import System.IO.Unsafe(unsafePerformIO)
 import Text.Regex.Base.RegexLike(RegexMaker(..),RegexContext(..),RegexLike(..),MatchOffset,MatchLength)
 import Text.Regex.Posix.Wrap -- all
