--- a/Console/Options/Flags.hs
+++ b/Console/Options/Flags.hs
@@ -27,2 +27,3 @@ import Data.List
 import Data.Monoid
+import qualified Data.Semigroup as S
 
@@ -75,2 +76,8 @@ flattenFragments frags =
 
+instance S.Semigroup FlagFrag where
+    (<>) (FlagMany l1) (FlagMany l2) = FlagMany (l1 ++ l2)
+    (<>) (FlagMany l1) o             = FlagMany (l1 ++ [o])
+    (<>) o             (FlagMany l2) = FlagMany (o : l2)
+    (<>) o1            o2            = FlagMany [o1,o2]
+
 instance Monoid FlagFrag where
