--- a/library/JSONBytesBuilder/Private/Builder.hs
+++ b/library/JSONBytesBuilder/Private/Builder.hs
@@ -8,3 +8,3 @@ import qualified Data.ByteString.Lazy as D
 import qualified JSONBytesBuilder.Private.ByteString.Builder as E
-
+import qualified Data.Semigroup as S
 
@@ -20,2 +20,15 @@ newtype Rows =
 
+instance S.Semigroup Rows where
+  {-# INLINE (<>) #-}
+  (<>) =
+    \case
+      Rows (Just left) ->
+        \case
+          Rows (Just right) ->
+            Rows (Just (left <> A.char8 ',' <> right))
+          _ ->
+            Rows (Just left)
+      Rows Nothing ->
+        id
+
 instance Monoid Rows where
@@ -41,2 +54,15 @@ newtype Elements =
 
+instance S.Semigroup Elements where
+  {-# INLINE (<>) #-}
+  (<>) =
+    \case
+      Elements (Just left) ->
+        \case
+          Elements (Just right) ->
+            Elements (Just (left <> A.char8 ',' <> right))
+          _ ->
+            Elements (Just left)
+      Elements Nothing ->
+        id
+
 instance Monoid Elements where
