--- a/src/Pretty.lhs
+++ b/src/Pretty.lhs
@@ -178,2 +178,3 @@ module Pretty (
 -- Don't import Util( assertPanic ) because it makes a loop in the module structure
+import Prelude hiding ((<>))
 
--- a/src/RuleUtils.hs
+++ b/src/RuleUtils.hs
@@ -7,2 +7,3 @@ import DataP (Statement(..),Data(..),Type(..),Name,Var,Class,
 		Body(..),Constructor)
+import Prelude hiding ((<>))
 
--- a/src/Rules/BitsBinary.hs
+++ b/src/Rules/BitsBinary.hs
@@ -5,2 +5,3 @@ import Data.List (intersperse)
 import RuleUtils -- useful to have a look at this too
+import Prelude hiding ((<>))
 
--- a/src/Rules/FunctorM.hs
+++ b/src/Rules/FunctorM.hs
@@ -5,2 +5,3 @@ import Data.List
 import RuleUtils
+import Prelude hiding ((<>))
 
--- a/src/Rules/Generic.hs
+++ b/src/Rules/Generic.hs
@@ -6,3 +6,3 @@ import RuleUtils
 import Data.List(intersperse)
-
+import Prelude hiding ((<>))
 
--- a/src/Rules/Monoid.hs
+++ b/src/Rules/Monoid.hs
@@ -4,2 +4,3 @@ module Rules.Monoid (rules) where
 import RuleUtils
+import Prelude hiding ((<>))
 
--- a/src/Rules/Standard.hs
+++ b/src/Rules/Standard.hs
@@ -5,3 +5,3 @@ import Data.List
 import GenUtil
-
+import Prelude hiding ((<>))
 
--- a/src/Rules/Xml.hs
+++ b/src/Rules/Xml.hs
@@ -5,2 +5,3 @@ import Data.List (nub,sortBy)
 import RuleUtils -- useful to have a look at this too
+import Prelude hiding ((<>))
 
