diff --git a/Mhailist.cabal b/Mhailist.cabal
index 6331390..67901b3 100644
--- a/Mhailist.cabal
+++ b/Mhailist.cabal
@@ -27,7 +27,7 @@ Executable mhailist
                      Mhailist.BuildMessage, Mhailist.Error, Mhailist.List,
                      Mhailist.Message, Mhailist.Receive, Mhailist.Transaction,
                      Util.String
-  Build-depends:     base < 4, process, directory, haskell98, filepath, mtl,
+  Build-depends:     base, process, directory, filepath, mtl,
                      bytestring, old-locale, time, binary
   if flag(Warnings)
     Ghc-options:     -Wall
diff --git a/Mhailist/Receive.hs b/Mhailist/Receive.hs
index a8c7d39..6d8dcf2 100644
--- a/Mhailist/Receive.hs
+++ b/Mhailist/Receive.hs
@@ -32,10 +32,11 @@ import Mhailist.Transaction
 
 import Control.Concurrent
 import Data.List
-import Monad
-import System
+import Control.Monad
 import System.Directory
 import System.IO
+import System.Exit
+import System.Environment
 import System.Process
 
 
