diff --git a/LTR/Transaction.hs b/LTR/Transaction.hs
index ce32871..9e64533 100644
--- a/LTR/Transaction.hs
+++ b/LTR/Transaction.hs
@@ -27,5 +27,5 @@ import Data.Word (Word32, Word16)
 import Util.String ()
 import System.IO (IOMode (..), withBinaryFile)
-import System.Locale (defaultTimeLocale)
+import Data.Time.Locale.Compat (defaultTimeLocale)
 
 #ifdef __TEST__
diff --git a/Mhailist.cabal b/Mhailist.cabal
index 67901b3..47583c0 100644
--- a/Mhailist.cabal
+++ b/Mhailist.cabal
@@ -29,5 +29,5 @@ Executable mhailist
                      Util.String
   Build-depends:     base, process, directory, filepath, mtl,
-                     bytestring, old-locale, time, binary
+                     bytestring, time-locale-compat, time, binary
   if flag(Warnings)
     Ghc-options:     -Wall
@@ -42,5 +42,5 @@ Executable Test
                      Util.String, UnitTest
     Build-depends:   base < 4, process, directory, haskell98, filepath, mtl,
-                     bytestring, old-locale, time, binary, HUnit
+                     bytestring, time-locale-compat, time, binary, HUnit
     Cpp-options:     -D__TEST__
     Ghc-options:     -fno-ignore-asserts -O0
diff --git a/Mhailist/Receive.hs b/Mhailist/Receive.hs
index 6d8dcf2..8722916 100644
--- a/Mhailist/Receive.hs
+++ b/Mhailist/Receive.hs
@@ -1,2 +1,3 @@
+{-# LANGUAGE FlexibleContexts #-}
 -- 
 -- mhailist-receive - process mail for the list
