diff --git a/Database/Groundhog/Core.hs b/Database/Groundhog/Core.hs
index bc24604..58ee49b 100644
--- a/Database/Groundhog/Core.hs
+++ b/Database/Groundhog/Core.hs
@@ -2,2 +2,3 @@
 {-# LANGUAGE UndecidableInstances #-} -- Required for Projection'
+{-# LANGUAGE UndecidableSuperClasses #-} -- Required for Projection'
 -- | This module defines the functions and datatypes used throughout the framework.
diff --git a/Database/Groundhog/Instances.hs b/Database/Groundhog/Instances.hs
index 48fbb44..07a2ce7 100644
--- a/Database/Groundhog/Instances.hs
+++ b/Database/Groundhog/Instances.hs
@@ -233,3 +233,3 @@ instance (DbDescriptor db, PersistEntity v, PersistField v) => PrimitivePersistF
 
-instance PrimitivePersistField a => PurePersistField a where
+instance (PersistField a, PrimitivePersistField a) => PurePersistField a where
   toPurePersistValues = primToPurePersistValues
@@ -237,3 +237,3 @@ instance PrimitivePersistField a => PurePersistField a where
 
-instance PrimitivePersistField a => SinglePersistField a where
+instance (PersistField a, PrimitivePersistField a) => SinglePersistField a where
   toSinglePersistValue = primToSinglePersistValue
