--- a/Database/Groundhog/Generic/Sql.hs
+++ b/Database/Groundhog/Generic/Sql.hs
@@ -48,2 +48,3 @@ import Data.Monoid
 import Data.String
+import qualified Data.Semigroup as S
 
@@ -59,2 +60,5 @@ data RenderS db r = RenderS {
 
+instance S.Semigroup Utf8 where
+  (<>) (Utf8 a) (Utf8 b) = Utf8 (mappend a b)
+
 instance Monoid Utf8 where
@@ -139,2 +143,5 @@ renderPersistValue a = RenderS (fromChar '?') (a:)
 
+instance S.Semigroup (RenderS db r) where
+  (RenderS f1 g1) <> (RenderS f2 g2) = RenderS (f1 `mappend` f2) (g1 . g2)
+
 instance Monoid (RenderS db r) where
