diff --git a/happstack-heist.cabal b/happstack-heist.cabal
index 95e2c3f..93b70e3 100644
--- a/happstack-heist.cabal
+++ b/happstack-heist.cabal
@@ -23,6 +23,6 @@ Library
                        blaze-builder >= 0.2 && <0.4,
                        bytestring == 0.9.*,
                        filepath,
-                       happstack-server >= 6.0 && < 6.1,
+                       happstack-server >= 6.0 && < 6.2,
                        heist >= 0.5 && < 0.6,
                        mtl == 2.*
