diff --git a/Monads.hs b/Monads.hs
index 449e2ca..718eb4e 100644
--- a/Monads.hs
+++ b/Monads.hs
@@ -11,2 +11,3 @@
   , ScopedTypeVariables
+  , CPP
   #-}
@@ -25,3 +26,3 @@ import Control.Monad
 -------------------------------------------------------------
-
+#if ! MIN_VERSION_base(4,4,0)
 instance Monad (Either String) where
@@ -30,3 +31,3 @@ instance Monad (Either String) where
   fail s = Left s
-
+#endif
 --------------------------------------------------------------
