diff --git a/Bind.hs b/Bind.hs
index bae8f2a..11058d8 100644
--- a/Bind.hs
+++ b/Bind.hs
@@ -79,3 +79,3 @@ instance Freshen Name where
 
-instance HasNext m => Fresh m where
+instance (Monad m, HasNext m) => Fresh m where
   fresh = do { n <- nextInteger; return (Nm n) }
diff --git a/RankN.hs b/RankN.hs
index 17c19a3..0d7b7ea 100644
--- a/RankN.hs
+++ b/RankN.hs
@@ -1457,3 +1457,3 @@ splitSkol vs (p:zs) = (p:ps,bs) where (ps,bs) = splitSkol vs zs
 instance (Show term, Exhibit (DispInfo Z) term,Typable m term Rho
-         ,Accumulates m Pred) => Typable m term Sigma where
+         ,Accumulates m Pred, TyCh m) => Typable m term Sigma where
 
