--- shakespeare-css-0.10.5-orig/shakespeare-css.cabal	2011-12-26 21:59:19.000000000 +1100
+++ shakespeare-css-0.10.5/shakespeare-css.cabal	2012-01-06 21:09:25.728521969 +1100
@@ -35,6 +35,7 @@
                    , text             >= 0.7     && < 0.12
                    , process          >= 1.0     && < 1.2
                    , parsec           >= 2       && < 4
+                   , mtl
 
     exposed-modules: Text.Cassius
                      Text.Lucius
--- shakespeare-css-0.10.5-orig/Text/Css.hs	2011-12-26 21:59:19.000000000 +1100
+++ shakespeare-css-0.10.5/Text/Css.hs	2012-01-06 21:03:57.706334818 +1100
@@ -15,7 +15,11 @@
 import Text.ParserCombinators.Parsec (Parser, parse)
 import Text.Shakespeare.Base hiding (Scope)
 import Language.Haskell.TH
+#if MIN_VERSION_mtl(2,0,0)
 import Control.Applicative ((<$>), (<*>))
+#else
+import Control.Applicative ((<$>), (<*>), Applicative(..), pure)
+#endif
 import Control.Arrow ((***))
 
 class ToCss a where
@@ -94,6 +98,18 @@
     b' <- b
     return $ a' ++ ContentRaw " " : b'
 
+#if !MIN_VERSION_mtl(2,0,0)
+instance Monad (Either e) where
+    return = Right
+    Left  l >>= _ = Left l
+    Right r >>= k = k r    
+
+instance Applicative (Either e) where
+    pure          = Right
+    Left  e <*> _ = Left e
+    Right f <*> r = fmap f r
+#endif
+
 blockRuntime :: [(Deref, CDData url)]
              -> (url -> [(Text, Text)] -> Text)
              -> Block
