--- MissingH-1.4.0.1-orig/MissingH.cabal	2016-06-15 12:19:37.000000000 +1000
+++ MissingH-1.4.0.1/MissingH.cabal	2018-10-11 14:33:35.208601932 +1100
@@ -81,6 +81,7 @@
    UndecidableInstances, CPP, Rank2Types,
    MultiParamTypeClasses, FlexibleInstances, FlexibleContexts,
    ScopedTypeVariables
+ Default-Extensions: NoMonadFailDesugaring
 
  Build-Depends: network, parsec, base,
                mtl, HUnit, regex-compat,
