diff -rN -u old-MissingH-1.0.3/src/Data/BinPacking.hs new-MissingH-1.0.3/src/Data/BinPacking.hs
--- old-MissingH-1.0.3/src/Data/BinPacking.hs	2009-02-14 10:45:58.000000000 +0100
+++ new-MissingH-1.0.3/src/Data/BinPacking.hs	2009-02-14 10:45:58.000000000 +0100
@@ -75,9 +75,9 @@
 you can pass @(repeat binSize)@ to pass an infinite list of bins if the
 same size.  Any surplus bins will simply be ignored. -}
 type BinPacker = (Num size, Ord size, Show size, Show obj) => 
-                  [size]        -- ^ The sizes of bins
-               -> [(size, obj)] -- ^ The sizes and objects
-               -> Either (BinPackerError size obj) [[(size, obj)]] -- ^ Either error or results
+                  [size]
+               -> [(size, obj)]
+               -> Either (BinPackerError size obj) [[(size, obj)]]
 
 
 {- | Pack objects into bins, preserving order.  Objects will be taken from the

