--- MissingH-1.3.0.1-orig/src/Data/BinPacking.hs	2014-10-28 23:46:24.000000000 +1100
+++ MissingH-1.3.0.1/src/Data/BinPacking.hs	2016-04-20 10:23:16.403505044 +1000
@@ -69,7 +69,7 @@
 > [(size, obj)] is the sizes and objects
 > result is Either error or results
 -}
-type BinPacker = (Num size, Ord size, Show size, Show obj) => 
+type BinPacker = forall size obj . (Num size, Ord size, Show size, Show obj) => 
                   [size]        -- The sizes of bins
                -> [(size, obj)] -- The sizes and objects
                -> Either (BinPackerError size obj) [[(size, obj)]] -- Either error or results
