diff --git a/monad-control.cabal b/monad-control.cabal
index 5beb414..2ed0672 100644
--- a/monad-control.cabal
+++ b/monad-control.cabal
@@ -55,7 +55,7 @@ Library
                    Control.Monad.IO.Control
                    Control.Exception.Control
 
-  Build-depends: base                 >= 3     && < 4.4
+  Build-depends: base                 >= 3     && < 4.5
                , base-unicode-symbols >= 0.1.1 && < 0.3
                , transformers         >= 0.2   && < 0.3
 
@@ -69,10 +69,10 @@ executable test-monad-control
   ghc-options: -Wall
 
   if flag(test)
-    build-depends: base                 >= 3     && < 4.4
+    build-depends: base                 >= 3     && < 4.5
                  , base-unicode-symbols >= 0.1.1 && < 0.3
                  , HUnit                >= 1.2.2 && < 1.3
-                 , test-framework       >= 0.2.4 && < 0.4
+                 , test-framework       >= 0.2.4 && < 0.5
                  , test-framework-hunit >= 0.2.4 && < 0.3
     buildable: True
   else
