diff --git a/Control/Monad/Stream.hs b/Control/Monad/Stream.hs
index 8fbeb4a..d83f342 100644
--- a/Control/Monad/Stream.hs
+++ b/Control/Monad/Stream.hs
@@ -79,3 +79,5 @@ import GHC.List
 import qualified Data.Stream as Stream
-import GHC.Base
+import GHC.Base hiding ( MonadPlus, sequence, mapM, mplus, mzero, (=<<)
+                       , ap, join, liftM, liftM2, liftM3, liftM4, liftM5
+                       , when)
 #endif
