diff --git a/Graphics/UI/GLFW.hs b/Graphics/UI/GLFW.hs
index b91e24e..5d47c2c 100644
--- a/Graphics/UI/GLFW.hs
+++ b/Graphics/UI/GLFW.hs
@@ -999,14 +999,14 @@ glfwMouseposfun      :: IORef (Maybe (FunPtr GLFWmouseposfun))
 glfwMousewheelfun    :: IORef (Maybe (FunPtr GLFWmousewheelfun))
 glfwKeyfun           :: IORef (Maybe (FunPtr GLFWkeyfun))
 glfwCharfun          :: IORef (Maybe (FunPtr GLFWcharfun))
-glfwWindowsizefun    = unsafePerformIO (newIORef Nothing)
-glfwWindowclosefun   = unsafePerformIO (newIORef Nothing)
-glfwWindowrefreshfun = unsafePerformIO (newIORef Nothing)
-glfwMousebuttonfun   = unsafePerformIO (newIORef Nothing)
-glfwMouseposfun      = unsafePerformIO (newIORef Nothing)
-glfwMousewheelfun    = unsafePerformIO (newIORef Nothing)
-glfwKeyfun           = unsafePerformIO (newIORef Nothing)
-glfwCharfun          = unsafePerformIO (newIORef Nothing)
+glfwWindowsizefun    = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwWindowclosefun   = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwWindowrefreshfun = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwMousebuttonfun   = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwMouseposfun      = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwMousewheelfun    = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwKeyfun           = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
+glfwCharfun          = System.IO.Unsafe.unsafePerformIO (newIORef Nothing)
 
 foreign import ccall unsafe "wrapper" glfwWrapFun2 :: (Int -> Int -> IO ()) -> IO (FunPtr (Int -> Int -> IO ()))
 foreign import ccall unsafe "wrapper" glfwWrapFun1 :: (Int -> IO ()) -> IO (FunPtr (Int -> IO ()))
@@ -1035,7 +1035,7 @@ glfwCleanup = do
 -- text rendering
 
 fontTextures :: IORef ([(BitmapFont, GL.TextureObject)])
-fontTextures = unsafePerformIO (newIORef [])
+fontTextures = System.IO.Unsafe.unsafePerformIO (newIORef [])
 
 loadFont :: BitmapFont -> IO GL.TextureObject
 loadFont name = do
