diff --git a/Setup.hs b/Setup.hs
index 17a047d..6067f42 100644
--- a/Setup.hs
+++ b/Setup.hs
@@ -21 +21,2 @@ import Foreign (sizeOf)
-import Control.Monad (when)
+import Control.Applicative
+import Control.Monad
@@ -241,0 +243,8 @@ modify f = get >>= put . f
+
+instance Monad m => Functor (StateT s m) where
+    fmap  = liftM
+
+instance Monad m => Applicative (StateT s m) where
+    pure  = return
+    (<*>) = ap  -- defined in Control.Monad
+
