diff --git a/src/Language/Haskell/Packages.hs b/src/Language/Haskell/Packages.hs
index d6f0be4..d485531 100644
--- a/src/Language/Haskell/Packages.hs
+++ b/src/Language/Haskell/Packages.hs
@@ -42,4 +42,6 @@ import qualified Control.Exception as Exception
 -- This was borrowed from the ghc-pkg source:
 #if MIN_VERSION_Cabal(1,22,0)
+type InstalledPackageInfoString = InstalledPackageInfo
+#elif MIN_VERSION_Cabal(1,22,0)
 type InstalledPackageInfoString = InstalledPackageInfo_ ModuleName
 #else
