diff --git a/Harpy/CodeGenMonad.hs b/Harpy/CodeGenMonad.hs
index 58d36e0..260ef13 100644
--- a/Harpy/CodeGenMonad.hs
+++ b/Harpy/CodeGenMonad.hs
@@ -168,6 +168,13 @@ data Label = Label Int String
 unCg :: CodeGen e s a -> ((e, CodeGenEnv) -> (s, CodeGenState) -> IO ((s, CodeGenState), Either ErrMsg a))
 unCg (CodeGen a) = a
 
+instance Functor (CodeGen e s) where
+    fmap  = liftM
+
+instance Applicative (CodeGen e s) where
+    pure  = return
+    (<*>) = ap
+
 instance Monad (CodeGen e s) where
     return x = cgReturn x
     fail err = cgFail err
