diff --git a/src/Nix/Eval.hs b/src/Nix/Eval.hs
index 0a50326..ffc4408 100644
--- a/src/Nix/Eval.hs
+++ b/src/Nix/Eval.hs
@@ -18,7 +18,7 @@ import           Control.Monad
 import           Control.Monad.Fix
 import           Control.Monad.Reader
 import           Control.Monad.State.Strict
-import           Data.Align.Key                 ( alignWithKey )
+import           Data.Semialign.Indexed         ( ialignWith )
 import           Data.Either                    ( isRight )
 import           Data.Fix                       ( Fix(Fix) )
 import           Data.HashMap.Lazy              ( HashMap )
@@ -386,7 +386,7 @@ buildArgument params arg = do
               Nothing -> id
               Just n  -> M.insert n $ const $ defer (withScopes scope arg)
         loebM
-          (inject $ M.mapMaybe id $ alignWithKey (assemble scope isVariadic)
+          (inject $ M.mapMaybe id $ ialignWith (assemble scope isVariadic)
                                                  args
                                                  (M.fromList s)
           )
