From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Needed for the library to compile

diff -urN hnix-0.14.0.2-new1/src/Nix/Json.hs hnix-0.14.0.2-new2/src/Nix/Json.hs
--- hnix-0.14.0.2-new1/src/Nix/Json.hs	2001-09-08 19:46:40.000000000 -0600
+++ hnix-0.14.0.2-new2/src/Nix/Json.hs	2021-10-30 19:11:44.340732400 -0600
@@ -19,6 +19,7 @@
 import           Nix.Utils
 import           Nix.Value
 import           Nix.Value.Monad
+import           Nix.Expr
 
 nvalueToJSONNixString :: MonadNix e t f m => NValue t f m -> m NixString
 nvalueToJSONNixString =
diff -urN hnix-0.14.0.2-new1/src/Nix/Utils.hs hnix-0.14.0.2-new2/src/Nix/Utils.hs
--- hnix-0.14.0.2-new1/src/Nix/Utils.hs	2001-09-08 19:46:40.000000000 -0600
+++ hnix-0.14.0.2-new2/src/Nix/Utils.hs	2021-10-30 18:50:42.367492562 -0600
@@ -160,7 +160,7 @@
 #if MIN_VERSION_aeson(2,0,0)
           AKM.toList
 #else
-          HM.toList
+          M.toList
 #endif
           m
   A.Array l -> A.list toEncodingSorted $ V.toList l
