diff --git a/main/Repl.hs b/main/Repl.hs
index f97aed0..42478a7 100644
--- a/main/Repl.hs
+++ b/main/Repl.hs
@@ -53,7 +53,7 @@ import           Control.Monad.Reader
 import           Control.Monad.State.Strict
 
 import           System.Console.Haskeline.MonadException
-import           System.Console.Repline
+import           System.Console.Repline hiding ( options )
 import           System.Environment
 import           System.Exit
 
