--- a/hnix.cabal
+++ b/hnix.cabal
@@ -433,8 +433,7 @@ library
     , ref-tf >= 0.4.0 && < 0.5
     , regex-tdfa >= 1.2.3 && < 1.4
     , scientific >= 0.3.6 && < 0.4
-    , semialign >= 1 && < 1.2
-    , semialign-indexed >= 1 && < 1.2
+    , semialign >= 1.2
     , some >= 1.0.1 && < 1.1
     , split >= 0.2.3 && < 0.3
     , syb >= 0.7 && < 0.8
--- a/src/Nix/Expr/Types.hs
+++ b/src/Nix/Expr/Types.hs
@@ -73,8 +73,10 @@ type VarName = Text
 hashAt :: VarName -> Lens' (AttrSet v) (Maybe v)
 hashAt = flip alterF
 
+#if ! MIN_VERSION_semialign(1,2,0)
 -- unfortunate orphans
 instance Hashable1 NonEmpty
+#endif
 
 -- | The main Nix expression type. As it is polimophic, has a functor,
 -- which allows to traverse expressions and map functions over them.
