From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Quick fix to get working with hashable-1.4

diff -urN hnix-0.14.0.8/src/Nix/Expr/Types/Annotated.hs hnix-0.14.0.8-r1/src/Nix/Expr/Types/Annotated.hs
--- hnix-0.14.0.8/src/Nix/Expr/Types/Annotated.hs	2001-09-08 19:46:40.000000000 -0600
+++ hnix-0.14.0.8-r1/src/Nix/Expr/Types/Annotated.hs	2022-03-22 16:14:12.427132294 -0600
@@ -102,8 +102,6 @@
 
 -- ** Instances
 
-instance Hashable ann => Hashable1 (Ann ann)
-
 instance NFData ann => NFData1 (Ann ann)
 
 instance (Binary ann, Binary a) => Binary (Ann ann a)
@@ -119,6 +117,8 @@
 $(deriveJSON1 defaultOptions ''Ann)
 $(deriveJSON2 defaultOptions ''Ann)
 
+instance Hashable ann => Hashable1 (Ann ann)
+
 instance (Serialise ann, Serialise a) => Serialise (Ann ann a)
 
 -- ** @NExprLoc{,F}@ - annotated Nix expression
diff -urN hnix-0.14.0.8/src/Nix/Expr/Types.hs hnix-0.14.0.8-r1/src/Nix/Expr/Types.hs
--- hnix-0.14.0.8/src/Nix/Expr/Types.hs	2001-09-08 19:46:40.000000000 -0600
+++ hnix-0.14.0.8-r1/src/Nix/Expr/Types.hs	2022-03-22 16:10:05.537132175 -0600
@@ -181,11 +181,6 @@
     , Show, Read, Hashable
     )
 
-instance Hashable2 Antiquoted where
-  liftHashWithSalt2 ha _  salt (Plain a)      = ha (salt `hashWithSalt` (0 :: Int)) a
-  liftHashWithSalt2 _  _  salt EscapedNewline =     salt `hashWithSalt` (1 :: Int)
-  liftHashWithSalt2 _  hb salt (Antiquoted b) = hb (salt `hashWithSalt` (2 :: Int)) b
-
 $(deriveShow1 ''Antiquoted)
 $(deriveShow2 ''Antiquoted)
 $(deriveRead1 ''Antiquoted)
@@ -196,6 +191,11 @@
 $(deriveOrd2  ''Antiquoted)
 $(deriveJSON2 defaultOptions ''Antiquoted)
 
+instance Hashable2 Antiquoted where
+  liftHashWithSalt2 ha _  salt (Plain a)      = ha (salt `hashWithSalt` (0 :: Int)) a
+  liftHashWithSalt2 _  _  salt EscapedNewline =     salt `hashWithSalt` (1 :: Int)
+  liftHashWithSalt2 _  hb salt (Antiquoted b) = hb (salt `hashWithSalt` (2 :: Int)) b
+
 deriving instance Hashable v => Hashable1 (Antiquoted v)
 
 -- *** Lens traversals
