--- buildwrapper-0.2.2-orig/test/Language/Haskell/BuildWrapper/Tests.hs	2011-12-15 07:23:46.000000000 +1100
+++ buildwrapper-0.2.2/test/Language/Haskell/BuildWrapper/Tests.hs	2011-12-29 12:35:12.501209391 +1100
@@ -14,6 +14,7 @@
 
 import Language.Haskell.BuildWrapper.Base
 
+import Data.ByteString.Lazy.Char8 (ByteString (..))
 import Data.Maybe
 import Data.Aeson
 import Data.Char
@@ -831,4 +832,4 @@
         let
                 n1'=n1{bwn_title=removeSpaces $ bwn_title n1}
                 n2'=n1{bwn_title=removeSpaces $ bwn_title n2}
-        assertEqual msg n1' n2'
\ No newline at end of file
+        assertEqual msg n1' n2'
