--- hoogle-4.2.11-orig/hoogle.cabal	2012-04-06 04:03:44.000000000 +1000
+++ hoogle-4.2.11/hoogle.cabal	2012-05-14 01:22:20.387918182 +1000
@@ -117,7 +117,7 @@
         conduit >= 0.2 && < 0.5,
         wai >= 1.1 && < 1.3,
         warp >= 1.1 && < 1.3,
-        Cabal >= 1.8 && < 1.15
+        Cabal >= 1.8 && < 1.16
 
     other-modules:
         CmdLine.All
--- hoogle-4.2.11-orig/src/Web/Server.hs	2012-04-06 04:03:44.000000000 +1000
+++ hoogle-4.2.11/src/Web/Server.hs	2012-05-17 10:17:13.271813644 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards #-}
+{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards, CPP #-}
 
 module Web.Server(server) where
 
@@ -15,6 +15,9 @@
 import Control.Exception
 import System.Time
 import Data.Time.Clock
+#if (__GLASGOW_HASKELL__>=705)
+import Data.Time.Clock.POSIX
+#endif
 
 import Network.Wai
 import Network.Wai.Handler.Warp
@@ -56,8 +59,18 @@
                 if dynamic then  buffer template get else return get
 
         modTime ext = unsafeInterleaveIO $ do
+-- In ghc-7.5.20120511 there is an API change in directory 1.1.0.1 with
+-- no version version number bump.
+-- directory 1.1.0.2 ghc 7.4.1          getModificationTime returns ClockTime
+-- directory 1.1.0.1 ghc 7.5.1.20120511 getModificationTime returns UTCTime
+-- Hence test is on ghc version to see which API it is here.
+#if (__GLASGOW_HASKELL__>=705)
+            a <- getModificationTime $ resources </> "hoogle" <.> ext
+            return $ show $ utcTimeToPOSIXSeconds a
+#else
             TOD a _ <- getModificationTime $ resources </> "hoogle" <.> ext
             return $ show a
+#endif
 
         args t = do
             css <- modTime "css"; js <- modTime "js"
