--- hoogle-4.2.10-orig/hoogle.cabal	2012-03-04 22:36:33.000000000 +1100
+++ hoogle-4.2.10/hoogle.cabal	2012-03-31 23:16:45.380401642 +1100
@@ -44,7 +44,7 @@
         parsec >= 2.1,
         transformers == 0.2.*,
         uniplate == 1.6.*,
-        haskell-src-exts >= 1.9 && < 1.12
+        haskell-src-exts >= 1.9 && < 1.14
 
     if !os(mingw32)
         build-depends: unix
--- hoogle-4.2.10-orig/src/Hoogle/Language/Haskell.hs	2012-03-04 22:36:33.000000000 +1100
+++ hoogle-4.2.10/src/Hoogle/Language/Haskell.hs	2012-03-31 23:28:00.519614872 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE PatternGuards #-}
+{-# LANGUAGE CPP, PatternGuards #-}
 
 module Hoogle.Language.Haskell(parseInputHaskell) where
 
@@ -62,7 +62,13 @@
         ParseOk y -> transDecl x $ fmap (subtractCols 16) y
         _ -> Nothing
 
-exts = [EmptyDataDecls,TypeOperators,ExplicitForall,GADTs,KindSignatures,MultiParamTypeClasses
+exts = [EmptyDataDecls,TypeOperators
+#if MIN_VERSION_haskell_src_exts(1,13,0)
+       ,ExplicitForAll,
+#else
+       ,ExplicitForall,
+#endif
+        GADTs,KindSignatures,MultiParamTypeClasses
        ,TypeFamilies,FlexibleContexts,FunctionalDependencies,ImplicitParams,MagicHash,UnboxedTuples]
 
 
