--- hoogle-4.2.14-orig/src/Web/Server.hs	2012-11-07 09:50:54.000000000 +1100
+++ hoogle-4.2.14/src/Web/Server.hs	2012-11-07 10:17:29.043109444 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards #-}
+{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards, CPP #-}
 
 module Web.Server(server) where
 
@@ -14,6 +14,9 @@
 import Control.Concurrent
 import Control.Exception
 import Data.Time.Clock
+#if MIN_VERSION_directory(1,2,0)
+import Data.Time.Clock.POSIX
+#endif
 
 import Network.Wai
 import Network.Wai.Handler.Warp
@@ -55,8 +58,13 @@
                 if dynamic then  buffer template get else return get
 
         modTime ext = unsafeInterleaveIO $ do
+#if MIN_VERSION_directory(1,2,0)
+            x <- getModificationTime $ resources </> "hoogle" <.> ext
+            return $ map (\x -> if isSpace x then '_' else x) $ show $ utcTimeToPOSIXSeconds x
+#else
             x <- getModificationTime $ resources </> "hoogle" <.> ext
             return $ map (\x -> if isSpace x then '_' else x) $ show x
+#endif
 
         args t = do
             css <- modTime "css"; js <- modTime "js"
--- hoogle-4.2.14-orig/src/CmdLine/Load.hs	2012-11-07 09:50:54.000000000 +1100
+++ hoogle-4.2.14/src/CmdLine/Load.hs	2012-11-07 10:09:19.419348339 +1100
@@ -4,7 +4,7 @@
 import Hoogle
 import General.Base
 import General.Util
-import General.System
+import General.System (doesDirectoryExist, doesFileExist, getDirectoryContents)
 
 
 -- | Given a list of search directories, and a query, load the databases you
