diff -ru hoogle-4.2.21.orig/src/Recipe/Download.hs hoogle-4.2.21/src/Recipe/Download.hs
--- hoogle-4.2.21.orig/src/Recipe/Download.hs   2013-09-01 21:41:32.000000000 +0400
+++ hoogle-4.2.21/src/Recipe/Download.hs        2013-09-28 15:49:02.000000000 +0400
@@ -43,8 +43,8 @@
                 , (platform, platform, "http://code.galois.com/darcs/haskell-platform/haskell-platform.cabal")
                 , ("download/base.txt", "download/base.txt", "http://www.haskell.org/hoogle/base.txt")
                 , ("download/ghc.txt",  "download/ghc.txt", "http://www.haskell.org/ghc/docs/latest/html/libraries/ghc/ghc.txt")
-                , (cabals <.> "txt", cabals <.> "tar.gz", "http://hackage.haskell.org/packages/archive/00-index.tar.gz")
-                , (inputs <.> "txt", inputs <.> "tar.gz", "http://hackage.haskell.org/packages/archive/00-hoogle.tar.gz")
+                , (cabals <.> "txt", cabals <.> "tar.gz", "http://old.hackage.haskell.org/packages/archive/00-index.tar.gz")
+                , (inputs <.> "txt", inputs <.> "tar.gz", "http://old.hackage.haskell.org/packages/archive/00-hoogle.tar.gz")
                 ]
     withDownloader opt downloader items
     extractTarball cabals