--- hoogle-4.2.12-orig/hoogle.cabal	2012-07-16 05:22:28.000000000 +1000
+++ hoogle-4.2.12/hoogle.cabal	2012-08-10 21:55:48.763211747 +1000
@@ -114,7 +114,7 @@
         conduit >= 0.2 && < 0.6,
         wai >= 1.1 && < 1.4,
         warp >= 1.1 && < 1.4,
-        Cabal >= 1.8 && < 1.15
+        Cabal >= 1.8 && < 1.18
 
     other-modules:
         CmdLine.All
--- hoogle-4.2.12-orig/src/Web/Server.hs	2012-07-16 05:22:28.000000000 +1000
+++ hoogle-4.2.12/src/Web/Server.hs	2012-08-10 21:55:13.205077645 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards #-}
+{-# LANGUAGE RecordWildCards, ScopedTypeVariables, PatternGuards, CPP #-}
 
 module Web.Server(server) where
 
@@ -15,6 +15,9 @@
 import Control.Exception
 import System.Time
 import Data.Time.Clock
+#if (__GLASGOW_HASKELL__>=705)
+import Data.Time.Clock.POSIX
+#endif
 
 import Network.Wai
 import Network.Wai.Handler.Warp
@@ -56,8 +59,18 @@
                 if dynamic then  buffer template get else return get
 
         modTime ext = unsafeInterleaveIO $ do
+-- In ghc-7.5.20120511 there is an API change in directory 1.1.0.1 with
+-- no version version number bump.
+-- directory 1.1.0.2 ghc 7.4.1          getModificationTime returns ClockTime
+-- directory 1.1.0.1 ghc 7.5.1.20120511 getModificationTime returns UTCTime
+-- Hence test is on ghc version to see which API it is here.
+#if (__GLASGOW_HASKELL__>=705)
+            a <- getModificationTime $ resources </> "hoogle" <.> ext
+            return $ show $ utcTimeToPOSIXSeconds a
+#else
             TOD a _ <- getModificationTime $ resources </> "hoogle" <.> ext
             return $ show a
+#endif
 
         args t = do
             css <- modTime "css"; js <- modTime "js"
--- hoogle-4.2.12-orig/src/CmdLine/Load.hs	2012-07-16 05:22:28.000000000 +1000
+++ hoogle-4.2.12/src/CmdLine/Load.hs	2012-08-10 21:55:13.206077678 +1000
@@ -4,7 +4,7 @@
 import Hoogle
 import General.Base
 import General.Util
-import General.System
+import General.System (doesDirectoryExist, doesFileExist, getDirectoryContents)
 
 
 -- | Given a list of search directories, and a query, load the databases you
