diff --git a/Setup.hs b/Setup.hs
index e4be656..95ecea0 100644
--- a/Setup.hs
+++ b/Setup.hs
@@ -3,7 +3,7 @@ import Distribution.PackageDescription.Parse
 import Distribution.Verbosity
 import Distribution.System
 import Distribution.Simple
-import Distribution.Simple.Utils
+import Distribution.Simple.Utils hiding (die)
 import Distribution.Simple.Setup
 import Distribution.Simple.Command
 import Distribution.Simple.Program
