diff --git a/Data/ConcreteTypeRep.hs b/Data/ConcreteTypeRep.hs
index 4340c5b..ce804a2 100644
--- a/Data/ConcreteTypeRep.hs
+++ b/Data/ConcreteTypeRep.hs
@@ -63,3 +63,7 @@ instance Hashable ConcreteTypeRep where
 #ifdef NEW_TYPEREP
-  hashWithSalt salt (CTR (TypeRep (Fingerprint w1 w2) _ _)) = salt `hashWithSalt` w1 `hashWithSalt` w2
+  hashWithSalt salt (CTR (TypeRep (Fingerprint w1 w2) _ _
+#if MIN_VERSION_ghc(7,10,1)
+                         _
+#endif
+                         )) = salt `hashWithSalt` w1 `hashWithSalt` w2
 #else
