diff --git a/src/Distribution/Hackage/DB/Parsed.hs b/src/Distribution/Hackage/DB/Parsed.hs
index 9ff416b..9d3ab7c 100644
--- a/src/Distribution/Hackage/DB/Parsed.hs
+++ b/src/Distribution/Hackage/DB/Parsed.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 {- |
@@ -24,3 +25,9 @@ import qualified Distribution.Hackage.DB.Unparsed as Unparsed
 import Distribution.PackageDescription
+#if MIN_VERSION_Cabal(2,2,0)
+import Distribution.PackageDescription.Parsec ( parseGenericPackageDescription, runParseResult, ParseResult(..) )
+import Distribution.ParseUtils ( ParseResult(..) )
+import qualified Data.ByteString.Lazy.Char8 as BLC8
+#else
 import Distribution.PackageDescription.Parse ( parsePackageDescription, ParseResult(..) )
+#endif
 import Distribution.Text ( display )
@@ -69,2 +76,7 @@ parsePackage name version buf = case parsePackage' buf of
 parsePackage' :: ByteString -> Either String GenericPackageDescription
+#if MIN_VERSION_Cabal(2,2,0)
+parsePackage' buf = case runParseResult (parseGenericPackageDescription (BLC8.toStrict buf)) of
+                     (_, Right a)    -> Right a
+                     (_, Left err)   -> Left (show err)
+#else
 parsePackage' buf = case parsePackageDescription (decodeUTF8 buf) of
@@ -72,2 +84,3 @@ parsePackage' buf = case parsePackageDescription (decodeUTF8 buf) of
                      ParseFailed err -> Left (show err)
+#endif
   where
