--- vector-algorithms-0.5.3-orig/vector-algorithms.cabal	2011-09-01 04:57:15.000000000 +1000
+++ vector-algorithms-0.5.3/vector-algorithms.cabal	2012-05-13 23:08:15.751953789 +1000
@@ -29,7 +29,7 @@
     Build-Depends: base >= 3 && < 5,
                    vector >= 0.6 && < 0.10,
                    primitive >=0.3 && <0.5,
-                   bytestring >= 0.9 && < 0.10
+                   bytestring >= 0.9 && < 0.11
 
     Exposed-Modules:
         Data.Vector.Algorithms.Optimal
@@ -52,9 +52,15 @@
         FlexibleContexts,
         CPP
 
-    GHC-Options:
-        -Odph
-        -funbox-strict-fields
+    if impl(ghc >= 7.4.0)
+        GHC-Options:
+            -Odph
+            -funbox-strict-fields
+            -fsimpl-tick-factor=153
+    else
+        GHC-Options:
+           -Odph
+           -funbox-strict-fields
 
     Include-Dirs:
         include
