diff -rN -u old-hsgnutls/src/Network/GnuTLS/Internals.chs new-hsgnutls/src/Network/GnuTLS/Internals.chs
--- old-hsgnutls/src/Network/GnuTLS/Internals.chs	2006-09-14 09:31:51.000000000 +0100
+++ new-hsgnutls/src/Network/GnuTLS/Internals.chs	2006-09-14 09:31:51.000000000 +0100
@@ -90,10 +90,15 @@
 withSession :: Session t -> (Ptr () -> IO a) -> IO a
 withSession (Session s _) = withForeignPtr s
 
-throwGnutlsIf :: CInt -> IO ()
+{-# SPECIALIZE throwGnutlsIf :: CInt  -> IO () #-}
+{-# SPECIALIZE throwGnutlsIf :: CLong -> IO () #-}
+
+throwGnutlsIf :: Integral n => n -> IO ()
 throwGnutlsIf 0     = return ()
-throwGnutlsIf v     = {#call gnutls_strerror #} v >>= safePeekCString >>= (\str -> fail (str++" ("++show v++")"))
+throwGnutlsIf v     = {#call gnutls_strerror #} (fromIntegral v) >>= safePeekCString >>= (\str -> fail (str++" ("++show v++")"))
 
+{-# SPECIALIZE throwGnutlsIfNeg :: CInt  -> IO Int #-}
+{-# SPECIALIZE throwGnutlsIfNeg :: CLong -> IO Int #-}
 throwGnutlsIfNeg v = if v < 0 then throwGnutlsIf v >> return 0 else return (fromIntegral v)
 
 {-# INLINE peekEnum #-}

