diff --git a/Web/Authenticate/OAuth.hs b/Web/Authenticate/OAuth.hs
index 62d0c9c..96fb52f 100644
--- a/Web/Authenticate/OAuth.hs
+++ b/Web/Authenticate/OAuth.hs
@@ -35,7 +35,7 @@ import Data.ByteString.Base64
 import Data.Time
 import Numeric
 import Codec.Crypto.RSA (rsassa_pkcs1_v1_5_sign, ha_SHA1)
-import Crypto.Types.PubKey.RSA (PrivateKey(..))
+import Crypto.Types.PubKey.RSA (PrivateKey(..), PublicKey(..))
 import Network.HTTP.Types (Header)
 import Blaze.ByteString.Builder (toByteString)
 import Control.Monad.IO.Class (MonadIO)
@@ -109,6 +109,7 @@ data SignMethod = PLAINTEXT
                 | HMACSHA1
                 | RSASHA1 PrivateKey
                   deriving (Show, Eq, Ord, Read, Data, Typeable)
+deriving instance Ord PublicKey
 deriving instance Ord PrivateKey
 -- | Data type for redential.
 data Credential = Credential { unCredential :: [(BS.ByteString, BS.ByteString)] }
