diff --git a/purescript.cabal b/purescript.cabal
index e3cb5b5..a58178f 100644
--- a/purescript.cabal
+++ b/purescript.cabal
@@ -121,7 +121,7 @@ library
                    fsnotify >= 0.2.1,
                    Glob >= 0.7 && < 0.8,
                    haskeline >= 0.7.0.0,
-                   http-client >= 0.4.30 && <0.5,
+                   http-client >= 0.4.30,
                    http-types -any,
                    language-javascript == 0.6.*,
                    lens == 4.*,
diff --git a/src/Language/PureScript/Ide/Pursuit.hs b/src/Language/PureScript/Ide/Pursuit.hs
index 90957fa..d200934 100644
--- a/src/Language/PureScript/Ide/Pursuit.hs
+++ b/src/Language/PureScript/Ide/Pursuit.hs
@@ -47,7 +47,6 @@ queryPursuit q = do
 
 
 handler :: HttpException -> IO [a]
-handler StatusCodeException{} = pure []
 handler _ = pure []
 
 searchPursuitForDeclarations :: Text -> IO [PursuitResponse]
