diff --git a/psc-package/Main.hs b/psc-package/Main.hs
index b6b7943..b147de0 100644
--- a/psc-package/Main.hs
+++ b/psc-package/Main.hs
@@ -14,3 +14,3 @@ import           Data.List (nub)
 import qualified Data.Map as Map
-import           Data.Maybe (mapMaybe)
+import           Data.Maybe (mapMaybe, fromJust)
 import qualified Data.Set as Set
@@ -29,2 +29,5 @@ import           Turtle hiding (fold)
 
+ttlU :: Text -> Line
+ttlU = fromJust . textToLine
+
 packageFile :: Path.FilePath
@@ -121,3 +124,3 @@ readPackageSet PackageConfig{ set } = do
   unless exists $ do
-    echo $ format (fp%" does not exist") dbFile
+    echo $ ttlU $ format (fp % " does not exist") dbFile
     exit (ExitFailure 1)
@@ -141,3 +144,3 @@ getTransitiveDeps db depends = do
       Nothing -> do
-        echo ("Package " <> pkg <> " does not exist in package set")
+        echo ("Package " <> (ttlU pkg) <> " does not exist in package set")
         exit (ExitFailure 1)
@@ -152,5 +155,5 @@ updateImpl config@PackageConfig{ depends } = do
   trans <- getTransitiveDeps db depends
-  echo ("Updating " <> pack (show (length trans)) <> " packages...")
+  echo ("Updating " <> ttlU (pack (show (length trans))) <> " packages...")
   for_ trans $ \(pkgName, pkg) -> do
-    echo ("Updating " <> pkgName)
+    echo ("Updating " <> (ttlU pkgName))
     installOrUpdate config pkgName pkg
@@ -196,3 +199,3 @@ listDependencies = do
   trans <- getTransitiveDeps db depends
-  traverse_ (echo . fst) trans
+  traverse_ (echo . ttlU . fst) trans
 
@@ -202,3 +205,3 @@ listPackages = do
   db <- readPackageSet pkg
-  traverse_ echo (fmt <$> Map.assocs db)
+  traverse_ echo $ map ttlU (fmt <$> Map.assocs db)
   where
@@ -224,3 +227,3 @@ listSourcePaths = do
   paths <- getSourcePaths pkg db depends
-  traverse_ (echo . pathToTextUnsafe) paths
+  traverse_ (echo . ttlU . pathToTextUnsafe) paths
 
