diff --git a/app/Command/REPL.hs b/app/Command/REPL.hs
index 1b81994..7653b8b 100644
--- a/app/Command/REPL.hs
+++ b/app/Command/REPL.hs
@@ -16 +16 @@ import           Prelude ()
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Control/Monad/Supply/Class.hs b/src/Control/Monad/Supply/Class.hs
index 64038a6..fe3aeb6 100644
--- a/src/Control/Monad/Supply/Class.hs
+++ b/src/Control/Monad/Supply/Class.hs
@@ -10 +10 @@ module Control.Monad.Supply.Class where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/AST/SourcePos.hs b/src/Language/PureScript/AST/SourcePos.hs
index f208dee..dbf813d 100644
--- a/src/Language/PureScript/AST/SourcePos.hs
+++ b/src/Language/PureScript/AST/SourcePos.hs
@@ -7 +7 @@ module Language.PureScript.AST.SourcePos where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/CodeGen/JS.hs b/src/Language/PureScript/CodeGen/JS.hs
index de11a87..bf50fb9 100644
--- a/src/Language/PureScript/CodeGen/JS.hs
+++ b/src/Language/PureScript/CodeGen/JS.hs
@@ -9 +9 @@ module Language.PureScript.CodeGen.JS
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/CodeGen/JS/Common.hs b/src/Language/PureScript/CodeGen/JS/Common.hs
index 0060b56..8344643 100644
--- a/src/Language/PureScript/CodeGen/JS/Common.hs
+++ b/src/Language/PureScript/CodeGen/JS/Common.hs
@@ -4 +4 @@ module Language.PureScript.CodeGen.JS.Common where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/CodeGen/JS/Printer.hs b/src/Language/PureScript/CodeGen/JS/Printer.hs
index d8b59dc..005b08b 100644
--- a/src/Language/PureScript/CodeGen/JS/Printer.hs
+++ b/src/Language/PureScript/CodeGen/JS/Printer.hs
@@ -7 +7 @@ module Language.PureScript.CodeGen.JS.Printer
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/CoreImp/Optimizer/Inliner.hs b/src/Language/PureScript/CoreImp/Optimizer/Inliner.hs
index d7dc998..2cd25b0 100644
--- a/src/Language/PureScript/CoreImp/Optimizer/Inliner.hs
+++ b/src/Language/PureScript/CoreImp/Optimizer/Inliner.hs
@@ -14 +14 @@ module Language.PureScript.CoreImp.Optimizer.Inliner
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/CoreImp/Optimizer/TCO.hs b/src/Language/PureScript/CoreImp/Optimizer/TCO.hs
index 0a5d949..fbe91ad 100644
--- a/src/Language/PureScript/CoreImp/Optimizer/TCO.hs
+++ b/src/Language/PureScript/CoreImp/Optimizer/TCO.hs
@@ -4 +4 @@ module Language.PureScript.CoreImp.Optimizer.TCO (tco) where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/AsMarkdown.hs b/src/Language/PureScript/Docs/AsMarkdown.hs
index 6cb3b4e..1a02de3 100644
--- a/src/Language/PureScript/Docs/AsMarkdown.hs
+++ b/src/Language/PureScript/Docs/AsMarkdown.hs
@@ -9 +9 @@ module Language.PureScript.Docs.AsMarkdown
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/Convert/ReExports.hs b/src/Language/PureScript/Docs/Convert/ReExports.hs
index 481fc60..54b7ab3 100644
--- a/src/Language/PureScript/Docs/Convert/ReExports.hs
+++ b/src/Language/PureScript/Docs/Convert/ReExports.hs
@@ -5 +5 @@ module Language.PureScript.Docs.Convert.ReExports
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/Render.hs b/src/Language/PureScript/Docs/Render.hs
index b60cae8..8fd1db5 100644
--- a/src/Language/PureScript/Docs/Render.hs
+++ b/src/Language/PureScript/Docs/Render.hs
@@ -12 +12 @@ module Language.PureScript.Docs.Render where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/RenderedCode/RenderKind.hs b/src/Language/PureScript/Docs/RenderedCode/RenderKind.hs
index 3539a12..1b82c92 100644
--- a/src/Language/PureScript/Docs/RenderedCode/RenderKind.hs
+++ b/src/Language/PureScript/Docs/RenderedCode/RenderKind.hs
@@ -10 +10 @@ module Language.PureScript.Docs.RenderedCode.RenderKind
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/RenderedCode/RenderType.hs b/src/Language/PureScript/Docs/RenderedCode/RenderType.hs
index 15f51dc..2aca7f4 100644
--- a/src/Language/PureScript/Docs/RenderedCode/RenderType.hs
+++ b/src/Language/PureScript/Docs/RenderedCode/RenderType.hs
@@ -13 +13 @@ module Language.PureScript.Docs.RenderedCode.RenderType
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Docs/RenderedCode/Types.hs b/src/Language/PureScript/Docs/RenderedCode/Types.hs
index 0d64e30..1180964 100644
--- a/src/Language/PureScript/Docs/RenderedCode/Types.hs
+++ b/src/Language/PureScript/Docs/RenderedCode/Types.hs
@@ -46 +46 @@ module Language.PureScript.Docs.RenderedCode.Types
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Errors.hs b/src/Language/PureScript/Errors.hs
index bd447fb..29b8342 100644
--- a/src/Language/PureScript/Errors.hs
+++ b/src/Language/PureScript/Errors.hs
@@ -9 +9 @@ module Language.PureScript.Errors
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Errors/JSON.hs b/src/Language/PureScript/Errors/JSON.hs
index d013235..3baec5c 100644
--- a/src/Language/PureScript/Errors/JSON.hs
+++ b/src/Language/PureScript/Errors/JSON.hs
@@ -5 +5 @@ module Language.PureScript.Errors.JSON where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Hierarchy.hs b/src/Language/PureScript/Hierarchy.hs
index 837fd3a..61f0390 100644
--- a/src/Language/PureScript/Hierarchy.hs
+++ b/src/Language/PureScript/Hierarchy.hs
@@ -18 +18 @@ module Language.PureScript.Hierarchy where
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Interactive.hs b/src/Language/PureScript/Interactive.hs
index ce051e9..15477dc 100644
--- a/src/Language/PureScript/Interactive.hs
+++ b/src/Language/PureScript/Interactive.hs
@@ -13 +13 @@ import           Prelude ()
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Interactive/Printer.hs b/src/Language/PureScript/Interactive/Printer.hs
index 38022a7..be618af 100644
--- a/src/Language/PureScript/Interactive/Printer.hs
+++ b/src/Language/PureScript/Interactive/Printer.hs
@@ -3 +3 @@ module Language.PureScript.Interactive.Printer where
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Linter.hs b/src/Language/PureScript/Linter.hs
index 7c7a275..6f42fb5 100644
--- a/src/Language/PureScript/Linter.hs
+++ b/src/Language/PureScript/Linter.hs
@@ -6 +6 @@ module Language.PureScript.Linter (lint, module L) where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Linter/Exhaustive.hs b/src/Language/PureScript/Linter/Exhaustive.hs
index 30ad256..d9d897e 100644
--- a/src/Language/PureScript/Linter/Exhaustive.hs
+++ b/src/Language/PureScript/Linter/Exhaustive.hs
@@ -11 +11 @@ module Language.PureScript.Linter.Exhaustive
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Make.hs b/src/Language/PureScript/Make.hs
index e2f9d46..3508064 100644
--- a/src/Language/PureScript/Make.hs
+++ b/src/Language/PureScript/Make.hs
@@ -23 +23 @@ module Language.PureScript.Make
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Names.hs b/src/Language/PureScript/Names.hs
index ea84f15..91b9092 100644
--- a/src/Language/PureScript/Names.hs
+++ b/src/Language/PureScript/Names.hs
@@ -9 +9 @@ module Language.PureScript.Names where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/PSString.hs b/src/Language/PureScript/PSString.hs
index 0dcb3b4..d476e41 100644
--- a/src/Language/PureScript/PSString.hs
+++ b/src/Language/PureScript/PSString.hs
@@ -16 +16 @@ module Language.PureScript.PSString
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Parser/Common.hs b/src/Language/PureScript/Parser/Common.hs
index 4eb5bb6..ecc823b 100644
--- a/src/Language/PureScript/Parser/Common.hs
+++ b/src/Language/PureScript/Parser/Common.hs
@@ -4 +4 @@ module Language.PureScript.Parser.Common where
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Parser/Lexer.hs b/src/Language/PureScript/Parser/Lexer.hs
index 9b62cc2..c2bc4bf 100644
--- a/src/Language/PureScript/Parser/Lexer.hs
+++ b/src/Language/PureScript/Parser/Lexer.hs
@@ -67 +67 @@ module Language.PureScript.Parser.Lexer
-import Prelude.Compat hiding (lex)
+import Prelude.Compat hiding (lex, (<>))
diff --git a/src/Language/PureScript/Pretty/Common.hs b/src/Language/PureScript/Pretty/Common.hs
index b728023..25ddbdd 100644
--- a/src/Language/PureScript/Pretty/Common.hs
+++ b/src/Language/PureScript/Pretty/Common.hs
@@ -8 +8 @@ module Language.PureScript.Pretty.Common where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Pretty/Kinds.hs b/src/Language/PureScript/Pretty/Kinds.hs
index 24d4451..53d21bb 100644
--- a/src/Language/PureScript/Pretty/Kinds.hs
+++ b/src/Language/PureScript/Pretty/Kinds.hs
@@ -8 +8 @@ module Language.PureScript.Pretty.Kinds
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Pretty/Types.hs b/src/Language/PureScript/Pretty/Types.hs
index bee62db..40c2956 100644
--- a/src/Language/PureScript/Pretty/Types.hs
+++ b/src/Language/PureScript/Pretty/Types.hs
@@ -17 +17 @@ module Language.PureScript.Pretty.Types
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Pretty/Values.hs b/src/Language/PureScript/Pretty/Values.hs
index 659e98c..4bb7c99 100644
--- a/src/Language/PureScript/Pretty/Values.hs
+++ b/src/Language/PureScript/Pretty/Values.hs
@@ -10 +10 @@ module Language.PureScript.Pretty.Values
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Publish/ErrorsWarnings.hs b/src/Language/PureScript/Publish/ErrorsWarnings.hs
index 8067395..f075320 100644
--- a/src/Language/PureScript/Publish/ErrorsWarnings.hs
+++ b/src/Language/PureScript/Publish/ErrorsWarnings.hs
@@ -16 +16 @@ module Language.PureScript.Publish.ErrorsWarnings
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Renamer.hs b/src/Language/PureScript/Renamer.hs
index cb28d1e..b08cabf 100644
--- a/src/Language/PureScript/Renamer.hs
+++ b/src/Language/PureScript/Renamer.hs
@@ -6 +6 @@ module Language.PureScript.Renamer (renameInModules) where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Sugar/TypeClasses/Deriving.hs b/src/Language/PureScript/Sugar/TypeClasses/Deriving.hs
index 6914f4b..3f874b0 100755
--- a/src/Language/PureScript/Sugar/TypeClasses/Deriving.hs
+++ b/src/Language/PureScript/Sugar/TypeClasses/Deriving.hs
@@ -4 +4 @@ module Language.PureScript.Sugar.TypeClasses.Deriving (deriveInstances) where
-import           Prelude.Compat
+import           Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/TypeChecker.hs b/src/Language/PureScript/TypeChecker.hs
index 63875ad..73416fd 100644
--- a/src/Language/PureScript/TypeChecker.hs
+++ b/src/Language/PureScript/TypeChecker.hs
@@ -12 +12 @@ module Language.PureScript.TypeChecker
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/TypeChecker/Entailment.hs b/src/Language/PureScript/TypeChecker/Entailment.hs
index 2ab1734..081789c 100644
--- a/src/Language/PureScript/TypeChecker/Entailment.hs
+++ b/src/Language/PureScript/TypeChecker/Entailment.hs
@@ -14 +14 @@ module Language.PureScript.TypeChecker.Entailment
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/TypeChecker/Skolems.hs b/src/Language/PureScript/TypeChecker/Skolems.hs
index 0b6ef54..9af4c06 100644
--- a/src/Language/PureScript/TypeChecker/Skolems.hs
+++ b/src/Language/PureScript/TypeChecker/Skolems.hs
@@ -11 +11 @@ module Language.PureScript.TypeChecker.Skolems
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/TypeChecker/Types.hs b/src/Language/PureScript/TypeChecker/Types.hs
index f96771a..581630a 100644
--- a/src/Language/PureScript/TypeChecker/Types.hs
+++ b/src/Language/PureScript/TypeChecker/Types.hs
@@ -27 +27 @@ module Language.PureScript.TypeChecker.Types
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/TypeClassDictionaries.hs b/src/Language/PureScript/TypeClassDictionaries.hs
index 70d138b..28c1cf3 100644
--- a/src/Language/PureScript/TypeClassDictionaries.hs
+++ b/src/Language/PureScript/TypeClassDictionaries.hs
@@ -6 +6 @@ module Language.PureScript.TypeClassDictionaries where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
diff --git a/src/Language/PureScript/Types.hs b/src/Language/PureScript/Types.hs
index d600bf4..a7af3da 100644
--- a/src/Language/PureScript/Types.hs
+++ b/src/Language/PureScript/Types.hs
@@ -10 +10 @@ module Language.PureScript.Types where
-import Prelude.Compat
+import Prelude.Compat hiding ((<>))
