diff --git a/Control/Monad/Ghc.hs b/Control/Monad/Ghc.hs
index e76cc96..d63e934 100644
--- a/Control/Monad/Ghc.hs
+++ b/Control/Monad/Ghc.hs
@@ -58,8 +58,10 @@ instance MTL.MonadIO Ghc where
     liftIO = GHC.liftIO
 #endif
 
-instance MonadCatch Ghc where
+instance MonadThrow Ghc where
     throwM  = liftIO . E.throwIO
+
+instance MonadCatch Ghc where
     catch   = GHC.gcatch
 
 #if __GLASGOW_HASKELL__ >= 700
