diff --git a/autoconf.mk.in b/autoconf.mk.in
index 86e0932..de4bdcf 100644
--- a/autoconf.mk.in
+++ b/autoconf.mk.in
@@ -24,7 +24,7 @@ LDFLAGS		:= @LDFLAGS@
 OPTLLDFLAGS     := $(addprefix -optl,$(LDFLAGS))
 LIBS            := @LIBS@
 OPTLLIBS        := $(filter-out -%,$(LIBS)) $(addprefix -optl,$(filter -%,$(LIBS)))
-GHCFLAGS	:= @GHCFLAGS@ @CPPFLAGS@ @DEFS@
+GHCFLAGS	:= -package base @GHCFLAGS@ @CPPFLAGS@ @DEFS@
 USE_GNULIBSHA1  := @USE_GNULIBSHA1@
 HAVE_HTTP       := @HAVE_HTTP@
 HAVE_CURSES     := @HAVE_CURSES@
