--- configure.ac	2006-10-09 21:44:18.000000000 +0200
+++ configure.ac.NEW	2006-10-19 02:09:47.000000000 +0200
@@ -109,7 +109,7 @@ WORKAROUND_POSIXSIGNALS([installHandler,
 
 dnl Look for Text.Regex
 
-GHC_CHECK_MODULE(Text.Regex( mkRegex, matchRegex, Regex ), text, mkRegex undefined)
+GHC_CHECK_MODULE(Text.Regex( mkRegex, matchRegex, Regex ), regex-compat, mkRegex undefined)
 
 dnl See if we need a package for QuickCheck
 
