diff -urwpN darcs-1.0.9-orig/darcs_completion darcs-1.0.9/darcs_completion
--- darcs-1.0.9-orig/darcs_completion	2007-06-07 21:42:57.000000000 +0200
+++ darcs-1.0.9/darcs_completion	2007-06-07 21:43:12.000000000 +0200
@@ -3,7 +3,6 @@
 # darcs command line completion.
 # Copyright 2002 "David Roundy" <droundy@abridgegame.org>
 #
-have darcs &&
 _darcs()
 {
     local cur
@@ -48,5 +47,5 @@ _darcs()
     return 0
 
 }
-[ "$have" ] && complete -F _darcs -o default darcs
+complete -F _darcs -o default darcs
 
