--- yesod-routes-1.0.1-orig/yesod-routes.cabal	2012-04-19 15:02:42.000000000 +1000
+++ yesod-routes-1.0.1/yesod-routes.cabal	2012-05-05 14:47:10.438175691 +1000
@@ -41,7 +41,7 @@
                  , yesod-routes
                  , text                      >= 0.5      && < 0.12
                  , HUnit                     >= 1.2      && < 1.3
-                 , hspec                     >= 0.6      && < 0.10
+                 , hspec                     >= 0.6      && < 1.1
                  , containers
                  , template-haskell
                  , path-pieces
