diff --git a/src/Numeric/Decidable/Units.hs b/src/Numeric/Decidable/Units.hs
index 9d0b92a..22482e5 100644
--- a/src/Numeric/Decidable/Units.hs
+++ b/src/Numeric/Decidable/Units.hs
@@ -1 +1,2 @@
+{-# LANGUAGE ConstrainedClassMethods #-}
 module Numeric.Decidable.Units 
diff --git a/src/Numeric/Domain/Euclidean.hs b/src/Numeric/Domain/Euclidean.hs
index d5e7c46..4da95bc 100644
--- a/src/Numeric/Domain/Euclidean.hs
+++ b/src/Numeric/Domain/Euclidean.hs
@@ -12,3 +12,3 @@ import Numeric.Natural (Natural)
 import Numeric.Ring.Class
-import Prelude (Eq (..), Integer, Maybe (..), abs)
+import Prelude (Eq (..), Integer, Maybe (..), abs, fail)
 import Prelude (fst, otherwise)
diff --git a/src/Numeric/Quadrance/Class.hs b/src/Numeric/Quadrance/Class.hs
index 1d5da8e..0c6a85f 100644
--- a/src/Numeric/Quadrance/Class.hs
+++ b/src/Numeric/Quadrance/Class.hs
@@ -21,3 +21,3 @@ instance Quadrance () a where
 
-instance Monoidal r => Quadrance r () where
+instance (Additive r, Monoidal r) => Quadrance r () where
   quadrance _ = zero
