--- a/Text/HTML/TagStream/Text.hs
+++ b/Text/HTML/TagStream/Text.hs
@@ -159,7 +159,7 @@ decodeEntitiesText =
         , decUncons  = T.uncons }
   where decodeEntity entity =
           CL.sourceList ["&",entity,";"]
-          $= XML.parseText def { XML.psDecodeEntities = XML.decodeHtmlEntities }
+          $= XML.parseTextPos def { XML.psDecodeEntities = XML.decodeHtmlEntities }
           $= CL.map snd
           $$ XML.content
 
