From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>

Compatibility patch for extra-1.7.10

diff -urN hlint-3.2.7/src/Hint/Fixities.hs hlint-3.2.7-r1/src/Hint/Fixities.hs
--- hlint-3.2.7/src/Hint/Fixities.hs	2020-11-29 14:13:03.000000000 -0700
+++ hlint-3.2.7-r1/src/Hint/Fixities.hs	2022-01-27 12:20:16.389095265 -0700
@@ -19,7 +19,7 @@
 import Hint.Type(DeclHint,Idea(..),rawIdea,toSS)
 import Config.Type
 import Control.Monad
-import Data.List.Extra
+import Data.List.Extra hiding ((!?))
 import Data.Map
 import Data.Generics.Uniplate.DataOnly
 import Refact.Types
