diff --git a/Bustle/UI/FilterDialog.hs b/Bustle/UI/FilterDialog.hs
index 248ce8d..6658d22 100644
--- a/Bustle/UI/FilterDialog.hs
+++ b/Bustle/UI/FilterDialog.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
 {-
diff --git a/Test/Renderer.hs b/Test/Renderer.hs
index a760b5a..08fac5a 100644
--- a/Test/Renderer.hs
+++ b/Test/Renderer.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
 module Main where
