diff --git a/clientsession.cabal b/clientsession.cabal
index 26f030a..9467bd8 100644
--- a/clientsession.cabal
+++ b/clientsession.cabal
@@ -29,7 +29,7 @@ library
                    , crypto-api          >= 0.6.4      && < 0.10
                    , cryptocipher        >= 0.2.5
                    , skein               >= 0.1        && < 0.2
-                   , base64-bytestring   >= 0.1.0.3    && < 0.1.1.0
+                   , base64-bytestring   >= 0.1.0.3    && < 0.2.1.0
                    , entropy             >= 0.2.1
                    , cprng-aes           >= 0.2
     exposed-modules: Web.ClientSession
