--- clientsession-0.8.0.1-orig/src/Web/ClientSession.hs	2012-09-10 23:42:50.000000000 +1000
+++ clientsession-0.8.0.1/src/Web/ClientSession.hs	2013-01-03 22:15:40.639113981 +1100
@@ -268,7 +268,7 @@
 aesRNG = do
   (bs, count) <-
       I.atomicModifyIORef aesRef $ \(ASt rng count) ->
-          let (bs', rng') = genRandomBytes rng 16
+          let (bs', rng') = genRandomBytes 16 rng
           in (ASt rng' (succ count), (bs', count))
   when (count == threshold) $ void $ forkIO aesReseed
   either (error . show) return $ decode bs
