--- epic-0.1.13-orig/epic.cabal	2011-09-12 20:49:00.000000000 +1000
+++ epic-0.1.13/epic.cabal	2011-09-18 22:45:16.672784917 +1000
@@ -28,7 +28,7 @@
                        Epic.Language Epic.Lexer Epic.CodegenC Epic.CodegenStack
                        Epic.OTTLang Epic.Simplify Epic.Stackcode 
                        Epic.Evaluator Paths_epic
-        Build-depends:	base >=4 && <5 , haskell98, mtl, Cabal, array, directory, process
+        Build-depends:	base >=4 && <5 , mtl, Cabal, array, directory, process
         Extensions:    BangPatterns
 
 Executable     epic
@@ -37,5 +37,5 @@
                        Epic.Language Epic.Lexer Epic.CodegenC Epic.CodegenStack
                        Epic.OTTLang Epic.Simplify Epic.Stackcode 
                        Epic.Evaluator Paths_epic
-               Build-depends: base >=4 && <5, mtl, array, haskell98, Cabal, directory, process
+               Build-depends: base >=4 && <5, mtl, array, Cabal, directory, process
                Extensions: BangPatterns
--- epic-0.1.13-orig/Epic/Epic.lhs	2011-09-12 20:49:00.000000000 +1000
+++ epic-0.1.13/Epic/Epic.lhs	2011-09-18 22:46:05.836099468 +1000
@@ -45,8 +45,8 @@
 Combinators for constructing an expression
 
 > import Control.Monad.State
-> import System
 > import System.IO
+> import System.Process (system)
 > import Debug.Trace
 
 > import Epic.Language
--- epic-0.1.13-orig/Epic/Stackcode.lhs	2011-09-12 20:49:00.000000000 +1000
+++ epic-0.1.13/Epic/Stackcode.lhs	2011-09-18 22:48:00.944177622 +1000
@@ -1,7 +1,7 @@
 > module Epic.Stackcode where
 
 > import Control.Monad.State
-> import List
+> import Data.List
 
 > import Epic.Language
 > import Debug.Trace
