diff -Naur helium-1.6.orig/helium/src/texthint/Main.hs helium-1.6/helium/src/texthint/Main.hs
--- helium-1.6.orig/helium/src/texthint/Main.hs	2006-02-05 17:34:36.000000000 +0100
+++ helium-1.6/helium/src/texthint/Main.hs	2006-04-18 19:13:45.000000000 +0200
@@ -18,6 +18,8 @@
 import OSSpecific(slash)
 import Directory
 
+import System.Console.Readline
+
 data State = 
     State
     { maybeModName :: Maybe String
@@ -75,6 +77,10 @@
         else
             return initialState
             
+    -- Initialize readline
+    initialize
+    -- Deactivate filename completion
+    setCompletionEntryFunction (Just $ const $ return [])
     -- Enter read-eval-print loop            
     loop stateAfterLoad
     
@@ -89,9 +95,9 @@
 
 loop :: State -> IO State
 loop state = do
-    putStr (prompt state)
-    hFlush stdout
-    command' <- getLine
+    command'' <- readline (prompt state)
+    let command' = maybe ":q" id command''
+    addHistory command'
     let command = trim command'
     newState <- case command of
         (':':cmd:rest) -> 
