--- wai-logger-0.1.2-orig/Network/Wai/Logger/Date.hs	2011-12-28 17:19:37.000000000 +1100
+++ wai-logger-0.1.2/Network/Wai/Logger/Date.hs	2011-12-31 16:54:10.271662713 +1100
@@ -1,4 +1,4 @@
-{-# LANGUAGE DoAndIfThenElse, BangPatterns #-}
+{-# LANGUAGE BangPatterns #-}
 {-# LANGUAGE CPP #-}
 
 module Network.Wai.Logger.Date (
@@ -41,9 +41,10 @@
     newEt <- GETTIME
     cache <- readIORef ref
     let oldEt = unixTime cache
-    if oldEt == newEt then
+    if oldEt == newEt 
+      then do
         return $ zonedDate cache
-    else do
+      else do
         newCache <- newDate newEt
         writeIORef ref newCache
         return $ zonedDate newCache
