commit dbe8f42b12b6ef5efb5b4fab2d54f2c9ff1e4a5e
Author: Dag Odenhall <dag.odenhall@gmail.com>
Date:   Tue Jan 17 17:48:15 2012 +0100

    apparently build-deps need repeating for test-suites

diff --git a/data-lens-ixset.cabal b/data-lens-ixset.cabal
index d2df7e8..7646282 100644
--- a/data-lens-ixset.cabal
+++ b/data-lens-ixset.cabal
@@ -26,4 +26,7 @@ Test-Suite test-data-lens-ixset
   Type          : exitcode-stdio-1.0
   GHC-Options   : -Wall
   Main-is       : test-data-lens-ixset.hs
-  Build-depends : QuickCheck == 2.4.*
+  Build-depends : base == 4.*,
+                  data-lens == 2.0.*,
+                  ixset == 1.0.*,
+                  QuickCheck == 2.4.*

