diff -urpwN hsshellscript-2.6.3/Makefile hsshellscript-2.6.3.chg/Makefile
--- hsshellscript-2.6.3/Makefile	2006-08-14 06:50:03.000000000 -0400
+++ hsshellscript-2.6.3.chg/Makefile	2006-10-06 01:54:04.000000000 -0400
@@ -64,9 +64,9 @@ VERSION		= 2.6.3
 HC 		= ghc
 CC		= gcc
 CFLAGS_GCC	= `pkg-config --cflags glib-2.0` -Ibuild -Isrc
-CFLAGS_GHC	= -ibuild:src/lib:src/test -I. -odir build -hidir build -fglasgow-exts -package-name hsshellscript -O2
+CFLAGS_GHC	= -ibuild:src/lib:src/test -I. -odir build -hidir build -fglasgow-exts -package-name hsshellscript-2.6.3 -O2
 PROFFLAGS	= -osuf p_o -hisuf p_hi -prof -auto-all
-LIBS   		= -syslib lang -syslib text -syslib util build/libhsshellscript.a `pkg-config --libs glib-2.0`
+LIBS   		= -syslib unix -syslib parsec build/libhsshellscript.a `pkg-config --libs glib-2.0`
 SRC_LIB_HS	= HsShellScript HsShellScript/Paths HsShellScript/Shell HsShellScript/Args HsShellScript/GetOpt
 SRC_LIB_CHS	= HsShellScript/Commands HsShellScript/ProcErr HsShellScript/Misc 
 LIB_HS		= $(foreach hs, $(SRC_LIB_HS), src/lib/$(hs).hs) $(foreach hs, $(SRC_LIB_CHS), build/$(hs).hs)
diff -urpwN hsshellscript-2.6.3/src/test/force_mv.hs hsshellscript-2.6.3.chg/src/test/force_mv.hs
--- hsshellscript-2.6.3/src/test/force_mv.hs	2006-08-14 06:50:03.000000000 -0400
+++ hsshellscript-2.6.3.chg/src/test/force_mv.hs	2006-10-06 00:57:09.000000000 -0400
@@ -10,14 +10,14 @@ import Data.Bits
 
 
 header = "force_mv <Parameter an mv>\n\
-         \Der Aufruf ist wie mv, auﬂer daﬂ genau eine Quelle und ein Ziel erforderlich ist (also auch kein --target-directory)."
+         \Der Aufruf ist wie mv, auer da genau eine Quelle und ein Ziel erforderlich ist (also auch kein --target-directory)."
 
 
 main = mainwrapper $ do
    (opts,pfade) <- fmap teilen $ getArgs
    case pfade of
       [quelle, ziel] -> force_cmd (\q z -> run "/bin/mv" (opts ++ [q, z])) quelle ziel
-      _              -> failIO "Es muﬂ genau eine Quelle und genau ein Ziel angegeben werden."
+      _              -> failIO "Es mu genau eine Quelle und genau ein Ziel angegeben werden."
 
 
 -- Optionen von Pfaden trennen
