diff --git a/IDE/System/Process/Internals.hs b/IDE/System/Process/Internals.hs
index 5ac49ed..d40ac8c 100644
--- a/IDE/System/Process/Internals.hs
+++ b/IDE/System/Process/Internals.hs
@@ -58,6 +58,7 @@ import Data.IORef
 #endif
 
 import System.IO 	( Handle )
+import qualified System.IO.Unsafe as U
 import System.Exit	( ExitCode )
 import Control.Concurrent
 import Control.Exception
@@ -329,7 +330,7 @@ runGenProcess_ fun CreateProcess{ cmdspec = cmdsp,
 
 {-# NOINLINE runInteractiveProcess_lock #-}
 runInteractiveProcess_lock :: MVar ()
-runInteractiveProcess_lock = unsafePerformIO $ newMVar ()
+runInteractiveProcess_lock = U.unsafePerformIO $ newMVar ()
 
 foreign import ccall unsafe "runInteractiveProcessLeksah"
   c_runInteractiveProcess
