--- process-leksah-1.0.1.4-orig/process-leksah.cabal	2010-12-04 01:58:24.000000000 +1100
+++ process-leksah-1.0.1.4/process-leksah.cabal	2012-12-08 22:22:24.628131215 +1100
@@ -55,8 +55,8 @@
      cpp-options: -Dbase3
   }
 
-  build-depends: directory >= 1.0 && < 1.2,
-                 filepath  >= 1.1 && < 1.3
+  build-depends: directory >= 1.0,
+                 filepath  >= 1.1
 
   extensions: CPP
 }
--- process-leksah-1.0.1.4-orig/IDE/System/Process/Internals.hs	2010-12-04 01:58:22.000000000 +1100
+++ process-leksah-1.0.1.4/IDE/System/Process/Internals.hs	2012-12-08 22:34:48.481478225 +1100
@@ -48,7 +48,7 @@
 
 #ifndef __HUGS__
 #if !defined(mingw32_HOST_OS) && !defined(__MINGW32__)
-import System.Posix.Types ( CPid )
+import System.Posix.Types ( CPid(..) )
 import System.Posix.Process.Internals ( pPrPr_disableITimers, c_execvpe )
 import System.IO 	( IOMode(..) )
 #else
@@ -540,8 +540,9 @@
                        False {-is_socket-}
                        False {-non-blocking-}
   fD <- FD.setNonBlockingMode fD True -- see #3316
+  le <- getLocaleEncoding
   mkHandleFromFD fD fd_type filepath mode False{-is_socket-}
-                       (Just localeEncoding)
+                       (Just le)
 #else
   fdToHandle' fd (Just Stream)
      False{-Windows: not a socket,  Unix: don't set non-blocking-}
--- process-leksah-1.0.1.4-orig/IDE/System/Process.hs	2010-12-04 01:58:22.000000000 +1100
+++ process-leksah-1.0.1.4/IDE/System/Process.hs	2012-12-08 22:42:39.759565463 +1100
@@ -78,6 +78,9 @@
 #ifdef __GLASGOW_HASKELL__
 #if __GLASGOW_HASKELL__ >= 611
 import GHC.IO.Exception	( ioException, IOErrorType(..) )
+#if __GLASGOW_HASKELL__ >= 706
+import Foreign.C.Types( CInt(..) )
+#endif
 #else
 import GHC.IOBase	( ioException, IOErrorType(..) )
 #endif
@@ -85,6 +88,7 @@
 import System.Win32.Console (generateConsoleCtrlEvent, cTRL_BREAK_EVENT)
 #else
 import System.Posix.Signals 
+import System.Posix.Types ( CPid(..) )
 #endif
 #endif
 
