--- exception-transformers-0.3.0.2-orig/exception-transformers.cabal	2012-04-24 07:54:39.000000000 +1000
+++ exception-transformers-0.3.0.2/exception-transformers.cabal	2012-07-07 23:55:58.757224249 +1000
@@ -25,7 +25,7 @@
 
   build-depends:
     base >=4 && <5,
-    stm >=2.1 && <2.4,
+    stm >=2.1 && <2.5,
     transformers >=0.2 && <0.4
 
   ghc-options:
--- exception-transformers-0.3.0.2-orig/Control/Monad/Exception.hs	2012-04-24 07:54:39.000000000 +1000
+++ exception-transformers-0.3.0.2/Control/Monad/Exception.hs	2012-07-07 23:59:00.750945649 +1000
@@ -56,7 +56,9 @@
     liftException
   ) where
 
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding (catch)
+#endif
 
 import Control.Applicative
 import qualified Control.Exception as E (Exception(..),
--- exception-transformers-0.3.0.2-orig/tests/unit/Main.hs	2012-04-24 07:54:39.000000000 +1000
+++ exception-transformers-0.3.0.2/tests/unit/Main.hs	2012-07-07 23:59:41.308774513 +1000
@@ -1,8 +1,10 @@
-{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE CPP, ScopedTypeVariables #-}
 
 module Main where
 
+#if !MIN_VERSION_base(4,6,0)
 import Prelude hiding (catch)
+#endif
 
 import Control.Monad.Exception
 import Control.Monad.Trans.Error
