diff --git a/XMonad/Prompt/MPD.hs b/XMonad/Prompt/MPD.hs
index b5eec7c..c1e8277 100644
--- a/XMonad/Prompt/MPD.hs
+++ b/XMonad/Prompt/MPD.hs
@@ -32,2 +32,3 @@ import Data.Maybe
 import Network.MPD
+import Data.String (fromString)
 import XMonad
@@ -102,3 +103,3 @@ findMatching :: RunMPD -> XPConfig -> [Metadata] -> X [Song]
 findMatching runMPD xp metas = do
-  resp <- io . runMPD . fmap extractSongs . listAllInfo $ Path B.empty
+  resp <- io . runMPD . fmap extractSongs . listAllInfo $ fromString ""
   case resp of
