--- maccatcher-2.1.2-orig/maccatcher.cabal	2011-08-03 06:15:40.000000000 +1000
+++ maccatcher-2.1.2/maccatcher.cabal	2011-08-12 21:38:36.000669438 +1000
@@ -26,8 +26,7 @@
     build-depends:    base >= 4 && < 5
   else
     build-depends:    base < 4
-  build-depends:      haskell98
-                    , binary
+  build-depends:      binary
                     , process
                     , parsec
   exposed-modules:  Data.MAC
