--- Chart-1.9-orig/Graphics/Rendering/Chart/State.hs	2015-03-31 07:57:27.000000000 +1100
+++ Chart-1.9/Graphics/Rendering/Chart/State.hs	2018-10-12 20:42:11.160485040 +1100
@@ -92,14 +92,14 @@
 -- | Pop and return the next color from the state
 takeColor :: EC l (AlphaColour Double)
 takeColor = liftCState $ do
-  (c:cs) <- use colors
+  ~(c:cs) <- use colors
   colors .= cs
   return c
 
 -- | Pop and return the next shape from the state
 takeShape :: EC l PointShape
 takeShape = liftCState $ do
-  (c:cs) <- use shapes
+  ~(c:cs) <- use shapes
   shapes .= cs
   return c
 
