diff -urN ormolu-0.1.4.1/ormolu.cabal ormolu-0.1.4.1-new/ormolu.cabal
--- ormolu-0.1.4.1/ormolu.cabal	2001-09-08 19:46:40.000000000 -0600
+++ ormolu-0.1.4.1-new/ormolu.cabal	2021-02-03 13:13:58.462579107 -0700
@@ -11,45 +11,6 @@
 description:     A formatter for Haskell source code.
 category:        Development, Formatting
 build-type:      Simple
-data-files:
-    data/examples/declaration/annotation/*.hs
-    data/examples/declaration/class/*.hs
-    data/examples/declaration/data/*.hs
-    data/examples/declaration/data/gadt/*.hs
-    data/examples/declaration/default/*.hs
-    data/examples/declaration/deriving/*.hs
-    data/examples/declaration/foreign/*.hs
-    data/examples/declaration/instance/*.hs
-    data/examples/declaration/rewrite-rule/*.hs
-    data/examples/declaration/role-annotation/*.hs
-    data/examples/declaration/signature/complete/*.hs
-    data/examples/declaration/signature/fixity/*.hs
-    data/examples/declaration/signature/inline/*.hs
-    data/examples/declaration/signature/minimal/*.hs
-    data/examples/declaration/signature/pattern/*.hs
-    data/examples/declaration/signature/set-cost-centre/*.hs
-    data/examples/declaration/signature/specialize/*.hs
-    data/examples/declaration/signature/type/*.hs
-    data/examples/declaration/splice/*.hs
-    data/examples/declaration/type-families/closed-type-family/*.hs
-    data/examples/declaration/type-families/data-family/*.hs
-    data/examples/declaration/type-families/type-family/*.hs
-    data/examples/declaration/type-synonyms/*.hs
-    data/examples/declaration/type/*.hs
-    data/examples/declaration/value/function/*.hs
-    data/examples/declaration/value/function/arrow/*.hs
-    data/examples/declaration/value/function/comprehension/*.hs
-    data/examples/declaration/value/function/do/*.hs
-    data/examples/declaration/value/function/infix/*.hs
-    data/examples/declaration/value/function/pattern/*.hs
-    data/examples/declaration/value/other/*.hs
-    data/examples/declaration/value/pattern-synonyms/*.hs
-    data/examples/declaration/warning/*.hs
-    data/examples/import/*.hs
-    data/examples/module-header/*.hs
-    data/examples/other/*.hs
-    data/diff-tests/inputs/*.hs
-    data/diff-tests/outputs/*.txt
 
 extra-doc-files:
     CONTRIBUTING.md
@@ -141,7 +102,13 @@
     else
         ghc-options: -O2 -Wall
 
+flag executable
+    description: Install the ormolu executable
+    default: False
+
 executable ormolu
+    if !flag(executable)
+        buildable: False
     main-is:          Main.hs
     hs-source-dirs:   app
     other-modules:    Paths_ormolu
