diff -urN ormolu-0.1.4.1/ormolu.cabal ormolu-0.1.4.1-new/ormolu.cabal
--- ormolu-0.1.4.1/ormolu.cabal	2001-09-08 19:46:40.000000000 -0600
+++ ormolu-0.1.4.1-new/ormolu.cabal	2021-02-03 13:11:58.480121392 -0700
@@ -141,7 +141,13 @@
     else
         ghc-options: -O2 -Wall
 
+flag executable
+    description: Install the ormolu executable
+    default: False
+
 executable ormolu
+    if !flag(executable)
+        buildable: False
     main-is:          Main.hs
     hs-source-dirs:   app
     other-modules:    Paths_ormolu
