--- yi-contrib-0.6.4.0-orig/src/Yi/Config/Users/Reiner.hs	2011-07-21 17:51:21.000000000 +1000
+++ yi-contrib-0.6.4.0/src/Yi/Config/Users/Reiner.hs	2011-07-22 01:27:19.845246755 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE NoMonomorphismRestriction, NamedFieldPuns, DoAndIfThenElse #-}
+{-# LANGUAGE NoMonomorphismRestriction, NamedFieldPuns #-}
 module Yi.Config.Users.Reiner (setup, main) where
 
 import Yi.Config.Simple
@@ -42,10 +42,10 @@
     let dir = takeDirectory bufInfoFileName
     exists <- io $ doesDirectoryExist dir
     if not exists 
-    then do
+     then do
             io $ createDirectoryIfMissing True dir
             withEditor $ printMsg $  "Created directory '" ++ dir ++ "'."
-    else withEditor $ printMsg $ "Directory already exists!"
+     else withEditor $ printMsg $ "Directory already exists!"
 
 sectionSize = 80
 
