--- hslogger-1.2.6-orig/hslogger.cabal	2014-10-28 08:33:01.000000000 +1100
+++ hslogger-1.2.6/hslogger.cabal	2014-12-31 23:44:34.133177851 +1100
@@ -47,7 +47,7 @@
         Build-Depends: unix
     if flag(small_base)
         build-depends: base >= 4 && < 5, containers, directory, process,
-                       time < 1.5, old-locale
+                       time < 1.6, old-locale
     else
         build-depends: base < 3, time < 1.5
     GHC-Options: -O2
--- hslogger-1.2.6-orig/src/System/Log/Handler/Log4jXML.hs	2014-10-28 08:33:01.000000000 +1100
+++ hslogger-1.2.6/src/System/Log/Handler/Log4jXML.hs	2014-12-31 23:38:09.703125933 +1100
@@ -109,7 +109,11 @@
 import Control.Concurrent.MVar
 import Data.List (isPrefixOf)
 import System.IO
+#if MIN_VERSION_time(1,5,0)
+import Data.Time (defaultTimeLocale)
+#else
 import System.Locale (defaultTimeLocale)
+#endif
 import Data.Time
 import System.Log
 import System.Log.Handler
--- hslogger-1.2.6-orig/src/System/Log/Formatter.hs	2014-10-28 08:33:01.000000000 +1100
+++ hslogger-1.2.6/src/System/Log/Formatter.hs	2014-12-31 23:37:02.032865470 +1100
@@ -28,8 +28,13 @@
 import System.Posix.Process (getProcessID)
 #endif
 
+
+#if MIN_VERSION_time(1,5,0)
+import Data.Time (getZonedTime,getCurrentTime,formatTime,defaultTimeLocale)
+#else
 import System.Locale (defaultTimeLocale)
 import Data.Time (getZonedTime,getCurrentTime,formatTime)
+#endif
 
 import System.Log
 
