--- hspec-0.9.0-orig/Test/Hspec/Core.hs	2011-10-01 10:52:27.000000000 +1000
+++ hspec-0.9.0/Test/Hspec/Core.hs	2011-12-20 12:17:02.821130417 +1100
@@ -124,6 +124,6 @@
 isFailure _        = False
 
 -- | Create a more readable display of a quantity of something.
-quantify :: Num a => a -> String -> String
+quantify :: (Eq a, Num a, Show a) => a -> String -> String
 quantify 1 s = "1 " ++ s
 quantify n s = show n ++ " " ++ s ++ "s"
