diff --git a/Graphics/HGL/Internals/Utilities.hs b/Graphics/HGL/Internals/Utilities.hs
index dc80fa0..f58b8e6 100644
--- a/Graphics/HGL/Internals/Utilities.hs
+++ b/Graphics/HGL/Internals/Utilities.hs
@@ -20,7 +20,7 @@ module Graphics.HGL.Internals.Utilities(
         modMVar, modMVar_
 	) where
 
-import qualified Control.Exception as E (bracket, try, IOException, tryJust, ioErrors)
+import qualified Control.OldException as E (bracket, try, IOException, tryJust, ioErrors)
 import Control.Concurrent( MVar, takeMVar, putMVar )
 
 bracket :: IO a -> (a -> IO b) -> (a -> IO c) -> IO c
diff --git a/HGL.cabal b/HGL.cabal
index 7c5b612..fbf89e8 100644
--- a/HGL.cabal
+++ b/HGL.cabal
@@ -22,7 +22,7 @@ flag split-base
 
 library
   if flag(split-base)
-    build-depends: base >= 3 && < 4, array
+    build-depends: base >= 4 && < 5, array
   else
     build-depends: base < 2
   exposed-modules:
