diff --git a/Test/TestMonoidSubclasses.hs b/Test/TestMonoidSubclasses.hs
index 78e81ed..3e219ee 100644
--- a/Test/TestMonoidSubclasses.hs
+++ b/Test/TestMonoidSubclasses.hs
@@ -764,2 +764,3 @@ instance TextualMonoid TestString where
 
+#if ! MIN_VERSION_QuickCheck(2,9,2)
 instance Arbitrary All where
@@ -784,2 +785,3 @@ instance Arbitrary a => Arbitrary (Sum a) where
    arbitrary = fmap Sum arbitrary
+#endif
 
@@ -803,2 +805,3 @@ instance (Arbitrary a, Arbitrary b) => Arbitrary (Stateful a b) where
 
+#if ! MIN_VERSION_QuickCheck(2,9,2)
 instance CoArbitrary All where
@@ -823,2 +826,3 @@ instance CoArbitrary a => CoArbitrary (Sum a) where
    coarbitrary (Sum a) = coarbitrary a
+#endif
 
