commit 7dcb635088e73b508dbe00ae7fe08dae99968416
Author: Tim King <taking@cs.nyu.edu>
Date:   Thu Jul 16 20:49:51 2015 +0200

    Fix for BOOST_SED_CPP for gcc-5.

diff --git a/config/boost.m4 b/config/boost.m4
index 80a6968..da49c5e 100644
--- a/config/boost.m4
+++ b/config/boost.m4
@@ -72,7 +72,8 @@ dnl boost_cv_lib_version='1_37\r' for instance, which breaks
 dnl everything else.
 dnl Cannot use 'dnl' after [$4] because a trailing dnl may break AC_CACHE_CHECK
 (eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
-  tr -d '\r' |
+  grep -v "^#" |
+  tr -d '\r\n ' |
   $SED -n -e "$1" >conftest.i 2>&1],
   [$3],
   [$4])
@@ -206,7 +207,7 @@ AC_LANG_POP([C++])dnl
   AC_CACHE_CHECK([for Boost's header version],
     [boost_cv_lib_version],
     [m4_pattern_allow([^BOOST_LIB_VERSION$])dnl
-     _BOOST_SED_CPP([/^boost-lib-version = /{s///;s/\"//g;p;q;}],
+     _BOOST_SED_CPP([/^boost-lib-version=/{s///;s/\"//g;p;q;}],
                     [#include <boost/version.hpp>
 boost-lib-version = BOOST_LIB_VERSION],
     [boost_cv_lib_version=`cat conftest.i`])])
