diff -rN -u old-darcs-fastconvert/darcs-fastconvert.cabal new-darcs-fastconvert/darcs-fastconvert.cabal
--- old-darcs-fastconvert/darcs-fastconvert.cabal	2011-12-04 16:33:04.317731889 +0300
+++ new-darcs-fastconvert/darcs-fastconvert.cabal	2011-12-04 16:33:04.318731889 +0300
@@ -68,7 +68,7 @@
   Build-depends: hashed-storage >= 0.5.3 && < 0.6,
                  cmdlib >= 0.3 && < 0.4,
                  base >= 4 && < 5,
-                 attoparsec >= 0.8 && < 0.9,
+                 attoparsec == 0.10.*,
                  datetime, old-time, filepath, bytestring, mtl,
                  directory, utf8-string, containers

diff -rN -u old-darcs-fastconvert/Import.hs new-darcs-fastconvert/Import.hs
--- old-darcs-fastconvert/Import.hs	2011-12-04 16:33:04.316731889 +0300
+++ new-darcs-fastconvert/Import.hs	2011-12-04 16:33:04.317731889 +0300
@@ -55,6 +55,7 @@
 import Marks

 import qualified Data.Attoparsec.Char8 as A
+import qualified Data.Attoparsec.Types as A
 import Data.Attoparsec.Char8( (<?>) )

 data RepoFormat = Darcs2Format | HashedFormat deriving (Eq, Data, Typeable)
