--- mueval-0.8.2-orig/mueval.cabal	2012-03-17 07:29:38.000000000 +1100
+++ mueval-0.8.2/mueval.cabal	2012-04-13 10:51:30.520598695 +1000
@@ -31,7 +31,7 @@
 library
         exposed-modules:     Mueval.Parallel, Mueval.Context, Mueval.Interpreter,
                              Mueval.ArgsParse, Mueval.Resources
-        build-depends:       base>=4 && < 5, containers, directory, mtl>2, filepath, unix, process,
+        build-depends:       base>=4 && < 5, containers, directory, mtl<2.2, filepath, unix, process,
                              hint>=0.3.1, show>=0.3, utf8-string, Cabal, extensible-exceptions
         ghc-options:         -Wall -static -O2
 
--- mueval-0.8.2-orig/Mueval/Interpreter.hs	2012-03-17 07:29:38.000000000 +1100
+++ mueval-0.8.2/Mueval/Interpreter.hs	2012-04-13 10:44:45.313405110 +1000
@@ -2,7 +2,7 @@
 -- TODO: suggest the convenience functions be put into Hint proper?
 module Mueval.Interpreter where
 
-import Control.Monad (guard,mplus,unless,when)
+import Control.Monad (guard,mplus,unless,when,Functor)
 import Control.Monad.Trans (MonadIO)
 import Control.Monad.Writer (Any(..),runWriterT,tell)
 import Data.Char (isDigit)
