diff --git a/doc-test.hs b/doc-test.hs
index 7fa4db4..3d3317e 100644
--- a/doc-test.hs
+++ b/doc-test.hs
@@ -15,7 +15,7 @@ module Main ( main ) where
 import Test.DocTest
 
 main :: IO ()
-main = doctest
+main = doctest $ [ "-ignore-package=monads-fd", "-ignore-package=monads-tf"] ++
        [ "Text/ParserCombinators/Parsec/Rfc2234.hs"
        , "Text/ParserCombinators/Parsec/Rfc2821.hs"
        , "Text/ParserCombinators/Parsec/Rfc2822.hs"
