diff --git a/src/main/Test.hs b/src/main/Test.hs
index 5a8484e..750fdbf 100644
--- a/src/main/Test.hs
+++ b/src/main/Test.hs
@@ -61,7 +61,7 @@ toSpec = go
                 shouldBeReadable (reformat cfg code) (L.fromStrict codeExpect)
               go next'
             _ ->
-              fail
+              error
                 "'haskell given' block must be followed by a 'haskell expect' block"
         "haskell pending" -> do
           it (UTF8.toString desc) pending
