--- aws-0.19-orig/aws.cabal	2018-02-12 05:02:03.000000000 +1100
+++ aws-0.19/aws.cabal	2018-04-25 19:47:26.257724413 +1000
@@ -153,6 +153,9 @@
   if !impl(ghc >= 7.6)
     Build-depends: ghc-prim
 
+  if !impl(ghc >= 8.0)
+    Build-depends: semigroups == 0.18.*
+
   GHC-Options: -Wall
 
   Default-Language: Haskell2010
--- aws-0.19-orig/Aws/Iam/Core.hs	2018-04-25 18:59:56.348205221 +1000
+++ aws-0.19/Aws/Iam/Core.hs	2018-04-25 19:26:50.629766975 +1000
@@ -28,7 +28,8 @@
 import           Data.IORef
 import           Data.List                      (intersperse, sort)
 import           Data.Maybe
-import           Data.Monoid
+import           Data.Monoid                    ()
+import qualified Data.Semigroup                 as Sem
 import           Data.Text                      (Text)
 import qualified Data.Text                      as Text
 import           Data.Time
@@ -60,9 +61,12 @@
 instance Loggable IamMetadata where
     toLogText (IamMetadata r) = "IAM: request ID=" <> fromMaybe "<none>" r
 
+instance Sem.Semigroup IamMetadata where
+    IamMetadata r1 <> IamMetadata r2 = IamMetadata (r1 `mplus` r2)
+
 instance Monoid IamMetadata where
     mempty = IamMetadata Nothing
-    IamMetadata r1 `mappend` IamMetadata r2 = IamMetadata (r1 `mplus` r2)
+    mappend = (<>)
 
 data IamConfiguration qt
     = IamConfiguration {
--- aws-0.19-orig/Aws/DynamoDb/Core.hs	2018-04-25 18:59:56.346205218 +1000
+++ aws-0.19/Aws/DynamoDb/Core.hs	2018-04-25 19:26:21.464720365 +1000
@@ -143,7 +143,8 @@
 import           Data.List
 import qualified Data.Map                     as M
 import           Data.Maybe
-import           Data.Monoid
+import           Data.Monoid                  ()
+import qualified Data.Semigroup               as Sem
 import           Data.Proxy
 import           Data.Scientific
 import qualified Data.Serialize               as Ser
@@ -749,9 +750,12 @@
         ", x-amz-id-2=" `mappend`
         fromMaybe "<none>" id2
 
+instance Sem.Semigroup DdbResponse where
+    a <> b = DdbResponse (ddbrCrc a `mplus` ddbrCrc b) (ddbrMsgId a `mplus` ddbrMsgId b)
+
 instance Monoid DdbResponse where
     mempty = DdbResponse Nothing Nothing
-    mappend a b = DdbResponse (ddbrCrc a `mplus` ddbrCrc b) (ddbrMsgId a `mplus` ddbrMsgId b)
+    mappend = (<>)
 
 
 data Region = Region {
@@ -1273,10 +1277,14 @@
                                    in runParser a kf' ks
     {-# INLINE mplus #-}
 
+instance Sem.Semigroup (Parser a) where
+    (<>) = mplus
+    {-# INLINE (<>) #-}
+
 instance Monoid (Parser a) where
     mempty  = fail "mempty"
     {-# INLINE mempty #-}
-    mappend = mplus
+    mappend = (<>)
     {-# INLINE mappend #-}
 
 apP :: Parser (a -> b) -> Parser a -> Parser b
--- aws-0.19-orig/Aws/Sqs/Core.hs	2018-04-25 18:59:56.355205232 +1000
+++ aws-0.19/Aws/Sqs/Core.hs	2018-04-25 19:25:55.550678935 +1000
@@ -17,6 +17,7 @@
 import           Data.List
 import           Data.Maybe
 import           Data.Monoid
+import qualified Data.Semigroup                 as Sem
 import           Data.Ord
 import qualified Data.Text                      as T
 import qualified Data.Text.Encoding             as T
@@ -65,9 +66,12 @@
                                       ", x-amz-id-2=" `mappend`
                                       fromMaybe "<none>" id2
 
+instance Sem.Semigroup SqsMetadata where
+    SqsMetadata a1 r1 <> SqsMetadata a2 r2 = SqsMetadata (a1 `mplus` a2) (r1 `mplus` r2)
+
 instance Monoid SqsMetadata where
     mempty = SqsMetadata Nothing Nothing
-    SqsMetadata a1 r1 `mappend` SqsMetadata a2 r2 = SqsMetadata (a1 `mplus` a2) (r1 `mplus` r2)
+    mappend = (<>)
 
 data SqsAuthorization 
     = SqsAuthorizationHeader 
--- aws-0.19-orig/Aws/Ses/Core.hs	2018-02-12 05:02:03.000000000 +1100
+++ aws-0.19/Aws/Ses/Core.hs	2018-04-25 19:25:02.632594285 +1000
@@ -33,6 +33,7 @@
 import           Data.IORef
 import           Data.Maybe
 import           Data.Monoid
+import qualified Data.Semigroup                 as Sem
 import           Data.Text                      (Text)
 import qualified Data.Text.Encoding             as TE
 import           Data.Typeable
@@ -61,9 +62,12 @@
 instance Loggable SesMetadata where
     toLogText (SesMetadata rid) = "SES: request ID=" `mappend` fromMaybe "<none>" rid
 
+instance Sem.Semigroup SesMetadata where
+    SesMetadata r1 <> SesMetadata r2 = SesMetadata (r1 `mplus` r2)
+
 instance Monoid SesMetadata where
     mempty = SesMetadata Nothing
-    SesMetadata r1 `mappend` SesMetadata r2 = SesMetadata (r1 `mplus` r2)
+    mappend = (<>)
 
 data SesConfiguration qt
     = SesConfiguration {
@@ -184,11 +188,13 @@
           s = Blaze.fromByteString
           one = 1 :: Int
 
-instance Monoid Destination where
-    mempty = Destination [] [] []
-    mappend (Destination a1 a2 a3) (Destination b1 b2 b3) =
+instance Sem.Semigroup Destination where
+    (Destination a1 a2 a3) <> (Destination b1 b2 b3) =
         Destination (a1 ++ b1) (a2 ++ b2) (a3 ++ b3)
 
+instance Monoid Destination where
+    mempty = Destination [] [] []
+    mappend = (<>)
 
 -- | An e-mail address.
 type EmailAddress = Text
--- aws-0.19-orig/Aws/SimpleDb/Core.hs	2018-02-12 05:02:03.000000000 +1100
+++ aws-0.19/Aws/SimpleDb/Core.hs	2018-04-25 19:23:37.304457651 +1000
@@ -12,6 +12,7 @@
 import           Data.List
 import           Data.Maybe
 import           Data.Monoid
+import qualified Data.Semigroup                 as Sem
 import qualified Data.Text                      as T
 import qualified Data.Text.Encoding             as T
 import           Data.Typeable
@@ -46,9 +47,12 @@
                                      ", box usage=" `mappend`
                                      fromMaybe "<not available>" bu
 
+instance Sem.Semigroup SdbMetadata where
+    SdbMetadata r1 b1 <> SdbMetadata r2 b2 = SdbMetadata (r1 `mplus` r2) (b1 `mplus` b2)
+
 instance Monoid SdbMetadata where
     mempty = SdbMetadata Nothing Nothing
-    SdbMetadata r1 b1 `mappend` SdbMetadata r2 b2 = SdbMetadata (r1 `mplus` r2) (b1 `mplus` b2)
+    mappend = (<>)
 
 data SdbConfiguration qt
     = SdbConfiguration {
--- aws-0.19-orig/Aws/S3/Core.hs	2018-04-25 18:59:56.351205226 +1000
+++ aws-0.19/Aws/S3/Core.hs	2018-04-25 19:21:10.647225357 +1000
@@ -14,6 +14,7 @@
 import           Data.List
 import           Data.Maybe
 import           Data.Monoid
+import qualified Data.Semigroup                 as Sem
 import           Control.Applicative            ((<|>))
 import           Data.Time
 import           Data.Typeable
@@ -164,9 +165,12 @@
       }
     deriving (Show, Typeable)
 
+instance Sem.Semigroup S3Metadata where
+    S3Metadata a1 r1 <> S3Metadata a2 r2 = S3Metadata (a1 `mplus` a2) (r1 `mplus` r2)
+
 instance Monoid S3Metadata where
     mempty = S3Metadata Nothing Nothing
-    S3Metadata a1 r1 `mappend` S3Metadata a2 r2 = S3Metadata (a1 `mplus` a2) (r1 `mplus` r2)
+    mappend = (<>)
 
 instance Loggable S3Metadata where
     toLogText (S3Metadata id2 rid) = "S3: request ID=" `mappend`
