--- HJScript-0.5.0-orig/src/HJScript/XMLGenerator.hs	2010-04-23 07:28:30.000000000 +1000
+++ HJScript-0.5.0/src/HJScript/XMLGenerator.hs	2012-02-19 00:46:58.106661136 +1100
@@ -23,9 +23,9 @@
 type Attribute = Exp AttributeNode
 
 instance HSX.XMLGen HJScript' where
- type HSX.XML          HJScript' = XML
- newtype HSX.Child     HJScript' = HJSChild Child
- newtype HSX.Attribute HJScript' = HJSAttr Attribute
+ type XML          HJScript' = XML
+ newtype Child     HJScript' = HJSChild Child
+ newtype Attribute HJScript' = HJSAttr Attribute
  genElement = element
  genEElement = eElement
  xmlToChild = HJSChild . castToNode
