--- netbsd-cvs/Makefile
+++ libedit-uNF/Makefile
@@ -5,6 +5,7 @@
 
 WARNS=	3
 LIB=	edit
+LIBS=-lncurses
 
 LIBDPLIBS=     termcap ${.CURDIR}/../libterm
 
@@ -25,9 +26,10 @@
 	editline.3 tok_line.3 editline.3 tok_str.3
 
 # For speed and debugging
-#SRCS=   ${OSRCS} tokenizer.c history.c readline.c
+SRCS=   ${OSRCS} tokenizer.c history.c readline.c
 # For protection
-SRCS=	editline.c tokenizer.c history.c readline.c
+#SRCS=	editline.c tokenizer.c history.c readline.c
+OOBJS= $(patsubst %.c,%.o,$(SRCS))
 
 LIBEDITDIR?=${.CURDIR}
 
@@ -51,54 +51,68 @@
 
 SUBDIR=	readline
 
+HOST_SH=sh
+AR=ar
+LIBEDITDIR=.
+_MKTARGET_CREATE=true
+_MKTARGET_LINK=true
+all: .depend libedit.a libedit.so
+
 vi.h: vi.c makelist Makefile
 	${_MKTARGET_CREATE}
 	${HOST_SH} ${LIBEDITDIR}/makelist -h ${LIBEDITDIR}/vi.c \
-	    > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	    > $@.tmp && \
+	    mv $@.tmp $@
 
 emacs.h: emacs.c makelist Makefile
 	${_MKTARGET_CREATE}
 	${HOST_SH} ${LIBEDITDIR}/makelist -h ${LIBEDITDIR}/emacs.c \
-	    > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	    > $@.tmp && \
+	    mv $@.tmp $@
 
 common.h: common.c makelist Makefile
 	${_MKTARGET_CREATE}
 	${HOST_SH} ${LIBEDITDIR}/makelist -h ${LIBEDITDIR}/common.c \
-	    > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	    > $@.tmp && \
+	    mv $@.tmp $@
 
 fcns.h: ${AHDR} makelist Makefile
 	${_MKTARGET_CREATE}
-	${HOST_SH} ${LIBEDITDIR}/makelist -fh ${AHDR} > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	${HOST_SH} ${LIBEDITDIR}/makelist -fh ${AHDR} > $@.tmp && \
+	    mv $@.tmp $@
 
 fcns.c: ${AHDR} fcns.h help.h makelist Makefile
 	${_MKTARGET_CREATE}
-	${HOST_SH} ${LIBEDITDIR}/makelist -fc ${AHDR} > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	${HOST_SH} ${LIBEDITDIR}/makelist -fc ${AHDR} > $@.tmp && \
+	    mv $@.tmp $@
 
 help.c: ${ASRC} makelist Makefile
 	${_MKTARGET_CREATE}
-	${HOST_SH} ${LIBEDITDIR}/makelist -bc ${ASRC} > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	${HOST_SH} ${LIBEDITDIR}/makelist -bc ${ASRC} > $@.tmp && \
+	    mv $@.tmp $@
 
 help.h: ${ASRC} makelist Makefile
 	${_MKTARGET_CREATE}
-	${HOST_SH} ${LIBEDITDIR}/makelist -bh ${ASRC} > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	${HOST_SH} ${LIBEDITDIR}/makelist -bh ${ASRC} > $@.tmp && \
+	    mv $@.tmp $@
 
 editline.c: ${OSRCS} makelist Makefile
 	${_MKTARGET_CREATE}
-	${HOST_SH} ${LIBEDITDIR}/makelist -e ${OSRCS:T} > ${.TARGET}.tmp && \
-	    mv ${.TARGET}.tmp ${.TARGET}
+	${HOST_SH} ${LIBEDITDIR}/makelist -e ${OSRCS:T} > $@.tmp && \
+	    mv $@.tmp $@
 
 tc1.o:	${LIBEDITDIR}/TEST/tc1.c
 	
-tc1:	libedit.a tc1.o 
+tc1:
 	${_MKTARGET_LINK}
-	${CC} ${LDFLAGS} ${.ALLSRC} -o ${.TARGET} libedit.a ${LDADD} -ltermcap
+	${CC} ${CFLAGS} ${LDFLAGS} ${.ALLSRC} -o $@ TEST/test.c libedit.a ${LDADD} ${LIBS}
 
-.include <bsd.lib.mk>
-.include <bsd.subdir.mk>
+.depend: vi.h emacs.h common.h fcns.h help.h help.c
+	-mkdir .a .so
+.c.o:
+	$(CC) $(CFLAGS) -c $< -o .a/$*
+	$(CC) $(CFLAGS) -fPIC -c $< -o .so/$*
+libedit.a: $(OOBJS)
+	$(AR) -r $@ .a/*
+libedit.so: $(OOBJS)
+	$(CC) --shared -o $@ .so/* $(LIBS) -Wl,-soname,libedit.so
--- netbsd-cvs/el.c
+++ libedit-uNF/el.c
@@ -439,8 +439,10 @@
 		static const char elpath[] = "/.editrc";
 		char path[MAXPATHLEN];
 
+#if 0
 		if (issetugid())
 			return (-1);
+#endif
 		if ((ptr = getenv("HOME")) == NULL)
 			return (-1);
 		if (strlcpy(path, ptr, sizeof(path)) >= sizeof(path))
