diff --git a/recursion-schemes.cabal b/recursion-schemes.cabal
index 36d203e..10d4b7b 100644
--- a/recursion-schemes.cabal
+++ b/recursion-schemes.cabal
@@ -56,10 +56,10 @@ library
 
   build-depends: 
     base                 >= 4 && < 5,
-    transformers         >= 0.2 && < 0.3,
+    transformers         >= 0.2 && < 0.4,
     comonad              >= 1.1.1 && < 1.2,
-    comonad-transformers >= 2.0 && < 2.1,
-    free                 >= 2.0 && < 2.1
+    comonad-transformers >= 2.0 && < 2.2,
+    free                 >= 2.0 && < 2.2
 
   exposed-modules:
     Data.Functor.Foldable
