diff --git a/TailHandle.hs b/TailHandle.hs
index bd499b9..d510211 100644
--- a/TailHandle.hs
+++ b/TailHandle.hs
@@ -68,7 +68,7 @@ closeTail th@TailHandle{ thFd = Just fd } = do
   }
 #ifdef INOTIFY
   where
-    rm_watch = INotify.removeWatch (Data.Maybe.fromJust (trINotify (thRuntime th)))
+    rm_watch = INotify.removeWatch
 #endif
 
 seekTail :: FileOffset -> TailHandle -> IO TailHandle
diff --git a/ztail.cabal b/ztail.cabal
index 523a43a..23e3853 100644
--- a/ztail.cabal
+++ b/ztail.cabal
@@ -19,5 +19,5 @@ Executable ztail
     Other-Modules:	Util, Display, TailTypes, TailHandle, Tail
     Build-Depends:	base == 4.*, regex-compat, unix, time, old-locale, process, array
     if flag(inotify)
-        Build-Depends:	hinotify
+        Build-Depends:	hinotify >= 0.3.2
         CPP-Options:	-DINOTIFY
