--- src/Text/XML/HaXml/Parse.hs	2006-06-25 20:58:45.000000000 +0200
+++ src/Text/XML/HaXml/Parse.hs	2006-06-25 20:58:54.000000000 +0200
@@ -562,7 +562,7 @@
           ] `onFail`
     do { t <- next
        ; failP ("Expected one of\ 
-               \ (ID, IDREF, IDREFS, ENTITY, ENTITIES, NMTOKEN, NMTOKENS)\ 
+                 (ID, IDREF, IDREFS, ENTITY, ENTITIES, NMTOKEN, NMTOKENS)\ 
                \\nbut got "++show t)
        }
 -- `adjustErr` (++"\nLooking for a tokenized type:\n\ 
--- src/Text/XML/HaXml/SAX.hs	2006-06-25 21:05:33.000000000 +0200
+++ src/Text/XML/HaXml/SAX.hs	2006-06-25 21:05:37.000000000 +0200
@@ -55,7 +55,7 @@
             , saxchardata
             ]
 	`adjustErr` (++"\nLooking for a SAX event:\n\ 
-                 \  elem-open, elem-close, PI, comment, DTD, ref, or chardata")
+                    elem-open, elem-close, PI, comment, DTD, ref, or chardata")
 
 saxelementopen :: XParser SaxElement
 saxelementopen = do
