diff --git a/test/Data/Conduit/Process/UnixSpec.hs b/test/Data/Conduit/Process/UnixSpec.hs
index fc5d3e4..4cabb50 100644
--- a/test/Data/Conduit/Process/UnixSpec.hs
+++ b/test/Data/Conduit/Process/UnixSpec.hs
@@ -81,6 +81,6 @@ spec = describe "unix-process-conduit" $ do
             (Just sink)
             Nothing
         res <- waitForProcess pid
-        lbs <- getLBS
+        lbs <- fmap (filter (L.isPrefixOf "foo=") . L.split 10) getLBS
         res `shouldBe` ExitSuccess
-        lbs `shouldBe` L.fromChunks ["foo=bar\n"]
+        lbs `shouldBe` [L.fromChunks ["foo=bar"]]
