--- haskell-mpi-1.2.0-orig/haskell-mpi.cabal	2012-01-13 07:47:41.000000000 +1100
+++ haskell-mpi-1.2.0/haskell-mpi.cabal	2012-01-15 19:26:10.259292470 +1100
@@ -114,7 +114,6 @@
       src
    build-depends:
       base > 3 && <= 5,
-      haskell98,
       bytestring,
       cereal,
       extensible-exceptions,
--- haskell-mpi-1.2.0-orig/src/C2HS.hs	2012-01-13 07:47:41.000000000 +1100
+++ haskell-mpi-1.2.0/src/C2HS.hs	2012-01-15 19:26:10.260292494 +1100
@@ -40,7 +40,7 @@
   module Foreign,
 
   -- * Re-export the C language component of the FFI
-  module CForeign,
+  module Foreign.C,
 
   -- * Composite marshalling functions
   withCStringLenIntConv, peekCStringLenIntConv, withIntConv, withFloatConv,
@@ -61,9 +61,9 @@
        hiding       (Word)
 		    -- Should also hide the Foreign.Marshal.Pool exports in
 		    -- compilers that export them
-import CForeign
+import Foreign.C
 
-import Monad        (liftM)
+import Control.Monad (liftM)
 
 
 -- Composite marshalling functions
@@ -208,7 +208,7 @@
 
 -- |Obtain Haskell 'Bool' from C value.
 --
-cToBool :: Num a => a -> Bool
+cToBool :: (Eq a, Num a) => a -> Bool
 cToBool  = toBool
 
 -- |Convert a C enumeration to Haskell.
