commit c9e9537a1d05480649e9f6df70d88b59eb53e5bb (HEAD -> patch-1, origin/patch-1)
Author: Ollie Charles <ollie@ocharles.org.uk>
Date:   Wed Oct 2 14:40:41 2019 +0100

    Update QQ.hs

diff --git a/src/Data/JSON/QQ.hs b/src/Data/JSON/QQ.hs
index ef974bd..c7eca25 100644
--- a/src/Data/JSON/QQ.hs
+++ b/src/Data/JSON/QQ.hs
@@ -75,7 +75,7 @@ jpNumber = JsonNumber <$> do
       d <- many digit
       return $ o:d
 
-    convert :: Monad m => String -> m Scientific
+    convert :: String -> Parser Scientific
     convert = either fail return . A.parseOnly (A.scientific <* A.endOfInput) . T.pack
 
 jpObject :: JsonParser
