--- hxt-9.1.5-orig/hxt.cabal	2011-11-11 03:20:43.000000000 +1100
+++ hxt-9.1.5/hxt.cabal	2011-12-18 20:02:50.529426004 +1100
@@ -166,7 +166,6 @@
  extensions: MultiParamTypeClasses DeriveDataTypeable FunctionalDependencies FlexibleInstances
 
  build-depends: base       >= 4   && < 5,
-                haskell98  >= 1   && < 2,
                 containers >= 0.2 && < 1,
                 directory  >= 1   && < 2,
                 filepath   >= 1   && < 2,
