--- a/src/Data/Binary/Tagged.hs
+++ b/src/Data/Binary/Tagged.hs
@@ -20,6 +20,8 @@
 {-# LANGUAGE TypeOperators #-}
 -- We need this for Interleave
 {-# LANGUAGE UndecidableInstances #-}
+
+{-# LANGUAGE RecordWildCards #-}
 -----------------------------------------------------------------------------
 -- |
 -- Module      :  Data.Binary.Tagged
@@ -325,7 +327,7 @@ sopStructuralInfoS :: forall xss. ( All2 HasStructuralInfo xss
                    => DatatypeInfo xss
                    -> StructuralInfo
 sopStructuralInfoS di@(Newtype _ _ ci)  = NominalNewtype (datatypeName di) (sopNominalNewtype ci)
-sopStructuralInfoS di@(ADT _ _ _)       = StructuralInfo (datatypeName di) (sopNominalAdtPOP (hpure Proxy :: POP Proxy xss))
+sopStructuralInfoS di@(ADT{})       = StructuralInfo (datatypeName di) (sopNominalAdtPOP (hpure Proxy :: POP Proxy xss))
 
 sopNominalNewtype :: forall x. HasStructuralInfo x => ConstructorInfo '[x] -> StructuralInfo
 sopNominalNewtype _ = structuralInfo (Proxy :: Proxy x)
