diff --git a/glade.cabal b/glade.cabal
index 211ed47..44cef2c 100644
--- a/glade.cabal
+++ b/glade.cabal
@@ -62,7 +62,7 @@ Source-Repository head
   location:     http://code.haskell.org/glade/
 
 Library
-        build-depends:  base >= 4 && < 5, array, containers, haskell98, mtl,
+        build-depends:  base >= 4 && < 5, array, containers, mtl,
                         glib >= 0.12 && < 0.13, 
                         pango >= 0.12 && < 0.13, 
                         cairo >= 0.12 && < 0.13,
