diff --git a/src/Text/Trifecta/Instances.hs b/src/Text/Trifecta/Instances.hs
index 2f6ad9c..48a154f 100644
--- a/src/Text/Trifecta/Instances.hs
+++ b/src/Text/Trifecta/Instances.hs
@@ -1 +1,2 @@
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
@@ -17,3 +18,5 @@ import qualified Data.Semigroup as Data
 
+#if ! MIN_VERSION_ansi_wl_pprint(0,6,8)
 instance Data.Semigroup Doc where
   (<>) = (<>)
+#endif
