diff --git a/src/Text/Trifecta/Highlight.hs b/src/Text/Trifecta/Highlight.hs
index 14f9df5..e39df36 100644
--- a/src/Text/Trifecta/Highlight.hs
+++ b/src/Text/Trifecta/Highlight.hs
@@ -43,7 +43,7 @@ import qualified Data.ByteString.Lazy.UTF8 as LazyUTF8
 
 -- | Convert a 'Highlight' into a coloration on a 'Doc'.
 withHighlight :: Highlight -> Doc -> Doc
-withHighlight Comment                     = blue
+withHighlight Text.Parser.Token.Highlight.Comment                     = blue
 withHighlight ReservedIdentifier          = magenta
 withHighlight ReservedConstructor         = magenta
 withHighlight EscapeCode                  = magenta
