--- boxes-0.1.2-orig/boxes.cabal	2011-08-23 13:23:37.000000000 +1000
+++ boxes-0.1.2/boxes.cabal	2013-05-10 19:20:58.043632084 +1000
@@ -13,7 +13,7 @@
 cabal-version:       >= 1.6
 
 library
-  build-depends:     base >= 3 && < 5, split >=0.1 && <0.2
+  build-depends:     base >= 3 && < 5, split >=0.2 && <0.3
   exposed-modules:   Text.PrettyPrint.Boxes
 
 source-repository head
--- boxes-0.1.2-orig/Text/PrettyPrint/Boxes.hs	2011-08-23 13:23:37.000000000 +1000
+++ boxes-0.1.2/Text/PrettyPrint/Boxes.hs	2013-05-10 19:21:34.296630605 +1000
@@ -212,7 +212,7 @@
 --   at most @h@, containing text @t@ flowed into as many columns as
 --   necessary.
 columns :: Alignment -> Int -> Int -> String -> [Box]
-columns a w h t = map (mkParaBox a h) . chunk h $ flow w t
+columns a w h t = map (mkParaBox a h) . chunksOf h $ flow w t
 
 -- | @mkParaBox a n s@ makes a box of height @n@ with the text @s@
 --   aligned according to @a@.
