diff --git a/HTTP.cabal b/HTTP.cabal
index 590af1e..6435e2f 100644
--- a/HTTP.cabal
+++ b/HTTP.cabal
@@ -80,7 +80,7 @@ Library
                  Network.HTTP.MD5Aux,
                  Network.HTTP.Utils
   GHC-options: -fwarn-missing-signatures -Wall
-  Build-depends: base >= 2 && < 4.4, network, parsec, mtl
+  Build-depends: base >= 2 && < 4.5, network, parsec, mtl
   if flag(old-base)
     Build-depends: base < 3
   else
diff --git a/Network/BufferType.hs b/Network/BufferType.hs
index bb70e85..523323b 100644
--- a/Network/BufferType.hs
+++ b/Network/BufferType.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -----------------------------------------------------------------------------
 -- |
 -- Module      :  Network.BufferType
diff --git a/Network/TCP.hs b/Network/TCP.hs
index 225ea15..70e66bc 100644
--- a/Network/TCP.hs
+++ b/Network/TCP.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 -----------------------------------------------------------------------------
 -- |
 -- Module      :  Network.TCP
