diff --git a/test/main.hs b/test/main.hs
index 685b166..deb8016 100644
--- a/test/main.hs
+++ b/test/main.hs
@@ -6,7 +6,7 @@ import Control.Monad (replicateM_)
 main :: IO ()
 main = do
     _ <- forkIO $ runTCPServer (ServerSettings 4000 Nothing) echo
-    threadDelay 1000
+    threadDelay 1000000
     replicateM_ 10000
         $ runTCPClient (ClientSettings 4000 "localhost") doNothing
 
