diff --git a/libmpd.cabal b/libmpd.cabal
index a72cda0..fe350f5 100644
--- a/libmpd.cabal
+++ b/libmpd.cabal
@@ -39,7 +39,7 @@ Library
                         utf8-string >= 0.3.1 && < 0.4,
                         old-locale >= 1.0 && < 2.0,
                         time >= 1.1 && < 2.0,
-                        containers >= 0.4 && < 0.5
+                        containers >= 0.3 && < 0.5
     Exposed-Modules:    Network.MPD, Network.MPD.Commands.Extensions,
                         Network.MPD.Core
     Other-Modules:      Network.MPD.Core.Class,
