diff --git a/Lambdabot/Pointful.hs b/Lambdabot/Pointful.hs
index 8c3e67e..7811332 100644
--- a/Lambdabot/Pointful.hs
+++ b/Lambdabot/Pointful.hs
@@ -44,7 +44,7 @@ succAlpha []       = "a"
 optimizeD :: Decl -> Decl
-optimizeD (PatBind locat (PVar fname) (UnGuardedRhs (Lambda _ pats rhs)) (BDecls []))
-        =  FunBind [Match locat fname pats Nothing (UnGuardedRhs rhs) (BDecls [])]
+optimizeD (PatBind locat (PVar fname) (UnGuardedRhs (Lambda _ pats rhs)) Nothing)
+        =  FunBind [Match locat fname pats Nothing (UnGuardedRhs rhs) Nothing]
 ---- combine function binding and lambda
-optimizeD (FunBind [Match locat fname pats1 Nothing (UnGuardedRhs (Lambda _ pats2 rhs)) (BDecls [])])
-        =  FunBind [Match locat fname (pats1 ++ pats2) Nothing (UnGuardedRhs rhs) (BDecls [])]
+optimizeD (FunBind [Match locat fname pats1 Nothing (UnGuardedRhs (Lambda _ pats2 rhs)) Nothing])
+        =  FunBind [Match locat fname (pats1 ++ pats2) Nothing (UnGuardedRhs rhs) Nothing]
 optimizeD x = x
@@ -164,3 +164,3 @@ combinators = M.fromList $ map declToTuple defs
           f@(ParseFailed _ _) -> error ("Combinator loading: " ++ show f)
-        declToTuple (PatBind _ (PVar fname) (UnGuardedRhs body) (BDecls []))
+        declToTuple (PatBind _ (PVar fname) (UnGuardedRhs body) Nothing)
           = (UnQual fname, Paren body)
