diff --git a/src/System/Taffybar/FreedesktopNotifications.hs b/src/System/Taffybar/FreedesktopNotifications.hs
index e1e5add..c054b15 100644
--- a/src/System/Taffybar/FreedesktopNotifications.hs
+++ b/src/System/Taffybar/FreedesktopNotifications.hs
@@ -73,3 +73,3 @@ initialNoteState wrapper l cfg = do
 
-getServerInformation :: IO (Text, Text, Text, Text)
+getServerInformation :: DBusR (Text, Text, Text, Text)
 getServerInformation =
@@ -80,3 +80,3 @@ getServerInformation =
 
-getCapabilities :: IO [Text]
+getCapabilities :: DBusR [Text]
 getCapabilities = return ["body", "body-markup"]
@@ -199,8 +199,11 @@ notificationDaemon onNote onCloseNote = do
   _ <- requestName client "org.freedesktop.Notifications" [nameAllowReplacement, nameReplaceExisting]
-  export client "/org/freedesktop/Notifications"
-    [ autoMethod "org.freedesktop.Notifications" "GetServerInformation" getServerInformation
-    , autoMethod "org.freedesktop.Notifications" "GetCapabilities" getCapabilities
-    , autoMethod "org.freedesktop.Notifications" "CloseNotification" onCloseNote
-    , autoMethod "org.freedesktop.Notifications" "Notify" onNote
+  export client "/org/freedesktop/Notifications" defaultInterface {
+    interfaceName = "org.freedesktop.Notifications",
+    interfaceMethods = [
+      autoMethod "GetServerInformation" getServerInformation
+    , autoMethod "GetCapabilities" getCapabilities
+    , autoMethod "CloseNotification" onCloseNote
+    , autoMethod "Notify" onNote
     ]
+  }
 
diff --git a/src/System/Taffybar/MPRIS2.hs b/src/System/Taffybar/MPRIS2.hs
index 13e2f57..46b6aa9 100644
--- a/src/System/Taffybar/MPRIS2.hs
+++ b/src/System/Taffybar/MPRIS2.hs
@@ -11,3 +11,3 @@ import Data.Maybe ( listToMaybe )
 import DBus
-import DBus.Client
+import DBus.Client hiding ( getProperty )
 import Data.List (isPrefixOf)
diff --git a/taffybar.cabal b/taffybar.cabal
index b939041..cf7e13d 100644
--- a/taffybar.cabal
+++ b/taffybar.cabal
@@ -38,3 +38,3 @@ library
                  cairo,
-                 dbus >= 0.10.1 && < 1.0,
+                 dbus >= 1,
                  gtk >= 0.12.1 && < 0.15,
