Get rid of UTF-8 from comments to convince haddock.
diff --git a/src/System/Taffybar/Weather.hs b/src/System/Taffybar/Weather.hs
index 0ccc922..8f150e2 100644
--- a/src/System/Taffybar/Weather.hs
+++ b/src/System/Taffybar/Weather.hs
@@ -52,7 +52,7 @@
 --
 -- As an example, a template like
 --
--- > "$tempF$ °F"
+-- > "$tempF$ F"
 --
 -- would yield a widget displaying the temperature in Farenheit with a
 -- small label after it.
diff --git a/src/System/Taffybar/Weather.hs b/src/System/Taffybar/Weather.hs
index 8f150e2..0384e16 100644
--- a/src/System/Taffybar/Weather.hs
+++ b/src/System/Taffybar/Weather.hs
@@ -259,7 +259,7 @@ data WeatherFormatter = WeatherFormatter (WeatherInfo -> String) -- ^ Specify a
 -- 'weatherFormatter' field.
 data WeatherConfig =
   WeatherConfig { weatherStation :: String   -- ^ The weather station to poll. No default
-                , weatherTemplate :: String  -- ^ Template string, as described above.  Default: $tempF$ °F
+                , weatherTemplate :: String  -- ^ Template string, as described above.  Default: $tempF$ F
                 , weatherFormatter :: WeatherFormatter -- ^ Default: substitute in all interpolated variables (above)
                 }
 
@@ -267,7 +267,7 @@ data WeatherConfig =
 -- renders the temperature.
 defaultWeatherConfig :: String -> WeatherConfig
 defaultWeatherConfig station = WeatherConfig { weatherStation = station
-                                             , weatherTemplate = "$tempF$ °F"
+                                             , weatherTemplate = "$tempF$ F"
                                              , weatherFormatter = DefaultWeatherFormatter
                                              }
 
diff --git a/taffybar.cabal b/taffybar.cabal
index f634be5..439ca61 100644
--- a/taffybar.cabal
+++ b/taffybar.cabal
@@ -22,9 +22,9 @@ description:
   .
     * More robust strut handling for multiple monitors of different sizes (contributed by Morgan Gibson)
   .
-    * New widgets from José A. Romero (network monitor, fs monitor, another CPU monitor)
+    * New widgets from Jose A. Romero (network monitor, fs monitor, another CPU monitor)
   .
-    * Allow the bar widget to grow vertically (also contributed by José A. Romero)
+    * Allow the bar widget to grow vertically (also contributed by Jose A. Romero)
   .
   Changes in v0.2.0:
   .
