diff --git a/taffybar.cabal b/taffybar.cabal
index 4de7ea1..dcef04d 100644
--- a/taffybar.cabal
+++ b/taffybar.cabal
@@ -20,13 +20,13 @@ description:
   .
   Changes in v0.3.0:
   .
-    * A new pager (System.Taffybar.TaffyPager) from José A. Romero L.  This pager is a drop-in replacement for the dbus-based XMonadLog widget.  It communicates via X atoms and EWMH like a real pager.  It even supports changing workspaces by clicking on them.  I recommend this over the old widget.
+    * A new pager (System.Taffybar.TaffyPager) from Jose A. Romero L.  This pager is a drop-in replacement for the dbus-based XMonadLog widget.  It communicates via X atoms and EWMH like a real pager.  It even supports changing workspaces by clicking on them.  I recommend this over the old widget.
   .
     * Added an MPRIS2 widget (contributed by Igor Babuschkin)
   .
     * Ported to use the newer merged dbus library instead of dbus-client/dbus-core (contributed by CJ van den Berg)
   .
-    * Finally have the calendar widget pop up over the date/time widget (contributed by José A. Romero)
+    * Finally have the calendar widget pop up over the date/time widget (contributed by Jose A. Romero)
   .
     * GHC 7.6 compatibility
   .
