diff --git a/src/System/Taffybar/Util.hs b/src/System/Taffybar/Util.hs
index c306b98..d5c1250 100644
--- a/src/System/Taffybar/Util.hs
+++ b/src/System/Taffybar/Util.hs
@@ -24,6 +24,7 @@ import           Control.Monad.Trans.Reader
 import           Data.Either.Combinators
 import           Data.GI.Base.GError
 import qualified Data.GI.Gtk.Threading as Gtk
+import           Data.Maybe (fromJust)
 import qualified Data.Text as T
 import           Data.Tuple.Sequence
 import qualified GI.GdkPixbuf.Objects.Pixbuf as Gdk
@@ -129,7 +130,7 @@ catchGErrorsAsLeft action = catch (Right <$> action) (return . Left)
 
 safePixbufNewFromFile :: FilePath -> IO (Either GError Gdk.Pixbuf)
 safePixbufNewFromFile filepath =
-  catchGErrorsAsLeft (Gdk.pixbufNewFromFile filepath)
+  catchGErrorsAsLeft (fromJust <$> Gdk.pixbufNewFromFile filepath)
 
 getPixbufFromFilePath :: FilePath -> IO (Maybe Gdk.Pixbuf)
 getPixbufFromFilePath filepath = do
diff --git a/src/System/Taffybar/Widget/Util.hs b/src/System/Taffybar/Widget/Util.hs
index 45dac2f..2839bbd 100644
--- a/src/System/Taffybar/Widget/Util.hs
+++ b/src/System/Taffybar/Widget/Util.hs
@@ -20,6 +20,7 @@ import           Control.Monad
 import           Control.Monad.IO.Class
 import           Data.Functor ( ($>) )
 import           Data.Int
+import           Data.Maybe (fromJust)
 import qualified Data.Text as T
 import qualified GI.Gdk as D
 import qualified GI.GdkPixbuf.Objects.Pixbuf as GI
@@ -157,7 +158,7 @@ vFillCenter widget =
 
 pixbufNewFromFileAtScaleByHeight :: Int32 -> String -> IO PB.Pixbuf
 pixbufNewFromFileAtScaleByHeight height name =
-  PB.pixbufNewFromFileAtScale name (-1) height True
+  fromJust <$> PB.pixbufNewFromFileAtScale name (-1) height True
 
 loadIcon :: Int32 -> String -> IO PB.Pixbuf
 loadIcon height name =
