--- singletons-0.8.1-orig/singletons.cabal	2012-07-22 04:54:24.000000000 +1000
+++ singletons-0.8.1/singletons.cabal	2012-07-22 10:32:15.692499743 +1000
@@ -28,7 +28,7 @@
   build-depends:      
       base >= 4 && < 5,
       mtl >= 2.1.1,
-      template-haskell >= 2.8,
+      template-haskell >= 2.6,
       containers >= 0.5,
       syb >= 0.3
   exposed-modules:    Singletons.Lib, Singletons.CustomStar,
