--- singletons-0.8-orig/Singletons/Singletons.hs	2012-06-03 08:18:36.000000000 +1000
+++ singletons-0.8/Singletons/Singletons.hs	2012-07-20 12:56:35.389638970 +1000
@@ -12,7 +12,10 @@
 module Singletons.Singletons where
 
 import Language.Haskell.TH.Lib
-import Language.Haskell.TH.Syntax hiding (lift)
+import Language.Haskell.TH.Syntax(Body(..), Clause(..), Con(..), Cxt(..),
+       Dec(..), Exp(..), Foreign(..), Info(..), Kind(..), Match(..),
+       Name(..), Pat(..), Pragma(..), Pred(..), Strict(..), TyVarBndr(..),
+       Q(..), Type(..), mkName, nameBase, newName, report, tupleDataName)
 import Singletons.Util
 import Singletons.Promote
 import qualified Data.Map as Map
