diff --git a/Data/Generics/SYB/WithClass/Derive.hs b/Data/Generics/SYB/WithClass/Derive.hs
index 7d3fe61..6d70079 100644
--- a/Data/Generics/SYB/WithClass/Derive.hs
+++ b/Data/Generics/SYB/WithClass/Derive.hs
@@ -246,4 +246,9 @@ typeInfo d
  = case d of
+#if MIN_VERSION_template_haskell(2,11,0)
+   DataD    _ n ps _ cs _ -> return (n, map varName ps, map conA cs)
+   NewtypeD _ n ps _ c  _ -> return (n, map varName ps, [conA c])
+#else
    DataD    _ n ps cs _ -> return (n, map varName ps, map conA cs)
    NewtypeD _ n ps c  _ -> return (n, map varName ps, [conA c])
+#endif
    _ -> error ("derive: not a data type declaration: " ++ show d)
