diff --git a/HN/Optimizer/ArgumentValues.hs b/HN/Optimizer/ArgumentValues.hs
index ecf057a..f60a606 100644
--- a/HN/Optimizer/ArgumentValues.hs
+++ b/HN/Optimizer/ArgumentValues.hs
@@ -1,2 +1,3 @@
 {-# LANGUAGE GADTs, StandaloneDeriving, FlexibleInstances, DeriveFoldable,  MultiParamTypeClasses, FlexibleContexts #-}
+{-# LANGUAGE CPP #-}
 module HN.Optimizer.ArgumentValues (runAv, ArgFact, AFType) where
@@ -23,3 +24,5 @@ type SingleArgLattice a = DataflowLattice (WithTopAndBot a)
 deriving instance Show ChangeFlag
+#if !MIN_VERSION_GLASGOW_HASKELL(8,0,1,1)
 deriving instance Foldable (ListF [a])
+#endif
 
