diff --git a/hoodle-parser.cabal b/hoodle-parser.cabal
index 4418084..655f6c0 100644
--- a/hoodle-parser.cabal
+++ b/hoodle-parser.cabal
@@ -24 +24 @@ Library
-                 either >= 3.1,
+                 errors,
diff --git a/src/Text/Hoodle/Migrate/V0_1_1_to_V0_2_2.hs b/src/Text/Hoodle/Migrate/V0_1_1_to_V0_2_2.hs
index cca04da..91c8bfe 100644
--- a/src/Text/Hoodle/Migrate/V0_1_1_to_V0_2_2.hs
+++ b/src/Text/Hoodle/Migrate/V0_1_1_to_V0_2_2.hs
@@ -20,0 +21 @@ import           Control.Lens
+import           Control.Error (hoistEither)
@@ -22 +23 @@ import           Control.Monad.Trans
-import           Control.Monad.Trans.Either
+import           Control.Monad.Trans.Except
@@ -95 +96 @@ migrate bstr = do
-  runEitherT $ do 
+  runExceptT $ do 
diff --git a/src/Text/Hoodle/Migrate/V0_2_2_to_V0_3.hs b/src/Text/Hoodle/Migrate/V0_2_2_to_V0_3.hs
index b357e0c..754eda3 100644
--- a/src/Text/Hoodle/Migrate/V0_2_2_to_V0_3.hs
+++ b/src/Text/Hoodle/Migrate/V0_2_2_to_V0_3.hs
@@ -22,0 +23 @@ import           Control.Lens
+import           Control.Error (hoistEither)
@@ -24 +25 @@ import           Control.Monad.Trans
-import           Control.Monad.Trans.Either
+import           Control.Monad.Trans.Except
@@ -125 +126 @@ migrate bstr = do
-  runEitherT $ do 
+  runExceptT $ do 
@@ -127 +128 @@ migrate bstr = do
-    if | v <= "0.1.1" -> do oh <- EitherT (Text.Hoodle.Migrate.V0_1_1_to_V0_2_2.migrate bstr)
+    if | v <= "0.1.1" -> do oh <- ExceptT (Text.Hoodle.Migrate.V0_1_1_to_V0_2_2.migrate bstr)
