--- HDBC-2.4.0.2-orig/Database/HDBC/Locale.hs	2017-08-22 01:17:54.000000000 +1000
+++ HDBC-2.4.0.2/Database/HDBC/Locale.hs	2018-04-17 22:00:49.586330905 +1000
@@ -7,7 +7,7 @@
 
 where
 
-#ifdef MIN_TIME_15
+#if MIN_VERSION_time(1,5,0)
 import Data.Time.Format (defaultTimeLocale)
 #else
 import System.Locale (defaultTimeLocale)
--- HDBC-2.4.0.2-orig/testsrc/TestSqlValue.hs	2017-08-22 01:17:54.000000000 +1000
+++ HDBC-2.4.0.2/testsrc/TestSqlValue.hs	2018-04-17 22:07:11.359792368 +1000
@@ -13,7 +13,7 @@
 import Test.QuickCheck.Tools
 import qualified Test.HUnit as HU
 import Database.HDBC
-import Data.Time.Format
+import qualified Data.Time.Format as TF
 import Data.Time.LocalTime
 import Database.HDBC.Locale (defaultTimeLocale, iso8601DateFormat)
 import Data.Maybe
@@ -31,12 +31,12 @@
 
 testZonedTimeStr = "1989-08-01 15:33:01 -0500"
 testZonedTime :: ZonedTime
-testZonedTime = fromJust $ parseTime defaultTimeLocale (iso8601DateFormat (Just "%T %z"))
+testZonedTime = fromJust $ TF.parseTime defaultTimeLocale (iso8601DateFormat (Just "%T %z"))
                 testZonedTimeStr
 
 testZonedTimeFracStr = "1989-08-01 15:33:01.536 -0500"
 testZonedTimeFrac :: ZonedTime
-testZonedTimeFrac = fromJust $ parseTime defaultTimeLocale (iso8601DateFormat (Just "%T%Q %z"))
+testZonedTimeFrac = fromJust $ TF.parseTime defaultTimeLocale (iso8601DateFormat (Just "%T%Q %z"))
                     testZonedTimeFracStr
 
 ztparsenf =
