diff --git a/Control/Monad/Maybe.hs b/Control/Monad/Maybe.hs
index b4c03e5..ebe4dda 100644
--- a/Control/Monad/Maybe.hs
+++ b/Control/Monad/Maybe.hs
@@ -33,3 +33,4 @@ module Control.Monad.Maybe (
 
-import Control.Monad()
+import Control.Applicative
+import Control.Monad
 import Control.Monad.Trans()
@@ -47,2 +48,6 @@ instance (Functor m) => Functor (MaybeT m) where
 
+instance (Monad m) => Applicative (MaybeT m) where
+    pure  = return
+    (<*>) = ap  -- defined in Control.Monad
+
 instance (Monad m) => Monad (MaybeT m) where
@@ -52,2 +57,6 @@ instance (Monad m) => Monad (MaybeT m) where
 
+instance (Monad m) => Alternative (MaybeT m) where
+    (<|>) = mplus
+    empty = mzero
+
 instance (Monad m) => MonadPlus (MaybeT m) where
