diff --git a/src/Text/Digestive/Blaze/Html5.hs b/src/Text/Digestive/Blaze/Html5.hs
index 95966bc..f2e86af 100644
--- a/src/Text/Digestive/Blaze/Html5.hs
+++ b/src/Text/Digestive/Blaze/Html5.hs
@@ -3,2 +3,3 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE CPP               #-}
 module Text.Digestive.Blaze.Html5
@@ -22,3 +23,3 @@ module Text.Digestive.Blaze.Html5
 import           Control.Monad               (forM_, when)
-import           Data.Maybe                  (fromMaybe)
+import           Data.Maybe                  (fromMaybe, listToMaybe)
 import           Data.Monoid                 (mappend, mempty)
@@ -145,3 +146,7 @@ inputFile ref view = H.input
     ref'  = absoluteRef ref view
-    value = fromMaybe "" $ fieldInputFile ref view
+    value = fromMaybe "" $
+#if MIN_VERSION_digestive_functors(0,8,1)
+              listToMaybe $
+#endif
+                fieldInputFile ref view
 
