--- test-shouldbe-0.2.1-orig/test-shouldbe.cabal	2012-06-19 17:35:11.000000000 +1000
+++ test-shouldbe-0.2.1/test-shouldbe.cabal	2012-06-28 00:21:22.032351616 +1000
@@ -20,7 +20,7 @@
   ghc-options:
       -Wall
   build-depends:
-      base < 4.6
+      base < 4.7
     , HUnit
   hs-source-dirs:
       src
@@ -38,7 +38,7 @@
   hs-source-dirs:
       src, test
   build-depends:
-      base        >= 4.0  && < 4.6
+      base        >= 4.0  && < 4.7
     , silently
     , hspec
     , hspec-discover
--- test-shouldbe-0.2.1-orig/src/Test/HUnit/ShouldBe.hs	2012-06-19 17:35:11.000000000 +1000
+++ test-shouldbe-0.2.1/src/Test/HUnit/ShouldBe.hs	2012-06-28 00:21:47.880952612 +1000
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 -- |
 -- Introductory documentation: <https://github.com/sol/test-shouldbe#readme>
 module Test.HUnit.ShouldBe (
@@ -34,7 +35,9 @@
 , errorCall
 ) where
 
+#if __GLASGOW_HASKELL__ < 705
 import           Prelude hiding (catch)
+#endif
 import           Test.HUnit
 import           Control.Exception
 import           Data.Typeable
