--- a/src/Language/Python/Common/LexerUtils.hs
+++ b/src/Language/Python/Common/LexerUtils.hs
@@ -18,3 +18,3 @@ import Control.Monad.Error.Class (throwError)
 import Data.List (foldl')
-import Data.Map as Map hiding (null, map, foldl')
+import Data.Map as Map hiding (null, map, foldl', take, drop)
 import Data.Word (Word8)
--- a/src/Language/Python/Common/PrettyAST.hs
+++ b/src/Language/Python/Common/PrettyAST.hs
@@ -16,3 +16,3 @@ import Language.Python.Common.Pretty
 import Language.Python.Common.AST 
-
+import Prelude hiding ((<>))
 --------------------------------------------------------------------------------
--- a/src/Language/Python/Common/SrcLocation.hs
+++ b/src/Language/Python/Common/SrcLocation.hs
@@ -38,2 +38,3 @@ import Language.Python.Common.Pretty
 import Data.Data
+import Prelude hiding ((<>))
 
