--- ansi-wl-pprint-0.6.7.1-orig/Text/PrettyPrint/ANSI/Leijen.hs	2013-12-26 07:20:10.000000000 +1100
+++ ansi-wl-pprint-0.6.7.1/Text/PrettyPrint/ANSI/Leijen.hs	2015-03-18 09:32:41.610373888 +1100
@@ -96,7 +96,7 @@
    align, hang, indent, encloseSep, list, tupled, semiBraces,
 
    -- * Operators
-   (<+>), (<$>), (</>), (<$$>), (<//>),
+   (<+>), (Text.PrettyPrint.ANSI.Leijen.<$>), (</>), (<$$>), (<//>),
 
    -- * List combinators
    hsep, vsep, fillSep, sep, hcat, vcat, fillCat, cat, punctuate,
@@ -309,7 +309,7 @@
 --      out
 -- @
 vsep :: [Doc] -> Doc
-vsep            = fold (<$>)
+vsep            = fold (Text.PrettyPrint.ANSI.Leijen.<$>)
 
 -- | The document @(cat xs)@ concatenates all documents @xs@ either
 -- horizontally with @(\<\>)@, if it fits the page, or vertically with
