diff --git a/Monadius.cabal b/Monadius.cabal
index f465515..2d5afb1 100644
--- a/Monadius.cabal
+++ b/Monadius.cabal
@@ -25,8 +25,8 @@ Executable monadius
   Build-depends:       base      >= 4 && < 5
                      , array     >= 0.3 && < 0.5
                      , directory >= 1.1 && < 1.2
-                     , GLUT      >= 2.2 && < 2.3
-                     , OpenGL    >= 2.4 && < 2.5
+                     , GLUT      >= 2.2 && < 2.4
+                     , OpenGL    >= 2.4 && < 2.6
 
   Hs-source-dirs:      Monadius
   Ghc-options:         -Wall
