From 755ee9e0aa3cb371359fade2111e45f843a0cc17 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Mon, 16 Jan 2012 23:40:06 +0300
Subject: [PATCH 2/2] allow array-0.4.* (for ghc-7.4.1-rc1)

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 Monadius.cabal |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Monadius.cabal b/Monadius.cabal
index 5d3ae16..f465515 100644
--- a/Monadius.cabal
+++ b/Monadius.cabal
@@ -23,7 +23,7 @@ Data-files:          readme-e.txt, readme-j.txt
 
 Executable monadius
   Build-depends:       base      >= 4 && < 5
-                     , array     >= 0.3 && < 0.4
+                     , array     >= 0.3 && < 0.5
                      , directory >= 1.1 && < 1.2
                      , GLUT      >= 2.2 && < 2.3
                      , OpenGL    >= 2.4 && < 2.5
-- 
1.7.8.3

