From 9cc26d973bdbbbb5fdf6ccd41f5dd0f426fc9c18 Mon Sep 17 00:00:00 2001
From: Jack Todaro <jackmtodaro@gmail.com>
Date: Sun, 19 Aug 2018 10:12:30 +1000
Subject: [PATCH] ghc 8.4 compatibility

---
 src/Yesod/Markdown.hs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/Yesod/Markdown.hs b/src/Yesod/Markdown.hs
index 62cef45..e44046f 100644
--- a/src/Yesod/Markdown.hs
+++ b/src/Yesod/Markdown.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE ExplicitForAll #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
@@ -48,8 +49,12 @@ import Yesod.Form.Types
 import qualified Data.ByteString as B
 import qualified Data.Text as T
 
+#if !MIN_VERSION_base(4,11,0)
+import Data.Semigroup
+#endif
+
 newtype Markdown = Markdown { unMarkdown :: Text }
-    deriving (Eq, Ord, Show, Read, PersistField, IsString, Monoid)
+    deriving (Eq, Ord, Show, Read, PersistField, IsString, Monoid, Semigroup)
 
 instance PersistFieldSql Markdown where
     sqlType _ = SqlString
-- 
2.18.0

