--- websockets-0.8.2.5-orig/websockets.cabal	2014-06-27 22:44:42.000000000 +1000
+++ websockets-0.8.2.5/websockets.cabal	2014-07-02 16:16:25.990877403 +1000
@@ -94,7 +94,7 @@
 
   Build-depends:
     HUnit                      >= 1.2 && < 1.3,
-    QuickCheck                 >= 2.4 && < 2.7,
+    QuickCheck                 >= 2.4 && < 2.8,
     test-framework             >= 0.4 && < 0.9,
     test-framework-hunit       >= 0.2 && < 0.4,
     test-framework-quickcheck2 >= 0.2 && < 0.4,
--- websockets-0.8.2.5-orig/tests/haskell/Network/WebSockets/Server/Tests.hs	2014-06-27 22:44:42.000000000 +1000
+++ websockets-0.8.2.5/tests/haskell/Network/WebSockets/Server/Tests.hs	2014-07-02 16:33:05.048485696 +1000
@@ -1,4 +1,5 @@
 --------------------------------------------------------------------------------
+{-# LANGUAGE CPP                 #-}
 {-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 module Network.WebSockets.Server.Tests
@@ -19,7 +20,12 @@
 --------------------------------------------------------------------------------
 import qualified Data.ByteString.Lazy           as BL
 import           Data.Text                      (Text)
+#if MIN_VERSION_QuickCheck(2,7,0)
+import           Test.QuickCheck.Random         (newQCGen)
+#else
 import           System.Random                  (newStdGen)
+#define          newQCGen                       newStdGen
+#endif
 import           Test.Framework                 (Test, testGroup)
 import           Test.Framework.Providers.HUnit (testCase)
 import           Test.HUnit                     (Assertion, assert, (@=?))
@@ -78,7 +84,7 @@
 --------------------------------------------------------------------------------
 sample :: Arbitrary a => IO [a]
 sample = do
-    gen <- newStdGen
+    gen <- newQCGen
     return $ (unGen arbitrary) gen 512
 
 
