diff --git a/Hakyll/Web/Agda.hs b/Hakyll/Web/Agda.hs
index 7caa871..4e0e4ff 100644
--- a/Hakyll/Web/Agda.hs
+++ b/Hakyll/Web/Agda.hs
@@ -47,3 +47,3 @@ checkFile file = do
     TCM.resetState
-    toTopLevelModuleName . TCM.iModuleName . fst <$> Imp.typeCheckMain file
+    toTopLevelModuleName . TCM.iModuleName . fst <$> Imp.typeCheckMain file Imp.TypeCheck
 
@@ -110,3 +110,3 @@ annotate m pos mi = anchor ! attributes
     attributes = [name (show pos)] ++
-                 fromMaybe [] (definitionSite mi >>= link) ++
+                 fromMaybe [] (definitionSite mi >>= \ds -> link (defSiteModule ds, defSitePos ds)) ++
                  (case classes of [] -> []; cs -> [theclass (unwords cs)])
