diff -urwpN gtk2hs-0.9.10.orig/configure.ac gtk2hs-0.9.10/configure.ac
--- gtk2hs-0.9.10.orig/configure.ac	2006-02-20 17:30:07.000000000 +0000
+++ gtk2hs-0.9.10/configure.ac	2006-02-20 17:57:55.000000000 +0000
@@ -381,7 +381,8 @@ dnl bindings for Haskell. So set the GTK
 dnl appropaiate value to substitute into the gtk package files
 
 if test "$HAVE_GTK_VERSION_2_8" = "yes" -a "$ENABLE_CAIRO" = "yes"; then
-  GTK_CAIRO_DEPEND=", cairo-${PACKAGE_VERSION}"
+  GTK_CAIRO_DEPEND=", cairo-${PACKAGE_VERSION}";
+  GTK_CAIRO_DEPEND_NO_CABAL=",\"cairo\"";
 fi
 
 dnl The configuration program for GTK is kind of stupid in that it
@@ -713,6 +714,7 @@ AC_SUBST(GHCPKG_USE_AUTOLIBS)
 AC_SUBST(GHCPKG_BUILD_GHCI_LIB)
 dnl Optional packages
 AC_SUBST(GTK_CAIRO_DEPEND)
+AC_SUBST(GTK_CAIRO_DEPEND_NO_CABAL)
 dnl Documentation
 AC_SUBST(HADDOCK)
 AC_SUBST(GHC_DOCDIR)
diff -urwpN gtk2hs-0.9.10.orig/gtk/gtk.pkg.in gtk2hs-0.9.10/gtk/gtk.pkg.in
--- gtk2hs-0.9.10.orig/gtk/gtk.pkg.in	2006-02-20 17:30:07.000000000 +0000
+++ gtk2hs-0.9.10/gtk/gtk.pkg.in	2006-02-20 17:32:33.000000000 +0000
@@ -8,7 +8,7 @@ Package {
      extra_libraries = [@GTK_LIBS_CQ@],
      include_dirs    = [@GTK_CFLAGS_CQ@],
      c_includes      = ["gtk/gtk.h"],
-     package_deps    = ["glib" @GTK_CAIRO_DEPEND@],
+     package_deps    = ["glib" @GTK_CAIRO_DEPEND_NO_CABAL@],
      extra_ghc_opts  = [],
      extra_cc_opts   = [],
      extra_ld_opts   = [@GTK_LIBEXTRA_CQ@]
