diff -urwpN gtk2hs-0.9.10.orig/mk/common.mk gtk2hs-0.9.10/mk/common.mk
--- gtk2hs-0.9.10.orig/mk/common.mk	2006-02-13 16:31:33.184111717 +0000
+++ gtk2hs-0.9.10/mk/common.mk	2006-02-13 17:26:41.972180579 +0000
@@ -79,6 +79,8 @@ noDeps   := $(strip $(findstring clean,$
 	$(strip $(HC) -c $< -o $@ $(INCLUDES) \
 		$(AM_CPPFLAGS) $($(PKG)_CPPFLAGS) $(CPPFLAGS) \
 		$(addprefix -optc,$(AM_CFLAGS) $($(PKG)_CFLAGS) \
+				  $(call getVar,$<,CFLAGS) $(CFLAGS)) \
+		$(addprefix -opta,$(AM_CFLAGS) $($(PKG)_CFLAGS) \
 				  $(call getVar,$<,CFLAGS) $(CFLAGS)))
 
 # The cheeky rule for .hi files says that .hi files can be created as
@@ -106,7 +108,8 @@ noDeps   := $(strip $(findstring clean,$
 	$(strip $(HSC2HS) $(HSCFLAGS) +RTS $(HSTOOLFLAGS) -RTS \
         $(addprefix -L-optl,$(AM_LDFLAGS) $(LDFLAGS) $($(PKG)_LIBS)) \
         $(addprefix -C,	$(filter-out -I%,$(AM_CPPFLAGS) $(CPPFLAGS)) \
-	$(addprefix -optc,$(AM_CFLAGS) $(CFLAGS) $($(PKG)_CFLAGS)))\
+	$(addprefix -optc,$(AM_CFLAGS) $(CFLAGS) $($(PKG)_CFLAGS)) \
+	$(addprefix -opta,$(AM_CFLAGS) $(CFLAGS) $($(PKG)_CFLAGS))) \
         $(filter -I%,$(AM_CPPFLAGS) $(CPPFLAGS)) $($(PKG)_CPPFLAGS)\
 	-C'-optc-include' -C'-optc$(CONFIG_HEADER)' \
 	--include $($(PKG)_HEADER) \
