diff --git a/bio.cabal b/bio.cabal
index 6497311..329dda7 100644
--- a/bio.cabal
+++ b/bio.cabal
@@ -38,7 +38,7 @@ Flag examples
   Default:           True
 
 Library
-  Build-Depends:     base>=4 && <5, binary >=0.4 && <0.5, tagsoup>=0.8, bytestring >= 0.9.1,
+  Build-Depends:     base>=4 && <5, binary >=0.4 && <0.7, tagsoup>=0.8, bytestring >= 0.9.1,
                      containers, array, parallel, parsec, mtl, directory, QuickCheck >= 2
 
   Exposed-modules:   Bio.Sequence,
