diff --git a/snap.cabal b/snap.cabal
index 7d02e19..4f05518 100644
--- a/snap.cabal
+++ b/snap.cabal
@@ -124,7 +124,7 @@ Library
     base                      >= 4        && < 5,
     bytestring                >= 0.9.1    && < 0.11,
     cereal                    >= 0.3      && < 0.4,
-    clientsession             >= 0.7.3.6  && < 0.8,
+    clientsession             >= 0.7.3.6  && < 0.9,
     configurator              >= 0.1      && < 0.3,
     containers                >= 0.3      && < 0.6,
     directory                 >= 1.0      && < 1.2,
