--- snap-0.7-orig/snap.cabal	2011-12-02 12:08:06.000000000 +1100
+++ snap-0.7/snap.cabal	2011-12-20 21:21:50.154543310 +1100
@@ -113,13 +113,13 @@
     directory-tree            >= 0.10     && < 0.11,
     data-lens                 >= 2.0.1    && < 2.1,
     data-lens-template        >= 2.1      && < 2.2,
-    filepath                  >= 1.1      && < 1.3,
+    filepath                  >= 1.1      && < 1.4,
     hashable                  >= 1.1      && < 1.2,
     heist                     >= 0.7      && < 0.8,
     logict                    >= 0.4.2    && < 0.6,
     mtl                       >  2.0      && < 2.1,
     mwc-random                >= 0.8      && < 0.11,
-    old-time                  >= 1.0      && < 1.1,
+    old-time                  >= 1.0      && < 1.2,
     pwstore-fast              >= 2.2      && < 2.3,
     safe                      >= 0.3      && < 0.4,
     skein                     >= 0.1.0.3  && < 0.2,
@@ -127,7 +127,7 @@
     snap-server               >= 0.7      && < 0.8,
     stm                       >= 2.2      && < 2.3,
     syb                       >= 0.1      && < 0.4,
-    template-haskell          >= 2.2      && < 2.7,
+    template-haskell          >= 2.2      && < 2.8,
     text                      >= 0.11     && < 0.12,
     time                      >= 1.1      && < 1.5,
     transformers              >= 0.2      && < 0.3,
@@ -175,10 +175,10 @@
     containers          >= 0.3     && < 0.5,
     directory           >= 1.0     && < 1.2,
     directory-tree      >= 0.10    && < 0.11,
-    filepath            >= 1.1     && < 1.3,
-    old-time            >= 1.0     && < 1.1,
+    filepath            >= 1.1     && < 1.4,
+    old-time            >= 1.0     && < 1.2,
     snap-server         >= 0.7     && < 0.8,
-    template-haskell    >= 2.2     && < 2.7,
+    template-haskell    >= 2.2     && < 2.8,
     text                >= 0.11    && < 0.12
 
   ghc-prof-options: -prof -auto-all
