--- DBus-0.4-orig/Setup.hs	2008-09-08 09:28:20.000000000 +1000
+++ DBus-0.4/Setup.hs	2012-04-06 21:20:16.206326717 +1000
@@ -1,2 +1,2 @@
 import Distribution.Simple
-main = defaultMain
+main = defaultMainWithHooks autoconfUserHooks
--- DBus-0.4-orig/DBus/Internal.hsc	2008-09-08 09:28:20.000000000 +1000
+++ DBus-0.4/DBus/Internal.hsc	2012-04-06 21:47:57.338506460 +1000
@@ -9,7 +9,7 @@
 
 module DBus.Internal where
 
-import Control.Exception (throwDyn)
+import Control.OldException (throwDyn)
 import Control.Monad (when)
 import DBus (Error(..))
 import Foreign
--- DBus-0.4-orig/DBus/Message.hsc	2008-09-08 09:28:20.000000000 +1000
+++ DBus-0.4/DBus/Message.hsc	2012-04-06 21:33:00.931142451 +1000
@@ -1,4 +1,4 @@
-{-# LANGUAGE PatternSignatures, TypeSynonymInstances, FlexibleInstances #-}
+{-# LANGUAGE ScopedTypeVariables, TypeSynonymInstances, FlexibleInstances #-}
 {-# OPTIONS -fglasgow-exts #-}
 -- HDBus -- Haskell bindings for D-Bus.
 -- Copyright (C) 2006 Evan Martin <martine@danga.com>
@@ -38,7 +38,8 @@
 import Data.Word
 import Data.Char
 import Data.Dynamic
-import Foreign
+import Foreign (ForeignPtr(), Ptr(), Storable(), alloca, allocaBytes, 
+                maybePeek, newForeignPtr_, nullPtr, peek, with, withForeignPtr)
 import Foreign.C.String
 import Foreign.C.Types (CInt)
 import System.IO.Unsafe
