Workarounds https://ghc.haskell.org/trac/ghc/ticket/12127
diff --git a/src/Values.hs b/src/Values.hs
index e21bacf..2859486 100644
--- a/src/Values.hs
+++ b/src/Values.hs
@@ -13,3 +13,3 @@ where
 import Data.Array.Repa (
-  Array, U,
+  Source(..), U,
   Z( Z ),
diff --git a/src/Volumetric.hs b/src/Volumetric.hs
index 5b24760..c24a64d 100644
--- a/src/Volumetric.hs
+++ b/src/Volumetric.hs
@@ -24,3 +24,3 @@ import Data.Array.Repa (
   (:.)( (:.) ),
-  Array,
+  Source(..),
   D,
